
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day1/ex5_evaluating_assemblies/">
      
      
        <link rel="prev" href="../ex4_assembly/">
      
      
        <link rel="next" href="../../day2/ex6_initial_binning/">
      
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.34">
    
    
      
        <title>Assembly evaluation - Metagenomics Summer School</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.35f28582.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#assembly-evaluation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Metagenomics Summer School" class="md-header__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Summer School
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Assembly evaluation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../ex1_bash_and_scheduler/" class="md-tabs__link">
          
  
  Day 1

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day2/ex6_initial_binning/" class="md-tabs__link">
          
  
  Day 2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-tabs__link">
          
  
  Day 3

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day4/ex15_gene_annotation_part3/" class="md-tabs__link">
          
  
  Day 4

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-tabs__link">
          
  
  Resources

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../supplementary/supplementary_1/" class="md-tabs__link">
          
  
  Supplementary

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Metagenomics Summer School" class="md-nav__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Metagenomics Summer School
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Day 1
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex1_bash_and_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction I: Shell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex2_1_intro_to_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction II: HPC and job scheduler
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex2_quality_filtering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filter raw reads by quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex3_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly I: Assembling contigs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex4_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly II: Varying the parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Assembly evaluation
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Assembly evaluation
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#evaluating-the-resource-consumption-of-various-assemblies" class="md-nav__link">
    <span class="md-ellipsis">
      Evaluating the resource consumption of various assemblies
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluating-the-assemblies-using-bbmap" class="md-nav__link">
    <span class="md-ellipsis">
      Evaluating the assemblies using BBMap
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reconstruct-rrna-using-phyloflash-emirge" class="md-nav__link">
    <span class="md-ellipsis">
      Reconstruct rRNA using PhyloFlash-EMIRGE
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequence-taxonomic-classification-using-kraken2" class="md-nav__link">
    <span class="md-ellipsis">
      Sequence taxonomic classification using Kraken2
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optional-evaluating-assemblies-using-metaquast" class="md-nav__link">
    <span class="md-ellipsis">
      (Optional) Evaluating assemblies using MetaQUAST
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 2
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex6_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to binning
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex7_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Binning with multiple tools
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex8_bin_dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bin dereplication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex9_refining_bins/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Manual bin refinement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex10.1_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifying viral contigs in metagenomic data
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 3
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assigning taxonomy to refined prokaryotic bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11.1_phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex10.2_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virus taxonomy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex12_gene_prediction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene prediction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex13_gene_annotation_part1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation I: BLAST-like and HMM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex14_gene_annotation_part2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation II: DRAM and coverage calculation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 4
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex15_gene_annotation_part3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation III: DRAM distillation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16a_data_presentation_Intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to data presentation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16b_data_presentation_Coverage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coverage heatmaps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ordinations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16d_data_presentation_KEGG_pathways/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    KEGG pathway maps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16e_data_presentation_Gene_synteny/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene synteny
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CAZy heatmaps
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Resources
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex8): Dereplicating data from multiple assemblies
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex9): Generating input files for VizBin from DAS_Tool curated bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11) : Viral taxonomy prediction via vContact2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/7_command_line_shortcuts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Useful command line shortcuts
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Supplementary
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Supplementary
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI Setup
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI File system, Working directory and Symlinks
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#evaluating-the-resource-consumption-of-various-assemblies" class="md-nav__link">
    <span class="md-ellipsis">
      Evaluating the resource consumption of various assemblies
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluating-the-assemblies-using-bbmap" class="md-nav__link">
    <span class="md-ellipsis">
      Evaluating the assemblies using BBMap
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reconstruct-rrna-using-phyloflash-emirge" class="md-nav__link">
    <span class="md-ellipsis">
      Reconstruct rRNA using PhyloFlash-EMIRGE
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequence-taxonomic-classification-using-kraken2" class="md-nav__link">
    <span class="md-ellipsis">
      Sequence taxonomic classification using Kraken2
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optional-evaluating-assemblies-using-metaquast" class="md-nav__link">
    <span class="md-ellipsis">
      (Optional) Evaluating assemblies using MetaQUAST
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="assembly-evaluation">Assembly evaluation<a class="headerlink" href="#assembly-evaluation" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#evaluating-the-resource-consumption-of-various-assemblies">Evaluating the resource consumption of various assemblies</a></li>
<li><a href="#evaluating-the-assemblies-using-bbmap">Evaluating the assemblies using <code>BBMap</code></a></li>
<li><a href="#sequence-taxonomic-classification-using-kraken2">Sequence taxonomic classification using <code>Kraken2</code></a></li>
<li><a href="#reconstruct-rrna-using-phyloflash-emirge">Reconstruct rRNA using <code>PhyloFlash</code>-<code>EMIRGE</code></a></li>
<li><a href="#optional-evaluating-assemblies-using-metaquast"><em>(Optional)</em> Evaluating assemblies using <code>MetaQUAST</code></a></li>
</ul>
</div>
<hr />
<p><center>
<a class="glightbox" href="../../theme_images/eval_assembly.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../theme_images/eval_assembly.png" width="450" /></a>
</center></p>
<h2 id="evaluating-the-resource-consumption-of-various-assemblies">Evaluating the resource consumption of various assemblies<a class="headerlink" href="#evaluating-the-resource-consumption-of-various-assemblies" title="Permanent link">&para;</a></h2>
<p>Check to see if your assembly jobs have completed. If you have multiple jobs running or queued, the easiest way to check this is to simply run the <code>squeue</code> command.</p>
<div class="admonition terminal-2">
<p class="admonition-title">Check job progress</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>squeue<span class="w"> </span>--me
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>JOBID         USER     ACCOUNT   NAME        CPUS MIN_MEM PARTITI START_TIME     TIME_LEFT STATE    NODELIST(REASON)    
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>39035482      jboe440  nesi02659 spawner-jupy   2      4G interac 2023-08-31T1     7:47:42 RUNNING  wbn004      
</code></pre></div>
</div>
<p>If there are no jobs besides your Jupyter session listed, either everything running has completed or failed. To get a list of all jobs we have run in the last day, we can use the <code>sacct</code> command. By default this will report all jobs for the day but we can add a parameter to tell the command to report all jobs run since the date we are specifying.</p>
<div class="admonition terminal-2">
<p class="admonition-title">Check progress for jobs started on specific date</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>sacct<span class="w"> </span>-S<span class="w"> </span><span class="m">2023</span>-08-12
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>JobID           JobName          Alloc     Elapsed     TotalCPU  ReqMem   MaxRSS State      
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>--------------- ---------------- ----- ----------- ------------ ------- -------- ---------- 
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>38483216        spawner-jupyter+     2    07:45:01     00:00:00      4G          NODE_FAIL  
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>38483216.batch  batch                2    07:45:01     00:00:00                  CANCELLED  
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>38483216.extern extern               2    07:45:01     00:00:00                  CANCELLED  
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>38485254        spades_assembly     12    00:14:38     01:56:40     10G          COMPLETED  
<a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>38485254.batch  batch               12    00:14:38     01:56:40         7227872K COMPLETED  
<a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a>38485254.extern extern              12    00:14:38     00:00:00                0 COMPLETED
</code></pre></div>
</div>
<p>Each job has been broken up into several lines, but the main ones to keep an eye on are the base <code>JobID</code> values. </p>
<details class="circle-info">
<summary>Using <code>srun</code></summary>
<p>If you use <code>srun</code>, the JobID will have values suffixed with <em>.0</em>. The first of these references the complete job. The later (and any subsequent suffixes like <em>.1</em>, <em>.2</em>) are the individual steps in the script that were called with the <code>srun</code> command.</p>
</details>
<p>We can see here the time elapsed for each job, and the number of CPU hours used during the run. If we want a more detailed breakdown of the job we can use the <code>nn_seff</code> command</p>
<div class="admonition terminal-2">
<p class="admonition-title">Check job resource use</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>nn_seff<span class="w"> </span><span class="m">38485254</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>Cluster: mahuika
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>Job ID: 38485254
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>State: COMPLETED
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>Cores: 6
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>Tasks: 1
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>Nodes: 1
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>Job Wall-time:   48.8%  00:14:38 of 00:30:00 time limit
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>CPU Efficiency: 132.9%  01:56:40 of 01:27:48 core-walltime
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>Mem Efficiency:  68.9%  6.89 GB of 10.00 GB0
</code></pre></div>
</div>
<p>Here we see some of the same information, but we also get some information regarding how well our job used the resources we allocated to it. You can see here that my CPU and memory usage was somewhat efficient but had high memory efficiency. In the future, I can request less time and retain the same RAM and still had the job run to completion.</p>
<p>CPU efficiency is harder to interpret as it can be impacted by the behaviour of the program. For example, mapping tools like <code>bowtie</code> and <code>BBMap</code> can more or less use all of their threads, all of the time and achieve nearly 100% efficiency. More complicated processes, like those performed in <code>SPAdes</code> go through periods of multi-thread processing and periods of single-thread processing, drawing the average efficiency down.</p>
<hr />
<h2 id="evaluating-the-assemblies-using-bbmap">Evaluating the assemblies using <code>BBMap</code><a class="headerlink" href="#evaluating-the-assemblies-using-bbmap" title="Permanent link">&para;</a></h2>
<p>Evaluating the quality of a raw metagenomic assembly is quite a tricky process. Since, by definition, our community is a mixture of different organisms, the genomes from some of these organisms assemble better than those of others. It is possible to have an assembly that looks 'bad' by traditional metrics that still yields high-quality genomes from individual species, and the converse is also true.</p>
<p>A few quick checks I recommend are to see how many contigs or scaffolds your data were assembled into, and then see how many contigs or scaffolds you have above a certain minimum length threshold. We will use <code>seqmagick</code> for performing the length filtering, and then just count sequence numbers using <code>grep</code>.</p>
<p>These steps will take place in the <code>4.evaluation/</code> folder, which contains copies of our <code>SPAdes</code> and <code>IDBA-UD</code> assemblies.</p>
<div class="admonition warning">
<p class="admonition-title">Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own folder</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="c1"># Load seqmagick</span>
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>module<span class="w"> </span>purge
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>module<span class="w"> </span>load<span class="w"> </span>seqmagick/0.8.4-gimkl-2020a-Python-3.8.2
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>
<a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a><span class="c1"># Navigate to working directory</span>
<a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/4.evaluation/
<a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a>
<a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a><span class="c1"># Filter assemblies and check number of contigs</span>
<a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>seqmagick<span class="w"> </span>convert<span class="w"> </span>--min-length<span class="w"> </span><span class="m">1000</span><span class="w"> </span>spades_assembly/spades_assembly.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a><span class="w">                                    </span>spades_assembly/spades_assembly.m1000.fna
<a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a>grep<span class="w"> </span>-c<span class="w"> </span><span class="s1">&#39;&gt;&#39;</span><span class="w"> </span>spades_assembly/spades_assembly.fna<span class="w"> </span>spades_assembly/spades_assembly.m1000.fna
<a id="__codelineno-6-12" name="__codelineno-6-12" href="#__codelineno-6-12"></a>
<a id="__codelineno-6-13" name="__codelineno-6-13" href="#__codelineno-6-13"></a>seqmagick<span class="w"> </span>convert<span class="w"> </span>--min-length<span class="w"> </span><span class="m">1000</span><span class="w"> </span>idbaud_assembly/idbaud_assembly.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-14" name="__codelineno-6-14" href="#__codelineno-6-14"></a><span class="w">                                    </span>idbaud_assembly/idbaud_assembly.m1000.fna
<a id="__codelineno-6-15" name="__codelineno-6-15" href="#__codelineno-6-15"></a>grep<span class="w"> </span>-c<span class="w"> </span><span class="s1">&#39;&gt;&#39;</span><span class="w"> </span>idbaud_assembly/idbaud_assembly.fna<span class="w"> </span>idbaud_assembly/idbaud_assembly.m1000.fna
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1"><code>SPAdes</code></label><label for="__tabbed_1_2"><code>IDBA-UD</code></label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>spades_assembly/spades_assembly.fna:1327
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>spades_assembly/spades_assembly.m1000.fna:933
</code></pre></div>
</div>
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>idbaud_assembly/idbaud_assembly.fna:5056
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>idbaud_assembly/idbaud_assembly.m1000.fna:1996
</code></pre></div>
</div>
</div>
</div>
</div>
<p>If you have your own assemblies and you want to try inspect them in the same way, try that now. Note that the file names will be slightly different to the files provided above. If you followed the exact commands in the previous exercise, you can use the following commands.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>seqmagick<span class="w"> </span>convert<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="w">    </span>--min-length<span class="w"> </span><span class="m">1000</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="w">    </span>../3.assembly/my_spades_assembly/scaffolds.fasta<span class="w"> </span>my_spades_assembly.m1000.fna
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>seqmagick<span class="w"> </span>convert<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a><span class="w">    </span>--min-length<span class="w"> </span><span class="m">1000</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a><span class="w">    </span>../3.assembly/my_idbaud_assembly/scaffold.fa<span class="w"> </span>my_idbaud_assembly.m1000.fna
</code></pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Choice of software: sequence file manipulation</p>
<p>The tool <code>seqtk</code> is also available on NeSI and performs many of the same functions as <code>seqmagick</code>. My choice of <code>seqmagick</code> is mostly cosmetic as the parameter names are more explicit so it's easier to understand what's happening in a command when I look back at my log files. Regardless of which tool you prefer, we strongly recommend getting familiar with either <code>seqtk</code> or <code>seqmagick</code> as both perform a lot of common FASTA and FASTQ file manipulations.</p>
</div>
<p>As we can see here, the <code>SPAdes</code> assembly has completed with fewer contigs assembled than the <code>IDBA-UD</code>, both in terms of total contigs assembled and contigs above the 1,000 bp size. This doesn't tell us a lot though - has <code>SPAdes</code> managed to assemble fewer reads, or has it managed to assemble the sequences into longer (and hence fewer) contigs? We can check this by looking at the N50/L50 (see more information about this statistic <a href="https://www.molecularecologist.com/2017/03/29/whats-n50/">here</a>) of the assembly with <code>BBMap</code>.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Load BBMap module</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>module<span class="w"> </span>purge
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>module<span class="w"> </span>load<span class="w"> </span>BBMap/39.01-GCC-11.3.0
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a>
<a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a><span class="c1"># Generate statistics for filtered SPAdes assembly</span>
<a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a>stats.sh<span class="w"> </span><span class="k">in</span><span class="o">=</span>spades_assembly/spades_assembly.m1000.fna
</code></pre></div>
</div>
<p>This gives quite a verbose output:</p>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>A       C       G       T       N       IUPAC   Other   GC      GC_stdev
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>0.2541  0.2475  0.2453  0.2531  0.0018  0.0000  0.0000  0.4928  0.0958
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>Main genome scaffold total:             934
<a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>Main genome contig total:               2703
<a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>Main genome scaffold sequence total:    34.293 MB
<a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a>Main genome contig sequence total:      34.231 MB       0.182% gap
<a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a>Main genome scaffold N/L50:             53/160.826 KB
<a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a>Main genome contig N/L50:               107/72.909 KB
<a id="__codelineno-11-10" name="__codelineno-11-10" href="#__codelineno-11-10"></a>Main genome scaffold N/L90:             302/15.325 KB
<a id="__codelineno-11-11" name="__codelineno-11-11" href="#__codelineno-11-11"></a>Main genome contig N/L90:               812/4.643 KB
<a id="__codelineno-11-12" name="__codelineno-11-12" href="#__codelineno-11-12"></a>Max scaffold length:                    1.222 MB
<a id="__codelineno-11-13" name="__codelineno-11-13" href="#__codelineno-11-13"></a>Max contig length:                      1.045 MB
<a id="__codelineno-11-14" name="__codelineno-11-14" href="#__codelineno-11-14"></a>Number of scaffolds &gt; 50 KB:            152
<a id="__codelineno-11-15" name="__codelineno-11-15" href="#__codelineno-11-15"></a>% main genome in scaffolds &gt; 50 KB:     77.10%
<a id="__codelineno-11-16" name="__codelineno-11-16" href="#__codelineno-11-16"></a>
<a id="__codelineno-11-17" name="__codelineno-11-17" href="#__codelineno-11-17"></a>
<a id="__codelineno-11-18" name="__codelineno-11-18" href="#__codelineno-11-18"></a>Minimum         Number          Number          Total           Total           Scaffold
<a id="__codelineno-11-19" name="__codelineno-11-19" href="#__codelineno-11-19"></a>Scaffold        of              of              Scaffold        Contig          Contig  
<a id="__codelineno-11-20" name="__codelineno-11-20" href="#__codelineno-11-20"></a>Length          Scaffolds       Contigs         Length          Length          Coverage
<a id="__codelineno-11-21" name="__codelineno-11-21" href="#__codelineno-11-21"></a>--------        --------------  --------------  --------------  --------------  --------
<a id="__codelineno-11-22" name="__codelineno-11-22" href="#__codelineno-11-22"></a>    All                    934           2,703      34,293,018      34,230,627    99.82%
<a id="__codelineno-11-23" name="__codelineno-11-23" href="#__codelineno-11-23"></a>   1 KB                    934           2,703      34,293,018      34,230,627    99.82%
<a id="__codelineno-11-24" name="__codelineno-11-24" href="#__codelineno-11-24"></a> 2.5 KB                    744           2,447      33,967,189      33,909,776    99.83%
<a id="__codelineno-11-25" name="__codelineno-11-25" href="#__codelineno-11-25"></a>   5 KB                    581           2,131      33,379,814      33,328,450    99.85%
<a id="__codelineno-11-26" name="__codelineno-11-26" href="#__codelineno-11-26"></a>  10 KB                    394           1,585      32,031,121      31,994,989    99.89%
<a id="__codelineno-11-27" name="__codelineno-11-27" href="#__codelineno-11-27"></a>  25 KB                    236             916      29,564,265      29,545,757    99.94%
<a id="__codelineno-11-28" name="__codelineno-11-28" href="#__codelineno-11-28"></a>  50 KB                    152             599      26,441,339      26,429,391    99.95%
<a id="__codelineno-11-29" name="__codelineno-11-29" href="#__codelineno-11-29"></a> 100 KB                     92             415      22,238,822      22,230,143    99.96%
<a id="__codelineno-11-30" name="__codelineno-11-30" href="#__codelineno-11-30"></a> 250 KB                     31             153      12,606,774      12,603,418    99.97%
<a id="__codelineno-11-31" name="__codelineno-11-31" href="#__codelineno-11-31"></a> 500 KB                      6              32       4,821,355       4,821,095    99.99%
<a id="__codelineno-11-32" name="__codelineno-11-32" href="#__codelineno-11-32"></a>   1 MB                      1               2       1,221,548       1,221,538   100.00%
</code></pre></div>
</div>
<div class="admonition danger">
<p class="admonition-title">N50 and L50 in <code>BBMap</code></p>
<p>Unfortunately, the N50 and L50 values generated by <code>stats.sh</code> are switched. N50 should be a length and L50 should be a count. The results table below shows the corrected values based on <code>stats.sh</code> outputs.</p>
</div>
<p>But what we can highlight here is that the statistics for the <code>SPAdes</code> assembly, with short contigs removed, yielded an N50 of 72.5 kbp at the contig level. We will now compute those same statistics from the other assembly options.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>stats.sh<span class="w"> </span><span class="k">in</span><span class="o">=</span>spades_assembly/spades_assembly.fna
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>
<a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>stats.sh<span class="w"> </span><span class="k">in</span><span class="o">=</span>idbaud_assembly/idbaud_assembly.m1000.fna
<a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>stats.sh<span class="w"> </span><span class="k">in</span><span class="o">=</span>idbaud_assembly/idbaud_assembly.fna
</code></pre></div>
</div>
<table>
<thead>
<tr>
<th style="text-align: left;">Assembly</th>
<th style="text-align: center;">N50 (contig)</th>
<th style="text-align: center;">L50 (contig)</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><strong>SPAdes</strong> (filtered)</td>
<td style="text-align: center;">72.9 kbp</td>
<td style="text-align: center;">107</td>
</tr>
<tr>
<td style="text-align: left;"><strong>SPAdes</strong> (unfiltered)</td>
<td style="text-align: center;">72.3 kbp</td>
<td style="text-align: center;">108</td>
</tr>
<tr>
<td style="text-align: left;"><strong>IDBA-UD</strong> (filtered)</td>
<td style="text-align: center;">103.9 kbp</td>
<td style="text-align: center;">82</td>
</tr>
<tr>
<td style="text-align: left;"><strong>IDBA-UD</strong> (unfiltered)</td>
<td style="text-align: center;">96.6 kbp</td>
<td style="text-align: center;">88</td>
</tr>
</tbody>
</table>
<h2 id="reconstruct-rrna-using-phyloflash-emirge">Reconstruct rRNA using <code>PhyloFlash</code>-<code>EMIRGE</code><a class="headerlink" href="#reconstruct-rrna-using-phyloflash-emirge" title="Permanent link">&para;</a></h2>
<p>Another method that we can use to obtain taxonomic composition of our metagenomic libraries is to extract/reconstruct the 16S and 18S ribosomal RNA gene. However, modern assemblers often struggle to produce contiguous 16S rRNA genes due to (1) taxa harbouring multiple copies of the gene and (2) repeat regions that short reads cannot span (<a href="https://github.com/ablab/spades/issues/803">see here for details</a>). As such, software such as <a href="https://github.com/csmiller/EMIRGE">EMIRGE</a> (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2011-12-5-r44">Miller et al., 2011</a>) and <a href="https://github.com/HRGV/phyloFlash">PhyloFlash</a> (<a href="https://journals.asm.org/doi/10.1128/msystems.00920-20">Gruber-Vodicka et al., 2020</a>) were developed to reconstruct 16S and 18S rRNA genes. These methods leverage the expansive catalogue of 16S rRNA genes available in databases such as SILVA in order to subset reads and then reconstruct the full-length gene.</p>
<p>For this workshop, we will use PhyloFlash to obtain sequences and abundances of 16S rRNA sequences. The reason being that PhyloFlash can also run EMIRGE as part of its routine, compare those reconstructions with sequences generated by PhyloFlash's map-assemble, and produce a set of 16S rRNA gene sequences as well as estimate relative abundances via coverage estimation.</p>
<div class="admonition warning">
<p class="admonition-title"><code>-lib</code></p>
<p>A quirk of PhyloFlash is that all outputs will be placed in the current running directory. The only punctuation allowed is "-" and "_". Make sure to name your "LIB" sensibly so you can track which library you're working with.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-13-1">1</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1"></a>nano<span class="w"> </span>phyloflash.sl
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-14-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-14-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-14-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-14-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-14-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-14-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-14-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-14-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-14-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-14-10">10</a></span>
<span class="normal"><a href="#__codelineno-14-11">11</a></span>
<span class="normal"><a href="#__codelineno-14-12">12</a></span>
<span class="normal"><a href="#__codelineno-14-13">13</a></span>
<span class="normal"><a href="#__codelineno-14-14">14</a></span>
<span class="normal"><a href="#__codelineno-14-15">15</a></span>
<span class="normal"><a href="#__codelineno-14-16">16</a></span>
<span class="normal"><a href="#__codelineno-14-17">17</a></span>
<span class="normal"><a href="#__codelineno-14-18">18</a></span>
<span class="normal"><a href="#__codelineno-14-19">19</a></span>
<span class="normal"><a href="#__codelineno-14-20">20</a></span>
<span class="normal"><a href="#__codelineno-14-21">21</a></span>
<span class="normal"><a href="#__codelineno-14-22">22</a></span>
<span class="normal"><a href="#__codelineno-14-23">23</a></span>
<span class="normal"><a href="#__codelineno-14-24">24</a></span>
<span class="normal"><a href="#__codelineno-14-25">25</a></span>
<span class="normal"><a href="#__codelineno-14-26">26</a></span>
<span class="normal"><a href="#__codelineno-14-27">27</a></span>
<span class="normal"><a href="#__codelineno-14-28">28</a></span>
<span class="normal"><a href="#__codelineno-14-29">29</a></span>
<span class="normal"><a href="#__codelineno-14-30">30</a></span>
<span class="normal"><a href="#__codelineno-14-31">31</a></span>
<span class="normal"><a href="#__codelineno-14-32">32</a></span>
<span class="normal"><a href="#__codelineno-14-33">33</a></span>
<span class="normal"><a href="#__codelineno-14-34">34</a></span>
<span class="normal"><a href="#__codelineno-14-35">35</a></span>
<span class="normal"><a href="#__codelineno-14-36">36</a></span>
<span class="normal"><a href="#__codelineno-14-37">37</a></span>
<span class="normal"><a href="#__codelineno-14-38">38</a></span>
<span class="normal"><a href="#__codelineno-14-39">39</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-14-2" name="__codelineno-14-2"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-14-3" name="__codelineno-14-3"></a><span class="c1">#SBATCH --job-name      phyloflash</span>
<a id="__codelineno-14-4" name="__codelineno-14-4"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-14-5" name="__codelineno-14-5"></a><span class="c1">#SBATCH --time          15:00</span>
<a id="__codelineno-14-6" name="__codelineno-14-6"></a><span class="c1">#SBATCH --mem           16G</span>
<a id="__codelineno-14-7" name="__codelineno-14-7"></a><span class="c1">#SBATCH --cpus-per-task 12</span>
<a id="__codelineno-14-8" name="__codelineno-14-8"></a><span class="c1">#SBATCH --error         %x.%j.err</span>
<a id="__codelineno-14-9" name="__codelineno-14-9"></a><span class="c1">#SBATCH --output        %x.%j.out</span>
<a id="__codelineno-14-10" name="__codelineno-14-10"></a>
<a id="__codelineno-14-11" name="__codelineno-14-11"></a>module<span class="w"> </span>purge
<a id="__codelineno-14-12" name="__codelineno-14-12"></a>module<span class="w"> </span>load<span class="w"> </span>Miniconda3/23.10.0-1
<a id="__codelineno-14-13" name="__codelineno-14-13"></a>module<span class="w"> </span>load<span class="w"> </span>USEARCH/11.0.667-i86linux32
<a id="__codelineno-14-14" name="__codelineno-14-14"></a>
<a id="__codelineno-14-15" name="__codelineno-14-15"></a><span class="c1"># Navigate to working directory</span>
<a id="__codelineno-14-16" name="__codelineno-14-16"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/4.evaluation/
<a id="__codelineno-14-17" name="__codelineno-14-17"></a>
<a id="__codelineno-14-18" name="__codelineno-14-18"></a><span class="c1"># Set conda variables</span>
<a id="__codelineno-14-19" name="__codelineno-14-19"></a><span class="nb">source</span><span class="w"> </span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span>/etc/profile.d/conda.sh
<a id="__codelineno-14-20" name="__codelineno-14-20"></a><span class="nb">export</span><span class="w"> </span><span class="nv">PYTHONNOUSERSITE</span><span class="o">=</span><span class="m">1</span>
<a id="__codelineno-14-21" name="__codelineno-14-21"></a>
<a id="__codelineno-14-22" name="__codelineno-14-22"></a><span class="c1"># Activate environment</span>
<a id="__codelineno-14-23" name="__codelineno-14-23"></a><span class="nb">export</span><span class="w"> </span><span class="nv">CONDA_ENVS_PATH</span><span class="o">=</span>/nesi/project/nesi02659/MGSS_2024/resources/tools
<a id="__codelineno-14-24" name="__codelineno-14-24"></a><span class="nb">source</span><span class="w"> </span>activate<span class="w"> </span>phyloflash
<a id="__codelineno-14-25" name="__codelineno-14-25"></a>
<a id="__codelineno-14-26" name="__codelineno-14-26"></a><span class="c1"># Run PhyloFlash per sample pair</span>
<a id="__codelineno-14-27" name="__codelineno-14-27"></a><span class="k">for</span><span class="w"> </span>r1<span class="w"> </span><span class="k">in</span><span class="w"> </span>../3.assembly/sample*_R1.fastq.gz<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-14-28" name="__codelineno-14-28"></a><span class="w">    </span>phyloFlash.pl<span class="w"> </span>-lib<span class="w"> </span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$r1</span><span class="w"> </span>_R1.fastq.gz<span class="k">)</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-29" name="__codelineno-14-29"></a><span class="w">                  </span>-read1<span class="w"> </span><span class="si">${</span><span class="nv">r1</span><span class="si">}</span><span class="w"> </span>-read2<span class="w"> </span><span class="si">${</span><span class="nv">r1</span><span class="p">/R1/R2</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-30" name="__codelineno-14-30"></a><span class="w">                  </span>-everything<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-31" name="__codelineno-14-31"></a><span class="w">                  </span>-CPUs<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span>
<a id="__codelineno-14-32" name="__codelineno-14-32"></a><span class="k">done</span>
<a id="__codelineno-14-33" name="__codelineno-14-33"></a>
<a id="__codelineno-14-34" name="__codelineno-14-34"></a>phyloFlash_compare.pl<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-35" name="__codelineno-14-35"></a><span class="w">    </span>--zip<span class="w"> </span><span class="k">$(</span>ls<span class="w"> </span>-1<span class="w"> </span>sample*.phyloFlash.tar.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>paste<span class="w"> </span>-sd<span class="w"> </span><span class="s2">&quot;,&quot;</span><span class="k">)</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-36" name="__codelineno-14-36"></a><span class="w">    </span>--task<span class="w"> </span>heatmap,barplot,matrix,ntu_table<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-37" name="__codelineno-14-37"></a><span class="w">    </span>--out<span class="w"> </span>allsamples_compare
<a id="__codelineno-14-38" name="__codelineno-14-38"></a>
<a id="__codelineno-14-39" name="__codelineno-14-39"></a>conda<span class="w"> </span>deactivate
</code></pre></div></td></tr></table></div>
</div>
<p>A successful run of PhyloFlash (lines 24-29) would have generated the following per-sample outputs:</p>
<ul>
<li><code>tar.gz</code> archive containing relevant data files</li>
<li>An <code>.html</code> page summary</li>
<li>A timestamped <code>.log</code> detailing what was done</li>
</ul>
<p>The reconstructed 16S rRNA sequences are stored in the per-sample archive. We will extract them to explore the data files inside.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="k">for</span><span class="w"> </span>pf<span class="w"> </span><span class="k">in</span><span class="w"> </span>*phyloFlash.tar.gz<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="w">    </span><span class="nv">outdir</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">pf</span><span class="si">}</span><span class="w"> </span>.tar.gz<span class="k">)</span>
<a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a><span class="w">    </span>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>
<a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="w">    </span>tar<span class="w"> </span>-xvzf<span class="w"> </span><span class="si">${</span><span class="nv">pf</span><span class="si">}</span><span class="w"> </span>-C<span class="w"> </span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>
<a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a><span class="k">done</span>
</code></pre></div>
</div>
<p>The reconstructed sequences based on the <code>phyloFlash</code> and <code>EMIRGE</code> workflows are stored in <code>sample?.phyloFlash/sample?.all.final.fasta</code>.</p>
<p>In lines 32-34 of our slurm script, we also ran a cross-sample comparison script. This generated 4 files:</p>
<ul>
<li><code>allsamples_compare.barplot.pdf</code> A relative abundance barplot of identified taxa</li>
<li><code>allsamples_compare.heatmap.pdf</code> A heatmap of identified taxa</li>
<li><code>allsamples_compare.matrix.tsv</code> A dissimilarity matrix (abundance-weighted, UniFrac-like)</li>
<li><code>allsamples_compare.ntu_table.tsv</code> (long/tidy-format table of relative abundance per sample per taxa)</li>
</ul>
<p>The <code>allsamples_compare.ntu_table.tsv</code> is especially useful if you intend to perform downstream diversity analyses in <code>R</code>.</p>
<h2 id="sequence-taxonomic-classification-using-kraken2">Sequence taxonomic classification using <code>Kraken2</code><a class="headerlink" href="#sequence-taxonomic-classification-using-kraken2" title="Permanent link">&para;</a></h2>
<p>Most, if not all, of the time, we never know the taxonomic composition of our metagenomic reads <em>a priori</em>. For some environments, we can make good guesses (e.g., <em>Prochlorococcus</em> in marine samples, members of the Actinobacteriota in soil samples, various Bacteroidota in the gut microbiome, etc.). Here, we can use <code>Kraken2</code>, a k-mer based taxonomic classifier to help us interrogate the taxonomic composition of our samples. This is helpful if there are targets we might be looking for (e.g. working hypotheses, well characterised microbiome). Furthermore, we can estimate the abundance of the classified taxa using <code>Bracken</code>.</p>
<!--
!!! terminal "code"

    <div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-16-1">1</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1"></a>nano<span class="w"> </span>kraken2.sl
</code></pre></div></td></tr></table></div>

!!! terminal "code"

    <div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-17-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-17-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-17-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-17-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-17-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-17-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-17-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-17-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-17-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-17-10">10</a></span>
<span class="normal"><a href="#__codelineno-17-11">11</a></span>
<span class="normal"><a href="#__codelineno-17-12">12</a></span>
<span class="normal"><a href="#__codelineno-17-13">13</a></span>
<span class="normal"><a href="#__codelineno-17-14">14</a></span>
<span class="normal"><a href="#__codelineno-17-15">15</a></span>
<span class="normal"><a href="#__codelineno-17-16">16</a></span>
<span class="normal"><a href="#__codelineno-17-17">17</a></span>
<span class="normal"><a href="#__codelineno-17-18">18</a></span>
<span class="normal"><a href="#__codelineno-17-19">19</a></span>
<span class="normal"><a href="#__codelineno-17-20">20</a></span>
<span class="normal"><a href="#__codelineno-17-21">21</a></span>
<span class="normal"><a href="#__codelineno-17-22">22</a></span>
<span class="normal"><a href="#__codelineno-17-23">23</a></span>
<span class="normal"><a href="#__codelineno-17-24">24</a></span>
<span class="normal"><a href="#__codelineno-17-25">25</a></span>
<span class="normal"><a href="#__codelineno-17-26">26</a></span>
<span class="normal"><a href="#__codelineno-17-27">27</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-17-2" name="__codelineno-17-2"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-17-3" name="__codelineno-17-3"></a><span class="c1">#SBATCH --job-name      kraken2</span>
<a id="__codelineno-17-4" name="__codelineno-17-4"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-17-5" name="__codelineno-17-5"></a><span class="c1">#SBATCH --time          10:00</span>
<a id="__codelineno-17-6" name="__codelineno-17-6"></a><span class="c1">#SBATCH --mem           80G</span>
<a id="__codelineno-17-7" name="__codelineno-17-7"></a><span class="c1">#SBATCH --cpus-per-task 20</span>
<a id="__codelineno-17-8" name="__codelineno-17-8"></a><span class="c1">#SBATCH --error         %x.%j.err</span>
<a id="__codelineno-17-9" name="__codelineno-17-9"></a><span class="c1">#SBATCH --output        %x.%j.out</span>
<a id="__codelineno-17-10" name="__codelineno-17-10"></a>
<a id="__codelineno-17-11" name="__codelineno-17-11"></a><span class="c1"># Load module</span>
<a id="__codelineno-17-12" name="__codelineno-17-12"></a>module<span class="w"> </span>purge
<a id="__codelineno-17-13" name="__codelineno-17-13"></a>module<span class="w"> </span>load<span class="w"> </span>Kraken2/2.1.3-GCC-11.3.0
<a id="__codelineno-17-14" name="__codelineno-17-14"></a>
<a id="__codelineno-17-15" name="__codelineno-17-15"></a><span class="c1"># Point to database</span>
<a id="__codelineno-17-16" name="__codelineno-17-16"></a><span class="nv">K2DB</span><span class="o">=</span>/nesi/project/nesi02659/MGSS_2024/resources/databases/k2_standard_20240605
<a id="__codelineno-17-17" name="__codelineno-17-17"></a>
<a id="__codelineno-17-18" name="__codelineno-17-18"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/4.evaluation
<a id="__codelineno-17-19" name="__codelineno-17-19"></a>
<a id="__codelineno-17-20" name="__codelineno-17-20"></a><span class="c1"># Run Kraken2</span>
<a id="__codelineno-17-21" name="__codelineno-17-21"></a>kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-22" name="__codelineno-17-22"></a><span class="w">        </span>--classified-out<span class="w"> </span>spades_assembly.m1000.k2_classified.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-23" name="__codelineno-17-23"></a><span class="w">        </span>--unclassified-out<span class="w"> </span>spades_assembly.m1000.k2_unclassified.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-24" name="__codelineno-17-24"></a><span class="w">        </span>--report<span class="w"> </span>spades_assembly.m1000.k2_report.txt<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-25" name="__codelineno-17-25"></a><span class="w">        </span>--output<span class="w"> </span>spades_assembly.m1000.k2_out<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-26" name="__codelineno-17-26"></a><span class="w">        </span>--db<span class="w"> </span><span class="si">${</span><span class="nv">K2DB</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-27" name="__codelineno-17-27"></a><span class="w">        </span>spades_assembly/spades_assembly.m1000.fna
</code></pre></div></td></tr></table></div>
-->

<!--
As `Kraken2` classifications are *k*-mer based, we can also classify reads. This can be helpful if trying to filter out reads that may belong to taxonomic classifications that you're not interested in. When using reads for classification, we can also estimate the abundance of reads that belong to those taxa using Bracken.
-->

<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>nano<span class="w"> </span>kraken2_bracken.sl
</code></pre></div>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-19-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-19-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-19-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-19-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-19-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-19-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-19-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-19-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-19-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-19-10">10</a></span>
<span class="normal"><a href="#__codelineno-19-11">11</a></span>
<span class="normal"><a href="#__codelineno-19-12">12</a></span>
<span class="normal"><a href="#__codelineno-19-13">13</a></span>
<span class="normal"><a href="#__codelineno-19-14">14</a></span>
<span class="normal"><a href="#__codelineno-19-15">15</a></span>
<span class="normal"><a href="#__codelineno-19-16">16</a></span>
<span class="normal"><a href="#__codelineno-19-17">17</a></span>
<span class="normal"><a href="#__codelineno-19-18">18</a></span>
<span class="normal"><a href="#__codelineno-19-19">19</a></span>
<span class="normal"><a href="#__codelineno-19-20">20</a></span>
<span class="normal"><a href="#__codelineno-19-21">21</a></span>
<span class="normal"><a href="#__codelineno-19-22">22</a></span>
<span class="normal"><a href="#__codelineno-19-23">23</a></span>
<span class="normal"><a href="#__codelineno-19-24">24</a></span>
<span class="normal"><a href="#__codelineno-19-25">25</a></span>
<span class="normal"><a href="#__codelineno-19-26">26</a></span>
<span class="normal"><a href="#__codelineno-19-27">27</a></span>
<span class="normal"><a href="#__codelineno-19-28">28</a></span>
<span class="normal"><a href="#__codelineno-19-29">29</a></span>
<span class="normal"><a href="#__codelineno-19-30">30</a></span>
<span class="normal"><a href="#__codelineno-19-31">31</a></span>
<span class="normal"><a href="#__codelineno-19-32">32</a></span>
<span class="normal"><a href="#__codelineno-19-33">33</a></span>
<span class="normal"><a href="#__codelineno-19-34">34</a></span>
<span class="normal"><a href="#__codelineno-19-35">35</a></span>
<span class="normal"><a href="#__codelineno-19-36">36</a></span>
<span class="normal"><a href="#__codelineno-19-37">37</a></span>
<span class="normal"><a href="#__codelineno-19-38">38</a></span>
<span class="normal"><a href="#__codelineno-19-39">39</a></span>
<span class="normal"><a href="#__codelineno-19-40">40</a></span>
<span class="normal"><a href="#__codelineno-19-41">41</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-19-2" name="__codelineno-19-2"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-19-3" name="__codelineno-19-3"></a><span class="c1">#SBATCH --job-name      kraken2_bracken</span>
<a id="__codelineno-19-4" name="__codelineno-19-4"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-19-5" name="__codelineno-19-5"></a><span class="c1">#SBATCH --time          30:00</span>
<a id="__codelineno-19-6" name="__codelineno-19-6"></a><span class="c1">#SBATCH --mem           80G</span>
<a id="__codelineno-19-7" name="__codelineno-19-7"></a><span class="c1">#SBATCH --cpus-per-task 20</span>
<a id="__codelineno-19-8" name="__codelineno-19-8"></a><span class="c1">#SBATCH --error         %x.%j.err</span>
<a id="__codelineno-19-9" name="__codelineno-19-9"></a><span class="c1">#SBATCH --output        %x.%j.out</span>
<a id="__codelineno-19-10" name="__codelineno-19-10"></a>
<a id="__codelineno-19-11" name="__codelineno-19-11"></a><span class="c1"># Load modules</span>
<a id="__codelineno-19-12" name="__codelineno-19-12"></a>module<span class="w"> </span>purge
<a id="__codelineno-19-13" name="__codelineno-19-13"></a>module<span class="w"> </span>load<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-14" name="__codelineno-19-14"></a><span class="w">    </span>Kraken2/2.1.3-GCC-11.3.0<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-15" name="__codelineno-19-15"></a><span class="w">    </span>Bracken/2.7-GCC-11.3.0
<a id="__codelineno-19-16" name="__codelineno-19-16"></a>
<a id="__codelineno-19-17" name="__codelineno-19-17"></a><span class="c1"># Navigate to working directory</span>
<a id="__codelineno-19-18" name="__codelineno-19-18"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/4.evaluation/
<a id="__codelineno-19-19" name="__codelineno-19-19"></a>
<a id="__codelineno-19-20" name="__codelineno-19-20"></a><span class="c1"># Point to database</span>
<a id="__codelineno-19-21" name="__codelineno-19-21"></a><span class="nv">K2DB</span><span class="o">=</span>/nesi/project/nesi02659/MGSS_2024/resources/databases/k2_standard_20240605
<a id="__codelineno-19-22" name="__codelineno-19-22"></a>
<a id="__codelineno-19-23" name="__codelineno-19-23"></a><span class="c1"># Create output directory</span>
<a id="__codelineno-19-24" name="__codelineno-19-24"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>read_classification/
<a id="__codelineno-19-25" name="__codelineno-19-25"></a>
<a id="__codelineno-19-26" name="__codelineno-19-26"></a><span class="c1"># Run Kraken2 and Bracken</span>
<a id="__codelineno-19-27" name="__codelineno-19-27"></a><span class="k">for</span><span class="w"> </span>r1<span class="w"> </span><span class="k">in</span><span class="w"> </span>../3.assembly/sample?_R1.fastq.gz<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-19-28" name="__codelineno-19-28"></a><span class="w">  </span><span class="c1"># Output basename</span>
<a id="__codelineno-19-29" name="__codelineno-19-29"></a><span class="w">  </span><span class="nv">outbase</span><span class="o">=</span>read_classification/<span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">r1</span><span class="si">}</span><span class="w"> </span>_R1.fastq.gz<span class="k">)</span>
<a id="__codelineno-19-30" name="__codelineno-19-30"></a><span class="w">  </span><span class="c1"># Taxonomic classification of reads</span>
<a id="__codelineno-19-31" name="__codelineno-19-31"></a><span class="w">  </span>kraken2<span class="w"> </span>--paired<span class="w"> </span>--threads<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-32" name="__codelineno-19-32"></a><span class="w">          </span>--classified-out<span class="w"> </span><span class="si">${</span><span class="nv">outbase</span><span class="si">}</span><span class="c1">#.k2_classified.fq \</span>
<a id="__codelineno-19-33" name="__codelineno-19-33"></a><span class="w">          </span>--unclassified-out<span class="w"> </span><span class="si">${</span><span class="nv">outbase</span><span class="si">}</span><span class="c1">#.k2_unclassified.fq \</span>
<a id="__codelineno-19-34" name="__codelineno-19-34"></a><span class="w">          </span>--report<span class="w"> </span><span class="si">${</span><span class="nv">outbase</span><span class="si">}</span>.k2_report.txt<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-35" name="__codelineno-19-35"></a><span class="w">          </span>--output<span class="w"> </span><span class="si">${</span><span class="nv">outbase</span><span class="si">}</span>.k2_out<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-36" name="__codelineno-19-36"></a><span class="w">          </span>--db<span class="w"> </span><span class="si">${</span><span class="nv">K2DB</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-37" name="__codelineno-19-37"></a><span class="w">          </span><span class="si">${</span><span class="nv">r1</span><span class="si">}</span><span class="w"> </span><span class="si">${</span><span class="nv">r1</span><span class="p">/R1/R2</span><span class="si">}</span>
<a id="__codelineno-19-38" name="__codelineno-19-38"></a><span class="w">  </span><span class="c1"># Estimate taxa abundance</span>
<a id="__codelineno-19-39" name="__codelineno-19-39"></a><span class="w">  </span>bracken<span class="w"> </span>-d<span class="w"> </span><span class="si">${</span><span class="nv">K2DB</span><span class="si">}</span><span class="w"> </span>-i<span class="w"> </span><span class="si">${</span><span class="nv">outbase</span><span class="si">}</span>.k2_report.txt<span class="w"> </span>-r<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-19-40" name="__codelineno-19-40"></a><span class="w">          </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">outbase</span><span class="si">}</span>.bracken
<a id="__codelineno-19-41" name="__codelineno-19-41"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
</div>
<p>The main output is quite dense and verbose with columns indicated <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#sample-report-output-format">here</a>. The report is more human-readable, with nicely spaced columns that indicate:</p>
<ol>
<li>Percentage reads mapped to taxon</li>
<li>Number of reads mapped to taxon</li>
<li>Number of reads directly assigned to taxon</li>
<li>Rank of taxon</li>
<li>NCBI taxonomy ID</li>
<li>Scientific name of taxon</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Remember to produce the <code>--report</code> as Bracken bases its estimation on the report</p>
</div>
<p>The code tells <code>Kraken2</code> that the inputs <code>${r1}</code> and <code>${r1/R1/R2}</code> are paired-end reads. The <code>#</code> after <code>${base}</code> will be replaced with the read orientation (i.e., forward <code>_1</code> or reverse <code>_2</code>).</p>
<p>The Bracken outputs provides us with adjusted number (columns <code>added_reads</code> and <code>new_est_reads</code>) and fraction of reads that were assigned to each species identified by Kraken2 for downstream analyses.</p>
<details class="warning">
<summary>Database construction for <code>Kraken2</code> and <code>Bracken</code></summary>
<p>For this workshop (and most applications), the standard or other extensive <a href="https://benlangmead.github.io/aws-indexes/k2">pre-built Kraken2 databases</a> are sufficient. However, if you require specific reference sequences to be present, the database building process (outlined <a href="https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases">here</a>) can be quite time and resource intensive.</p>
<p>The Kraken2 database also comes with a few pre-built Bracken databases (filenames look like this: <code>database100mers.kmer_distrib</code>). Note that the "100mers" doesn't refer to k-mers (oligonucleotide frequency), but it is the length of the read the database was built for. In this workshop, we've used the 100bp database to estimate abundances for 126bp reads. <a href="https://github.com/jenniferlu717/Bracken/issues/260">According to the developers, this is acceptable</a>. As long as the read length of the built database is <span class="arithmatex">\(\le\)</span> than the length of reads to be classified, Bracken will provide good enough estimates.</p>
<p>If you want to build your own Bracken database based on your library minimum read lengths, you can follow the process <a href="https://github.com/jenniferlu717/Bracken#step-0-build-a-kraken-1krakenuniqkraken2-database">here</a>. Take note that you will require the initial sequence and taxonomy files used for building the <code>Kraken2</code> database (not part of the files in the pre-built databases) and these (especially for NCBI nt and bacterial databses) will take a long time to download and lots of memory to build.</p>
</details>
<h2 id="optional-evaluating-assemblies-using-metaquast"><em>(Optional)</em> Evaluating assemblies using <code>MetaQUAST</code><a class="headerlink" href="#optional-evaluating-assemblies-using-metaquast" title="Permanent link">&para;</a></h2>
<p>For more genome-informed evaluation of the assembly, we can use the <code>MetaQUAST</code> tool to view our assembled metagenome. This is something of an optional step because, like <code>QUAST</code>, <code>MetaQUAST</code> aligns your assembly against a set of reference genomes. Under normal circumstances we wouldn't know the composition of the metagenome that led to our assembly. In this instance determining the optimal reference genomes for a <code>MetaQUAST</code> evaluation is a bit of a problem. For your own work, the following tools could be used to generate taxonomic summaries of your metagenomes to inform your reference selection:</p>
<div class="admonition abstract">
<ol>
<li><a href="http://clark.cs.ucr.edu/">CLARK</a> (DNA based. <em>k</em>-mer classification)</li>
<li><a href="http://kaiju.binf.ku.dk/">Kaiju</a> (Protein based, BLAST classification)</li>
<li><a href="https://ccb.jhu.edu/software/centrifuge/manual.shtml">Centrifuge</a> (DNA based, sequence alignment classification)</li>
<li><a href="https://microbiology.se/software/metaxa2/">MeTaxa2</a> or <a href="https://github.com/wwood/singlem">SingleM</a> (DNA based, 16S rRNA recovery and classification)</li>
<li><a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a> (DNA based, clade-specific marker gene classification)</li>
</ol>
</div>
<p>A good summary and comparison of these tools (and more) was published by <a href="https://www.ncbi.nlm.nih.gov/pubmed/31398336">Ye <em>et al.</em></a>.</p>
<p>However, since we <strong><em>do</em></strong> know the composition of the original communities used to build this mock metagenome, <code>MetaQUAST</code> will work very well for us today. In your <code>4.evaluation/</code> directory you will find a file called <code>ref_genomes.txt</code>. This file contains the names of the genomes used to build these mock metagenomes. We will provide these as the reference input for <code>MetaQUAST</code>.</p>
<div class="admonition warning">
<p class="admonition-title">Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own folder</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-20-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-20-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-20-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-20-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-20-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-20-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-20-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-20-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-20-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-20-10">10</a></span>
<span class="normal"><a href="#__codelineno-20-11">11</a></span>
<span class="normal"><a href="#__codelineno-20-12">12</a></span>
<span class="normal"><a href="#__codelineno-20-13">13</a></span>
<span class="normal"><a href="#__codelineno-20-14">14</a></span>
<span class="normal"><a href="#__codelineno-20-15">15</a></span>
<span class="normal"><a href="#__codelineno-20-16">16</a></span>
<span class="normal"><a href="#__codelineno-20-17">17</a></span>
<span class="normal"><a href="#__codelineno-20-18">18</a></span>
<span class="normal"><a href="#__codelineno-20-19">19</a></span>
<span class="normal"><a href="#__codelineno-20-20">20</a></span>
<span class="normal"><a href="#__codelineno-20-21">21</a></span>
<span class="normal"><a href="#__codelineno-20-22">22</a></span>
<span class="normal"><a href="#__codelineno-20-23">23</a></span>
<span class="normal"><a href="#__codelineno-20-24">24</a></span>
<span class="normal"><a href="#__codelineno-20-25">25</a></span>
<span class="normal"><a href="#__codelineno-20-26">26</a></span>
<span class="normal"><a href="#__codelineno-20-27">27</a></span>
<span class="normal"><a href="#__codelineno-20-28">28</a></span>
<span class="normal"><a href="#__codelineno-20-29">29</a></span>
<span class="normal"><a href="#__codelineno-20-30">30</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-20-2" name="__codelineno-20-2"></a>
<a id="__codelineno-20-3" name="__codelineno-20-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-20-4" name="__codelineno-20-4"></a><span class="c1">#SBATCH --job-name      metaquast</span>
<a id="__codelineno-20-5" name="__codelineno-20-5"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-20-6" name="__codelineno-20-6"></a><span class="c1">#SBATCH --time          00:15:00</span>
<a id="__codelineno-20-7" name="__codelineno-20-7"></a><span class="c1">#SBATCH --mem           10GB</span>
<a id="__codelineno-20-8" name="__codelineno-20-8"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a id="__codelineno-20-9" name="__codelineno-20-9"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-20-10" name="__codelineno-20-10"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-20-11" name="__codelineno-20-11"></a>
<a id="__codelineno-20-12" name="__codelineno-20-12"></a><span class="c1"># Load module</span>
<a id="__codelineno-20-13" name="__codelineno-20-13"></a>module<span class="w"> </span>purge
<a id="__codelineno-20-14" name="__codelineno-20-14"></a>module<span class="w"> </span>load<span class="w"> </span>QUAST/5.0.2-gimkl-2018b
<a id="__codelineno-20-15" name="__codelineno-20-15"></a>
<a id="__codelineno-20-16" name="__codelineno-20-16"></a><span class="c1"># Working directory</span>
<a id="__codelineno-20-17" name="__codelineno-20-17"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/4.evaluation
<a id="__codelineno-20-18" name="__codelineno-20-18"></a>
<a id="__codelineno-20-19" name="__codelineno-20-19"></a><span class="c1"># Run metaquast    </span>
<a id="__codelineno-20-20" name="__codelineno-20-20"></a>metaquast.py<span class="w"> </span>--references-list<span class="w"> </span>ref_genomes.txt<span class="w"> </span>--max-ref-number<span class="w"> </span><span class="m">21</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-21" name="__codelineno-20-21"></a><span class="w">             </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-22" name="__codelineno-20-22"></a><span class="w">             </span>--labels<span class="w"> </span>SPAdes,SPAdes.m1000,IDBAUD,IDBAUD.m1000<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-23" name="__codelineno-20-23"></a><span class="w">             </span>--output-dir<span class="w"> </span>metaquast_results/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-24" name="__codelineno-20-24"></a><span class="w">             </span>spades_assembly/spades_assembly.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-25" name="__codelineno-20-25"></a><span class="w">             </span>spades_assembly/spades_assembly.m1000.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-26" name="__codelineno-20-26"></a><span class="w">             </span>idbaud_assembly/idbaud_assembly.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-27" name="__codelineno-20-27"></a><span class="w">             </span>idbaud_assembly/idbaud_assembly.m1000.fna
<a id="__codelineno-20-28" name="__codelineno-20-28"></a>
<a id="__codelineno-20-29" name="__codelineno-20-29"></a><span class="c1"># Create archive for download</span>
<a id="__codelineno-20-30" name="__codelineno-20-30"></a>tar<span class="w"> </span>-cvzf<span class="w"> </span>metaquast_results.tar.gz<span class="w"> </span>metaquast_results
</code></pre></div></td></tr></table></div>
</div>
<p>By now, you should be getting familiar enough with the console to understand what most of the parameters here refer to. The one parameter that needs explanation is the <code>--max-ref-number</code> flag, which we have set to 21. This caps the maximum number of reference genomes to be downloaded from NCBI which we do in the interest of speed. Since there are 21 taxa in the file <code>ref_genomes.txt</code> (10 prokaryote species and 11 viruses), <code>MetaQUAST</code> will download one reference genome for each. If we increase  the <code>--max-ref-number</code> flag we will start to get multiple reference genomes per taxa provided which is usually desirable.</p>
<p>We will now look at a few interesting assembly comparisons.</p>
<div class="admonition note">
<p class="admonition-title">Viewing HTML in Jupyter Hub</p>
<p>The NeSI Jupyter hub does not currently support viewing HTML that require Javascript (even if the browser you are running it in does). To view a basic version of the report, download the report file by navigating to the <code>4.evaluation/</code> directory, right-click <code>metaquast_results.tar.gz</code> and select download. You will need to decompress and extract the archive on your computer. Navigate to the folder with the extracted contents to open the <code>report.html</code>.</p>
</div>
<div class="admonition success">
<div class="tabbed-set tabbed-alternate" data-tabs="2:4"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><input id="__tabbed_2_3" name="__tabbed_2" type="radio" /><input id="__tabbed_2_4" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1">Brief summary<br>of assemblies</label><label for="__tabbed_2_2">Comparison of NGA50<br>between assemblies</label><label for="__tabbed_2_3">Comparison of<br>aligned contigs</label><label for="__tabbed_2_4">Inspection of<br>unaligned contigs</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p><a class="glightbox" href="../../figures/ex5_fig1_shortsummary_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex5_fig1_shortsummary_2022.png" /></a></p>
</div>
<div class="tabbed-block">
<p><a class="glightbox" href="../../figures/ex5_fig2_nga50_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex5_fig2_nga50_2022.png" /></a></p>
</div>
<div class="tabbed-block">
<p><a class="glightbox" href="../../figures/ex5_fig3_contigsmatched_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex5_fig3_contigsmatched_2022.png" /></a></p>
</div>
<div class="tabbed-block">
<p><a class="glightbox" href="../../figures/ex5_fig4_contigsunmatched_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex5_fig4_contigsunmatched_2022.png" /></a></p>
</div>
</div>
</div>
</div>
<hr />












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.07f07601.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.56dfad97.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>