# Metagenomics Summer School

Course materials for the Genomics Aotearoa Metagenomics Summer School, to be hosted at the University of Auckland in December.

A draft timetable for the day is provided below, but please keep in mind that this is subject to change as we evaluate our course material.

---

## Workshop exercises

### Day 1

1. Bash scripting
1. The metagenomics decision tree
1. [Quality filtering raw reads](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/day1/ex3_quality_filtering.md)
1. [Assembly](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/day1/ex4_assembly.md)

### Day 2

1. [Evaluating the overnight assembly](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/day2/ex5_evaluating_assemblies.md)
1. Overview of binning history
1. [Binning](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/day2/ex7_initial_binning.md)

### Day 3

1. Bin refinement
1. Gene prediction
1. Gene annotation

### Day 4

1. Gene annotation (refresher)
1. Presentation of data

---

## Timetable

### Day 1 - 10<sup>th</sup> December 2019

|Time|Event|
|:---|:---|
|9:00 am – 9:45 am|**Introduction**<br>Welcome<br>Logging into *NeSI*|
|9:45 am – 10:30 am|**Bash scripting**|
|10:30 am – 10:50 am|**Morning tea break**|
|10:50 am – 11:30 am|**Bash scripting** (continued)|
|11:30 am – 12:00 pm|**The metagenomics decision tree**<br>Proposing a hypothesis for your project<br>Dividing into working groups|
|12:00 pm – 12:45 pm|**Break for lunch**|
|12:45 pm – 1:45 pm|**Quality filtering raw reads**<br>Visualisation with *FastQC*<br>Read trimming and adapter removal<br>Diagnosing poor libraries<br>Common issues and best practice|
|1:45 pm – 3:00 pm|**Assembly**<br>Choice of assemblers<br>Considerations for parameters, and when to stop!|
|3:00 pm – 3:20 pm|**Afternoon tea break**|
|3:20 pm – 3:45 pm|**Assembly** (continued)<br>Submitting jobs to *NeSI* via *slurm*|
|4:00 pm – 5:00 pm|**End of day wrap up**<br>Attendees can work with their own data, if available|

### Day 2 - 11<sup>th</sup> December 2019

|Time|Event|
|:---|:---|
|9:00 am – 9:30 am|**Introduction**<br>Overview of yesterday, questions<br>Overview of today|
|9:30 am – 10:30 am|**Evaluating the overnight assembly**<br>Future considerations – co-assembly vs single assemblies<br>Run evaluation tool/script|
|10:30 am – 10:50 am|**Morning tea break**|
|10:50 am – 12:00 pm|**Overview of binning history**<br>ESOM, differetial coverage, tetranucleotide frequencies<br>Strategies for binning|
|12:00 pm – 12:45 pm|**Break for lunch**|
|12:45 pm – 1:45 pm|**Binning**<br>Short contig removal<br>Read mapping<br>Multi-binning strategy (*Metabat* and *Maxbin*)|
|1:45 pm – 3:00 pm|**Binning** (continued)<br>Bin dereplication via *DAS_Tool*<br>Evaluating bins using *CheckM*|
|3:00 pm – 3:20 pm|**Afternoon tea break**|
|3:20 pm – 4:00 pm|**Binning** (continued)<br>Discuss additional dereplication strategies, such as *dRep*<br>How to work with viral and eukaryotic bins<br>Dealing with organisms which possess minimal genomes|
|4:00 pm – 5:00 pm|**End of day wrap up**<br>Attendees can work with their own data, if available|

### Day 3 - 12<sup>th</sup> December 2019

|Time|Event|
|:---|:---|
|9:00 am – 9:30 am|**Introduction**<br>Overview of yesterday, questions<br>Overview of today|
|9:30 am – 11:30 am|**Bin refinement**<br>Working with *VizBin* (hands on)<br>Other strategies (*ESOMana*)|
|10:30 am – 10:50 am|**Morning tea break**|
|10:50 am – 11:30 am|**Gene prediction**<br>Introduce *prodigal*, discuss single vs meta mode<br>Discuss what *prodigal* can't find, discuss other tools (*RNAmer*, *Aragorn*, etc)|
|11:30 am – 12:00 pm|**Gene annotation**<br>BLAST-like gene annotation using *usearch* or *diamond*<br>Introduce the different databases, highlight our reasons for *KEGG*<br>Evaluating the quality of gene assignment<br>Differences in taxonomies (*GTDB*, *NCBI* etc)|
|12:00 pm – 12:45 pm|**Break for lunch**|
|12:45 pm – 3:00 pm|**Gene annotation** (continued)<br>Looking at gene networks in *MEGAN* (unassembled and assembled reads)<br>Tie findings to your initial hypothesis|
|3:00 pm – 3:20 pm|**Afternoon tea break**|
|3:20 pm – 4:00 pm|**Gene annotation** (continued)<br>Using online resources - *KEGG*, *BioCyc*, *MetaCyc*, *HydDB*, *PSORT*)<br>Tie findings to your initial hypothesis|
|4:00 pm – 5:00 pm|**End of day wrap up**<br>Attendees can work with their own data, if available|

### Day 4 - 13<sup>th</sup> December 2019

|Time|Event|
|:---|:---|
|9:00 am – 9:30 am|**Introduction**<br>Overview of yesterday, questions<br>Overview of today|
|9:30 am – 10:30 am|**Gene annotation** (refresher)<br>Tie findings to your initial hypothesis|
|10:30 am – 10:50 am|**Morning tea break**<br>Fill in form on hypothesis for own work|
|10:50 am – 12:00 pm|**Present and discuss findings**<br>Each group to give a casual discussion of their data<br>What they were looking for, what did they find?<br>Which databases were most helpful?|
|12:00 pm – 12:45 pm|**Break for lunch**|
|12:45 pm – 3:00 pm|**Presentation of data**<br>How do visualise findings - Metabolism maps, heatmaps, gene trees*<br>Gene synteny alignments (*R*)|
|3:00 pm – 3:20 pm|**Afternoon tea break**|
|3:20 pm – 4:00 pm|**End of day wrap up**<br>Final discussion|

----
