
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day4/ex15_gene_annotation_part3/">
      
      
        <link rel="prev" href="../../day3/ex14_gene_annotation_part2/">
      
      
        <link rel="next" href="../ex16a_data_presentation_Intro/">
      
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.34">
    
    
      
        <title>Gene annotation III: DRAM distillation - Metagenomics Summer School</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.35f28582.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#gene-annotation-iii-dram-distillation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Metagenomics Summer School" class="md-header__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Summer School
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Gene annotation III: DRAM distillation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day1/ex1_bash_and_scheduler/" class="md-tabs__link">
          
  
  Day 1

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day2/ex6_initial_binning/" class="md-tabs__link">
          
  
  Day 2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-tabs__link">
          
  
  Day 3

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="./" class="md-tabs__link">
          
  
  Day 4

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-tabs__link">
          
  
  Resources

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../supplementary/supplementary_1/" class="md-tabs__link">
          
  
  Supplementary

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Metagenomics Summer School" class="md-nav__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Metagenomics Summer School
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 1
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex1_bash_and_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction I: Shell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_1_intro_to_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction II: HPC and job scheduler
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_quality_filtering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filter raw reads by quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex3_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly I: Assembling contigs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex4_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly II: Varying the parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex5_evaluating_assemblies/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly evaluation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 2
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex6_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to binning
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex7_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Binning with multiple tools
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex8_bin_dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bin dereplication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex9_refining_bins/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Manual bin refinement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex10.1_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifying viral contigs in metagenomic data
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 3
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assigning taxonomy to refined prokaryotic bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11.1_phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex10.2_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virus taxonomy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex12_gene_prediction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene prediction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex13_gene_annotation_part1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation I: BLAST-like and HMM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex14_gene_annotation_part2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation II: DRAM and coverage calculation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Day 4
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Gene annotation III: DRAM distillation
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Gene annotation III: DRAM distillation
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview-of-drampy-annotate-output" class="md-nav__link">
    <span class="md-ellipsis">
      Overview of DRAM.py annotate output
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview-of-dram-vpy-annotate-output" class="md-nav__link">
    <span class="md-ellipsis">
      Overview of DRAM-v.py annotate output
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dram-and-dram-v-distillation-of-the-results" class="md-nav__link">
    <span class="md-ellipsis">
      DRAM and DRAM-v distillation of the results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#drampy-distill-output-files" class="md-nav__link">
    <span class="md-ellipsis">
      DRAM.py distill output files
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dram-vpy-distill-output-files" class="md-nav__link">
    <span class="md-ellipsis">
      DRAM-v.py distill output files
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16a_data_presentation_Intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to data presentation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16b_data_presentation_Coverage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coverage heatmaps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16c_OPTIONAL_data_presentation_Ordination/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ordinations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16d_data_presentation_KEGG_pathways/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    KEGG pathway maps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16e_data_presentation_Gene_synteny/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene synteny
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CAZy heatmaps
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Resources
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex8): Dereplicating data from multiple assemblies
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex9): Generating input files for VizBin from DAS_Tool curated bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11) : Viral taxonomy prediction via vContact2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/7_command_line_shortcuts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Useful command line shortcuts
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Supplementary
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Supplementary
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI Setup
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI File system, Working directory and Symlinks
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview-of-drampy-annotate-output" class="md-nav__link">
    <span class="md-ellipsis">
      Overview of DRAM.py annotate output
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview-of-dram-vpy-annotate-output" class="md-nav__link">
    <span class="md-ellipsis">
      Overview of DRAM-v.py annotate output
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dram-and-dram-v-distillation-of-the-results" class="md-nav__link">
    <span class="md-ellipsis">
      DRAM and DRAM-v distillation of the results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#drampy-distill-output-files" class="md-nav__link">
    <span class="md-ellipsis">
      DRAM.py distill output files
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dram-vpy-distill-output-files" class="md-nav__link">
    <span class="md-ellipsis">
      DRAM-v.py distill output files
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="gene-annotation-iii-dram-distillation">Gene annotation III: DRAM distillation<a class="headerlink" href="#gene-annotation-iii-dram-distillation" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#overview-of-drampy-annotate-output">Overview of <code>DRAM.py annotate</code> and <code>DRAM-v.py annotate</code> output</a></li>
<li><a href="#dram-and-dram-v-distillation-of-the-results"><code>DRAM</code> and <code>DRAM-v</code> distillation step and visualization of results</a></li>
<li><a href="#tie-findings-to-your-initial-goal">Tie findings to your initial goal</a></li>
</ul>
</div>
<hr />
<h2 id="overview-of-drampy-annotate-output">Overview of <code>DRAM.py annotate</code> output<a class="headerlink" href="#overview-of-drampy-annotate-output" title="Permanent link">&para;</a></h2>
<p>The submitted jobs from the previous session should now be completed. If we examine the output directory <code>10.gene_annotation_and_coverage/dram_annotations/</code> we will see the following files:</p>
<table>
<thead>
<tr>
<th style="text-align: left;">File name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><code>genes.faa</code> and <code>genes.fna</code></td>
<td style="text-align: left;">FASTA files with all the genes called by <code>Prodigal</code>, with additional header information gained from the annotation as nucleotide and amino acid records, respectively</td>
</tr>
<tr>
<td style="text-align: left;"><code>genes.gff</code></td>
<td style="text-align: left;">GFF3 file with the same annotation information as well as gene locations</td>
</tr>
<tr>
<td style="text-align: left;"><code>scaffolds.fna</code></td>
<td style="text-align: left;">A collection of all scaffolds/contigs given as input to <code>DRAM.py annotate</code> with added bin information</td>
</tr>
<tr>
<td style="text-align: left;"><code>annotations.tsv</code></td>
<td style="text-align: left;">This file includes all annotation information about every gene from all MAGs</td>
</tr>
<tr>
<td style="text-align: left;"><code>trnas.tsv</code></td>
<td style="text-align: left;">Summary of the tRNAs found in each MAG</td>
</tr>
<tr>
<td style="text-align: left;"><code>rrnas.tsv</code></td>
<td style="text-align: left;">Summary of the rRNAs found in each MAG</td>
</tr>
</tbody>
</table>
<p>If we inspect the head of the annotation file we will see the following</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage/
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>dram_annotations/annotations.tsv<span class="w"> </span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>        fasta   scaffold        gene_position   start_position  end_position    strandednessrank     ko_id   kegg_hit        peptidase_id    peptidase_family        peptidase_hit   peptidase_RBH        peptidase_identity      peptidase_bitScore      peptidase_eVal  pfam_hits   cazy_id  cazy_hits       heme_regulatory_motif_count     bin_taxonomy    bin_completeness    bin_contamination
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>bin_0_bin_0_NODE_11_length_360679_cov_0.995524_1        bin_0   bin_0_NODE_11_length_360679_cov_0.995524     1       1       213     -1      C       K00873  pyruvate kinase [EC:2.7.1.40]Pyruvate kinase, barrel domain [PF00224.24]                     0       d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis       99.59   3.39
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>bin_0_bin_0_NODE_11_length_360679_cov_0.995524_2        bin_0   bin_0_NODE_11_length_360679_cov_0.995524     2       253     909     -1      C       K22293  GntR family transcriptional regulator, rspAB operon transcriptional repressor                                            FCD domain [PF07729.15]; Bacterial regulatory proteins, gntR family [PF00392.24]             0d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis  99.59   3.39
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>bin_0_bin_0_NODE_11_length_360679_cov_0.995524_3        bin_0   bin_0_NODE_11_length_360679_cov_0.995524     3       1010    1843    -1      D                                           Universal stress protein family [PF00582.29]                     0       d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis       99.59   3.39
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>bin_0_bin_0_NODE_11_length_360679_cov_0.995524_4        bin_0   bin_0_NODE_11_length_360679_cov_0.995524     4       1865    3397    -1      C       K01708  galactarate dehydratase [EC:4.2.1.42]                                                                D-galactarate dehydratase / Altronate hydrolase, C terminus [PF04295.16]                     0       d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis       99.59   3.39
</code></pre></div>
</div>
<p>For each gene annotated, <code>DRAM</code> provides a summary rank (from A to E), representing the confidence of the annotation based on reciprocal best hits (RBH). The following figure briefly explains how this summary rank is calculated:</p>
<p><center><a class="glightbox" href="../../figures/ex14_DRAM_annotation_rank.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex14_DRAM_annotation_rank.png" width="500" /></a></center></p>
<h2 id="overview-of-dram-vpy-annotate-output">Overview of <code>DRAM-v.py annotate</code> output<a class="headerlink" href="#overview-of-dram-vpy-annotate-output" title="Permanent link">&para;</a></h2>
<p><code>DRAM-v</code> generates the same output files as <code>DRAM</code>, but this time for the viral contigs. These files can be viewed in the output directory <code>10.gene_annotation_and_coverage/dramv_annotations/</code>. In this case, <code>annotations.tsv</code> also includes some viral-specific columns, including viral gene database matches (<code>vogdb</code>), and categories that are used by <code>DRAM-v.py distill</code> to identify putative auxiliary metabolic genes (AMGs) (<code>virsorter_category</code>, <code>auxiliary_score</code>,  <code>is_transposon</code>, <code>amg_flags</code>)</p>
<hr />
<h2 id="dram-and-dram-v-distillation-of-the-results"><code>DRAM</code> and <code>DRAM-v</code> distillation of the results<a class="headerlink" href="#dram-and-dram-v-distillation-of-the-results" title="Permanent link">&para;</a></h2>
<p>After the annotation is finished, we will summarise and visualise these annotations with the so-called <em>distillation</em> step. We do so by running the following commands directly in the terminal. This will generate the distillate and liquor files for each dataset.</p>
<p>For the viral annotations, we will also include the parameters <code>--remove_transposons</code> ("Do not consider genes on scaffolds with transposons as potential AMGs") and <code>--remove_fs</code> ("Do not consider genes near ends of scaffolds as potential AMGs") to filter out some potential false positives for auxiliary metabolic gene identification.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-2-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-2-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-2-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-2-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-2-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-2-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-2-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-2-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-2-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-2-10">10</a></span>
<span class="normal"><a href="#__codelineno-2-11">11</a></span>
<span class="normal"><a href="#__codelineno-2-12">12</a></span>
<span class="normal"><a href="#__codelineno-2-13">13</a></span>
<span class="normal"><a href="#__codelineno-2-14">14</a></span>
<span class="normal"><a href="#__codelineno-2-15">15</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1"></a>module<span class="w"> </span>purge
<a id="__codelineno-2-2" name="__codelineno-2-2"></a>module<span class="w"> </span>load<span class="w"> </span>DRAM/1.3.5-Miniconda3
<a id="__codelineno-2-3" name="__codelineno-2-3"></a>
<a id="__codelineno-2-4" name="__codelineno-2-4"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage/
<a id="__codelineno-2-5" name="__codelineno-2-5"></a>
<a id="__codelineno-2-6" name="__codelineno-2-6"></a><span class="c1"># Prokaryote annotations</span>
<a id="__codelineno-2-7" name="__codelineno-2-7"></a>DRAM.py<span class="w"> </span>distill<span class="w"> </span>-i<span class="w"> </span>dram_annotations/annotations.tsv<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-8" name="__codelineno-2-8"></a><span class="w">                </span>-o<span class="w"> </span>dram_distillation<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-9" name="__codelineno-2-9"></a><span class="w">                </span>--trna_path<span class="w"> </span>dram_annotations/trnas.tsv<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-10" name="__codelineno-2-10"></a><span class="w">                </span>--rrna_path<span class="w"> </span>dram_annotations/rrnas.tsv
<a id="__codelineno-2-11" name="__codelineno-2-11"></a>
<a id="__codelineno-2-12" name="__codelineno-2-12"></a><span class="c1"># viral annotations</span>
<a id="__codelineno-2-13" name="__codelineno-2-13"></a>DRAM-v.py<span class="w"> </span>distill<span class="w"> </span>--remove_transposons<span class="w"> </span>--remove_fs<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-14" name="__codelineno-2-14"></a><span class="w">                  </span>-i<span class="w"> </span>dramv_annotations/annotations.tsv<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-15" name="__codelineno-2-15"></a><span class="w">                  </span>-o<span class="w"> </span>dramv_distillation
</code></pre></div></td></tr></table></div>
</div>
<h2 id="drampy-distill-output-files"><code>DRAM.py distill</code> output files<a class="headerlink" href="#drampy-distill-output-files" title="Permanent link">&para;</a></h2>
<p>The <code>DRAM</code> distillation step generates the following files that can be found within the <code>dram_distillation</code> directory :</p>
<table>
<thead>
<tr>
<th style="text-align: left;">File name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><code>genome_stats.tsv</code></td>
<td style="text-align: left;">Genome quality information required for <a href="https://www.nature.com/articles/nbt.3893">MIMAG</a></td>
</tr>
<tr>
<td style="text-align: left;"><code>metabolism_summary.xlsx</code></td>
<td style="text-align: left;">Summarised metabolism table containing number of genes with specific metabolic function identifiers</td>
</tr>
<tr>
<td style="text-align: left;"><code>product.html</code></td>
<td style="text-align: left;">HTML file displaying a heatmap summarising pathway coverage, electron transport chain component completion, and presence/absence of specific functions</td>
</tr>
<tr>
<td style="text-align: left;"><code>product.tsv</code></td>
<td style="text-align: left;">Data table visualised in <code>product.html</code></td>
</tr>
</tbody>
</table>
<p>First, let's have a look at the <code>genome_stats.tsv</code> file to check the assembly quality of our bins by double-clicking the file within the <code>Jupyter</code> environment, viewing from the terminal via <code>less</code> or <code>cat</code>, or downloading the files from <a href="../../resources/dram_distillation.zip">here</a> and opening locally (e.g. via Excel).</p>
<details class="magnifying-glass">
<summary>Content of <code>genome_stats.tsv</code></summary>
<table>
<thead>
<tr>
<th style="text-align: left;">genome</th>
<th style="text-align: left;">number of scaffolds</th>
<th style="text-align: left;">taxonomy</th>
<th style="text-align: left;">completeness score</th>
<th style="text-align: left;">contamination score</th>
<th style="text-align: left;">5S rRNA</th>
<th style="text-align: left;">16S rRNA</th>
<th style="text-align: left;">23S rRNA</th>
<th style="text-align: left;">tRNA count</th>
<th style="text-align: left;">assembly quality</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">bin_0</td>
<td style="text-align: left;">23</td>
<td style="text-align: left;">d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis</td>
<td style="text-align: left;">99.59</td>
<td style="text-align: left;">3.39</td>
<td style="text-align: left;">2 present</td>
<td style="text-align: left;">bin_0_bin_0_NODE_25_length_262222_cov_1.020389 (260223, 261736)</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">44</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_1</td>
<td style="text-align: left;">10</td>
<td style="text-align: left;">d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Nautiliales;f__Nautiliaceae;g__Nautilia;s__Nautilia profundicola</td>
<td style="text-align: left;">99.59</td>
<td style="text-align: left;">0.41</td>
<td style="text-align: left;">bin_1_bin_1_NODE_179_length_42779_cov_1.586174 (42578, 42683)</td>
<td style="text-align: left;">bin_1_bin_1_NODE_179_length_42779_cov_1.586174 (37726, 39229)</td>
<td style="text-align: left;">bin_1_bin_1_NODE_179_length_42779_cov_1.586174 (39618, 42521)</td>
<td style="text-align: left;">43</td>
<td style="text-align: left;">high</td>
</tr>
<tr>
<td style="text-align: left;">bin_2</td>
<td style="text-align: left;">14</td>
<td style="text-align: left;">d__Bacteria;p__Cyanobacteriota;c__Cyanobacteriia;o__PCC-6307;f__Cyanobiaceae;g__Prochlorococcus_C;s__</td>
<td style="text-align: left;">99.73</td>
<td style="text-align: left;">0.27</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">bin_2_bin_2_NODE_16_length_329212_cov_0.767174 (155033, 156512)</td>
<td style="text-align: left;">bin_2_bin_2_NODE_16_length_329212_cov_0.767174 (151325, 154198)</td>
<td style="text-align: left;">51</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_3</td>
<td style="text-align: left;">375</td>
<td style="text-align: left;">d__Bacteria;p__Planctomycetota;c__Brocadiia;o__Brocadiales;f__Brocadiaceae;g__Kuenenia;s__Kuenenia stuttgartiensis_A</td>
<td style="text-align: left;">91.21</td>
<td style="text-align: left;">0.61</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">bin_3_bin_3_NODE_465_length_7730_cov_0.184781 (1, 929)</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">37</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_4</td>
<td style="text-align: left;">12</td>
<td style="text-align: left;">d__Bacteria;p__Desulfobacterota;c__Desulfovibrionia;o__Desulfovibrionales;f__Desulfovibrionaceae;g__Desulfovibrio;s__Desulfovibrio desulfuricans</td>
<td style="text-align: left;">94.67</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">46</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_5</td>
<td style="text-align: left;">59</td>
<td style="text-align: left;">d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Burkholderiales;f__Nitrosomonadaceae;g__Nitrosomonas;s__Nitrosomonas europaea</td>
<td style="text-align: left;">99.97</td>
<td style="text-align: left;">0.9</td>
<td style="text-align: left;">bin_5_bin_5_NODE_53_length_160826_cov_1.123462 (86680, 86787)</td>
<td style="text-align: left;">bin_5_bin_5_NODE_53_length_160826_cov_1.123462 (90154, 91687)</td>
<td style="text-align: left;">bin_5_bin_5_NODE_53_length_160826_cov_1.123462 (86863, 89745)</td>
<td style="text-align: left;">42</td>
<td style="text-align: left;">high</td>
</tr>
<tr>
<td style="text-align: left;">bin_6</td>
<td style="text-align: left;">73</td>
<td style="text-align: left;">d__Bacteria;p__Bacillota;c__Bacilli;o__Staphylococcales;f__Staphylococcaceae;g__Staphylococcus;s__Staphylococcus aureus</td>
<td style="text-align: left;">98.58</td>
<td style="text-align: left;">0.11</td>
<td style="text-align: left;">3 present</td>
<td style="text-align: left;">bin_6_bin_6_NODE_110_length_77108_cov_0.318002 (42475, 44023)</td>
<td style="text-align: left;">bin_6_bin_6_NODE_110_length_77108_cov_0.318002 (39078, 41996)</td>
<td style="text-align: left;">34</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_7</td>
<td style="text-align: left;">57</td>
<td style="text-align: left;">d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Pseudomonadales;f__Pseudomonadaceae;g__Pseudomonas;s__Pseudomonas aeruginosa</td>
<td style="text-align: left;">96.87</td>
<td style="text-align: left;">0.11</td>
<td style="text-align: left;">2 present</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">42</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_8</td>
<td style="text-align: left;">16</td>
<td style="text-align: left;">d__Bacteria;p__Pseudomonadota;c__Gammaproteobacteria;o__Enterobacterales;f__Vibrionaceae;g__Vibrio;s__Vibrio cholerae</td>
<td style="text-align: left;">99.41</td>
<td style="text-align: left;">0.03</td>
<td style="text-align: left;">6 present</td>
<td style="text-align: left;">bin_8_bin_8_NODE_68_length_133755_cov_0.640623 (22138, 23676)</td>
<td style="text-align: left;">bin_8_bin_8_NODE_68_length_133755_cov_0.640623 (24392, 27264)</td>
<td style="text-align: left;">77</td>
<td style="text-align: left;">med</td>
</tr>
<tr>
<td style="text-align: left;">bin_9</td>
<td style="text-align: left;">70</td>
<td style="text-align: left;">d__Bacteria;p__Pseudomonadota;c__Alphaproteobacteria;o__Rhizobiales;f__Xanthobacteraceae;g__Nitrobacter;s__Nitrobacter winogradskyi</td>
<td style="text-align: left;">99.8</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">bin_9_bin_9_NODE_50_length_179569_cov_0.671348 (103643, 103751)</td>
<td style="text-align: left;">bin_9_bin_9_NODE_50_length_179569_cov_0.671348 (98442, 99927)</td>
<td style="text-align: left;">bin_9_bin_9_NODE_50_length_179569_cov_0.671348 (100757, 103567)</td>
<td style="text-align: left;">49</td>
<td style="text-align: left;">high</td>
</tr>
</tbody>
</table>
</details>
<p>To finish, we visualize the <em>Product</em>, an .HTML file produced in the distillation step, by double-clicking on it in our <em>Jupyter</em> lab notebook or downloading from <a href="../../resources/dram_distillation.zip">here</a>. The <em>Product</em> has three primary parts:</p>
<div class="admonition circle-check">
<p class="admonition-title"><em>Product</em> visualisation</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:3"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Modules</label><label for="__tabbed_1_2">ETC Complexes</label><label for="__tabbed_1_3">Other functions</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>Central metabolism pathways coverage. Completion of pathways is based on the structure of KEGG modules, with the pathway coverage calculated as the percent of steps with at least one gene present.</p>
<p><center>
<a class="glightbox" href="../../figures/ex15_fig1_DRAM_product_met_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex15_fig1_DRAM_product_met_2022.png" /></a>
</center></p>
</div>
<div class="tabbed-block">
<p>Electron Transport Chain component completion</p>
<p><center>
<a class="glightbox" href="../../figures/ex15_fig2_DRAM_product_ETC_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex15_fig2_DRAM_product_ETC_2022.png" /></a>
</center></p>
</div>
<div class="tabbed-block">
<p>Presence of specific functions, including CAZy, Nitrogen metabolism, Sulfur metabolism and Photosynthesis. Note that the taxonomic classification of each of the bins is also shown in the first figure</p>
<p><center>
<a class="glightbox" href="../../figures/ex15_fig3_DRAM_product_2022.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/ex15_fig3_DRAM_product_2022.png" /></a>
</center></p>
</div>
</div>
</div>
</div>
<h2 id="dram-vpy-distill-output-files"><code>DRAM-v.py distill</code> output files<a class="headerlink" href="#dram-vpy-distill-output-files" title="Permanent link">&para;</a></h2>
<p>The <code>DRAM-v</code> distillation step for the viral contigs generates the following files that can be found within the <code>dramv_distillation/</code> directory :</p>
<table>
<thead>
<tr>
<th style="text-align: left;"><div style="width:200px">File name</div></th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><code>vMAG_stats.tsv</code></td>
<td style="text-align: left;">"Genome" (in this case viral contigs of varying completeness) information including: total gene counts, viral vs host gene counts, and counts of genes related to viral replication, structure, and those with presumed viral or host benefits</td>
</tr>
<tr>
<td style="text-align: left;"><code>amg_summary.tsv</code></td>
<td style="text-align: left;">Genes identified as putative auxiliary metabolic genes (AMGs) and various columns for metabolic characterisation of each gene</td>
</tr>
<tr>
<td style="text-align: left;"><code>product.html</code></td>
<td style="text-align: left;">HTML file displaying a heatmap summarising AMG counts and presence/absence for different broad metabolic categories for each viral contig</td>
</tr>
</tbody>
</table>
<p>When viewing these files, see if you can find the following information:</p>
<div class="admonition quote">
<ul>
<li>What are some annotations of interest within the output annotations file? </li>
<li><em>NOTE: the *VirSorter2</em> annotations file includes multiple columns for both <strong>prokaryote</strong> and <strong>viral</strong> protein predictions. Be careful as to which column you are looking at (as well as its associated confidence score) when assessing viral annotations vs. AMGs*.</li>
<li>Among these annotations, how many were flagged as AMGs by <code>DRAM-v</code>?</li>
<li>What broad metabolic categories did the AMGs fall into? </li>
<li>Discussion point: How might we investigate whether identified putative AMGs are actually <em>within</em> the viral genomes, rather than residual contaminating host genomic sequence attached to the end of integrated prophage (but incompletely trimmed off in the excision process)?</li>
</ul>
</div>
<!--
---

## Tie findings to your initial goal

It is now time to explore the genomes and try to address your original goal!

You were tasked with identifying one of the following.

!!! abstract ""

    1. Denitrification (Nitrate or nitrite to nitrogen)
    2. Ammonia oxidation (Ammonia to nitrite or nitrate)
    3. Anammox (Ammonia and nitrite to nitrogen)
    4. Sulfur oxidation (SOX pathway, thiosulfate to sulfate)
    5. Sulfur reduction (DSR pathway, sulfate to sulfide)
    6. Photosynthetic carbon fixation
    7. Non-photosynthetic carbon fixation (Reverse TCA or Wood-Ljundahl)
    8. Non-polar flagella expression due to a chromosomal deletion
    9. Plasmid-encoded antibiotic resistance
    10. Aerobic (versus anaerobic) metabolism

Depending on what you are looking for, you will either be trying to find gene(s) of relevance to a particular functional pathway, or the omission of genes that might be critical in function. In either case, make sure to use the taxonomy of each MAG to determine whether it is likely to be a worthwhile candidate for exploration, as some of these traits are quite restricted in terms of which organisms carry them.

To conduct this exersise, you should use the information generated with `DRAM` as well as the annotation files we created yesterday and that are available in the directory `10.gene_annotation_and_coverage/gene_annotations`. 

Please note that we have also provided further annotation files within the directory `10.gene_annotation_and_coverage/example_annotation_tables` that contain information obtained after annotating the MAGs against additional databases (UniProt, UniRef100, KEGG, PFAM and TIGRfam). These example files can also be downloaded from [here](../resources/example_annotation_tables.zip). 

---

-->












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.07f07601.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.56dfad97.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>