
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/" rel="canonical"/>
<link href="../1_APPENDIX_ex8_Dereplication/" rel="prev"/>
<link href="../3_APPENDIX_ex11_Normalise_coverage_example/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#appendix-ex9-generating-input-files-for-vizbin-from-das_tool-curated-bins">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day 1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day 2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Day 3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day 4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction I: Shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction II: HPC and job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Filter raw reads by quality
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly I: Assembling contigs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly II: Varying the parameters
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Assembly evaluation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning with multiple tools
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manual bin refinement
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11.1_phylogenomics/">
        Phylogenomics
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation I: BLAST-like and HMM
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation II: DRAM and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation III: DRAM distillation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Introduction to data presentation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        CAZy heatmaps
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#1-generalise-bin-naming-and-add-bin-ids-to-sequence-headers">
    1. Generalise bin naming and add bin IDs to sequence headers
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-fragment-contigs">
    2. Fragment contigs
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#3-concatenate-fragmented-bins">
    3. Concatenate fragmented bins
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#4-read-mapping-of-subcontigs-fragmented-contigs-based-on-20k-length">
    4. Read mapping of subcontigs (fragmented contigs based on 20k length)
  </a>
<nav aria-label="4. Read mapping of subcontigs (fragmented contigs based on 20k length)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#4a-build-mapping-index">
    4a. Build mapping index
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#4b-map-sample-reads-to-index">
    4b. Map sample reads to index
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#5-generate-coverage-table-of-subcontigs-contigs-fragmented-based-on-20k-length">
    5. Generate coverage table of subcontigs (contigs fragmented based on 20k length)
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#6-generate-annotation-table-for-vizbin">
    6. Generate annotation table for VizBin
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#1-generalise-bin-naming-and-add-bin-ids-to-sequence-headers">
    1. Generalise bin naming and add bin IDs to sequence headers
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-fragment-contigs">
    2. Fragment contigs
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#3-concatenate-fragmented-bins">
    3. Concatenate fragmented bins
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#4-read-mapping-of-subcontigs-fragmented-contigs-based-on-20k-length">
    4. Read mapping of subcontigs (fragmented contigs based on 20k length)
  </a>
<nav aria-label="4. Read mapping of subcontigs (fragmented contigs based on 20k length)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#4a-build-mapping-index">
    4a. Build mapping index
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#4b-map-sample-reads-to-index">
    4b. Map sample reads to index
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#5-generate-coverage-table-of-subcontigs-contigs-fragmented-based-on-20k-length">
    5. Generate coverage table of subcontigs (contigs fragmented based on 20k length)
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#6-generate-annotation-table-for-vizbin">
    6. Generate annotation table for VizBin
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="appendix-ex9-generating-input-files-for-vizbin-from-das_tool-curated-bins">APPENDIX (ex9): Generating input files for <em>VizBin</em> from <em>DAS_Tool</em> curated bins<a class="headerlink" href="#appendix-ex9-generating-input-files-for-vizbin-from-das_tool-curated-bins" title="Permanent link">¶</a></h1>
<p>The final bins that we obtained in the previous step (output from <code>DAS_Tool</code>) have been copied into <code>6.bin_refinement/dastool_out/_DASTool_bins/</code></p>
<h2 id="1-generalise-bin-naming-and-add-bin-ids-to-sequence-headers">1. Generalise bin naming and add bin IDs to sequence headers<a class="headerlink" href="#1-generalise-bin-naming-and-add-bin-ids-to-sequence-headers" title="Permanent link">¶</a></h2>
<p>We will first modify the names of our bins to be simply numbered 1 to n bins. We will use a loop to do this, using the wildcard ( <code>*</code> ) to loop over all files in the <code>_DASTool_bins</code> folder, copying to the new <code>example_data_unchopped/</code> folder and renaming as <code>bin_[1-n].fna</code>. The <code>sed</code> command then adds the bin ID to the start of sequence headers in each of the new bin files (this will be handy information to have in the sequence headers for downstream processing).</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/6.bin_refinement/
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a><span class="c1"># Make a new directory the renamed bins</span>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>mkdir<span class="w"> </span>example_data_unchopped/
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="c1"># Copy and rename bins into generic &lt;bin_[0-n].fna&gt; filenames</span>
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a><span class="k">for</span><span class="w"> </span>file<span class="w"> </span><span class="k">in</span><span class="w"> </span>dastool_out/_DASTool_bins/*<span class="p">;</span>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="k">do</span>
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a><span class="w">    </span><span class="c1"># Copy and rename bin file</span>
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a><span class="w">    </span>cp<span class="w"> </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="w"> </span>example_data_unchopped/bin_<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.fna
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a><span class="w">    </span><span class="c1"># extract bin ID</span>
<a href="#__codelineno-0-13" id="__codelineno-0-13" name="__codelineno-0-13"></a><span class="w">    </span><span class="nv">binID</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span>example_data_unchopped/bin_<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.fna<span class="w"> </span>.fna<span class="k">)</span>
<a href="#__codelineno-0-14" id="__codelineno-0-14" name="__codelineno-0-14"></a><span class="w">    </span><span class="c1"># Add bin ID to sequence headers</span>
<a href="#__codelineno-0-15" id="__codelineno-0-15" name="__codelineno-0-15"></a><span class="w">    </span>sed<span class="w"> </span>-i<span class="w"> </span>-e<span class="w"> </span><span class="s2">"s/&gt;/&gt;</span><span class="si">${</span><span class="nv">binID</span><span class="si">}</span><span class="s2">_/g"</span><span class="w"> </span>example_data_unchopped/bin_<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.fna
<a href="#__codelineno-0-16" id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="w">    </span><span class="c1"># Increment i</span>
<a href="#__codelineno-0-17" id="__codelineno-0-17" name="__codelineno-0-17"></a><span class="w">    </span><span class="o">((</span><span class="nv">i</span><span class="o">+=</span><span class="m">1</span><span class="o">))</span>
<a href="#__codelineno-0-18" id="__codelineno-0-18" name="__codelineno-0-18"></a><span class="k">done</span>
</code></pre></div>
</div>
<h2 id="2-fragment-contigs">2. Fragment contigs<a class="headerlink" href="#2-fragment-contigs" title="Permanent link">¶</a></h2>
<p>Using the <code>cut_up_fasta.py</code> script that comes with the binning tool <code>CONCOCT</code>, cut contigs into 20k fragments to add better density to the cluster.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="c1"># Load CONCONCT module</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>module<span class="w"> </span>load<span class="w"> </span>CONCOCT/1.0.0-gimkl-2018b-Python-2.7.16
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="c1"># Make directory to add chopped bin data into</span>
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>mkdir<span class="w"> </span>example_data_20k/
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a>
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a><span class="c1"># loop over .fna files to generate chopped (fragmented) files using CONCONT's cut_up_fasta.py</span>
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>example_data_unchopped/*<span class="p">;</span>
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a><span class="k">do</span>
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a><span class="w">    </span><span class="nv">bin_name</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>.fna<span class="k">)</span>
<a href="#__codelineno-1-11" id="__codelineno-1-11" name="__codelineno-1-11"></a><span class="w">    </span>cut_up_fasta.py<span class="w"> </span>-c<span class="w"> </span><span class="m">20000</span><span class="w"> </span>-o<span class="w"> </span><span class="m">0</span><span class="w"> </span>--merge_last<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>&gt;<span class="w"> </span>example_data_20k/<span class="si">${</span><span class="nv">bin_name</span><span class="si">}</span>.chopped.fna
<a href="#__codelineno-1-12" id="__codelineno-1-12" name="__codelineno-1-12"></a><span class="k">done</span>
</code></pre></div>
</div>
<h2 id="3-concatenate-fragmented-bins">3. Concatenate fragmented bins<a class="headerlink" href="#3-concatenate-fragmented-bins" title="Permanent link">¶</a></h2>
<p>Concatenate chopped bins into a single <em>fastA</em> file.</p>
<p>Concatenate chopped bins into a single <code>all_bins.fna</code> file to use as input for both subcontig read mapping via <code>Bowtie2</code> and visualisation via <code>VizBin</code>.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>cat<span class="w"> </span>example_data_20k/*.fna<span class="w"> </span>&gt;<span class="w"> </span>all_bins.fna
</code></pre></div>
</div>
<h2 id="4-read-mapping-of-subcontigs-fragmented-contigs-based-on-20k-length">4. Read mapping of subcontigs (fragmented contigs based on 20k length)<a class="headerlink" href="#4-read-mapping-of-subcontigs-fragmented-contigs-based-on-20k-length" title="Permanent link">¶</a></h2>
<h3 id="4a-build-mapping-index">4a. Build mapping index<a class="headerlink" href="#4a-build-mapping-index" title="Permanent link">¶</a></h3>
<p>Build <code>Bowtie2</code> mapping index based on the concatenated chopped bins.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>read_mapping/
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>module<span class="w"> </span>load<span class="w"> </span>Bowtie2/2.4.5-GCC-11.3.0
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>bowtie2-build<span class="w"> </span>all_bins.fna<span class="w"> </span>read_mapping/bw_bins
</code></pre></div>
</div>
<h3 id="4b-map-sample-reads-to-index">4b. Map sample reads to index<a class="headerlink" href="#4b-map-sample-reads-to-index" title="Permanent link">¶</a></h3>
<p>Map quality filtered reads to the index using <code>Bowtie2</code>.</p>
<p>Example slurm script:</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-4-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-4-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-4-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-4-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-4-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-4-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-4-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-4-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-4-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-4-10">10</a></span>
<span class="normal"><a href="#__codelineno-4-11">11</a></span>
<span class="normal"><a href="#__codelineno-4-12">12</a></span>
<span class="normal"><a href="#__codelineno-4-13">13</a></span>
<span class="normal"><a href="#__codelineno-4-14">14</a></span>
<span class="normal"><a href="#__codelineno-4-15">15</a></span>
<span class="normal"><a href="#__codelineno-4-16">16</a></span>
<span class="normal"><a href="#__codelineno-4-17">17</a></span>
<span class="normal"><a href="#__codelineno-4-18">18</a></span>
<span class="normal"><a href="#__codelineno-4-19">19</a></span>
<span class="normal"><a href="#__codelineno-4-20">20</a></span>
<span class="normal"><a href="#__codelineno-4-21">21</a></span>
<span class="normal"><a href="#__codelineno-4-22">22</a></span>
<span class="normal"><a href="#__codelineno-4-23">23</a></span>
<span class="normal"><a href="#__codelineno-4-24">24</a></span>
<span class="normal"><a href="#__codelineno-4-25">25</a></span>
<span class="normal"><a href="#__codelineno-4-26">26</a></span>
<span class="normal"><a href="#__codelineno-4-27">27</a></span>
<span class="normal"><a href="#__codelineno-4-28">28</a></span>
<span class="normal"><a href="#__codelineno-4-29">29</a></span>
<span class="normal"><a href="#__codelineno-4-30">30</a></span>
<span class="normal"><a href="#__codelineno-4-31">31</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-4-2" name="__codelineno-4-2"></a>
<a id="__codelineno-4-3" name="__codelineno-4-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="c1">#SBATCH --job-name      6.bin_refinement_mapping</span>
<a id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="c1">#SBATCH --time          00:05:00</span>
<a id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="c1">#SBATCH --mem           1GB</span>
<a id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a id="__codelineno-4-9" name="__codelineno-4-9"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-4-10" name="__codelineno-4-10"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-4-11" name="__codelineno-4-11"></a>
<a id="__codelineno-4-12" name="__codelineno-4-12"></a>
<a id="__codelineno-4-13" name="__codelineno-4-13"></a>module<span class="w"> </span>purge
<a id="__codelineno-4-14" name="__codelineno-4-14"></a>module<span class="w"> </span>load<span class="w"> </span>Bowtie2/2.4.5-GCC-11.3.0<span class="w"> </span>SAMtools/1.15.1-GCC-11.3.0
<a id="__codelineno-4-15" name="__codelineno-4-15"></a>
<a id="__codelineno-4-16" name="__codelineno-4-16"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/6.bin_refinement/
<a id="__codelineno-4-17" name="__codelineno-4-17"></a>
<a id="__codelineno-4-18" name="__codelineno-4-18"></a><span class="c1"># Step 1</span>
<a id="__codelineno-4-19" name="__codelineno-4-19"></a><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span>sample1<span class="w"> </span>sample2<span class="w"> </span>sample3<span class="w"> </span>sample4<span class="p">;</span>
<a id="__codelineno-4-20" name="__codelineno-4-20"></a><span class="k">do</span>
<a id="__codelineno-4-21" name="__codelineno-4-21"></a>
<a id="__codelineno-4-22" name="__codelineno-4-22"></a><span class="w">  </span><span class="c1"># Step 2</span>
<a id="__codelineno-4-23" name="__codelineno-4-23"></a><span class="w">  </span>bowtie2<span class="w"> </span>--minins<span class="w"> </span><span class="m">200</span><span class="w"> </span>--maxins<span class="w"> </span><span class="m">800</span><span class="w"> </span>--threads<span class="w"> </span><span class="m">10</span><span class="w"> </span>--sensitive<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-4-24" name="__codelineno-4-24"></a><span class="w">          </span>-x<span class="w"> </span>read_mapping/bw_bins<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-4-25" name="__codelineno-4-25"></a><span class="w">          </span>-1<span class="w"> </span>../3.assembly/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span>-2<span class="w"> </span>../3.assembly/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-4-26" name="__codelineno-4-26"></a><span class="w">          </span>-S<span class="w"> </span>read_mapping/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam
<a id="__codelineno-4-27" name="__codelineno-4-27"></a>
<a id="__codelineno-4-28" name="__codelineno-4-28"></a><span class="w">  </span><span class="c1"># Step 3</span>
<a id="__codelineno-4-29" name="__codelineno-4-29"></a><span class="w">  </span>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">10</span><span class="w"> </span>-o<span class="w"> </span>read_mapping/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.bam<span class="w"> </span>read_mapping/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam
<a id="__codelineno-4-30" name="__codelineno-4-30"></a>
<a id="__codelineno-4-31" name="__codelineno-4-31"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition note">
<p class="admonition-title">Resource allocations</p>
<p>These settings are appropriate for this workshop's mock data. Full data sets will likely require considerably greater memory and time allocations.</p>
</div>
<h2 id="5-generate-coverage-table-of-subcontigs-contigs-fragmented-based-on-20k-length">5. Generate coverage table of subcontigs (contigs fragmented based on 20k length)<a class="headerlink" href="#5-generate-coverage-table-of-subcontigs-contigs-fragmented-based-on-20k-length" title="Permanent link">¶</a></h2>
<p>Use <code>MetaBAT</code>'s <code>jgi_summarise_bam_contig_depths</code> to generate a coverage table.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="c1"># Load module</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>module<span class="w"> </span>load<span class="w"> </span>MetaBAT/2.15-GCC-11.3.0
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="c1"># calculate coverage table</span>
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>jgi_summarize_bam_contig_depths<span class="w"> </span>--outputDepth<span class="w"> </span>example_data_20k_cov.txt<span class="w"> </span>read_mapping/sample*.bam
</code></pre></div>
</div>
<h2 id="6-generate-annotation-table-for-vizbin">6. Generate annotation table for <code>VizBin</code><a class="headerlink" href="#6-generate-annotation-table-for-vizbin" title="Permanent link">¶</a></h2>
<p>Using the chopped bin files (<code>example_data_20k/</code>) and the coverage table generated above (<code>example_data_20k_cov.txt</code>), we can use the following script to generate an annotation table in the format that <code>VizBin</code> is expecting. Note that here we are including columns for per-sub-contig coverage based on <em>sample1</em> (see note at the start of this exercise), label (bin ID), and length values, and storing this in <code>all_bins.sample1.vizbin.ann</code>.</p>
<p>What this script is doing is taking each fasta file and picking out the names of the contigs found in that file (bin). It is then looking for any coverage information for sample1 which is associated with that contig in the <code>example_data_20k_cov.txt</code> file, and adding that as a new column to the file. (If you wished to instead look based on sample2, you would need to modify the <code>cut</code> command in the line <code>sample1_cov=$(grep -P "${contigID}\t" example_data_20k_cov.txt | cut -f4)</code> accordingly (e.g. <code>cut -f6</code>).</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="c1"># Set up annotation file headers</span>
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">"coverage,label,length"</span><span class="w"> </span>&gt;<span class="w"> </span>all_bins.sample1.vizbin.ann
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a>
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="c1"># loop through bin .fna files</span>
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>example_data_20k/*.fna<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a><span class="w">    </span><span class="c1"># extract bin ID</span>
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a><span class="w">    </span><span class="nv">binID</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>.fna<span class="k">)</span>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a><span class="w">    </span><span class="c1"># loop through each sequence header in bin_file</span>
<a href="#__codelineno-6-9" id="__codelineno-6-9" name="__codelineno-6-9"></a><span class="w">    </span><span class="k">for</span><span class="w"> </span>header<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="sb">`</span>grep<span class="w"> </span><span class="s2">"&gt;"</span><span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="sb">`</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-6-10" id="__codelineno-6-10" name="__codelineno-6-10"></a><span class="w">        </span><span class="nv">contigID</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">header</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span><span class="s1">'s/&gt;//g'</span><span class="k">)</span>
<a href="#__codelineno-6-11" id="__codelineno-6-11" name="__codelineno-6-11"></a><span class="w">        </span><span class="c1"># identify this line from the coverage table (example_data_20k_cov.txt), and extract contigLen (column 2) and coverage for sample1.bam (column 4)</span>
<a href="#__codelineno-6-12" id="__codelineno-6-12" name="__codelineno-6-12"></a><span class="w">        </span><span class="nv">contigLen</span><span class="o">=</span><span class="k">$(</span>grep<span class="w"> </span>-P<span class="w"> </span><span class="s2">"</span><span class="si">${</span><span class="nv">contigID</span><span class="si">}</span><span class="s2">\t"</span><span class="w"> </span>example_data_20k_cov.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f2<span class="k">)</span>
<a href="#__codelineno-6-13" id="__codelineno-6-13" name="__codelineno-6-13"></a><span class="w">        </span><span class="nv">sample1_cov</span><span class="o">=</span><span class="k">$(</span>grep<span class="w"> </span>-P<span class="w"> </span><span class="s2">"</span><span class="si">${</span><span class="nv">contigID</span><span class="si">}</span><span class="s2">\t"</span><span class="w"> </span>example_data_20k_cov.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f4<span class="k">)</span>
<a href="#__codelineno-6-14" id="__codelineno-6-14" name="__codelineno-6-14"></a><span class="w">        </span><span class="c1"># Add to vizbin.ann file</span>
<a href="#__codelineno-6-15" id="__codelineno-6-15" name="__codelineno-6-15"></a><span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">"</span><span class="si">${</span><span class="nv">sample1_cov</span><span class="si">}</span><span class="s2">,</span><span class="si">${</span><span class="nv">binID</span><span class="si">}</span><span class="s2">,</span><span class="si">${</span><span class="nv">contigLen</span><span class="si">}</span><span class="s2">"</span><span class="w"> </span>&gt;&gt;<span class="w"> </span>all_bins.sample1.vizbin.ann
<a href="#__codelineno-6-16" id="__codelineno-6-16" name="__codelineno-6-16"></a><span class="w">    </span><span class="k">done</span>
<a href="#__codelineno-6-17" id="__codelineno-6-17" name="__codelineno-6-17"></a><span class="k">done</span>
</code></pre></div>
</div>
<p>We now have the <code>all_bins.fna</code> and <code>all_bins.sample1.vizbin.ann</code> files that were provided at the <a href="../../day2/ex9_refining_bins/">start of the VizBin exercise</a>.</p>
<hr/>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>