
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/" rel="canonical"/>
<link href="day1/ex1_bash_and_scheduler/" rel="next"/>
<link href="theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Metagenomics Summer School</title>
<link href="assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="." title="Metagenomics Summer School">
<img alt="logo" src="theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Home
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href=".">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="day1/ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="day2/ex6_initial_binning/">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="day3/ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="day4/ex15_gene_annotation_part3/">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="." title="Metagenomics Summer School">
<img alt="logo" src="theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<a class="md-nav__link md-nav__link--active" href=".">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="day1/ex1_bash_and_scheduler/">
        Introduction to shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day1/ex2_1_intro_to_scheduler/">
        Introduction to HPC &amp; HPC job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day1/ex2_quality_filtering/">
        Quality filtering raw reads
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day1/ex3_assembly/">
        Assembly
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day1/ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day1/ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day2/ex7_initial_binning/">
        Binning (continued)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day2/ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day3/ex13_gene_annotation_part1/">
        Gene annotation (part 1)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day3/ex14_gene_annotation_part2/">
        Gene annotation (part 2) and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex15_gene_annotation_part3/">
        Gene annotation (part 3)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex16b_data_presentation_Coverage/">
        Presentation of data: Per-sample coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="theme_images/uoa_ga_nesi_LOGOS.png"><img alt="image" src="theme_images/uoa_ga_nesi_LOGOS.png" width="350"/></a>
</center></p>
<style>h1 {text-align: center;}</style>
<h1><b>Metagenomics Summer School</b></h1>
<hr/>
<table>
<thead>
<tr>
<th align="left"><div style="width:350px"> <strong>Day</strong></div></th>
<th align="left"><strong>Lesson overview</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><a class="md-button md-button--primary" href="" title="">  Day 1  </a></td>
<td align="left"><a href="day1/ex1_bash_and_scheduler/">1. Intro Session I (pre-summer school): Shell </a><br/><a href="day1/ex2_1_intro_to_scheduler/">2. Intro Session II. : HPC and HPC Job Scheduler</a><br/><a href="day1/ex2_quality_filtering/">3. Quality filtering raw reads</a><br/><a href="day1/ex3_assembly/">4. Assembly</a><br/><a href="day1/ex4_assembly/">5. Assembly (part 2)</a><br/><a href="day1/ex5_evaluating_assemblies/">6. Evaluating the assemblies</a></td>
</tr>
<tr>
<td align="left"><a class="md-button md-button--primary" href="" title="">  Day 2  </a></td>
<td align="left"><a href="day2/ex6_initial_binning/">1. Introduction to binning</a><br/><a href="day2/ex7_initial_binning/">2. Binning (continued)</a><br/><a href="day2/ex8_bin_dereplication/">3. Bin dereplication</a><br/><a href="day2/ex9_refining_bins/">4. Manually refining bins</a></td>
</tr>
<tr>
<td align="left"><a class="md-button md-button--primary" href="" title="">  Day 3  </a></td>
<td align="left"><a href="day3/ex10_viruses/">1. Identifying viral contigs in metagenomic data</a><br/><a href="day3/ex11_coverage_and_taxonomy/">2. Assigning taxonomy to refined prokaryotic bins</a><br/><a href="day3/ex12_gene_prediction/">3. Gene prediction</a><br/><a href="day3/ex12_gene_prediction/">4. Gene annotation (part 1)</a><br/><a href="day3/ex14_gene_annotation_part2/">5. Gene annotation (part 2) and coverage calculation</a></td>
</tr>
<tr>
<td align="left"><a class="md-button md-button--primary" href="" title="">  Day 4  </a></td>
<td align="left"><a href="day4/ex15_gene_annotation_part3/">1. Gene annotation (part 3)</a><br/><a href="day4/ex16a_data_presentation_Intro/">2. Presentation of data - Intro</a><br/><a href="day4/ex16b_data_presentation_Coverage/">3. Presentation of data: Per-sample coverage heatmaps</a><br/><a href="day4/ex16c_OPTIONAL_data_presentation_Ordination/">4. Presentation of data: Ordinations</a><br/><a href="day4/ex16d_data_presentation_KEGG_pathways/">5. Presentation of data: KEGG pathway maps</a><br/><a href="day4/ex16e_data_presentation_Gene_synteny/">6. Presentation of data: Gene synteny</a><br/><a href="day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">7. Presentation of data: CAZY annotations heatmap</a></td>
</tr>
</tbody>
</table>
<p><br/></p>
<details class="calendar-days">
<summary>Timetable 2022</summary>
<div class="tabbed-set tabbed-alternate" data-tabs="1:4"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio"><input id="__tabbed_1_2" name="__tabbed_1" type="radio"><input id="__tabbed_1_3" name="__tabbed_1" type="radio"><input id="__tabbed_1_4" name="__tabbed_1" type="radio"><div class="tabbed-labels"><label for="__tabbed_1_1">Day 1 : 29<sup>th</sup> Nov</label><label for="__tabbed_1_2">Day 2 : 30<sup>th</sup> Nov</label><label for="__tabbed_1_3">Day 3 : 1<sup>st</sup> Dec</label><label for="__tabbed_1_4">Day 4 : 2<sup>nd</sup> Dec</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<table>
<thead>
<tr>
<th align="left">Time</th>
<th align="left">Event</th>
<th align="left">Session leader</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">9:00 am – 9:30 am</td>
<td align="left"><strong>Introduction</strong><br/>- Welcome<br/>- Logging into NeSI</td>
<td align="left">Jian Sheng Boey</td>
</tr>
<tr>
<td align="left">9:30 am – 10:30 am</td>
<td align="left"><strong>TASK:</strong> Bash scripting and introduction to Slurm scheduler</td>
<td align="left">Dinindu Senanayake</td>
</tr>
<tr>
<td align="left">10:30 am – 10:50 am</td>
<td align="left"><strong>Morning tea break</strong></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">10:50 am – 11:10 am</td>
<td align="left"><strong>TASK:</strong> Bash scripting (continued)</td>
<td align="left">Dinindu Senanayake</td>
</tr>
<tr>
<td align="left">11:10 am – 11:40 pm</td>
<td align="left"><strong>TALK:</strong> The metagenomics decision tree</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">11:40 am – 12:00 pm</td>
<td align="left"><strong>DISCUSSION</strong> Guided discussions on approaches to metagenomics analyses</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">12:00 pm – 12:45 pm</td>
<td align="left"><strong>Break for lunch</strong> (<em>lunch not provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">12:45 pm – 1:45 pm</td>
<td align="left"><strong>TALK:</strong> Quality filtering raw reads<br/><strong>TASK:</strong> Visualisation with <em>FastQC</em><br/><strong>TASK:</strong> Read trimming and adapter removal<br/><strong>TASK:</strong> Diagnosing poor libraries<br/><strong>TALK:</strong> Common issues and best practice<br/><strong>TASK (<em>Optional</em>):</strong> Filtering out host DNA</td>
<td align="left">Annie West</td>
</tr>
<tr>
<td align="left">1:45 pm – 3:00 pm</td>
<td align="left"><strong>TALK:</strong> Assembly<br/>- Choice of assemblers<br/>- Considerations for parameters, and when to stop!<br/><strong>TASK:</strong> Prepare data for assembly<br/><strong>TASK:</strong> Exploring assembler options<br/><strong>TASK:</strong> Submitting jobs to NeSI via slurm<br/><strong>TASK:</strong> Run SPAdes and IDBA-UD assembly<br/><strong>TASK (<em>Optional</em>):</strong> Submitting variant assemblies to NeSI</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">3:00 pm – 3:20 pm</td>
<td align="left"><strong>Afternoon tea break</strong> (<em>Tea, coffee, and snacks provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">3:20 pm – 5:00 pm</td>
<td align="left"><strong>TALK:</strong> Future considerations - co-assembly vs. single assemblies<br/><strong>TASK:</strong> Assembly evaluation<br/><strong>TASK:</strong> Short contig removal</td>
<td align="left">Jian Sheng Boey<br/> Kim Handley</td>
</tr>
</tbody>
</table>
</div>
<div class="tabbed-block">
<table>
<thead>
<tr>
<th align="left">Time</th>
<th align="left">Event</th>
<th align="left">Session leader</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">9:00 am – 9:15 am</td>
<td align="left"><strong>Introduction</strong><br/>- Overview of yesterday, questions<br/>- Overview of today</td>
<td align="left">Annie West</td>
</tr>
<tr>
<td align="left">9:15 am – 9:30 am</td>
<td align="left"><strong>RECORDED PRESENTATION:</strong> Genomic adaptations enabling <em>Acidithiobacillus</em> distribution across wide ranging hot spring temperatures and pHs (Chanenath Sriaporn)</td>
<td align="left"><br/> Kim Handley</td>
</tr>
<tr>
<td align="left">9:30 am – 10:30 am</td>
<td align="left"><strong>Binning (part 1)</strong><br/><strong>TALK:</strong> Overview of binning history<br/>- Key parameters and strategies for binning<br/><strong>TASK:</strong> Read mapping</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">10:30 am – 10:50 am</td>
<td align="left"><strong>Morning tea break</strong> (<em>Tea, coffee, and snacks provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">10:50 am – 11:20 am</td>
<td align="left"><strong>Binning (part 2)</strong><br/><strong>TASK:</strong> Multi-binning strategy (<em>Metabat</em> and <em>Maxbin</em>)</td>
<td align="left">Cesar Facimoto</td>
</tr>
<tr>
<td align="left">11:20 am – 12:00 pm</td>
<td align="left"><strong>TALK:</strong> Overview of binning history (<em>continued</em>)<br/>- Key parameters and strategies for binning</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">12:00 pm – 12:45 pm</td>
<td align="left"><strong>Break for lunch</strong> (<em>lunch not provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">12:45 pm – 2:00 pm</td>
<td align="left"><strong>Binning (part 3)</strong><br/><strong>TASK:</strong> Bin dereplication via <em>DAS_Tool</em><br/><strong>TASK:</strong> Evaluating bins using <em>CheckM</em></td>
<td align="left">Cesar Facimoto</td>
</tr>
<tr>
<td align="left">2:00 pm - 2:30 pm</td>
<td align="left"><strong>Binning (part 4)</strong><br/><strong>TALK:</strong> Discuss additional dereplication strategies, such as <em>dRep</em><br/><strong>TALK:</strong> How to work with viral and eukaryotic bins<br/><strong>TALK:</strong> Dealing with organisms which possess minimal genomes<br/></td>
<td align="left">Jian Sheng Boey<br/> Annie West</td>
</tr>
<tr>
<td align="left">2:30 pm - 3:00 pm</td>
<td align="left"><strong>TALK:</strong> Bin refinement<br/>- Refinement strategies</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">3:00 pm – 3:20 pm</td>
<td align="left"><strong>Afternoon tea break</strong> (<em>Tea, coffee, and snacks provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">3:20 pm – 5:00 pm</td>
<td align="left"><strong>TASK:</strong> Working with <em>VizBin</em><br/></td>
<td align="left">Jian Sheng Boey<br/> Annie West</td>
</tr>
</tbody>
</table>
</div>
<div class="tabbed-block">
<table>
<thead>
<tr>
<th align="left">Time</th>
<th align="left">Event</th>
<th align="left">Session leader</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">9:00 am – 9:20 am</td>
<td align="left"><strong>Introduction</strong><br/>- Overview of yesterday, questions<br/>- Overview of today</td>
<td align="left">Jian Sheng Boey</td>
</tr>
<tr>
<td align="left">9:20 am – 10:30 am</td>
<td align="left"><strong>TALK:</strong> Bin taxonomic classification<br/>- Bin and species determination<br/><strong>TASK:</strong> Taxonomic classification using GTDB-Tk<br/><strong>TASK:</strong> View phylogenetic trait distribution using <em>ANNOTREE</em></td>
<td align="left">David Waite</td>
</tr>
<tr>
<td align="left">10:30 am – 10:50 am</td>
<td align="left"><strong>Morning tea break</strong> (<em>Tea, coffee, and snacks provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">10:50 am – 12:00 pm</td>
<td align="left"><strong>TALK:</strong> Identifying viruses from metagenomic data<br/><strong>TASK:</strong> Identifying viral contigs using <em>VIBRANT</em><br/><strong>RECORDED PRESENTATION:</strong> Genetic exchange in ultra-small Patescibacteria (Emilie Gios)<br/><strong>TASK:</strong> QC of viral contigs using <em>CheckV</em><br/><strong>TASK:</strong> Taxonomic classification of viruses using <em>vContact2</em></td>
<td align="left">Jian Sheng Boey<br/>Annie West<br/><br/></td>
</tr>
<tr>
<td align="left">12:00 pm – 12:45 pm</td>
<td align="left"><strong>Break for lunch</strong> (<em>lunch not provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">12:45 pm - 1:30 pm</td>
<td align="left"><strong>TALK:</strong> Gene prediction using <em>prodigal</em> and other tools (<em>RNAmer</em>, <em>Aragorn</em>, etc)<br/><strong>TASK:</strong> Predict open reading frames and protein sequences</td>
<td align="left">Jian Sheng Boey <br/> Cesar Facimoto</td>
</tr>
<tr>
<td align="left">1:30 pm - 2:00 pm</td>
<td align="left"><strong>TALK:</strong> Gene annotation (part 1)<br/> - Methods<br/><strong>TASK:</strong> Gene annotation using <em>DIAMOND</em> and <em>HMMER3</em><br/><strong>Discussion:</strong> Evaluating the quality of gene assignment<br/></td>
<td align="left">Jian Sheng Boey<br/>Cesar Facimoto</td>
</tr>
<tr>
<td align="left">2:00 pm – 3:00 pm</td>
<td align="left"><strong>TALK:</strong> Gene annotation (part 2)<br/>- Using online resources (e.g. <em>KEGG, BioCyc, MetaCyc, HydDB, PSORT</em>)<br/><strong>TASK:</strong> View KEGG annotation in KEGG website</td>
<td align="left">Jian Sheng Boey<br/>Cesar Facimoto</td>
</tr>
<tr>
<td align="left">3:00 pm – 3:20 pm</td>
<td align="left"><strong>Afternoon tea break</strong></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">3:20 pm – 4:30 pm</td>
<td align="left"><strong>TASK:</strong> MAG annotation with <em>DRAM</em><br/><strong>TASK:</strong> Coverage calculation using <em>Bowtie2</em><br/><strong>TASK:</strong> Introduce group project goals<br/><strong>TASK:</strong> Dividing into working groups / get a group name<br/><strong>TASK:</strong> Select a goal from your project</td>
<td align="left">Annie West<br/>Kim Handley</td>
</tr>
<tr>
<td align="left">4:30 pm – 5:00 pm</td>
<td align="left"><strong>End of day wrap up</strong></td>
<td align="left">Kim Handley</td>
</tr>
</tbody>
</table>
</div>
<div class="tabbed-block">
<table>
<thead>
<tr>
<th align="left">Time</th>
<th align="left">Event</th>
<th align="left">Session leader</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">9:00 am – 9:15 am</td>
<td align="left"><strong>Introduction</strong><br/>- Overview of yesterday, questions<br/>- Overview of today</td>
<td align="left">Jian Sheng Boey</td>
</tr>
<tr>
<td align="left">9:15 am – 10:00 am</td>
<td align="left"><strong>Presentation of data</strong><br/><strong>TALK:</strong> <em>DRAM</em> results overview<br/><strong>TASK:</strong> Explore <em>DRAM</em> results</td>
<td align="left">Annie West</td>
</tr>
<tr>
<td align="left">10:00 am – 10:30 am</td>
<td align="left"><strong>Presentation of data (genome distributions)</strong><br/><strong>TALK:</strong> Visualising findings (environmental distribution)<br/><strong>TASK:</strong> Coverage heatmap and nMDS ordination</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">10:30 am – 10:50 am</td>
<td align="left"><strong>Morning tea break</strong> (<em>Tea, coffee, and snacks provided</em>)<br/><strong>TASK:</strong> Workshop survey</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">10:50 am – 12:00 pm</td>
<td align="left"><strong>Presentation of data (metabolism)</strong><br/><strong>TALK:</strong> Visualising findings (metabolic maps, genome features, metabolic schematics, and gene trees)<br/><strong>TASK:</strong> KEGG metabolic pathways<br/><strong>TASK:</strong> Gene synteny<br/><strong>TASK:</strong> CAZy heatmaps</td>
<td align="left">Cesar Facimoto<br/>Jian Sheng Boey</td>
</tr>
<tr>
<td align="left">12:00 pm – 12:45 pm</td>
<td align="left"><strong>Break for lunch</strong> (<em>lunch not provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">12:45 pm – 2:30 pm</td>
<td align="left"><strong>TASK:</strong> Analyse data for group work<br/><strong>TASK:</strong> Prepare group presentation</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">2:30 pm – 3:00 pm</td>
<td align="left"><strong>Present and discuss findings</strong><br/><strong>TASK:</strong> Each group to give an informal presentation of their data</td>
<td align="left">Kim Handley</td>
</tr>
<tr>
<td align="left">3:00 pm – 3:20 pm</td>
<td align="left"><strong>Afternoon tea break</strong> (<em>Tea, coffee, and snacks provided</em>)</td>
<td align="left"></td>
</tr>
<tr>
<td align="left">3:20 pm – 3:40 pm</td>
<td align="left"><strong>Present and discuss findings (continued)</strong><br/><strong>TASK:</strong> Each group to give an informal presentation of their data</td>
<td align="left">Annie West</td>
</tr>
<tr>
<td align="left">3:40 pm – 4:00 pm</td>
<td align="left"><strong>End of day wrap up</strong><br/>- Final discussion</td>
<td align="left">Kim Handley<br/>Jian Sheng Boey</td>
</tr>
</tbody>
</table>
</div>
</div>
</input></input></input></input></div>
</details>
<p><br/></p>
<details class="example">
<summary>Appendices</summary>
<table>
<thead>
<tr>
<th align="left">Appendix ID</th>
<th align="left"></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><a class="md-button" href="" title="">Appendix 1</a></td>
<td align="left"><a href="resources/1_APPENDIX_ex8_Dereplication/">Dereplicating data from multiple assemblies</a></td>
</tr>
<tr>
<td align="left"><a class="md-button" href="" title="">Appendix 2</a></td>
<td align="left"><a href="resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">Generating input files for "VizBin" from "DAS_Tool" curated bins</a></td>
</tr>
<tr>
<td align="left"><a class="md-button" href="" title="">Appendix 3</a></td>
<td align="left"><a href="resources/3_APPENDIX_ex11_Normalise_coverage_example/">Normalise per-sample coverage values by average library size</a></td>
</tr>
<tr>
<td align="left"><a class="md-button" href="" title="">Appendix 4</a></td>
<td align="left"><a href="resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">Viral taxonomy prediction via vContact2</a></td>
</tr>
<tr>
<td align="left"><a class="md-button" href="" title="">Appendix 5</a></td>
<td align="left"><a href="resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">Preparing input files for visualising gene synteny</a></td>
</tr>
</tbody>
</table>
</details>
<p><br/></p>
<div class="admonition comment-dots">
<p class="admonition-title">Post-workshop survey</p>
<p>Thank you for attending Metagenomics Summer School 2022! We would like your feedback on how we have done and what we can improve on. You can provide feedback <a href="https://auckland.au1.qualtrics.com/jfe/form/SV_3W4gkA3XZ0qK2do">here</a>.</p>
</div>
<p><br/></p>
<div class="admonition note">
<p class="admonition-title">License</p>
<p>Genomics Aotearoa / New Zealand eScience Infrastructure/ University of Auckland <strong>Metagenomics Summer School</strong> material is licensed under the <strong>GNU General Public License v3.0, 29 June 2007</strong> . (<a href="https://github.com/GenomicsAotearoa/shell-for-bioinformatics/blob/main/LICENSE">Follow this link for more information</a>)</p>
</div>
<p><br/></p>
<div class="admonition note">
<p class="admonition-title">Slides for workshop</p>
<p>You can find a copy of the slides presented during the workshop, with published figures removed, in the <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/tree/master/slides">slides/</a> folder.</p>
</div>
<p><br/></p>
<div class="admonition desktop-download-24">
<p class="admonition-title">Snapshots of results to download</p>
<p>If you are having trouble downloading files using <code>scp</code>, we are providing exemplar output files, which you can download through your browser, <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/tree/master/docs/resources">here</a>, or via the following links: </p>
<ul>
<li><a href="resources/fastqc_results.zip">FastQC results</a></li>
<li><a href="resources/quast_results_sans_reference.zip">Quast results</a> and <a href="resources/quast_references.zip">required references</a></li>
<li><a href="resources/vizbin_files.zip">Input files for VizBin</a></li>
<li><a href="resources/example_annotation_tables.zip">Gene annotation tables</a></li>
<li><a href="resources/dram_distillation.zip">DRAM output files</a></li>
</ul>
</div>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": ".", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>