# Getting started

Prior to testing out the assembly and binning workflows, it is necessary to create mock metagenomes that can be assembled into a predictable outcome.

---

### Bacterial genomes 

For the purposes of this workshop, we wanted a mixed community of bacterial species that covered the following functional traits:

1. Denitrification
1. Ammonia oxidation
1. Anaerobic ammonia oxidation (Anamox)
1. Oxidation of sulfur compounds
1. Reduction of sulfur compounds
1. Carbon fixation via photosynthetic pathway
1. Carbon fixation via non-photosynthetic pathway (reverse-TCA cycle or Wood–Ljungdahl pathway)
1. Presence of flagella, from which polar or lateral placement can be inferred
1. Plasmid-encoded antibiotic resistance
1. Oxidative phosphorylation pathways

The trick here is that not only does each of the eventual bins need to encode some of these properties, they also must be able to be identified in the *KEGG* ontology. To do this, I downloaded organisms from which these traits should be easy to identify, then annotated them using our lab pipeline (which you can read [here](https://github.com/GenomicsAotearoa/environmental_metagenomics)) to ensure that the traits were present in the genome.

### Identifying appropriate genomes

I selected the following bacterial genomes, based on their following genomic features:

|Feature|Species|Genome|Pathway/genes|
|:---|:---|:---|:---|
|Denitrification|*Pseudomonas aeruginosa*|[GCF_000152525.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000152525.1)|Possess the complete denitrification pathway|
||*Nitrobacter winogradskyi*|[GCF_000012725.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000012725.1)|Possess an incomplete denitrification pathway<br>Lacking *norBC* and *nosZ*, so can only convert nitrate to nitric oxide|
|Ammonia oxidation|*Nitrosomonas europaea*|[GCF_000009145.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000009145.1)|Possesses *amoCAB* and *Hao*, so is capable of oxidising ammonia to nitrite<br>Lacks *nxrAB*, so cannot oxidise nitrite to nitrate|
|Ammonia oxidation (Anamox)|*Kuenenia stuttgartiensis*|[GCF_900232105.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_900232105.1)|Possesses *Hzs* and *Hdh* so can oxidise ammonia to nitrogen<br>Also possess *nirS*, so can convert nitrite to hydrazine for additional nitrogen production|
|Sulfur oxidation|*Arcobacter nitrofigilis*|[GCF_000092245.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000092245.1)|Complete SOX pathway for thiosulfate oxidation<br>Also encodes a complete assimilatory sulfate reduction pathway|
|Sulfur reduction|*Desulfovibrio desulfuricans*|[GCF_000420465.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000420465.1)|Encodes a complete dissimilatory sulfate reduction pathway|
|Photosynthetic carbon fixation|*Prochlorococcus marinus*|[GCA_000015705.1](https://www.ncbi.nlm.nih.gov/assembly/GCA_000015705.1)|Encodes the majority of genes needed for Photosystem I, all of Photosystem II, and RuBisCO is easily detectable|
|Non-photosynthetic carbon fixation|*Nautilia profundicola*|[GCF_000021725.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000021725.1)|Possesses ATP-dependent citrate lyase, completing the reverse-TCA cycle<br>Lacks a complete Wood-Ljundahl pathway, as all Campylobacterota do|
|Flagella placement|*Vibrio cholerae*|[GCF_000006745.1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000006745.1)|Possess full machinery for flagella production<br>The difference between polar and lateral flagella placement is controlled by two genes, which I will delete from the genome|
|Plasmid-encoded antibiotic resistance|*Staphylococcus aureus*|[GCA_000027045.1](https://www.ncbi.nlm.nih.gov/assembly/GCA_000027045.1)|Contains a vancomycin resistance pathway in the genome<br>Lacks any plasmid sequence|
||plasmid pPR9|[GU237136.1](https://www.ncbi.nlm.nih.gov/nuccore/281307459/)|*S. aureus* plasmid that carries the Class A β-lactamase *blaZ*, and regulatory features|

A few notes about this selection above:

1. Oxidative phosphorylation pathways are present in all of these genomes, although there is variation between which complexes they encode. For this reason, I didn't pull down any genomes for this specific pathway.
1. The *Staphylococcus aureus* genome and pPR9 plasmid will be merged together to create a genome with plasmid-encoded antibiotic resistnace. The phylogenetic history of the plasmid should make it easy to associate with the bin.
1. In order to create the lateral flagella phenotype in the *V. cholerae* genome, *flhF* and *flhG* will be manually excised from the genome.

---

### Viral genomes (Update for MGSS_2020)

The following viral reference genomes were selected:

* achromobacter_virus_83_24_uid307926/NC_028834
* bacillus_virus_bps13_uid177519/NC_018857
* corynebacterium_phage_bfk20_uid20757/NC_009799
* prochlorococcus_phage_p_ssp10_uid195499/NC_020835
* pseudomonas_phage_pabg_uid215670/NC_022096
* ralstonia_virus_rs603_uid266715/NC_025454
* staphylococcus_phage_dw2_uid253383/NC_024391
* streptococcus_phage_tp_778l_uid227111/NC_022776
* vibrio_virus_ksf1_uid14562/NC_006294
* archaeal_bj1_virus_uid18503/NC_008695
* methanobacterium_phage_psim2_uid14160/NC_001902

---
