# Introduction

As an update for 2020, we have decided to include viral genomes in the mock metagenome data sets. To do this, mock viral metagenomes were created using 11 bacterial and archaeal viruses in differing distributions and abundance skews over four mock metagenomes as per the process outlined in the documentation for building the [bacterial mock metagenomes](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/MGSS2020_DEV/building_resources/1.building_mock_metagenomes.md). These viral metagenomes (of 100k reads/metagenome, in contrast to the bacterial 1m reads/metagenome) were then concatenated with 2019's bacterial metagenome files, resulting in the new forward and reverse read files for samples 1-4 for 2020.

As per the previous documentation, this work was conducted on the *NeSI* platform, using the programs `wget` and `InSilicoSeq`.

---

### Acquiring genomes

The genomes chosen in the [Introduction notebook](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/building_resources/0.introduction.md) were pulled from the NCBI repository using `wget`.

```bash
# Achromobacter phage 83-24 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/achromobacter_virus_83_24_uid307926/NC_028834.fna

# Bacillus phage BPS13 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/bacillus_virus_bps13_uid177519/NC_018857.fna
    
# Corynebacterium~phage~BFK20 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/corynebacterium_phage_bfk20_uid20757/NC_009799.fna
    
# Prochlorococcus~phage~P-SSP10 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/prochlorococcus_phage_p_ssp10_uid195499/NC_020835.fna

# Pseudomonas~phage~PaBG (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/pseudomonas_phage_pabg_uid215670/NC_022096.fna

# Ralstonia~phage~RS603 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/ralstonia_virus_rs603_uid266715/NC_025454.fna
    
# Staphylococcus~phage~DW2 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/staphylococcus_phage_dw2_uid253383/NC_024391.fna
    
# Streptococcus~phage~TP-778L (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/streptococcus_phage_tp_778l_uid227111/NC_022776.fna
    
# Vibrio~phage~KSF1 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/vibrio_virus_ksf1_uid14562/NC_006294.fna
    
# Archaeal~BJ1~virus (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/archaeal_bj1_virus_uid18503/NC_008695.fna
    
# Methanobacterium phage psiM2 (complete)
wget https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/methanobacterium_phage_psim2_uid14160/NC_001902.fna
```

Before proceeding, all genomes were procesed to have more succinct contig headers.

```bash
for i in *.fna;
do
    sed -i 's/ .*//g' ${i}
    sed -i 's/.*ref_//g' ${i}
    sed -i 's/_//g' ${i}
    sed -i 's/|//g' ${i}
done
```

---

### Building the metagenomes

As with the [bacterial metagenomes preparation](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/MGSS2020_DEV/building_resources/1.building_mock_metagenomes.md), `InSilicoSeq` was used to generate mock metagenomes from the downloaded viral genomes. 

The viral genomes were divided into four "communities" as follows:

```bash
com1="NC_001902.fna NC_006294.fna NC_008695.fna NC_009799.fna NC_018857.fna NC_020835.fna NC_022096.fna"
com2="NC_008695.fna NC_009799.fna NC_018857.fna NC_020835.fna NC_022096.fna NC_022776.fna NC_024391.fna NC_025454.fna"
com3="NC_009799.fna NC_018857.fna NC_020835.fna NC_022096.fna NC_022776.fna NC_024391.fna NC_025454.fna NC_028834.fna"
com4="NC_001902.fna NC_006294.fna NC_018857.fna NC_020835.fna NC_025454.fna NC_028834.fna"
```

Four metagenomes were then generated via `InSilicoSeq` based on these "communities", and each with a different abudance skew.

```bash
mkdir abund_skew1/ abund_skew2/ abund_skew3/ abund_skew4/

iss generate --cpus 8 \
--genomes ${com1} \
--abundance lognormal -m HiSeq --n_reads 100000 \
-o abund_skew1/viral_sample1 --compress

iss generate --cpus 8 \
--genomes ${com2} \
--abundance halfnormal -m HiSeq --n_reads 100000 \
-o abund_skew2/viral_sample2 --compress

iss generate --cpus 8 \
--genomes ${com3} \
--abundance exponential -m HiSeq --n_reads 100000 \
-o abund_skew3/viral_sample3 --compress

iss generate --cpus 8 \
--genomes ${com4} \
--abundance zero_inflated_lognormal -m HiSeq --n_reads 100000 \
-o abund_skew4/viral_sample4 --compress
```

Finally, these viral metagenomes were concatenated together with 2019's bacterial metagenome files.

```bash
for i in {1..4};
do
    for j in R1 R2;
    do
        zcat 2019_metagenomes/sample${i}_${j}.fastq.gz viral_genomes/abund_skew${i}/viral_sample${i}_${j}.fastq.gz | gzip -c > 2020_metagenomes/sample${i}_${j}.fastq.gz
    done
done
```

---
