
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day2/ex6_initial_binning/">
      
      
        <link rel="prev" href="../../day1/ex5_evaluating_assemblies/">
      
      
        <link rel="next" href="../ex7_initial_binning/">
      
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.34">
    
    
      
        <title>Introduction to binning - Metagenomics Summer School</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.35f28582.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#introduction-to-binning" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Metagenomics Summer School" class="md-header__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Summer School
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Introduction to binning
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day1/ex1_bash_and_scheduler/" class="md-tabs__link">
          
  
  Day 1

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="./" class="md-tabs__link">
          
  
  Day 2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-tabs__link">
          
  
  Day 3

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day4/ex15_gene_annotation_part3/" class="md-tabs__link">
          
  
  Day 4

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-tabs__link">
          
  
  Resources

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../supplementary/supplementary_1/" class="md-tabs__link">
          
  
  Supplementary

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Metagenomics Summer School" class="md-nav__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Metagenomics Summer School
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 1
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex1_bash_and_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction I: Shell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_1_intro_to_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction II: HPC and job scheduler
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_quality_filtering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filter raw reads by quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex3_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly I: Assembling contigs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex4_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly II: Varying the parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex5_evaluating_assemblies/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly evaluation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Day 2
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Introduction to binning
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Introduction to binning
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#remove-short-contigs-from-the-data-set" class="md-nav__link">
    <span class="md-ellipsis">
      Remove short contigs from the data set
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping" class="md-nav__link">
    <span class="md-ellipsis">
      Obtain coverage profiles for assembled contigs via read mapping
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Obtain coverage profiles for assembled contigs via read mapping">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-a-mapping-index" class="md-nav__link">
    <span class="md-ellipsis">
      Create a mapping index
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mapping-the-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Mapping the reads
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-1-loop-through-the-sample-files" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1 - Loop through the sample files
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-map-the-reads-using-bowtie2" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2 - Map the reads using bowtie2
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-sort-and-compress-results" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3 - Sort and compress results
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optional-read-mapping-using-an-array" class="md-nav__link">
    <span class="md-ellipsis">
      (Optional) Read mapping using an array
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex7_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Binning with multiple tools
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex8_bin_dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bin dereplication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex9_refining_bins/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Manual bin refinement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex10.1_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifying viral contigs in metagenomic data
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 3
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assigning taxonomy to refined prokaryotic bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11.1_phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex10.2_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virus taxonomy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex12_gene_prediction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene prediction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex13_gene_annotation_part1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation I: BLAST-like and HMM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex14_gene_annotation_part2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation II: DRAM and coverage calculation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 4
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex15_gene_annotation_part3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation III: DRAM distillation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16a_data_presentation_Intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to data presentation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16b_data_presentation_Coverage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coverage heatmaps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ordinations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16d_data_presentation_KEGG_pathways/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    KEGG pathway maps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16e_data_presentation_Gene_synteny/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene synteny
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CAZy heatmaps
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Resources
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex8): Dereplicating data from multiple assemblies
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex9): Generating input files for VizBin from DAS_Tool curated bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11) : Viral taxonomy prediction via vContact2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/7_command_line_shortcuts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Useful command line shortcuts
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Supplementary
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Supplementary
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI Setup
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI File system, Working directory and Symlinks
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#remove-short-contigs-from-the-data-set" class="md-nav__link">
    <span class="md-ellipsis">
      Remove short contigs from the data set
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping" class="md-nav__link">
    <span class="md-ellipsis">
      Obtain coverage profiles for assembled contigs via read mapping
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Obtain coverage profiles for assembled contigs via read mapping">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-a-mapping-index" class="md-nav__link">
    <span class="md-ellipsis">
      Create a mapping index
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mapping-the-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Mapping the reads
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-1-loop-through-the-sample-files" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1 - Loop through the sample files
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-map-the-reads-using-bowtie2" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2 - Map the reads using bowtie2
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-sort-and-compress-results" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3 - Sort and compress results
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optional-read-mapping-using-an-array" class="md-nav__link">
    <span class="md-ellipsis">
      (Optional) Read mapping using an array
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="introduction-to-binning">Introduction to binning<a class="headerlink" href="#introduction-to-binning" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#remove-short-contigs-from-the-data-set">Remove short contigs from the data set</a></li>
<li><a href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping">Obtain coverage profiles for assembled contigs via read mapping</a></li>
<li><a href="#optional-read-mapping-using-an-array"><em>Optional:</em> Read mapping using a slurm array</a></li>
</ul>
</div>
<hr />
<h2 id="remove-short-contigs-from-the-data-set">Remove short contigs from the data set<a class="headerlink" href="#remove-short-contigs-from-the-data-set" title="Permanent link">&para;</a></h2>
<p>Ideally, we do not want to be creating bins from all of the assembled contigs, as there is often a long tail of contigs which are only several <em>k</em>-mers long. These have little biological meaning, as they are too short for robust gene annotation, and they can introduce a significant degree of noise in the clustering algorithms used for binning. We therefore identify a suitable threshold for a minimum length of contigs to be considered for binning.</p>
<p>We have already done this in the <a href="../../day1/ex5_evaluating_assemblies/">previous exercise</a> so we could either use the existing filtering at 1,000 bp in length, or move to something stricter. Most binning tools have a default cut-off for minimum contig size - <code>MetaBAT</code> uses a default minimum of 2,500 bp, and recommends at least 1,500 bp. By contrast, <code>MaxBin</code> sets the minimum length at 1,000 bp.</p>
<hr />
<h2 id="obtain-coverage-profiles-for-assembled-contigs-via-read-mapping">Obtain coverage profiles for assembled contigs via read mapping<a class="headerlink" href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping" title="Permanent link">&para;</a></h2>
<p>Binning is done using a combination of information encoded in the <strong><em>composition</em></strong> and <strong><em>coverage</em></strong> of the assembled contigs. <strong><em>Composition</em></strong> refers to <em>k</em>-mer (usually tetranucleotide) frequency profiles of the contigs, which are generally conserved within a genome. By contrast, <strong><em>coverage</em></strong> is a reflection of the abundance of the contigs in the assembly. Organisms which are more abundant will contribute more genomic material to the metagenome, and hence their DNA will be, on average, more abundant in the sample. When binning, we can look for pieces of DNA which are not assembled together, but have similar <strong><em>composition</em></strong> and occur at approximately equal abundances in the sample to identify contigs which likely originate in the same genome.</p>
<p>The composition of the contigs is calculated by the binning tool at run time, but to obtain coverage information we must map our unassembled reads from each sample against the assembly to generate the differential abundance profiles for each contig. This is achieved using <code>bowtie2</code> to map the reads against the assembly, then <code>samtools</code> to sort and compress the resulting file.</p>
<h3 id="create-a-mapping-index">Create a mapping index<a class="headerlink" href="#create-a-mapping-index" title="Permanent link">&para;</a></h3>
<p>Before we can map reads, we need to create a <code>bowtie2</code> index file from the assembly, for use in read mapping. Navigate into the <code>5.binning/</code> folder to begin.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Load modules</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>module<span class="w"> </span>purge
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>module<span class="w"> </span>load<span class="w"> </span>Bowtie2/2.5.4-GCC-12.3.0
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a><span class="c1"># Navigate to working directory</span>
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/5.binning/
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a><span class="c1"># Build index</span>
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>bowtie2-build<span class="w"> </span>spades_assembly/spades_assembly.m1000.fna<span class="w"> </span>spades_assembly/bw_spades
</code></pre></div>
</div>
<p>If you look inside the <code>spades_assembly/</code> folder you will now see the following:</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>ls<span class="w"> </span>spades_assembly/
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>bw_spades.1.bt2  bw_spades.3.bt2  bw_spades.rev.1.bt2  spades_assembly.m1000.fna
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>bw_spades.2.bt2  bw_spades.4.bt2  bw_spades.rev.2.bt2
</code></pre></div>
</div>
<p>These files ending in <em>.bt2</em> are the index files for <code>bowtie2</code>, and are specific to this tool. If you wish to map using an alternate tool (for example <code>bowtie</code> or <code>BBMap</code>) you will need to create index/database files using these programs.</p>
<p>Generally speaking, we don't need to know the names of the index files, as they are simply referred to be the output name we specified (<em>bw_spades</em>) when running <code>bowtie2</code>.</p>
<h3 id="mapping-the-reads">Mapping the reads<a class="headerlink" href="#mapping-the-reads" title="Permanent link">&para;</a></h3>
<p>We will create a slurm script to perform the mapping steps, as these benefit greatly from the multithreaded capacity of NeSI and we will use a <em>for loop</em> to iterate over each set of reads to simplify our script.</p>
<p>The full script is provided here, and we will discuss it below.</p>
<p>Open a new script using nano:</p>
<div class="admonition terminal-2">
<p class="admonition-title">Create script named <code>spades_mapping.sl</code></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>nano<span class="w"> </span>spades_mapping.sl
</code></pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own folder.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-4-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-4-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-4-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-4-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-4-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-4-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-4-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-4-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-4-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-4-10">10</a></span>
<span class="normal"><a href="#__codelineno-4-11">11</a></span>
<span class="normal"><a href="#__codelineno-4-12">12</a></span>
<span class="normal"><a href="#__codelineno-4-13">13</a></span>
<span class="normal"><a href="#__codelineno-4-14">14</a></span>
<span class="normal"><a href="#__codelineno-4-15">15</a></span>
<span class="normal"><a href="#__codelineno-4-16">16</a></span>
<span class="normal"><a href="#__codelineno-4-17">17</a></span>
<span class="normal"><a href="#__codelineno-4-18">18</a></span>
<span class="normal"><a href="#__codelineno-4-19">19</a></span>
<span class="normal"><a href="#__codelineno-4-20">20</a></span>
<span class="normal"><a href="#__codelineno-4-21">21</a></span>
<span class="normal"><a href="#__codelineno-4-22">22</a></span>
<span class="normal"><a href="#__codelineno-4-23">23</a></span>
<span class="normal"><a href="#__codelineno-4-24">24</a></span>
<span class="normal"><a href="#__codelineno-4-25">25</a></span>
<span class="normal"><a href="#__codelineno-4-26">26</a></span>
<span class="normal"><a href="#__codelineno-4-27">27</a></span>
<span class="normal"><a href="#__codelineno-4-28">28</a></span>
<span class="normal"><a href="#__codelineno-4-29">29</a></span>
<span class="normal"><a href="#__codelineno-4-30">30</a></span>
<span class="normal"><a href="#__codelineno-4-31">31</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-4-2" name="__codelineno-4-2"></a>
<a id="__codelineno-4-3" name="__codelineno-4-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="c1">#SBATCH --job-name      spades_mapping</span>
<a id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="c1">#SBATCH --time          00:05:00</span>
<a id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="c1">#SBATCH --mem           1GB</span>
<a id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a id="__codelineno-4-9" name="__codelineno-4-9"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-4-10" name="__codelineno-4-10"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-4-11" name="__codelineno-4-11"></a>
<a id="__codelineno-4-12" name="__codelineno-4-12"></a>module<span class="w"> </span>purge
<a id="__codelineno-4-13" name="__codelineno-4-13"></a>module<span class="w"> </span>load<span class="w"> </span>Bowtie2/2.5.4-GCC-12.3.0<span class="w"> </span>SAMtools/1.19-GCC-12.3.0
<a id="__codelineno-4-14" name="__codelineno-4-14"></a>
<a id="__codelineno-4-15" name="__codelineno-4-15"></a><span class="c1"># Working directory</span>
<a id="__codelineno-4-16" name="__codelineno-4-16"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/5.binning/
<a id="__codelineno-4-17" name="__codelineno-4-17"></a>
<a id="__codelineno-4-18" name="__codelineno-4-18"></a><span class="c1"># Step 1</span>
<a id="__codelineno-4-19" name="__codelineno-4-19"></a><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span>sample1<span class="w"> </span>sample2<span class="w"> </span>sample3<span class="w"> </span>sample4<span class="p">;</span>
<a id="__codelineno-4-20" name="__codelineno-4-20"></a><span class="k">do</span>
<a id="__codelineno-4-21" name="__codelineno-4-21"></a>
<a id="__codelineno-4-22" name="__codelineno-4-22"></a><span class="w">  </span><span class="c1"># Step 2</span>
<a id="__codelineno-4-23" name="__codelineno-4-23"></a><span class="w">  </span>bowtie2<span class="w"> </span>--minins<span class="w"> </span><span class="m">200</span><span class="w"> </span>--maxins<span class="w"> </span><span class="m">800</span><span class="w"> </span>--threads<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>--sensitive<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-4-24" name="__codelineno-4-24"></a><span class="w">          </span>-x<span class="w"> </span>spades_assembly/bw_spades<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-4-25" name="__codelineno-4-25"></a><span class="w">          </span>-1<span class="w"> </span>../3.assembly/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span>-2<span class="w"> </span>../3.assembly/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-4-26" name="__codelineno-4-26"></a><span class="w">          </span>-S<span class="w"> </span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam
<a id="__codelineno-4-27" name="__codelineno-4-27"></a>
<a id="__codelineno-4-28" name="__codelineno-4-28"></a><span class="w">  </span><span class="c1"># Step 3</span>
<a id="__codelineno-4-29" name="__codelineno-4-29"></a><span class="w">  </span>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>.bam<span class="w"> </span><span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam
<a id="__codelineno-4-30" name="__codelineno-4-30"></a>
<a id="__codelineno-4-31" name="__codelineno-4-31"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition terminal-2">
<p class="admonition-title">Submit the script using <code>sbatch</code></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>sbatch<span class="w"> </span>spades_mapping.sl
</code></pre></div>
</div>
<h3 id="step-1-loop-through-the-sample-files">Step 1 - Loop through the sample files<a class="headerlink" href="#step-1-loop-through-the-sample-files" title="Permanent link">&para;</a></h3>
<p>Since we just want to perform the same set of operations on each file, we can use a <em>for loop</em> to repeat each operation on a set of files. The structure of the loop, and use of variables was covered on the first day.</p>
<p>For large sets of files, it can be beneficial to use a slurm <em>array</em> to send the jobs out to different nodes and distribute the process across many independent jobs. An example of how we could modify the above script is given at the bottom of this exercise, but is not necessary for the purposes of this workshop.</p>
<h3 id="step-2-map-the-reads-using-bowtie2">Step 2 - Map the reads using <em>bowtie2</em><a class="headerlink" href="#step-2-map-the-reads-using-bowtie2" title="Permanent link">&para;</a></h3>
<div class="admonition abstract">
<p class="admonition-title">This is performed using the following parameters</p>
<table>
<thead>
<tr>
<th style="text-align: left;"><div style="width:100px">Parameter</div></th>
<th style="text-align: left;">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><code>--minins</code></td>
<td style="text-align: left;">Minimum insert size, determines the minimum distance between the start of each read pair</td>
</tr>
<tr>
<td style="text-align: left;"><code>--maxins</code></td>
<td style="text-align: left;">Maximum insert size, determines the maximum distance between the start of each read pair</td>
</tr>
<tr>
<td style="text-align: left;"><code>--threads</code></td>
<td style="text-align: left;">Number of threads to use in read mapping</td>
</tr>
<tr>
<td style="text-align: left;"><code>--sensitive</code></td>
<td style="text-align: left;">Specifies where we want to be positioned in the trade-off between speed and sensitivity. See the manual for more information</td>
</tr>
<tr>
<td style="text-align: left;"><code>-x</code></td>
<td style="text-align: left;">The base name of our assembly index. Should be exactly the same as what was specified when running <code>bowtie2-build</code></td>
</tr>
<tr>
<td style="text-align: left;"><code>-1</code> / <code>-2</code></td>
<td style="text-align: left;">The forward and reverse read pairs to map to the assembly</td>
</tr>
<tr>
<td style="text-align: left;"><code>-S</code></td>
<td style="text-align: left;">Name of the output file, to be written in <em>sam</em> format</td>
</tr>
</tbody>
</table>
</div>
<h3 id="step-3-sort-and-compress-results">Step 3 - Sort and compress results<a class="headerlink" href="#step-3-sort-and-compress-results" title="Permanent link">&para;</a></h3>
<p>The default output format for most mapping tools is the Sequence Alignment/Map (<em>sam</em>) format. This is a compact text representation of where each short read sits in the contigs. You can view this file using any text viewer, although owing to the file size <code>less</code> is a good idea.</p>
<p>Generally I wouldn't bother with this - there is a lot of information in here and unless you are looking to extract specific information from the alignment directly, this is just an intermediate file in our workflow. In order to save disk space, and prepare the file for downstream analysis we now perform two final steps:</p>
<ol>
<li>Sort the mapping information</li>
<li>Compress the <em>sam</em> file into its binary equivalent, <em>bam</em></li>
</ol>
<div class="admonition abstract">
<p class="admonition-title">Which is achieved with the following parameters</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Parameter</th>
<th style="text-align: left;">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><code>sort</code></td>
<td style="text-align: left;">Subcommand for <code>samtools</code> to invoke the <code>sort</code> operation</td>
</tr>
<tr>
<td style="text-align: left;"><code>-@</code></td>
<td style="text-align: left;">Number of threads to use for sorting and compressing</td>
</tr>
<tr>
<td style="text-align: left;"><code>-o</code></td>
<td style="text-align: left;">Output file name. When we specify the <em>bam</em> extension <code>samtools</code> automatically compresses the output</td>
</tr>
</tbody>
</table>
</div>
<div class="admonition failure">
<p class="admonition-title">Do not open a <code>.bam</code> file, your terminal might crash!</p>
</div>
<p>Compressing the <em>sam</em> files to <em>bam</em> format is an important step as <em>sam</em> files can be massive. It is also helpful to sort the mapping information so that reads mapped to a contig are listed in order of their start position. For example</p>
<section style="display: flex;">
<div class="my-block" style="width:300px">
<div class="admonition terminal-2">
<p class="admonition-title">Unsorted reads</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Ref: REFERENCECONTIG
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>Map: --------ECONT--
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>Map: REFE-----------
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>Map: --FERENCECO----
<a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>Map: -----------NTIG
</code></pre></div>
</div>
</div>
<div class="my-block" style="width:30px"></div>
<div class="my-block" style="width:300px">
<div class="admonition terminal-2">
<p class="admonition-title">Sorted reads</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>Ref: REFERENCECONTIG
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>Map: REFE-----------
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>Map: --FERENCECO----
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>Map: --------ECONT--
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>Map: -----------NTIG
</code></pre></div>
</div>
</div>
</section>
<p>Reads will initially be mapped in an unsorted order, as they are added to the <em>sam</em> file in more or less the same order as they are encountered in the original <em>fastQ</em> files.</p>
<p>Sorting the mapping information is an important prerequisite for performing certain downstream processes. Not every tool we use requires reads to be sorted, but it can be frustrating having to debug the instances where read sorting matters, so we typically just get it done as soon as possible and then we don't have to worry about it again.</p>
<hr />
<h2 id="optional-read-mapping-using-an-array"><em>(Optional)</em> Read mapping using an array<a class="headerlink" href="#optional-read-mapping-using-an-array" title="Permanent link">&para;</a></h2>
<p>If you have a large number of files to process, it might be worth using a slurm array to distribute your individual mapping jobs across many separate nodes. An example script for how to perform this is given below. We do not need to use an array for read mapping in this workshop, but we will revisit array jobs in further lessons.</p>
<p>Open a new script using nano:</p>
<div class="admonition terminal-2">
<p class="admonition-title">Create script named <code>spades_mapping_array.sl</code></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>nano<span class="w"> </span>spades_mapping_array.sl
</code></pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own folder.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-9-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-9-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-9-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-9-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-9-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-9-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-9-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-9-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-9-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-9-10">10</a></span>
<span class="normal"><a href="#__codelineno-9-11">11</a></span>
<span class="normal"><a href="#__codelineno-9-12">12</a></span>
<span class="normal"><a href="#__codelineno-9-13">13</a></span>
<span class="normal"><a href="#__codelineno-9-14">14</a></span>
<span class="normal"><a href="#__codelineno-9-15">15</a></span>
<span class="normal"><a href="#__codelineno-9-16">16</a></span>
<span class="normal"><a href="#__codelineno-9-17">17</a></span>
<span class="normal"><a href="#__codelineno-9-18">18</a></span>
<span class="normal"><a href="#__codelineno-9-19">19</a></span>
<span class="normal"><a href="#__codelineno-9-20">20</a></span>
<span class="normal"><a href="#__codelineno-9-21">21</a></span>
<span class="normal"><a href="#__codelineno-9-22">22</a></span>
<span class="normal"><a href="#__codelineno-9-23">23</a></span>
<span class="normal"><a href="#__codelineno-9-24">24</a></span>
<span class="normal"><a href="#__codelineno-9-25">25</a></span>
<span class="normal"><a href="#__codelineno-9-26">26</a></span>
<span class="normal"><a href="#__codelineno-9-27">27</a></span>
<span class="normal"><a href="#__codelineno-9-28">28</a></span>
<span class="normal"><a href="#__codelineno-9-29">29</a></span>
<span class="normal"><a href="#__codelineno-9-30">30</a></span>
<span class="normal"><a href="#__codelineno-9-31">31</a></span>
<span class="normal"><a href="#__codelineno-9-32">32</a></span>
<span class="normal"><a href="#__codelineno-9-33">33</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-9-2" name="__codelineno-9-2"></a>
<a id="__codelineno-9-3" name="__codelineno-9-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-9-4" name="__codelineno-9-4"></a><span class="c1">#SBATCH --job-name      spades_mapping_array</span>
<a id="__codelineno-9-5" name="__codelineno-9-5"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-9-6" name="__codelineno-9-6"></a><span class="c1">#SBATCH --time          00:20:00</span>
<a id="__codelineno-9-7" name="__codelineno-9-7"></a><span class="c1">#SBATCH --mem           20GB</span>
<a id="__codelineno-9-8" name="__codelineno-9-8"></a><span class="c1">#SBATCH --array         0-3</span>
<a id="__codelineno-9-9" name="__codelineno-9-9"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a id="__codelineno-9-10" name="__codelineno-9-10"></a><span class="c1">#SBATCH --error         %x_%A_%a.err</span>
<a id="__codelineno-9-11" name="__codelineno-9-11"></a><span class="c1">#SBATCH --output        %x_%A_%a.out</span>
<a id="__codelineno-9-12" name="__codelineno-9-12"></a>
<a id="__codelineno-9-13" name="__codelineno-9-13"></a><span class="c1"># Load modules</span>
<a id="__codelineno-9-14" name="__codelineno-9-14"></a>module<span class="w"> </span>purge
<a id="__codelineno-9-15" name="__codelineno-9-15"></a>module<span class="w"> </span>load<span class="w"> </span>Bowtie2/2.5.4-GCC-12.3.0<span class="w"> </span>SAMtools/1.19-GCC-12.3.0
<a id="__codelineno-9-16" name="__codelineno-9-16"></a>
<a id="__codelineno-9-17" name="__codelineno-9-17"></a><span class="c1"># Working directory</span>
<a id="__codelineno-9-18" name="__codelineno-9-18"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/5.binning/
<a id="__codelineno-9-19" name="__codelineno-9-19"></a>
<a id="__codelineno-9-20" name="__codelineno-9-20"></a><span class="c1"># Load the sample names into a bash array</span>
<a id="__codelineno-9-21" name="__codelineno-9-21"></a><span class="nv">samples</span><span class="o">=(</span>sample1<span class="w"> </span>sample2<span class="w"> </span>sample3<span class="w"> </span>sample4<span class="o">)</span>
<a id="__codelineno-9-22" name="__codelineno-9-22"></a>
<a id="__codelineno-9-23" name="__codelineno-9-23"></a><span class="c1"># Run Bowtie2 and SAMTools, using the SLURM_ARRAY_TASK_ID variable to</span>
<a id="__codelineno-9-24" name="__codelineno-9-24"></a><span class="c1"># identify which position in the `samples` array to use</span>
<a id="__codelineno-9-25" name="__codelineno-9-25"></a>bowtie2<span class="w"> </span>--minins<span class="w"> </span><span class="m">200</span><span class="w"> </span>--maxins<span class="w"> </span><span class="m">800</span><span class="w"> </span>--threads<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>--sensitive<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-26" name="__codelineno-9-26"></a><span class="w">        </span>-x<span class="w"> </span>spades_assembly/bw_spades<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-27" name="__codelineno-9-27"></a><span class="w">        </span>-1<span class="w"> </span>../3.assembly/<span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-28" name="__codelineno-9-28"></a><span class="w">        </span>-2<span class="w"> </span>../3.assembly/<span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-29" name="__codelineno-9-29"></a><span class="w">        </span>-S<span class="w"> </span><span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_a.sam
<a id="__codelineno-9-30" name="__codelineno-9-30"></a>
<a id="__codelineno-9-31" name="__codelineno-9-31"></a>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-32" name="__codelineno-9-32"></a><span class="w">              </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_a.bam<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-9-33" name="__codelineno-9-33"></a><span class="w">              </span><span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_a.sam
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition terminal-2">
<p class="admonition-title">Submit the job to slurm</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>sbatch<span class="w"> </span>spades_mapping_array.sl
</code></pre></div>
</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.07f07601.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.56dfad97.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>