
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day2/ex6_initial_binning/" rel="canonical"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.1, mkdocs-material-8.3.9" name="generator"/>
<title>Introduction to binning - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.1d29e8d0.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.cbb835fc.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gdesc-inner { font-size: 0.75rem; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="" data-md-color-primary="indigo" data-md-color-scheme="" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#introduction-to-binning">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Introduction to binning
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-tabs__inner md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="./">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day1" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction to shell &amp; HPC Job Scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Quality filtering raw reads
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day2" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" data-md-toggle="toc" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Introduction to binning
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Introduction to binning
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#remove-short-contigs-from-the-data-set">
    Remove short contigs from the data set
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping">
    Obtain coverage profiles for assembled contigs via read mapping
  </a>
<nav aria-label="Obtain coverage profiles for assembled contigs via read mapping" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#creating-a-mapping-index">
    Creating a mapping index
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mapping-the-reads">
    Mapping the reads
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#step-1-loop-through-the-sample-files">
    Step 1 - Loop through the sample files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#step-2-map-the-reads-using-bowtie2">
    Step 2 - Map the reads using bowtie2
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#step-3-sorting-and-compressing-results">
    Step 3 - Sorting and compressing results
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-read-mapping-using-an-array">
    Optional: Read mapping using an array
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex7_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_4" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day3" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Per-sample coverage and assigning taxonomy
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation (part 1)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation (part 2)
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day4" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation (part 3)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Presentation of data: Per-sample coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        (*Optional*) Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: *Optional exercise*: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_6" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Resources" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8):Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): How to generate the blast files provided in [ex15_gene_synteny](../day4/ex16e_data_presentation_Gene_synteny.md)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/6_APPENDIX_ex15_gene_synteny_grab_GOI/">
        APPENDIX (ex15): Grab Gene of Interest (generate the *cys.txt* files provided in [ex15_gene_synteny](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/day4/ex16e_data_presentation_Gene_synteny.md))
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#remove-short-contigs-from-the-data-set">
    Remove short contigs from the data set
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping">
    Obtain coverage profiles for assembled contigs via read mapping
  </a>
<nav aria-label="Obtain coverage profiles for assembled contigs via read mapping" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#creating-a-mapping-index">
    Creating a mapping index
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mapping-the-reads">
    Mapping the reads
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#step-1-loop-through-the-sample-files">
    Step 1 - Loop through the sample files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#step-2-map-the-reads-using-bowtie2">
    Step 2 - Map the reads using bowtie2
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#step-3-sorting-and-compressing-results">
    Step 3 - Sorting and compressing results
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-read-mapping-using-an-array">
    Optional: Read mapping using an array
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/edit/master/docs/day2/ex6_initial_binning.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg>
</a>
<h1 id="introduction-to-binning">Introduction to binning<a class="headerlink" href="#introduction-to-binning" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#remove-short-contigs-from-the-data-set">Remove short contigs from the data set</a></li>
<li><a href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping">Obtain coverage profiles for assembled contigs via read mapping</a></li>
<li><a href="#optional-read-mapping-using-an-array"><em>Optional:</em> Read mapping using a slurm array</a></li>
</ul>
</div>
<hr/>
<h3 id="remove-short-contigs-from-the-data-set">Remove short contigs from the data set<a class="headerlink" href="#remove-short-contigs-from-the-data-set" title="Permanent link">¶</a></h3>
<p>Ideally, we do not want to be creating bins from all of the assembled contigs, as there is often a long tail of contigs which are only several <span class="arithmatex">\(k\)</span>-mers long. These have little biological meaning, as they are too short for robust gene annotation, and they can introduce a significant degree of noise in the clustering algorithms used for binning. We therefore identify a suitable threshold for a minimum length of contigs to be considered for binning.</p>
<p>We have already done this in the <a href="../../day1/ex5_evaluating_assemblies/">previous exercise</a> so we could either use the existing filtering at 1,000 bp in length, or move to something stricter. Most binning tools have a default cut-off for minimum contig size - <code>MetaBAT</code> uses a default minimum of 2,500 bp, and recommends at least 1,500 bp. By contrast, <code>MaxBin</code> sets the minimum length at 1,000 bp.</p>
<hr/>
<h3 id="obtain-coverage-profiles-for-assembled-contigs-via-read-mapping">Obtain coverage profiles for assembled contigs via read mapping<a class="headerlink" href="#obtain-coverage-profiles-for-assembled-contigs-via-read-mapping" title="Permanent link">¶</a></h3>
<p>Binning is done using a combination of information encoded in the <em>composition</em> and <em>coverage</em> of the assembled contigs. <em>Composition</em> refers to <em>k</em>-mer (usually tetranucleotide) frequency profiles of the contigs, which are generally conserved within a genome. By contrast, <em>coverage</em> is a reflection of the abundance of the contigs in the assembly. Organisms which are more abundant will contribute more genomic material to the metagenome, and hence their DNA will be, on average, more abundant in the sample. When binning, we can look for pieces of DNA which are not assembled together, but have similar <em>composition</em> and occur at approximately equal abundances in the sample to identify contigs which likely originate in the same genome.</p>
<p>The composition of the contigs is calculated by the binning tool at run time, but to obtain coverage information we must map our unassembled reads from each sample against the assembly to generate the differential abundance profiles for each contig. This is achieved using <code>bowtie2</code> to map the reads against the assembly, then <code>samtools</code> to sort and compress the resulting file.</p>
<h4 id="creating-a-mapping-index">Creating a mapping index<a class="headerlink" href="#creating-a-mapping-index" title="Permanent link">¶</a></h4>
<p>Before we can map reads, we need to create a <code>bowtie2</code> index file from the assembly, for use in read mapping. Navigate into the <code>5.binning/</code> folder to begin.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>module load Bowtie2/2.4.5-GCC-11.3.0
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a><span class="nb">cd</span> /nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/5.binning/
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>bowtie2-build spades_assembly/spades_assembly.m1000.fna spades_assembly/bw_spades
</code></pre></div>
<p>If you look inside the <code>spades_assembly/</code> folder you will now see the following:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>ls spades_assembly/
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="c1"># bw_spades.1.bt2  bw_spades.3.bt2  bw_spades.rev.1.bt2  spades_assembly.fna</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="c1"># bw_spades.2.bt2  bw_spades.4.bt2  bw_spades.rev.2.bt2  spades_assembly.m1000.fna</span>
</code></pre></div>
<p>These files ending in <em>.bt2</em> are the index files for <code>bowtie2</code>, and are specific to this tool. If you wish to map using an alternate tool (for example <code>bowtie</code> or <code>BBMap</code>) you will need to create index/database files using these programs.</p>
<p>Generally speaking, we don't need to know the names of the index files, as they are simply referred to be the output name we specified (<em>bw_spades</em>) when running <code>bowtie2</code>.</p>
<h4 id="mapping-the-reads">Mapping the reads<a class="headerlink" href="#mapping-the-reads" title="Permanent link">¶</a></h4>
<p>We will create a slurm script to perform the mapping steps, as these benefit greatly from the multithreaded capacity of NeSI and we will use a <em>for loop</em> to iterate over each set of reads to simplify our script.</p>
<p>The full script is provided here, and we will discuss it below.</p>
<p>Open a new script using nano:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>nano spades_mapping.sl
</code></pre></div>
<p>Paste in the following script. Remember to replace <your folder=""> with your own folder.</your></p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="c1">#SBATCH --job-name      spades_mapping</span>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a><span class="c1">#SBATCH --res           SummerSchool</span>
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a><span class="c1">#SBATCH --time          00:05:00</span>
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a><span class="c1">#SBATCH --mem           1GB</span>
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a><span class="c1">#SBATCH --error         spades_mapping.err</span>
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a><span class="c1">#SBATCH --output        spades_mapping.out</span>
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a>module purge
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a>module load Bowtie2/2.4.5-GCC-11.3.0 SAMtools/1.15.1-GCC-11.3.0
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a>
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a><span class="nb">cd</span> /nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/5.binning/
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a>
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a><span class="c1"># Step 1</span>
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a><span class="k">for</span> i <span class="k">in</span> sample1 sample2 sample3 sample4<span class="p">;</span>
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a><span class="k">do</span>
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a>
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a>  <span class="c1"># Step 2</span>
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a>  bowtie2 --minins <span class="m">200</span> --maxins <span class="m">800</span> --threads <span class="m">10</span> --sensitive <span class="se">\</span>
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a>          -x spades_assembly/bw_spades <span class="se">\</span>
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a>          -1 ../3.assembly/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R1.fastq.gz -2 ../3.assembly/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R2.fastq.gz <span class="se">\</span>
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>          -S <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam
<a href="#__codelineno-3-26" id="__codelineno-3-26" name="__codelineno-3-26"></a>
<a href="#__codelineno-3-27" id="__codelineno-3-27" name="__codelineno-3-27"></a>  <span class="c1"># Step 3</span>
<a href="#__codelineno-3-28" id="__codelineno-3-28" name="__codelineno-3-28"></a>  samtools sort -@ <span class="m">10</span> -o <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.bam <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam
<a href="#__codelineno-3-29" id="__codelineno-3-29" name="__codelineno-3-29"></a>
<a href="#__codelineno-3-30" id="__codelineno-3-30" name="__codelineno-3-30"></a><span class="k">done</span>
</code></pre></div>
<p>Now run the script using <code>sbatch</code></p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>sbatch spades_mapping.sl
</code></pre></div>
<h4 id="step-1-loop-through-the-sample-files">Step 1 - Loop through the sample files<a class="headerlink" href="#step-1-loop-through-the-sample-files" title="Permanent link">¶</a></h4>
<p>Since we just want to perform the same set of operations on each file, we can use a <em>for loop</em> to repeat each operation on a set of files. The structure of the loop, and use of variables was covered on the first day.</p>
<p>For large sets of files, it can be beneficial to use a slurm <em>array</em> to send the jobs out to different nodes and distribute the process across many independent jobs. An example of how we could modify the above script is given at the bottom of this exercise, but is not necessary for the purposes of this workshop.</p>
<h4 id="step-2-map-the-reads-using-bowtie2">Step 2 - Map the reads using <em>bowtie2</em><a class="headerlink" href="#step-2-map-the-reads-using-bowtie2" title="Permanent link">¶</a></h4>
<div class="admonition abstract">
<p class="admonition-title">This is performed using the following parameters</p>
<table>
<thead>
<tr>
<th align="left">Parameter</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>--minins</strong></td>
<td align="left">Minimum insert size, determines the minimum distance between the start of each read pair</td>
</tr>
<tr>
<td align="left"><strong>--maxins</strong></td>
<td align="left">Maximum insert size, determines the maximum distance between the start of each read pair</td>
</tr>
<tr>
<td align="left"><strong>--threads</strong></td>
<td align="left">Number of threads to use in read mapping</td>
</tr>
<tr>
<td align="left"><strong>--sensitive</strong></td>
<td align="left">Specifies where we want to be positioned in the trade-off between speed and sensitivity. See the manual for more information</td>
</tr>
<tr>
<td align="left"><strong>-x</strong></td>
<td align="left">The base name of our assembly index. Should be exactly the same as what was specified when running <strong>bowtie2-build</strong></td>
</tr>
<tr>
<td align="left"><strong>-1</strong> / <strong>-2</strong></td>
<td align="left">The forward and reverse read pairs to map to the assembly</td>
</tr>
<tr>
<td align="left"><strong>-S</strong></td>
<td align="left">Name of the output file, to be written in <em>sam</em> format</td>
</tr>
</tbody>
</table>
</div>
<h4 id="step-3-sorting-and-compressing-results">Step 3 - Sorting and compressing results<a class="headerlink" href="#step-3-sorting-and-compressing-results" title="Permanent link">¶</a></h4>
<p>The default output format for most mapping tools is the Sequence Alignment/Map (<em>sam</em>) format. This is a compact text representation of where each short read sits in the contigs. You can view this file using any text viewer, although owing to the file size <code>less</code> is a good idea.</p>
<p>Generally I wouldn't bother with this - there is a lot of information in here and unless you are looking to extract specific information from the alignment directly, this is just an intermediate file in our workflow. In order to save disk space, and prepare the file for downstream analysis we now perform two final steps:</p>
<ol>
<li>Sort the mapping information</li>
<li>Compress the <em>sam</em> file into its binary equivalent, <em>bam</em></li>
</ol>
<div class="admonition abstract">
<p class="admonition-title">Which is achieved with the following parameters</p>
<table>
<thead>
<tr>
<th align="left">Parameter</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>sort</strong></td>
<td align="left">Subcommand for <code>samtools</code> to invoke the <code>sort</code> operation</td>
</tr>
<tr>
<td align="left"><strong>-@</strong></td>
<td align="left">Number of threads to use for sorting and compressing</td>
</tr>
<tr>
<td align="left"><strong>-o</strong></td>
<td align="left">Output file name. When we specify the <em>bam</em> extension <code>samtools</code> automatically compresses the output</td>
</tr>
</tbody>
</table>
</div>
<p>Compressing the file to the <em>bam</em> format is an important step as when working with real data <em>sam</em> files can be massive and our storage capacity on NeSI is limited. It is also helpful to sort the mapping information so that reads mapped to a contig are listed in order of their start position. For example</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="c1"># Unsorted reads</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>Ref: REFERENCECONTIG
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>Map: --------ECONT--
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>Map: REFE-----------
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>Map: --FERENCECO----
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a>Map: -----------NTIG
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a>
<a href="#__codelineno-5-8" id="__codelineno-5-8" name="__codelineno-5-8"></a><span class="c1"># Sorted reads</span>
<a href="#__codelineno-5-9" id="__codelineno-5-9" name="__codelineno-5-9"></a>Ref: REFERENCECONTIG
<a href="#__codelineno-5-10" id="__codelineno-5-10" name="__codelineno-5-10"></a>Map: REFE-----------
<a href="#__codelineno-5-11" id="__codelineno-5-11" name="__codelineno-5-11"></a>Map: --FERENCECO----
<a href="#__codelineno-5-12" id="__codelineno-5-12" name="__codelineno-5-12"></a>Map: --------ECONT--
<a href="#__codelineno-5-13" id="__codelineno-5-13" name="__codelineno-5-13"></a>Map: -----------NTIG
</code></pre></div>
<p>Reads will initially be mapped in an unsorted order, as they are added to the <em>sam</em> file in more or less the same order as they are encountered in the original <em>fastQ</em> files.</p>
<p>Sorting the mapping information is an important prerequisite for performing certain downstream processes. Not every tool we use requires reads to be sorted, but it can be frustrating having to debug the instances where read sorting matters, so we typically just get it done as soon as possible and then we don't have to worry about it again.</p>
<p>In newer versions of <code>samtools</code> we can perform the sorting and compressing in a single operation (as shown in the script above). For older versions of <code>samtools</code>, you may need to use a command of the following form.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>samtools view -bS sample1.sam <span class="p">|</span> samtools sort -o sample1.bam
</code></pre></div>
<hr/>
<h3 id="optional-read-mapping-using-an-array"><em>Optional: Read mapping using an array</em><a class="headerlink" href="#optional-read-mapping-using-an-array" title="Permanent link">¶</a></h3>
<p>If you have a large number of files to process, it might be worth using a slurm array to distribute you individual mapping jobs across many separate nodes. An example script for how to perform this is given below, although it will not be covered in this workshop.</p>
<p>Open a new script using nano:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>nano spades_mapping_array.sl
</code></pre></div>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a>
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a><span class="c1">#SBATCH --job-name      spades_mapping_array</span>
<a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a><span class="c1">#SBATCH --res           SummerSchool</span>
<a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a><span class="c1">#SBATCH --time          00:20:00</span>
<a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a><span class="c1">#SBATCH --mem           20GB</span>
<a href="#__codelineno-8-8" id="__codelineno-8-8" name="__codelineno-8-8"></a><span class="c1">#SBATCH --array         0-3</span>
<a href="#__codelineno-8-9" id="__codelineno-8-9" name="__codelineno-8-9"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a href="#__codelineno-8-10" id="__codelineno-8-10" name="__codelineno-8-10"></a><span class="c1">#SBATCH --error         spades_mapping_array.%j.err</span>
<a href="#__codelineno-8-11" id="__codelineno-8-11" name="__codelineno-8-11"></a><span class="c1">#SBATCH --output        spades_mapping_array.%j.out</span>
<a href="#__codelineno-8-12" id="__codelineno-8-12" name="__codelineno-8-12"></a>
<a href="#__codelineno-8-13" id="__codelineno-8-13" name="__codelineno-8-13"></a>module purge
<a href="#__codelineno-8-14" id="__codelineno-8-14" name="__codelineno-8-14"></a>module load Bowtie2/2.4.5-GCC-11.3.0 SAMtools/1.15.1-GCC-11.3.0
<a href="#__codelineno-8-15" id="__codelineno-8-15" name="__codelineno-8-15"></a>
<a href="#__codelineno-8-16" id="__codelineno-8-16" name="__codelineno-8-16"></a><span class="nb">cd</span> /nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/5.binning/
<a href="#__codelineno-8-17" id="__codelineno-8-17" name="__codelineno-8-17"></a>
<a href="#__codelineno-8-18" id="__codelineno-8-18" name="__codelineno-8-18"></a>srun bowtie2-build spades_assembly/spades_assembly.m1000.fna spades_assembly/bw_spades
<a href="#__codelineno-8-19" id="__codelineno-8-19" name="__codelineno-8-19"></a>
<a href="#__codelineno-8-20" id="__codelineno-8-20" name="__codelineno-8-20"></a><span class="c1"># Load the sample names into a bash array</span>
<a href="#__codelineno-8-21" id="__codelineno-8-21" name="__codelineno-8-21"></a><span class="nv">samples</span><span class="o">=(</span>sample1 sample2 sample3 sample4<span class="o">)</span>
<a href="#__codelineno-8-22" id="__codelineno-8-22" name="__codelineno-8-22"></a>
<a href="#__codelineno-8-23" id="__codelineno-8-23" name="__codelineno-8-23"></a><span class="c1"># Activate the srun command, using the SLURM_ARRAY_TASK_ID variable to</span>
<a href="#__codelineno-8-24" id="__codelineno-8-24" name="__codelineno-8-24"></a><span class="c1"># identify which position in the `samples` array to use</span>
<a href="#__codelineno-8-25" id="__codelineno-8-25" name="__codelineno-8-25"></a>srun bowtie2 --minins <span class="m">200</span> --maxins <span class="m">800</span> --threads <span class="m">10</span> --sensitive -x spades_assembly/bw_spades <span class="se">\</span>
<a href="#__codelineno-8-26" id="__codelineno-8-26" name="__codelineno-8-26"></a>             -1 ../3.assembly/<span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_R1.fastq.gz <span class="se">\</span>
<a href="#__codelineno-8-27" id="__codelineno-8-27" name="__codelineno-8-27"></a>             -2 ../3.assembly/<span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>_R2.fastq.gz <span class="se">\</span>
<a href="#__codelineno-8-28" id="__codelineno-8-28" name="__codelineno-8-28"></a>             -S <span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>.sam
<a href="#__codelineno-8-29" id="__codelineno-8-29" name="__codelineno-8-29"></a>
<a href="#__codelineno-8-30" id="__codelineno-8-30" name="__codelineno-8-30"></a>srun samtools sort -o <span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>.bam <span class="si">${</span><span class="nv">samples</span><span class="p">[ </span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p"> ]</span><span class="si">}</span>.sam
</code></pre></div>
<p>submit the job to slurm</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>sbatch spades_mapping_array.sl
</code></pre></div>
</article>
</div>
</div>
<a class="md-top md-icon" data-md-component="top" hidden="" href="#">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </a>
</main>
<footer class="md-footer">
<nav aria-label="Footer" class="md-footer__inner md-grid">
<a aria-label="Previous: Evaluating the assemblies" class="md-footer__link md-footer__link--prev" href="../../day1/ex5_evaluating_assemblies/" rel="prev">
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</div>
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Previous
              </span>
              Evaluating the assemblies
            </div>
</div>
</a>
<a aria-label="Next: Introduction to binning" class="md-footer__link md-footer__link--next" href="../ex7_initial_binning/" rel="next">
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Next
              </span>
              Introduction to binning
            </div>
</div>
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"></path></svg>
</div>
</a>
</nav>
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.6c7ad80a.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom"});})</script></body>
</html>