
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day4/ex16e_data_presentation_Gene_synteny/">
      
      
        <link rel="prev" href="../ex16d_data_presentation_KEGG_pathways/">
      
      
        <link rel="next" href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
      
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.34">
    
    
      
        <title>Gene synteny - Metagenomics Summer School</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.35f28582.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#gene-synteny" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Metagenomics Summer School" class="md-header__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Summer School
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Gene synteny
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day1/ex1_bash_and_scheduler/" class="md-tabs__link">
          
  
  Day 1

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day2/ex6_initial_binning/" class="md-tabs__link">
          
  
  Day 2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-tabs__link">
          
  
  Day 3

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../ex15_gene_annotation_part3/" class="md-tabs__link">
          
  
  Day 4

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-tabs__link">
          
  
  Resources

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../supplementary/supplementary_1/" class="md-tabs__link">
          
  
  Supplementary

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Metagenomics Summer School" class="md-nav__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Metagenomics Summer School
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 1
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex1_bash_and_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction I: Shell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_1_intro_to_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction II: HPC and job scheduler
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_quality_filtering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filter raw reads by quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex3_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly I: Assembling contigs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex4_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly II: Varying the parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex5_evaluating_assemblies/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly evaluation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 2
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex6_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to binning
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex7_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Binning with multiple tools
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex8_bin_dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bin dereplication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex9_refining_bins/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Manual bin refinement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day2/ex10.1_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Identifying viral contigs in metagenomic data
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 3
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assigning taxonomy to refined prokaryotic bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11.1_phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex10.2_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virus taxonomy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex12_gene_prediction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene prediction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex13_gene_annotation_part1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation I: BLAST-like and HMM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex14_gene_annotation_part2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation II: DRAM and coverage calculation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Day 4
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex15_gene_annotation_part3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation III: DRAM distillation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16a_data_presentation_Intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to data presentation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16b_data_presentation_Coverage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coverage heatmaps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16c_OPTIONAL_data_presentation_Ordination/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ordinations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16d_data_presentation_KEGG_pathways/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    KEGG pathway maps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Gene synteny
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Gene synteny
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r" class="md-nav__link">
    <span class="md-ellipsis">
      Build a sulfur assimilation gene alignment figure to investigate gene synteny using R
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Build a sulfur assimilation gene alignment figure to investigate gene synteny using R">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-obtain-contigs-containing-genes-of-interest-r" class="md-nav__link">
    <span class="md-ellipsis">
      1. Obtain contigs containing genes of interest (R)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-check-contiguity-of-genes" class="md-nav__link">
    <span class="md-ellipsis">
      2. Check contiguity of genes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-subset-relevant-genes-and-positions" class="md-nav__link">
    <span class="md-ellipsis">
      3. Subset relevant genes and positions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-subset-relevant-scaffolds-from-bin-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      4. Subset relevant scaffolds from bin sequences
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-compare-scaffolds-using-tblastx" class="md-nav__link">
    <span class="md-ellipsis">
      5. Compare scaffolds using tBLASTx
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-plot-gene-synteny" class="md-nav__link">
    <span class="md-ellipsis">
      6. Plot gene synteny
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CAZy heatmaps
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Resources
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex8): Dereplicating data from multiple assemblies
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex9): Generating input files for VizBin from DAS_Tool curated bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11) : Viral taxonomy prediction via vContact2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/7_command_line_shortcuts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Useful command line shortcuts
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Supplementary
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Supplementary
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI Setup
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI File system, Working directory and Symlinks
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r" class="md-nav__link">
    <span class="md-ellipsis">
      Build a sulfur assimilation gene alignment figure to investigate gene synteny using R
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Build a sulfur assimilation gene alignment figure to investigate gene synteny using R">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-obtain-contigs-containing-genes-of-interest-r" class="md-nav__link">
    <span class="md-ellipsis">
      1. Obtain contigs containing genes of interest (R)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-check-contiguity-of-genes" class="md-nav__link">
    <span class="md-ellipsis">
      2. Check contiguity of genes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-subset-relevant-genes-and-positions" class="md-nav__link">
    <span class="md-ellipsis">
      3. Subset relevant genes and positions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-subset-relevant-scaffolds-from-bin-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      4. Subset relevant scaffolds from bin sequences
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-compare-scaffolds-using-tblastx" class="md-nav__link">
    <span class="md-ellipsis">
      5. Compare scaffolds using tBLASTx
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-plot-gene-synteny" class="md-nav__link">
    <span class="md-ellipsis">
      6. Plot gene synteny
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="gene-synteny">Gene synteny<a class="headerlink" href="#gene-synteny" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r">Build a sulfur assimilation gene alignment figure to investigate gene synteny using <code>R</code></a></li>
</ul>
</div>
<hr />
<h2 id="build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r">Build a sulfur assimilation gene alignment figure to investigate gene synteny using <code>R</code><a class="headerlink" href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r" title="Permanent link">&para;</a></h2>
<p>When investigating the evolution of genomes, we sometimes want to consider not only the presence/absence of genes in a genome, but also how they are arranged (e.g. in an operon). For this exercise, we are going to visualise several sulfur assimilation genes from our MAGs and compare their arrangements.</p>
<p>This exercise involves the use of commands in <code>bash</code> and <code>R</code>. Before starting, make sure you have an active terminal, then navigate to the directory <code>11.data_presentation/gene_synteny/</code>. Here, you'll find a copy of the annotations that came from DRAM (<code>dram_annotations.tsv</code>) and a directory of dereplicated bins in <code>dastool_bins/</code>. </p>
<h3 id="1-obtain-contigs-containing-genes-of-interest-r">1. Obtain contigs containing genes of interest (R)<a class="headerlink" href="#1-obtain-contigs-containing-genes-of-interest-r" title="Permanent link">&para;</a></h3>
<p>In an open RStudio session, set your working directory to <code>11.data_presentation/gene_synteny/</code> and then load the required libraries and annotation file.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-0-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-0-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-0-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-0-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-0-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-0-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-0-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-0-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-0-10">10</a></span>
<span class="normal"><a href="#__codelineno-0-11">11</a></span>
<span class="normal"><a href="#__codelineno-0-12">12</a></span>
<span class="normal"><a href="#__codelineno-0-13">13</a></span>
<span class="normal"><a href="#__codelineno-0-14">14</a></span>
<span class="normal"><a href="#__codelineno-0-15">15</a></span>
<span class="normal"><a href="#__codelineno-0-16">16</a></span>
<span class="normal"><a href="#__codelineno-0-17">17</a></span>
<span class="normal"><a href="#__codelineno-0-18">18</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1"># Working directory</span>
<a id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/11.data_presentation/gene_synteny/&quot;</span><span class="p">)</span>
<a id="__codelineno-0-3" name="__codelineno-0-3"></a>
<a id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="c1"># Tidyverse packages</span>
<a id="__codelineno-0-5" name="__codelineno-0-5"></a><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<a id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="nf">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<a id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="nf">library</span><span class="p">(</span><span class="n">purrr</span><span class="p">)</span>
<a id="__codelineno-0-8" name="__codelineno-0-8"></a><span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<a id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<a id="__codelineno-0-10" name="__codelineno-0-10"></a>
<a id="__codelineno-0-11" name="__codelineno-0-11"></a><span class="c1"># Gene synteny</span>
<a id="__codelineno-0-12" name="__codelineno-0-12"></a><span class="nf">library</span><span class="p">(</span><span class="n">genoPlotR</span><span class="p">)</span>
<a id="__codelineno-0-13" name="__codelineno-0-13"></a>
<a id="__codelineno-0-14" name="__codelineno-0-14"></a><span class="c1"># KEGG API</span>
<a id="__codelineno-0-15" name="__codelineno-0-15"></a><span class="nf">library</span><span class="p">(</span><span class="n">KEGGREST</span><span class="p">)</span>
<a id="__codelineno-0-16" name="__codelineno-0-16"></a>
<a id="__codelineno-0-17" name="__codelineno-0-17"></a><span class="c1"># Import annotations</span>
<a id="__codelineno-0-18" name="__codelineno-0-18"></a><span class="n">dram</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">&quot;dram_annotations.tsv&quot;</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<p>In order to be able to assimilate sulfate, prokaryotic genomes need to encode for the following KEGG orthologies which encode for an ATP-dependent thiosulfate/sulfate uptake system:</p>
<ul>
<li>K02045 (<em>cysA</em>)</li>
<li>K02046 (<em>cysU</em>)</li>
<li>K02047 (<em>cysW</em>)</li>
<li>K02048 (<em>cysP</em>) or K23163 (<em>sbp</em>)</li>
</ul>
<p>We'll subset the annotation file to obtain relevant rows and columns. In addtion to the bin and KO information, we require the gene position relative to the contig, strandedness, and the start and end position of the gene. </p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-1-1">1</a></span>
<span class="normal"><a href="#__codelineno-1-2">2</a></span>
<span class="normal"><a href="#__codelineno-1-3">3</a></span>
<span class="normal"><a href="#__codelineno-1-4">4</a></span>
<span class="normal"><a href="#__codelineno-1-5">5</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="n">genes_of_interest</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;K02048&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;K23163&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;K02046&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;K02047&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;K02045&quot;</span><span class="p">)</span>
<a id="__codelineno-1-2" name="__codelineno-1-2"></a>
<a id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="n">sulfate_assimilation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dram</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">ko_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes_of_interest</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-1-5" name="__codelineno-1-5"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">fasta</span><span class="p">,</span><span class="w"> </span><span class="n">scaffold</span><span class="p">,</span><span class="w"> </span><span class="n">gene_position</span><span class="p">,</span><span class="w"> </span><span class="n">ko_id</span><span class="p">,</span><span class="w"> </span><span class="n">strandedness</span><span class="p">,</span><span class="w"> </span><span class="n">start_position</span><span class="p">,</span><span class="w"> </span><span class="n">end_position</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<h3 id="2-check-contiguity-of-genes">2. Check contiguity of genes<a class="headerlink" href="#2-check-contiguity-of-genes" title="Permanent link">&para;</a></h3>
<p>We need to check that the genes are contiguous (all on the same contig) and sequential. If there gaps between genes, we will need to fill them.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-2-1">1</a></span>
<span class="normal"><a href="#__codelineno-2-2">2</a></span>
<span class="normal"><a href="#__codelineno-2-3">3</a></span>
<span class="normal"><a href="#__codelineno-2-4">4</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1"></a><span class="n">sulfate_assimilation</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-2-2" name="__codelineno-2-2"></a><span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">fasta</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-2-3" name="__codelineno-2-3"></a><span class="w">  </span><span class="nf">distinct</span><span class="p">(</span><span class="n">scaffold</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-2-4" name="__codelineno-2-4"></a><span class="w">  </span><span class="nf">tally</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;n_contigs&quot;</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a># A tibble: 4 × 2
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>  fasta n_contigs
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>  &lt;chr&gt;     &lt;int&gt;
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>1 bin_5         1
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>2 bin_7         2
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>3 bin_8         1
<a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>4 bin_9         1
</code></pre></div>
</div>
<p>We find that <code>bin_7</code> is discontiguous. We need to identify how the genes of interest are distributed across the 2 contigs.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-4-1">1</a></span>
<span class="normal"><a href="#__codelineno-4-2">2</a></span>
<span class="normal"><a href="#__codelineno-4-3">3</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="n">sulfate_assimilation</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-4-2" name="__codelineno-4-2"></a><span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">fasta</span><span class="p">,</span><span class="w"> </span><span class="n">scaffold</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-4-3" name="__codelineno-4-3"></a><span class="w">  </span><span class="nf">tally</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;n_genes&quot;</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a># A tibble: 5 × 3
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a># Groups:   fasta [4]
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>  fasta scaffold                                 n_genes
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>  &lt;chr&gt; &lt;chr&gt;                                      &lt;int&gt;
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>1 bin_5 bin_5_NODE_55_length_158395_cov_1.135272       4
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>2 bin_7 bin_7_NODE_54_length_158668_cov_0.373555       1
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>3 bin_7 bin_7_NODE_96_length_91726_cov_0.379302        4
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>4 bin_8 bin_8_NODE_12_length_355954_cov_0.516970       3
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>5 bin_9 bin_9_NODE_62_length_149231_cov_0.651774       4
</code></pre></div>
</div>
<p>Let's also check which gene is not contiguous.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-6-1">1</a></span>
<span class="normal"><a href="#__codelineno-6-2">2</a></span>
<span class="normal"><a href="#__codelineno-6-3">3</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="n">sulfate_assimilation</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-6-2" name="__codelineno-6-2"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">fasta</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;bin_7&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-6-3" name="__codelineno-6-3"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">scaffold</span><span class="p">,</span><span class="w"> </span><span class="n">gene_position</span><span class="p">,</span><span class="w"> </span><span class="n">ko_id</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a># A tibble: 5 × 3
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>  scaffold                                 gene_position ko_id 
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>  &lt;chr&gt;                                            &lt;dbl&gt; &lt;chr&gt; 
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>1 bin_7_NODE_54_length_158668_cov_0.373555           136 K02048
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>2 bin_7_NODE_96_length_91726_cov_0.379302             18 K02045
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>3 bin_7_NODE_96_length_91726_cov_0.379302             19 K02047
<a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>4 bin_7_NODE_96_length_91726_cov_0.379302             20 K02046
<a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>5 bin_7_NODE_96_length_91726_cov_0.379302             21 K23163
</code></pre></div>
</div>
<p>Looks like there is a an operon that consists of <em>sbp</em> and <em>cysAUW</em>, then a discontiguous <em>cysP</em>. </p>
<p>We also need to check for gaps between genes of interest.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="nf">select</span><span class="p">(</span><span class="n">sulfate_assimilation</span><span class="p">,</span><span class="w"> </span><span class="n">scaffold</span><span class="p">,</span><span class="w"> </span><span class="n">gene_position</span><span class="p">)</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a># A tibble: 16 × 2
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>   scaffold                                 gene_position
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>   &lt;chr&gt;                                            &lt;dbl&gt;
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a> 1 bin_5_NODE_55_length_158395_cov_1.135272           128
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a> 2 bin_5_NODE_55_length_158395_cov_1.135272           132
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a> 3 bin_5_NODE_55_length_158395_cov_1.135272           133
<a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a> 4 bin_5_NODE_55_length_158395_cov_1.135272           134
<a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a> 5 bin_7_NODE_54_length_158668_cov_0.373555           136
<a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a> 6 bin_7_NODE_96_length_91726_cov_0.379302             18
<a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a> 7 bin_7_NODE_96_length_91726_cov_0.379302             19
<a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a> 8 bin_7_NODE_96_length_91726_cov_0.379302             20
<a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a> 9 bin_7_NODE_96_length_91726_cov_0.379302             21
<a id="__codelineno-9-13" name="__codelineno-9-13" href="#__codelineno-9-13"></a>10 bin_8_NODE_12_length_355954_cov_0.516970           148
<a id="__codelineno-9-14" name="__codelineno-9-14" href="#__codelineno-9-14"></a>11 bin_8_NODE_12_length_355954_cov_0.516970           149
<a id="__codelineno-9-15" name="__codelineno-9-15" href="#__codelineno-9-15"></a>12 bin_8_NODE_12_length_355954_cov_0.516970           150
<a id="__codelineno-9-16" name="__codelineno-9-16" href="#__codelineno-9-16"></a>13 bin_9_NODE_62_length_149231_cov_0.651774            55
<a id="__codelineno-9-17" name="__codelineno-9-17" href="#__codelineno-9-17"></a>14 bin_9_NODE_62_length_149231_cov_0.651774            56
<a id="__codelineno-9-18" name="__codelineno-9-18" href="#__codelineno-9-18"></a>15 bin_9_NODE_62_length_149231_cov_0.651774            57
<a id="__codelineno-9-19" name="__codelineno-9-19" href="#__codelineno-9-19"></a>16 bin_9_NODE_62_length_149231_cov_0.651774            58
</code></pre></div>
</div>
<p>Here, we can see there is a gap in bin_5 where we need to fill 3 genes.</p>
<h3 id="3-subset-relevant-genes-and-positions">3. Subset relevant genes and positions<a class="headerlink" href="#3-subset-relevant-genes-and-positions" title="Permanent link">&para;</a></h3>
<p>Based on the explorations above, we need to:</p>
<ul>
<li>Add genes 129-131 for scaffold <code>bin_5_NODE_55_length_158395_cov_1.135272</code></li>
<li>Remove <code>bin_7_NODE_54_length_158668_cov_0.373555</code></li>
</ul>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-10-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-10-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-10-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-10-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-10-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-10-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-10-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-10-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-10-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-10-10">10</a></span>
<span class="normal"><a href="#__codelineno-10-11">11</a></span>
<span class="normal"><a href="#__codelineno-10-12">12</a></span>
<span class="normal"><a href="#__codelineno-10-13">13</a></span>
<span class="normal"><a href="#__codelineno-10-14">14</a></span>
<span class="normal"><a href="#__codelineno-10-15">15</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1"></a><span class="c1"># Add genes to bridge genes of interest for bin_5</span>
<a id="__codelineno-10-2" name="__codelineno-10-2"></a><span class="n">bin_5_additions</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dram</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-10-3" name="__codelineno-10-3"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">scaffold</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;bin_5_NODE_55_length_158395_cov_1.135272&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">gene_position</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="m">129</span><span class="o">:</span><span class="m">131</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-10-4" name="__codelineno-10-4"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">fasta</span><span class="p">,</span><span class="w"> </span><span class="n">scaffold</span><span class="p">,</span><span class="w"> </span><span class="n">gene_position</span><span class="p">,</span><span class="w"> </span><span class="n">ko_id</span><span class="p">,</span><span class="w"> </span><span class="n">strandedness</span><span class="p">,</span><span class="w"> </span><span class="n">start_position</span><span class="p">,</span><span class="w"> </span><span class="n">end_position</span><span class="p">)</span>
<a id="__codelineno-10-5" name="__codelineno-10-5"></a>
<a id="__codelineno-10-6" name="__codelineno-10-6"></a><span class="c1"># Remove extra scaffold from bin_7</span>
<a id="__codelineno-10-7" name="__codelineno-10-7"></a><span class="n">sulfate_assimilation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sulfate_assimilation</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-10-8" name="__codelineno-10-8"></a><span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">scaffold</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-10-9" name="__codelineno-10-9"></a><span class="w">  </span><span class="c1"># Our discontiguous scaffold only has one entry</span>
<a id="__codelineno-10-10" name="__codelineno-10-10"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">n</span><span class="p">()</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
<a id="__codelineno-10-11" name="__codelineno-10-11"></a>
<a id="__codelineno-10-12" name="__codelineno-10-12"></a><span class="c1"># Combine both tables</span>
<a id="__codelineno-10-13" name="__codelineno-10-13"></a><span class="n">sulfate_assimilation_coords</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bind_rows</span><span class="p">(</span><span class="n">sulfate_assimilation</span><span class="p">,</span><span class="w"> </span><span class="n">bin_5_additions</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-10-14" name="__codelineno-10-14"></a><span class="w">  </span><span class="nf">arrange</span><span class="p">(</span><span class="n">scaffold</span><span class="p">,</span><span class="w"> </span><span class="n">gene_position</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-10-15" name="__codelineno-10-15"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">gene_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">scaffold</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_position</span><span class="p">))</span>
</code></pre></div></td></tr></table></div>
</div>
<p>We will write out the results for processing in <code>bash</code>.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-11-1">1</a></span>
<span class="normal"><a href="#__codelineno-11-2">2</a></span>
<span class="normal"><a href="#__codelineno-11-3">3</a></span>
<span class="normal"><a href="#__codelineno-11-4">4</a></span>
<span class="normal"><a href="#__codelineno-11-5">5</a></span>
<span class="normal"><a href="#__codelineno-11-6">6</a></span>
<span class="normal"><a href="#__codelineno-11-7">7</a></span>
<span class="normal"><a href="#__codelineno-11-8">8</a></span>
<span class="normal"><a href="#__codelineno-11-9">9</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1"></a><span class="nf">walk</span><span class="p">(</span>
<a id="__codelineno-11-2" name="__codelineno-11-2"></a><span class="w">  </span><span class="nf">unique</span><span class="p">(</span><span class="n">sulfate_assimilation_coords</span><span class="o">$</span><span class="n">fasta</span><span class="p">),</span>
<a id="__codelineno-11-3" name="__codelineno-11-3"></a><span class="w">  </span><span class="nf">function</span><span class="p">(</span><span class="n">bin</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-11-4" name="__codelineno-11-4"></a><span class="w">    </span><span class="nf">write_tsv</span><span class="p">(</span>
<a id="__codelineno-11-5" name="__codelineno-11-5"></a><span class="w">      </span><span class="nf">filter</span><span class="p">(</span><span class="n">sulfate_assimilation_coords</span><span class="p">,</span><span class="w"> </span><span class="n">fasta</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">bin</span><span class="p">),</span>
<a id="__codelineno-11-6" name="__codelineno-11-6"></a><span class="w">      </span><span class="nf">paste0</span><span class="p">(</span><span class="n">bin</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.goi.tsv&quot;</span><span class="p">)</span>
<a id="__codelineno-11-7" name="__codelineno-11-7"></a><span class="w">    </span><span class="p">)</span>
<a id="__codelineno-11-8" name="__codelineno-11-8"></a><span class="w">  </span><span class="p">}</span>
<a id="__codelineno-11-9" name="__codelineno-11-9"></a><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<h3 id="4-subset-relevant-scaffolds-from-bin-sequences">4. Subset relevant scaffolds from bin sequences<a class="headerlink" href="#4-subset-relevant-scaffolds-from-bin-sequences" title="Permanent link">&para;</a></h3>
<p>Now, we need to return to the terminal. </p>
<div class="admonition warning">
<p class="admonition-title">Check that you're in <code>11.data_presentation/gene_synteny/</code></p>
</div>
<p>We'll use the previously generated table information to extract the scaffolds (using <code>seqtk</code>) we need from bin files.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-12-1">1</a></span>
<span class="normal"><a href="#__codelineno-12-2">2</a></span>
<span class="normal"><a href="#__codelineno-12-3">3</a></span>
<span class="normal"><a href="#__codelineno-12-4">4</a></span>
<span class="normal"><a href="#__codelineno-12-5">5</a></span>
<span class="normal"><a href="#__codelineno-12-6">6</a></span>
<span class="normal"><a href="#__codelineno-12-7">7</a></span>
<span class="normal"><a href="#__codelineno-12-8">8</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1"></a>module<span class="w"> </span>purge
<a id="__codelineno-12-2" name="__codelineno-12-2"></a>module<span class="w"> </span>load<span class="w"> </span>seqtk/1.4-GCC-11.3.0
<a id="__codelineno-12-3" name="__codelineno-12-3"></a>
<a id="__codelineno-12-4" name="__codelineno-12-4"></a><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span>*.goi.tsv<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-12-5" name="__codelineno-12-5"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="w"> </span>.goi.tsv<span class="k">)</span>
<a id="__codelineno-12-6" name="__codelineno-12-6"></a><span class="w">  </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span><span class="s1">&#39;1d&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>tmp.scaffolds
<a id="__codelineno-12-7" name="__codelineno-12-7"></a><span class="w">  </span>seqtk<span class="w"> </span>subseq<span class="w"> </span>dastool_bins/<span class="si">${</span><span class="nv">bin</span><span class="si">}</span>.fna<span class="w"> </span>tmp.scaffolds<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>.goi.fna
<a id="__codelineno-12-8" name="__codelineno-12-8"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
</div>
<h3 id="5-compare-scaffolds-using-tblastx">5. Compare scaffolds using tBLASTx<a class="headerlink" href="#5-compare-scaffolds-using-tblastx" title="Permanent link">&para;</a></h3>
<p>With our scaffolds ready, we can now perform pairwise comparisons between them. You can choose to generate complete combinations for comparisons. Here, we will only compare between bins sequentially. As we are using the <code>-subject</code> flag, the comparisons are single-threaded (we cannot change this).</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-13-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-13-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-13-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-13-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-13-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-13-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-13-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-13-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-13-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-13-10">10</a></span>
<span class="normal"><a href="#__codelineno-13-11">11</a></span>
<span class="normal"><a href="#__codelineno-13-12">12</a></span>
<span class="normal"><a href="#__codelineno-13-13">13</a></span>
<span class="normal"><a href="#__codelineno-13-14">14</a></span>
<span class="normal"><a href="#__codelineno-13-15">15</a></span>
<span class="normal"><a href="#__codelineno-13-16">16</a></span>
<span class="normal"><a href="#__codelineno-13-17">17</a></span>
<span class="normal"><a href="#__codelineno-13-18">18</a></span>
<span class="normal"><a href="#__codelineno-13-19">19</a></span>
<span class="normal"><a href="#__codelineno-13-20">20</a></span>
<span class="normal"><a href="#__codelineno-13-21">21</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1"></a>module<span class="w"> </span>purge
<a id="__codelineno-13-2" name="__codelineno-13-2"></a>module<span class="w"> </span>load<span class="w"> </span>BLAST/2.16.0-GCC-12.3.0
<a id="__codelineno-13-3" name="__codelineno-13-3"></a>
<a id="__codelineno-13-4" name="__codelineno-13-4"></a><span class="nv">bin_array</span><span class="o">=(</span><span class="k">$(</span>ls<span class="w"> </span>*.goi.fna<span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/.goi.fna//g&#39;</span><span class="k">)</span><span class="o">)</span>
<a id="__codelineno-13-5" name="__codelineno-13-5"></a>
<a id="__codelineno-13-6" name="__codelineno-13-6"></a><span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<a id="__codelineno-13-7" name="__codelineno-13-7"></a><span class="nv">max_i</span><span class="o">=</span><span class="k">$((</span><span class="w"> </span><span class="si">${#</span><span class="nv">bin_array</span><span class="p">[@]</span><span class="si">}</span><span class="o">-</span><span class="m">1</span><span class="w"> </span><span class="k">))</span>
<a id="__codelineno-13-8" name="__codelineno-13-8"></a>
<a id="__codelineno-13-9" name="__codelineno-13-9"></a><span class="k">while</span><span class="w"> </span><span class="o">[</span><span class="w"> </span><span class="nv">$i</span><span class="w"> </span>-lt<span class="w"> </span><span class="nv">$max_i</span><span class="w"> </span><span class="o">]</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-13-10" name="__codelineno-13-10"></a><span class="w">  </span><span class="nv">qbin</span><span class="o">=</span><span class="si">${</span><span class="nv">bin_array</span><span class="p">[</span><span class="nv">$i</span><span class="p">]</span><span class="si">}</span><span class="w"> </span><span class="c1"># Current bin as query</span>
<a id="__codelineno-13-11" name="__codelineno-13-11"></a><span class="w">  </span><span class="nv">sbin</span><span class="o">=</span><span class="si">${</span><span class="nv">bin_array</span><span class="p">[</span><span class="k">$((</span><span class="nv">$i</span><span class="o">+</span><span class="m">1</span><span class="k">))</span><span class="p">]</span><span class="si">}</span><span class="w"> </span><span class="c1"># Next bin as subject </span>
<a id="__codelineno-13-12" name="__codelineno-13-12"></a>
<a id="__codelineno-13-13" name="__codelineno-13-13"></a><span class="w">  </span><span class="nb">printf</span><span class="w"> </span><span class="s2">&quot;Comparing %s against %s\n&quot;</span><span class="w"> </span><span class="si">${</span><span class="nv">qbin</span><span class="si">}</span><span class="w"> </span><span class="si">${</span><span class="nv">sbin</span><span class="si">}</span>
<a id="__codelineno-13-14" name="__codelineno-13-14"></a>
<a id="__codelineno-13-15" name="__codelineno-13-15"></a><span class="w">  </span>tblastx<span class="w"> </span>-query<span class="w"> </span><span class="si">${</span><span class="nv">qbin</span><span class="si">}</span>.goi.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-13-16" name="__codelineno-13-16"></a><span class="w">          </span>-subject<span class="w"> </span><span class="si">${</span><span class="nv">sbin</span><span class="si">}</span>.goi.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-13-17" name="__codelineno-13-17"></a><span class="w">          </span>-out<span class="w"> </span>blast_<span class="si">${</span><span class="nv">qbin</span><span class="si">}</span>_<span class="si">${</span><span class="nv">sbin</span><span class="si">}</span>.txt<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-13-18" name="__codelineno-13-18"></a><span class="w">          </span>-outfmt<span class="w"> </span><span class="s2">&quot;7 std ppos frames&quot;</span>
<a id="__codelineno-13-19" name="__codelineno-13-19"></a>
<a id="__codelineno-13-20" name="__codelineno-13-20"></a><span class="w">  </span><span class="o">((</span>i++<span class="o">))</span><span class="w"> </span><span class="c1"># Increment i after each comparison</span>
<a id="__codelineno-13-21" name="__codelineno-13-21"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
</div>
<details class="hint">
<summary><code>-outfmt 7</code></summary>
<p>You may be familiar with BLAST output format 6 as the tabular output from BLAST. Here, format 7 is the one with a commented header lines. This allows <code>genoPlotR</code> downstream to parse the columns appropriately. The proceeding arguments specifies a standard output <code>std</code> with additional columns for percentage of positive matches <code>ppos</code> and the translated frames for query/subject <code>frames</code>.</p>
</details>
<p>Now we have all the necessary files. We will return to the RStudio session for plotting.</p>
<h3 id="6-plot-gene-synteny">6. Plot gene synteny<a class="headerlink" href="#6-plot-gene-synteny" title="Permanent link">&para;</a></h3>
<p>We first need to import all the relevant per bin files.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-14-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-14-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-14-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-14-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-14-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-14-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-14-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-14-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-14-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-14-10">10</a></span>
<span class="normal"><a href="#__codelineno-14-11">11</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1"></a><span class="c1"># Gene coordinates</span>
<a id="__codelineno-14-2" name="__codelineno-14-2"></a><span class="n">goi_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.*.goi.tsv&quot;</span><span class="p">)</span>
<a id="__codelineno-14-3" name="__codelineno-14-3"></a><span class="n">goi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">goi_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-14-4" name="__codelineno-14-4"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">nm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">goi_files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\\..*&quot;</span><span class="p">))</span>
<a id="__codelineno-14-5" name="__codelineno-14-5"></a>
<a id="__codelineno-14-6" name="__codelineno-14-6"></a><span class="c1"># BLAST results</span>
<a id="__codelineno-14-7" name="__codelineno-14-7"></a><span class="n">blast_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blast.*&quot;</span><span class="p">)</span>
<a id="__codelineno-14-8" name="__codelineno-14-8"></a><span class="n">blast</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">blast_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-14-9" name="__codelineno-14-9"></a><span class="w">  </span><span class="nf">read_comparison_from_blast</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<a id="__codelineno-14-10" name="__codelineno-14-10"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-14-11" name="__codelineno-14-11"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">nm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">blast_files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.txt&quot;</span><span class="p">))</span>
</code></pre></div></td></tr></table></div>
</div>
<p>We will also create a data frame that has relevant information about the gene. In this case, we are using the gene symbol.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-15-1">1</a></span>
<span class="normal"><a href="#__codelineno-15-2">2</a></span>
<span class="normal"><a href="#__codelineno-15-3">3</a></span>
<span class="normal"><a href="#__codelineno-15-4">4</a></span>
<span class="normal"><a href="#__codelineno-15-5">5</a></span>
<span class="normal"><a href="#__codelineno-15-6">6</a></span>
<span class="normal"><a href="#__codelineno-15-7">7</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1"></a><span class="n">ko</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="nf">bind_rows</span><span class="p">(</span><span class="n">goi</span><span class="p">)</span><span class="o">$</span><span class="n">ko_id</span><span class="p">)</span>
<a id="__codelineno-15-2" name="__codelineno-15-2"></a><span class="n">kegg_records</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">keggGet</span><span class="p">(</span><span class="n">ko</span><span class="p">[</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">ko</span><span class="p">)])</span>
<a id="__codelineno-15-3" name="__codelineno-15-3"></a><span class="n">ko_annot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">kegg_records</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">entry</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-15-4" name="__codelineno-15-4"></a><span class="w">    </span><span class="nf">data.frame</span><span class="p">(</span><span class="s">&quot;ko_id&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">entry</span><span class="p">[[</span><span class="s">&quot;ENTRY&quot;</span><span class="p">]],</span><span class="w"> </span>
<a id="__codelineno-15-5" name="__codelineno-15-5"></a><span class="w">               </span><span class="s">&quot;gene_symbol&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">entry</span><span class="p">[[</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">]])</span>
<a id="__codelineno-15-6" name="__codelineno-15-6"></a><span class="w">  </span><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-15-7" name="__codelineno-15-7"></a><span class="w">  </span><span class="nf">bind_rows</span><span class="p">()</span>
</code></pre></div></td></tr></table></div>
</div>
<p>We then create tracks of DNA segments and annotations that can be parsed by <code>genoPlotR</code>.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-16-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-16-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-16-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-16-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-16-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-16-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-16-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-16-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-16-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-16-10">10</a></span>
<span class="normal"><a href="#__codelineno-16-11">11</a></span>
<span class="normal"><a href="#__codelineno-16-12">12</a></span>
<span class="normal"><a href="#__codelineno-16-13">13</a></span>
<span class="normal"><a href="#__codelineno-16-14">14</a></span>
<span class="normal"><a href="#__codelineno-16-15">15</a></span>
<span class="normal"><a href="#__codelineno-16-16">16</a></span>
<span class="normal"><a href="#__codelineno-16-17">17</a></span>
<span class="normal"><a href="#__codelineno-16-18">18</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1"></a><span class="c1"># DNA segments</span>
<a id="__codelineno-16-2" name="__codelineno-16-2"></a><span class="n">ds_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">goi</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-16-3" name="__codelineno-16-3"></a><span class="w">  </span><span class="n">coords</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-16-4" name="__codelineno-16-4"></a><span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">start_position</span><span class="p">,</span><span class="w"> </span><span class="n">end_position</span><span class="p">,</span><span class="w"> </span><span class="n">strandedness</span><span class="p">)</span>
<a id="__codelineno-16-5" name="__codelineno-16-5"></a><span class="w">  </span><span class="nf">colnames</span><span class="p">(</span><span class="n">coords</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;name&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;start&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;end&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;strand&quot;</span><span class="p">)</span>
<a id="__codelineno-16-6" name="__codelineno-16-6"></a><span class="w">  </span><span class="nf">dna_seg</span><span class="p">(</span><span class="n">coords</span><span class="p">)</span>
<a id="__codelineno-16-7" name="__codelineno-16-7"></a><span class="p">})</span>
<a id="__codelineno-16-8" name="__codelineno-16-8"></a>
<a id="__codelineno-16-9" name="__codelineno-16-9"></a><span class="c1"># Annotations that include gene symbols</span>
<a id="__codelineno-16-10" name="__codelineno-16-10"></a><span class="n">annot_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">goi</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-16-11" name="__codelineno-16-11"></a><span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">ko_annot</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ko_id&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-16-12" name="__codelineno-16-12"></a><span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">start_position</span><span class="p">,</span><span class="w"> </span><span class="n">end_position</span><span class="p">,</span><span class="w"> </span><span class="n">strandedness</span><span class="p">,</span><span class="w"> </span><span class="n">ko_id</span><span class="p">,</span><span class="w"> </span><span class="n">gene_symbol</span><span class="p">)</span>
<a id="__codelineno-16-13" name="__codelineno-16-13"></a><span class="w">  </span><span class="nf">annotation</span><span class="p">(</span>
<a id="__codelineno-16-14" name="__codelineno-16-14"></a><span class="w">    </span><span class="n">x1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">start_position</span><span class="p">,</span>
<a id="__codelineno-16-15" name="__codelineno-16-15"></a><span class="w">    </span><span class="n">x2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">end_position</span><span class="p">,</span>
<a id="__codelineno-16-16" name="__codelineno-16-16"></a><span class="w">    </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">gene_symbol</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">gene_symbol</span><span class="p">)</span>
<a id="__codelineno-16-17" name="__codelineno-16-17"></a><span class="w">  </span><span class="p">)</span>
<a id="__codelineno-16-18" name="__codelineno-16-18"></a><span class="p">})</span>
</code></pre></div></td></tr></table></div>
</div>
<p>Finally, we can plot!</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-17-1">1</a></span>
<span class="normal"><a href="#__codelineno-17-2">2</a></span>
<span class="normal"><a href="#__codelineno-17-3">3</a></span>
<span class="normal"><a href="#__codelineno-17-4">4</a></span>
<span class="normal"><a href="#__codelineno-17-5">5</a></span>
<span class="normal"><a href="#__codelineno-17-6">6</a></span>
<span class="normal"><a href="#__codelineno-17-7">7</a></span>
<span class="normal"><a href="#__codelineno-17-8">8</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="nf">plot_gene_map</span><span class="p">(</span>
<a id="__codelineno-17-2" name="__codelineno-17-2"></a><span class="w">  </span><span class="n">dna_segs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ds_list</span><span class="p">,</span>
<a id="__codelineno-17-3" name="__codelineno-17-3"></a><span class="w">  </span><span class="n">comparisons</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast</span><span class="p">,</span>
<a id="__codelineno-17-4" name="__codelineno-17-4"></a><span class="w">  </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_list</span><span class="p">,</span>
<a id="__codelineno-17-5" name="__codelineno-17-5"></a><span class="w">  </span><span class="n">dna_seg_labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">ds_list</span><span class="p">),</span>
<a id="__codelineno-17-6" name="__codelineno-17-6"></a><span class="w">  </span><span class="n">dna_seg_scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<a id="__codelineno-17-7" name="__codelineno-17-7"></a><span class="w">  </span><span class="n">gene_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;arrows&quot;</span>
<a id="__codelineno-17-8" name="__codelineno-17-8"></a><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<p><a class="glightbox" href="../../figures/day4_synteny.01.synteny_all.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/day4_synteny.01.synteny_all.png" /></a></p>
<p>Seems a little messy, let's clean it up a little by only keeping those with the best matches (i.e., minimum E-value).</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-18-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-18-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-18-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-18-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-18-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-18-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-18-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-18-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-18-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-18-10">10</a></span>
<span class="normal"><a href="#__codelineno-18-11">11</a></span>
<span class="normal"><a href="#__codelineno-18-12">12</a></span>
<span class="normal"><a href="#__codelineno-18-13">13</a></span>
<span class="normal"><a href="#__codelineno-18-14">14</a></span>
<span class="normal"><a href="#__codelineno-18-15">15</a></span>
<span class="normal"><a href="#__codelineno-18-16">16</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1"></a><span class="n">filt_blast</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">blast</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-18-2" name="__codelineno-18-2"></a><span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">name1</span><span class="p">,</span><span class="w"> </span><span class="n">name2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-18-3" name="__codelineno-18-3"></a><span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">e_value</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">e_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-18-4" name="__codelineno-18-4"></a><span class="w">    </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-18-5" name="__codelineno-18-5"></a><span class="w">    </span><span class="nf">as.comparison</span><span class="p">()</span>
<a id="__codelineno-18-6" name="__codelineno-18-6"></a><span class="p">})</span>
<a id="__codelineno-18-7" name="__codelineno-18-7"></a>
<a id="__codelineno-18-8" name="__codelineno-18-8"></a><span class="c1"># Re-plot</span>
<a id="__codelineno-18-9" name="__codelineno-18-9"></a><span class="nf">plot_gene_map</span><span class="p">(</span>
<a id="__codelineno-18-10" name="__codelineno-18-10"></a><span class="w">  </span><span class="n">dna_segs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ds_list</span><span class="p">,</span>
<a id="__codelineno-18-11" name="__codelineno-18-11"></a><span class="w">  </span><span class="n">comparisons</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">filt_blast</span><span class="p">,</span>
<a id="__codelineno-18-12" name="__codelineno-18-12"></a><span class="w">  </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_list</span><span class="p">,</span>
<a id="__codelineno-18-13" name="__codelineno-18-13"></a><span class="w">  </span><span class="n">dna_seg_labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">ds_list</span><span class="p">),</span>
<a id="__codelineno-18-14" name="__codelineno-18-14"></a><span class="w">  </span><span class="n">dna_seg_scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<a id="__codelineno-18-15" name="__codelineno-18-15"></a><span class="w">  </span><span class="n">gene_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;arrows&quot;</span>
<a id="__codelineno-18-16" name="__codelineno-18-16"></a><span class="p">)</span>
</code></pre></div></td></tr></table></div>
</div>
<p><a class="glightbox" href="../../figures/day4_synteny.03.synteny_final.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="image" src="../../figures/day4_synteny.03.synteny_final.png" /></a></p>
<p>Much better! To further improve the plot, you might want to rearrange the tBLASTx comparisons and perhaps add additional information to the un-annotated genes based on hits to other databases (e.g., Pfam, TIGRFam, UniProt, etc.).</p>
<!--

You have been provided with a copy of the `prodigal` gene predictions for each of the bins (`.faa` files), an annotation output table using multiple databases (`.aa` files), a small table of the annotation of some key genes of interest (`cys.txt` files), and tBLASTx output (`blast*.txt`) comparing the genes of interest from these organisms. The annotation files were created by manually searching the annotations obtained in the previous exercises.

!!! note "Note"

    Refer to [this appendix](../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs.md) for detailed information on how to generate input data.

### Part 1 - Parsing files in *bash*

We will be performing this exercise in two stages. Firstly, in `bash`, we will use `cut` and `tail` to pull out the genes of interest listed in the `*cys.txt` files from the `prodigal` files. The gene names will then be used to create a table of gene coordinates from the `prodigal` output using `grep`, `cut`, and `sed`.

For these `bash` steps, we will need to return to our logged in NeSI terminal. Switch over to a NeSI `Jupyter hub` terminal or log in to a fresh session in a new terminal. 

Navigate to the `11.data_presentation/gene_synteny/` folder, and then perform the `cut` and `tail` steps outlined above.

!!! terminal "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/11.data_presentation/gene_synteny/
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a>
<a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>*_cys.txt<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>_cys.txt<span class="k">)</span>
<a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a><span class="w">  </span>cut<span class="w"> </span>-f1<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tail<span class="w"> </span>-n+2<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.genes
<a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a><span class="k">done</span>
</code></pre></div>

We now have three new files, ending with the `.genes` suffix which are simply a list of the genes that we wish to extract from the `prodigal` files. We can then use each of these files as input for `grep` to pull out the FASTA entries that correspond to these genes.

!!! terminal "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="k">for</span><span class="w"> </span>gene_list<span class="w"> </span><span class="k">in</span><span class="w"> </span>*.genes<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">gene_list</span><span class="si">}</span><span class="w"> </span>_cys.genes<span class="k">)</span>
<a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a><span class="w">  </span>grep<span class="w"> </span>-f<span class="w"> </span><span class="si">${</span><span class="nv">gene_list</span><span class="si">}</span><span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>.filtered.genes.faa<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-4" name="__codelineno-20-4" href="#__codelineno-20-4"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/ # /\t/g&#39;</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/&gt;//g&#39;</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-20-5" name="__codelineno-20-5" href="#__codelineno-20-5"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>-4<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.coords
<a id="__codelineno-20-6" name="__codelineno-20-6" href="#__codelineno-20-6"></a><span class="k">done</span>
</code></pre></div>

As previously, we will quickly go through the steps of this command:

* `grep -f ${gene_list} ${bin}.filtered.genes.faa` searches the `prodigal` FASTA amino acid sequence output for headers that match those in our `.genes` file
* `sed -e 's/ # /\t/g' -e 's/>//g'` replaces the default `prodigal` header line delimiter " # " with a tab, then removes `>` from the header.
* `cut -f 1-4` selects the first 4 columns delimited by a tab (default for `cut`).

Check the content of the `.coords` files now. You should see something like the following:

!!! terminal "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a>cat<span class="w"> </span>bin_3_cys.coords
</code></pre></div>

!!! circle-check "Terminal output"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a>bin_3_NODE_53_length_158395_cov_1.135272_128    135928  136935  1
<a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a>bin_3_NODE_53_length_158395_cov_1.135272_129    136994  137299  1
<a id="__codelineno-22-3" name="__codelineno-22-3" href="#__codelineno-22-3"></a>bin_3_NODE_53_length_158395_cov_1.135272_130    137411  138322  1
<a id="__codelineno-22-4" name="__codelineno-22-4" href="#__codelineno-22-4"></a>bin_3_NODE_53_length_158395_cov_1.135272_131    138413  139201  1
<a id="__codelineno-22-5" name="__codelineno-22-5" href="#__codelineno-22-5"></a>bin_3_NODE_53_length_158395_cov_1.135272_132    139267  140100  1
<a id="__codelineno-22-6" name="__codelineno-22-6" href="#__codelineno-22-6"></a>bin_3_NODE_53_length_158395_cov_1.135272_133    140110  140988  1
<a id="__codelineno-22-7" name="__codelineno-22-7" href="#__codelineno-22-7"></a>bin_3_NODE_53_length_158395_cov_1.135272_134    140985  142073  1
</code></pre></div>

If you recall from the previous exercise on gene prediction, we have taken the first four entries from each line of the `prodigal` output, which consists of:

!!! quote ""
    1. Gene name, written as [CONTIG]\_[GENE]
    2. Start position of the gene
    3. Stop position of the gene
    4. Orientation of the gene

We will now use these tables, together with the annotation tables to create the gene synteny view in `R`.

### Part 2 - Producing the figure in `R`

First, move back to the [Jupyter hub](https://jupyter.nesi.org.nz/hub/login) pane and start an `RStudio` session

#### 2.1 Prepare environment

Similar to previous sessions, we will begin by preparing the environment by setting our working directory and loading required libraries.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="c1"># Set working directory</span>
<a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/11.data_presentation/gene_synteny/&#39;</span><span class="p">)</span>
<a id="__codelineno-23-3" name="__codelineno-23-3" href="#__codelineno-23-3"></a>
<a id="__codelineno-23-4" name="__codelineno-23-4" href="#__codelineno-23-4"></a><span class="c1"># Load libraries</span>
<a id="__codelineno-23-5" name="__codelineno-23-5" href="#__codelineno-23-5"></a><span class="c1"># Tidyverse packages</span>
<a id="__codelineno-23-6" name="__codelineno-23-6" href="#__codelineno-23-6"></a><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<a id="__codelineno-23-7" name="__codelineno-23-7" href="#__codelineno-23-7"></a><span class="nf">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<a id="__codelineno-23-8" name="__codelineno-23-8" href="#__codelineno-23-8"></a><span class="nf">library</span><span class="p">(</span><span class="n">purrr</span><span class="p">)</span>
<a id="__codelineno-23-9" name="__codelineno-23-9" href="#__codelineno-23-9"></a><span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<a id="__codelineno-23-10" name="__codelineno-23-10" href="#__codelineno-23-10"></a><span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<a id="__codelineno-23-11" name="__codelineno-23-11" href="#__codelineno-23-11"></a>
<a id="__codelineno-23-12" name="__codelineno-23-12" href="#__codelineno-23-12"></a><span class="c1"># Gene synteny</span>
<a id="__codelineno-23-13" name="__codelineno-23-13" href="#__codelineno-23-13"></a><span class="nf">library</span><span class="p">(</span><span class="n">genoPlotR</span><span class="p">)</span>
</code></pre></div>

#### 2.2 Import files

Using genoPlotR requires files with 3 different types of information:

* Gene coordinates (from `prodigal`)
* Gene annotations (from our annotation workflows)
* Contig comparisons (from tBLASTx output)

First, we will import the `.coords` files, and set column names to the files.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="c1"># Gene coordinates</span>
<a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a><span class="n">coords_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.coords&quot;</span><span class="p">)</span>
<a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a><span class="n">coords_header</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;name&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;start&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;end&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;strand&quot;</span><span class="p">)</span>
<a id="__codelineno-24-4" name="__codelineno-24-4" href="#__codelineno-24-4"></a><span class="n">coords_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">coords_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-24-5" name="__codelineno-24-5" href="#__codelineno-24-5"></a><span class="w">  </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">coords_header</span><span class="p">)</span>
<a id="__codelineno-24-6" name="__codelineno-24-6" href="#__codelineno-24-6"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-24-7" name="__codelineno-24-7" href="#__codelineno-24-7"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">coords_files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cys.&quot;</span><span class="p">))</span>
</code></pre></div>

Take a look at the content of each of these data.frames, by entering their names into the terminal. You should notice that the coordinates occur at quite different positions between the genomes. If we were looking at complete genomes, this would indicate their position relative to the *origin of replication*, but as these are unclosed genomes obtained from MAGs, they reflect the coordinates upon their particular *contig*.

Lets continue to import the other files

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="c1"># Gene annotations</span>
<a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a><span class="n">annot_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^bin_.*.txt&quot;</span><span class="p">)</span>
<a id="__codelineno-25-3" name="__codelineno-25-3" href="#__codelineno-25-3"></a><span class="n">annot_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">annot_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-25-4" name="__codelineno-25-4" href="#__codelineno-25-4"></a><span class="w">  </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<a id="__codelineno-25-5" name="__codelineno-25-5" href="#__codelineno-25-5"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-25-6" name="__codelineno-25-6" href="#__codelineno-25-6"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">str_replace</span><span class="p">(</span><span class="n">annot_files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cys.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;annot&quot;</span><span class="p">))</span>
<a id="__codelineno-25-7" name="__codelineno-25-7" href="#__codelineno-25-7"></a>
<a id="__codelineno-25-8" name="__codelineno-25-8" href="#__codelineno-25-8"></a><span class="c1"># BLAST outputs</span>
<a id="__codelineno-25-9" name="__codelineno-25-9" href="#__codelineno-25-9"></a><span class="n">blast_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^blast_.*.txt&quot;</span><span class="p">)</span>
<a id="__codelineno-25-10" name="__codelineno-25-10" href="#__codelineno-25-10"></a><span class="n">blast_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">blast_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-25-11" name="__codelineno-25-11" href="#__codelineno-25-11"></a><span class="w">  </span><span class="nf">read_comparison_from_blast</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<a id="__codelineno-25-12" name="__codelineno-25-12" href="#__codelineno-25-12"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-25-13" name="__codelineno-25-13" href="#__codelineno-25-13"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">blast_files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.txt&quot;</span><span class="p">))</span>
</code></pre></div>

Notice that the file reading function for BLAST files are different (`read_comparison_from_blast()` versus `read_tsv()`). This is a function specific to `genoPlotR` for parsing BLAST outputs. If you have some custom comparison pipeline, `genoPlotR` can also read tab-delimited files via `read_comparison_from_tab()`

#### 2.3 Wrangle data

We now need to do the following:

* Parse coordinate data into a format genoPlotR can use
* Tidy up annotation tables so they will show the correct labels

We start by converting our coordinate data into a `dna_seg` (DNA segment) data class, then check what is in one of the objects in the list.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a><span class="c1"># Transform coordinate tables into &quot;dna_seg&quot; objects</span>
<a id="__codelineno-26-2" name="__codelineno-26-2" href="#__codelineno-26-2"></a><span class="n">ds_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">coords_list</span><span class="p">,</span><span class="w"> </span><span class="n">dna_seg</span><span class="p">)</span>
<a id="__codelineno-26-3" name="__codelineno-26-3" href="#__codelineno-26-3"></a>
<a id="__codelineno-26-4" name="__codelineno-26-4" href="#__codelineno-26-4"></a><span class="n">ds_list</span><span class="o">$</span><span class="n">bin_5_coords</span>
</code></pre></div>

|                                       name | start |   end | strand |  col | fill | lty | lwd | pch | cex | gene_type |
|-------------------------------------------:|------:|------:|-------:|-----:|-----:|----:|----:|----:|----:|----------:|
| bin_5_NODE_95_length_91726_cov_0.379302_67 | 71608 | 72606 |      1 | blue | blue |   1 |   1 |   8 |   1 |    arrows |
| bin_5_NODE_95_length_91726_cov_0.379302_68 | 72768 | 73586 |      1 | blue | blue |   1 |   1 |   8 |   1 |    arrows |
| bin_5_NODE_95_length_91726_cov_0.379302_69 | 73597 | 74466 |      1 | blue | blue |   1 |   1 |   8 |   1 |    arrows |
| bin_5_NODE_95_length_91726_cov_0.379302_70 | 74470 | 75459 |      1 | blue | blue |   1 |   1 |   8 |   1 |    arrows |

Now we tidy our annotations table and create a joint coordinate-annotation table. But first, lets take a look at our annotation tables to get a feel for what needs to change.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a><span class="n">annot_list</span>
</code></pre></div>

!!! circle-check "Console output"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a>$bin_3_annot
<a id="__codelineno-28-2" name="__codelineno-28-2" href="#__codelineno-28-2"></a># A tibble: 7 × 3
<a id="__codelineno-28-3" name="__codelineno-28-3" href="#__codelineno-28-3"></a>  `Query gene`                                 KO                                      Annotation                                 
<a id="__codelineno-28-4" name="__codelineno-28-4" href="#__codelineno-28-4"></a>  &lt;chr&gt;                                        &lt;chr&gt;                                   &lt;chr&gt;                                      
<a id="__codelineno-28-5" name="__codelineno-28-5" href="#__codelineno-28-5"></a>1 bin_3_NODE_53_length_158395_cov_1.135272_128 K02048                                  sbp1; Prokaryotic sulfate-/thiosulfate-bin…
<a id="__codelineno-28-6" name="__codelineno-28-6" href="#__codelineno-28-6"></a>2 bin_3_NODE_53_length_158395_cov_1.135272_129 possible predicted diverged CheY-domain possible predicted diverged CheY-domain    
<a id="__codelineno-28-7" name="__codelineno-28-7" href="#__codelineno-28-7"></a>3 bin_3_NODE_53_length_158395_cov_1.135272_130 Domain of unknown function 2            Domain of unknown function 2               
<a id="__codelineno-28-8" name="__codelineno-28-8" href="#__codelineno-28-8"></a>4 bin_3_NODE_53_length_158395_cov_1.135272_131 Domain of unknown function 2            Domain of unknown function 2               
<a id="__codelineno-28-9" name="__codelineno-28-9" href="#__codelineno-28-9"></a>5 bin_3_NODE_53_length_158395_cov_1.135272_132 K02046                                  cysU; cysU; sulfate transport ABC transpor…
<a id="__codelineno-28-10" name="__codelineno-28-10" href="#__codelineno-28-10"></a>6 bin_3_NODE_53_length_158395_cov_1.135272_133 K02047                                  cysW; cysW; sulfate transport ABC transpor…
<a id="__codelineno-28-11" name="__codelineno-28-11" href="#__codelineno-28-11"></a>7 bin_3_NODE_53_length_158395_cov_1.135272_134 K02045                                  cysA; cysA; sulfate transport ATP-binding …
<a id="__codelineno-28-12" name="__codelineno-28-12" href="#__codelineno-28-12"></a>
<a id="__codelineno-28-13" name="__codelineno-28-13" href="#__codelineno-28-13"></a>$bin_5_annot
<a id="__codelineno-28-14" name="__codelineno-28-14" href="#__codelineno-28-14"></a># A tibble: 4 × 3
<a id="__codelineno-28-15" name="__codelineno-28-15" href="#__codelineno-28-15"></a>  `Query gene`                               KO     Annotation                                                        
<a id="__codelineno-28-16" name="__codelineno-28-16" href="#__codelineno-28-16"></a>  &lt;chr&gt;                                      &lt;chr&gt;  &lt;chr&gt;                                                             
<a id="__codelineno-28-17" name="__codelineno-28-17" href="#__codelineno-28-17"></a>1 bin_5_NODE_95_length_91726_cov_0.379302_67 K02048 sbp; sulfate-binding protein                                      
<a id="__codelineno-28-18" name="__codelineno-28-18" href="#__codelineno-28-18"></a>2 bin_5_NODE_95_length_91726_cov_0.379302_68 K02046 cysT; sulfate transporter CysT                                    
<a id="__codelineno-28-19" name="__codelineno-28-19" href="#__codelineno-28-19"></a>3 bin_5_NODE_95_length_91726_cov_0.379302_69 K02047 cysW; sulfate transporter CysW                                    
<a id="__codelineno-28-20" name="__codelineno-28-20" href="#__codelineno-28-20"></a>4 bin_5_NODE_95_length_91726_cov_0.379302_70 K02045 cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA
<a id="__codelineno-28-21" name="__codelineno-28-21" href="#__codelineno-28-21"></a>
<a id="__codelineno-28-22" name="__codelineno-28-22" href="#__codelineno-28-22"></a>$bin_8_annot
<a id="__codelineno-28-23" name="__codelineno-28-23" href="#__codelineno-28-23"></a># A tibble: 4 × 3
<a id="__codelineno-28-24" name="__codelineno-28-24" href="#__codelineno-28-24"></a>  `Query gene`                                KO     Annotation                                    
<a id="__codelineno-28-25" name="__codelineno-28-25" href="#__codelineno-28-25"></a>  &lt;chr&gt;                                       &lt;chr&gt;  &lt;chr&gt;                                         
<a id="__codelineno-28-26" name="__codelineno-28-26" href="#__codelineno-28-26"></a>1 bin_8_NODE_60_length_149231_cov_0.651774_55 K02048 Thiosulphate-binding protein                  
<a id="__codelineno-28-27" name="__codelineno-28-27" href="#__codelineno-28-27"></a>2 bin_8_NODE_60_length_149231_cov_0.651774_56 K02046 sulfate ABC transporter permease subunit CysT 
<a id="__codelineno-28-28" name="__codelineno-28-28" href="#__codelineno-28-28"></a>3 bin_8_NODE_60_length_149231_cov_0.651774_57 K02047 Sulfate ABC transporter, permease protein CysW
<a id="__codelineno-28-29" name="__codelineno-28-29" href="#__codelineno-28-29"></a>4 bin_8_NODE_60_length_149231_cov_0.651774_58 K02045 sulphate transport system permease protein 1  
<a id="__codelineno-28-30" name="__codelineno-28-30" href="#__codelineno-28-30"></a>
<a id="__codelineno-28-31" name="__codelineno-28-31" href="#__codelineno-28-31"></a>$bin_9_annot
<a id="__codelineno-28-32" name="__codelineno-28-32" href="#__codelineno-28-32"></a># A tibble: 4 × 3
<a id="__codelineno-28-33" name="__codelineno-28-33" href="#__codelineno-28-33"></a>  `Query gene`                                 KO     Annotation                                           
<a id="__codelineno-28-34" name="__codelineno-28-34" href="#__codelineno-28-34"></a>  &lt;chr&gt;                                        &lt;chr&gt;  &lt;chr&gt;                                                
<a id="__codelineno-28-35" name="__codelineno-28-35" href="#__codelineno-28-35"></a>1 bin_9_NODE_12_length_355673_cov_0.516990_147 K02048 thiosulfate ABC transporter substrate-binding protein
<a id="__codelineno-28-36" name="__codelineno-28-36" href="#__codelineno-28-36"></a>2 bin_9_NODE_12_length_355673_cov_0.516990_148 K02046 sulfate ABC transporter permease                     
<a id="__codelineno-28-37" name="__codelineno-28-37" href="#__codelineno-28-37"></a>3 bin_9_NODE_12_length_355673_cov_0.516990_149 K02047 sulfate ABC transporter permease                     
<a id="__codelineno-28-38" name="__codelineno-28-38" href="#__codelineno-28-38"></a>4 bin_9_NODE_12_length_355673_cov_0.516990_150 K02045 sulfate ABC transporter ATP-binding protein   
</code></pre></div>

Immediately, we can see that there are some inconsistencies in the annotations. 

* In the annotations for bin 3, there are repeat gene names in the same annotation line.
* Annotations for the same KO number is inconsistent across tables. For instance, K02046 is cysU in bin 3 and cysT in bins 5 and 8.
* There is a complete lack of gene names in bin 9 annotations. 

We need to remedy this by having a consistent label system. First, create a data frame of labels you want to show in your synteny plot.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a><span class="n">KO_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<a id="__codelineno-29-2" name="__codelineno-29-2" href="#__codelineno-29-2"></a><span class="w">  </span><span class="s">&quot;KO&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;K0204&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="o">:</span><span class="m">8</span><span class="p">),</span>
<a id="__codelineno-29-3" name="__codelineno-29-3" href="#__codelineno-29-3"></a><span class="w">  </span><span class="s">&quot;gene_name&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;cysA&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cysU&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cysW&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sbp1&quot;</span><span class="p">)</span>
<a id="__codelineno-29-4" name="__codelineno-29-4" href="#__codelineno-29-4"></a><span class="p">)</span>
<a id="__codelineno-29-5" name="__codelineno-29-5" href="#__codelineno-29-5"></a>
<a id="__codelineno-29-6" name="__codelineno-29-6" href="#__codelineno-29-6"></a><span class="c1"># Check what we have created</span>
<a id="__codelineno-29-7" name="__codelineno-29-7" href="#__codelineno-29-7"></a><span class="n">KO_genes</span>
</code></pre></div>

!!! circle-check "Console output"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a>      KO gene_name
<a id="__codelineno-30-2" name="__codelineno-30-2" href="#__codelineno-30-2"></a>1 K02045      cysA
<a id="__codelineno-30-3" name="__codelineno-30-3" href="#__codelineno-30-3"></a>2 K02046      cysU
<a id="__codelineno-30-4" name="__codelineno-30-4" href="#__codelineno-30-4"></a>3 K02047      cysW
<a id="__codelineno-30-5" name="__codelineno-30-5" href="#__codelineno-30-5"></a>4 K02048      sbp1
</code></pre></div>

Then, we join the KO_genes, annotation, and coordinate tables. We also replace rows that contain `NA` with `"Unknown domain"`.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a><span class="n">ann_coords_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map2</span><span class="p">(</span><span class="n">annot_list</span><span class="p">,</span><span class="w"> </span><span class="n">coords_list</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">ann</span><span class="p">,</span><span class="w"> </span><span class="n">coord</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-31-2" name="__codelineno-31-2" href="#__codelineno-31-2"></a><span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">ann</span><span class="p">,</span><span class="w"> </span><span class="n">coord</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Query gene&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;name&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-31-3" name="__codelineno-31-3" href="#__codelineno-31-3"></a><span class="w">    </span><span class="nf">left_join</span><span class="p">(</span><span class="n">KO_genes</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-31-4" name="__codelineno-31-4" href="#__codelineno-31-4"></a><span class="w">    </span><span class="nf">mutate</span><span class="p">(</span>
<a id="__codelineno-31-5" name="__codelineno-31-5" href="#__codelineno-31-5"></a><span class="w">      </span><span class="n">gene_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">replace_na</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Unknown domain&quot;</span><span class="p">)</span>
<a id="__codelineno-31-6" name="__codelineno-31-6" href="#__codelineno-31-6"></a><span class="w">    </span><span class="p">)</span>
<a id="__codelineno-31-7" name="__codelineno-31-7" href="#__codelineno-31-7"></a><span class="p">})</span>
<a id="__codelineno-31-8" name="__codelineno-31-8" href="#__codelineno-31-8"></a>
<a id="__codelineno-31-9" name="__codelineno-31-9" href="#__codelineno-31-9"></a><span class="c1"># Check one of the tables</span>
<a id="__codelineno-31-10" name="__codelineno-31-10" href="#__codelineno-31-10"></a><span class="nf">View</span><span class="p">(</span><span class="n">ann_coords_list</span><span class="o">$</span><span class="n">bin_3_annot</span><span class="p">)</span><span class="w"> </span><span class="c1"># In RStudio, View() will open a data viewing tab on the top left pane!</span>
</code></pre></div>

Much better! We can use the `gene_name` column as our annotation on our gene synteny plot.

We also need to convert the tidy annotation table into something that `genoPlotR` can use (i.e. an `annotations` class object).

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a><span class="n">annotations_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">ann_coords_list</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-32-2" name="__codelineno-32-2" href="#__codelineno-32-2"></a><span class="w">  </span><span class="nf">annotation</span><span class="p">(</span>
<a id="__codelineno-32-3" name="__codelineno-32-3" href="#__codelineno-32-3"></a><span class="w">    </span><span class="n">x1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="c1"># Where each gene starts</span>
<a id="__codelineno-32-4" name="__codelineno-32-4" href="#__codelineno-32-4"></a><span class="w">    </span><span class="n">x2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">end</span><span class="p">,</span><span class="w"> </span><span class="c1"># Gene end coordinate</span>
<a id="__codelineno-32-5" name="__codelineno-32-5" href="#__codelineno-32-5"></a><span class="w">    </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">gene_name</span><span class="w"> </span><span class="c1"># Labels to show</span>
<a id="__codelineno-32-6" name="__codelineno-32-6" href="#__codelineno-32-6"></a><span class="w">  </span><span class="p">)</span>
<a id="__codelineno-32-7" name="__codelineno-32-7" href="#__codelineno-32-7"></a><span class="p">})</span>
<a id="__codelineno-32-8" name="__codelineno-32-8" href="#__codelineno-32-8"></a>
<a id="__codelineno-32-9" name="__codelineno-32-9" href="#__codelineno-32-9"></a><span class="c1"># Check one of the annotations_list objects</span>
<a id="__codelineno-32-10" name="__codelineno-32-10" href="#__codelineno-32-10"></a><span class="n">annotations_list</span><span class="o">$</span><span class="n">bin_5_annot</span>
</code></pre></div>

!!! circle-check "Console output"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a>     x1    x2 text color rot
<a id="__codelineno-33-2" name="__codelineno-33-2" href="#__codelineno-33-2"></a>1 71608 72606 sbp1 black   0
<a id="__codelineno-33-3" name="__codelineno-33-3" href="#__codelineno-33-3"></a>2 72768 73586 cysU black   0
<a id="__codelineno-33-4" name="__codelineno-33-4" href="#__codelineno-33-4"></a>3 73597 74466 cysW black   0
<a id="__codelineno-33-5" name="__codelineno-33-5" href="#__codelineno-33-5"></a>4 74470 75459 cysA black   0
</code></pre></div>

#### 2.4 Plot data

Lets plot our data to see what it looks like and if it need tweaking.

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a><span class="nf">plot_gene_map</span><span class="p">(</span>
<a id="__codelineno-34-2" name="__codelineno-34-2" href="#__codelineno-34-2"></a><span class="w">  </span><span class="n">dna_segs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ds_list</span><span class="p">,</span>
<a id="__codelineno-34-3" name="__codelineno-34-3" href="#__codelineno-34-3"></a><span class="w">  </span><span class="n">comparisons</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast_list</span><span class="p">,</span>
<a id="__codelineno-34-4" name="__codelineno-34-4" href="#__codelineno-34-4"></a><span class="w">  </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotations_list</span><span class="p">,</span>
<a id="__codelineno-34-5" name="__codelineno-34-5" href="#__codelineno-34-5"></a><span class="w">  </span><span class="n">dna_seg_labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">ds_list</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;_coords&quot;</span><span class="p">),</span>
<a id="__codelineno-34-6" name="__codelineno-34-6" href="#__codelineno-34-6"></a><span class="w">  </span><span class="n">dna_seg_scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<a id="__codelineno-34-7" name="__codelineno-34-7" href="#__codelineno-34-7"></a><span class="w">  </span><span class="n">gene_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;arrows&quot;</span>
<a id="__codelineno-34-8" name="__codelineno-34-8" href="#__codelineno-34-8"></a><span class="p">)</span>
</code></pre></div>

![image](../figures/day4_synteny.01.synteny_all.png)

We have a plot! However, it is rather cluttered and could be further improved:

* Remove gene names in the middle tracks
* Concatenate the "Unknown domain" portion into a single annotation
* Showing only the best tBLASTx hits, reducing short alignments

!!! r-project "code"

    <div class="highlight"><pre><span></span><code><a id="__codelineno-35-1" name="__codelineno-35-1" href="#__codelineno-35-1"></a><span class="c1"># Filter BLAST results</span>
<a id="__codelineno-35-2" name="__codelineno-35-2" href="#__codelineno-35-2"></a><span class="n">filt_blast_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">blast_list</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a id="__codelineno-35-3" name="__codelineno-35-3" href="#__codelineno-35-3"></a><span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-35-4" name="__codelineno-35-4" href="#__codelineno-35-4"></a><span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">name1</span><span class="p">,</span><span class="w"> </span><span class="n">name2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-35-5" name="__codelineno-35-5" href="#__codelineno-35-5"></a><span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">e_value</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">e_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<a id="__codelineno-35-6" name="__codelineno-35-6" href="#__codelineno-35-6"></a><span class="w">    </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-35-7" name="__codelineno-35-7" href="#__codelineno-35-7"></a><span class="w">    </span><span class="nf">as.comparison</span><span class="p">()</span>
<a id="__codelineno-35-8" name="__codelineno-35-8" href="#__codelineno-35-8"></a><span class="p">})</span>
<a id="__codelineno-35-9" name="__codelineno-35-9" href="#__codelineno-35-9"></a>
<a id="__codelineno-35-10" name="__codelineno-35-10" href="#__codelineno-35-10"></a><span class="c1"># Concatenate &quot;Unknown domain&quot; in bin 3</span>
<a id="__codelineno-35-11" name="__codelineno-35-11" href="#__codelineno-35-11"></a><span class="n">annotations_bin_3_mod</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotations_list</span><span class="o">$</span><span class="n">bin_3_annot</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-35-12" name="__codelineno-35-12" href="#__codelineno-35-12"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Unknown domain&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a id="__codelineno-35-13" name="__codelineno-35-13" href="#__codelineno-35-13"></a><span class="w">  </span><span class="nf">add_row</span><span class="p">(</span>
<a id="__codelineno-35-14" name="__codelineno-35-14" href="#__codelineno-35-14"></a><span class="w">    </span><span class="n">x1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">136994</span><span class="p">,</span><span class="w"> </span><span class="n">x2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">139201</span><span class="p">,</span><span class="w"> </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Unknown domain&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">rot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span>
<a id="__codelineno-35-15" name="__codelineno-35-15" href="#__codelineno-35-15"></a><span class="w">  </span><span class="p">)</span>
<a id="__codelineno-35-16" name="__codelineno-35-16" href="#__codelineno-35-16"></a>
<a id="__codelineno-35-17" name="__codelineno-35-17" href="#__codelineno-35-17"></a><span class="c1"># Remove gene names in middle tracks</span>
<a id="__codelineno-35-18" name="__codelineno-35-18" href="#__codelineno-35-18"></a><span class="nf">plot_gene_map</span><span class="p">(</span>
<a id="__codelineno-35-19" name="__codelineno-35-19" href="#__codelineno-35-19"></a><span class="w">  </span><span class="n">dna_segs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ds_list</span><span class="p">,</span>
<a id="__codelineno-35-20" name="__codelineno-35-20" href="#__codelineno-35-20"></a><span class="w">  </span><span class="n">comparisons</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">filt_blast_list</span><span class="p">,</span>
<a id="__codelineno-35-21" name="__codelineno-35-21" href="#__codelineno-35-21"></a><span class="w">  </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">annotations_bin_3_mod</span><span class="p">,</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">annotations_list</span><span class="p">[[</span><span class="m">4</span><span class="p">]]),</span>
<a id="__codelineno-35-22" name="__codelineno-35-22" href="#__codelineno-35-22"></a><span class="w">  </span><span class="n">dna_seg_labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">ds_list</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;_coords&quot;</span><span class="p">),</span>
<a id="__codelineno-35-23" name="__codelineno-35-23" href="#__codelineno-35-23"></a><span class="w">  </span><span class="n">dna_seg_scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<a id="__codelineno-35-24" name="__codelineno-35-24" href="#__codelineno-35-24"></a><span class="w">  </span><span class="n">gene_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;arrows&quot;</span>
<a id="__codelineno-35-25" name="__codelineno-35-25" href="#__codelineno-35-25"></a><span class="p">)</span>
</code></pre></div>

![image](../figures/day4_synteny.03.synteny_final.png)

This looks much cleaner!

??? success "Results at a glance"

    From the last plot, we see that:

    * The arrangement sulfur assimilation genes are well conserved
    * Gene sequences of *sbp1*, *cysW*, and *cysA* are very conserved between bins 3, 5, and 8
    * Bin 3 has 3 unknown domains between its sulfur binding protein and the other sulfur assimilation genes
    * *cysW* in bin 9 shows some sequence homology with *cysU* from bin 8
-->












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.07f07601.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.56dfad97.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>