
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day1/ex2_quality_filtering/" rel="canonical"/>
<link href="../ex2_1_intro_to_scheduler/" rel="prev"/>
<link href="../ex3_assembly/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Quality filtering raw reads - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#quality-filtering-raw-reads">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Quality filtering raw reads
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../ex1_bash_and_scheduler/">
        Introduction to shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex2_1_intro_to_scheduler/">
        Introduction to HPC &amp; HPC job scheduler
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Quality filtering raw reads
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Quality filtering raw reads
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#visualising-raw-reads">
    Visualising raw reads
  </a>
<nav aria-label="Visualising raw reads" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#loading-fastqc">
    Loading FastQC
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#running-fastqc">
    Running FastQC
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#viewing-the-outputs-from-fastqc">
    Viewing the outputs from FastQC
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#fastqc-outputs-for-libraries-with-errors">
    FastQC outputs for libraries with errors
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#read-trimming-and-adapter-removal-with-trimmomatic">
    Read trimming and adapter removal with trimmomatic
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#considerations-when-working-with-trimmomatic">
    Considerations when working with trimmomatic
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-working-with-the-illuminaclip-command">
    Optional: Working with the ILLUMINACLIP command
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#diagnosing-poor-libraries">
    Diagnosing poor libraries
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#understand-common-issues-and-best-practices">
    Understand common issues and best practices
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-filtering-out-host-dna">
    Optional: Filtering out host DNA
  </a>
<nav aria-label="Optional: Filtering out host DNA" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#downloading-the-masked-human-reference-genome">
    Downloading the masked human reference genome
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#indexing-the-reference-genome-and-read-mapping-with-bbmap">
    Indexing the reference genome and read mapping with BBMap
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex3_assembly/">
        Assembly
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning (continued)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation (part 1)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation (part 2) and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation (part 3)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Presentation of data: Per-sample coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#visualising-raw-reads">
    Visualising raw reads
  </a>
<nav aria-label="Visualising raw reads" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#loading-fastqc">
    Loading FastQC
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#running-fastqc">
    Running FastQC
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#viewing-the-outputs-from-fastqc">
    Viewing the outputs from FastQC
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#fastqc-outputs-for-libraries-with-errors">
    FastQC outputs for libraries with errors
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#read-trimming-and-adapter-removal-with-trimmomatic">
    Read trimming and adapter removal with trimmomatic
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#considerations-when-working-with-trimmomatic">
    Considerations when working with trimmomatic
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-working-with-the-illuminaclip-command">
    Optional: Working with the ILLUMINACLIP command
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#diagnosing-poor-libraries">
    Diagnosing poor libraries
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#understand-common-issues-and-best-practices">
    Understand common issues and best practices
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-filtering-out-host-dna">
    Optional: Filtering out host DNA
  </a>
<nav aria-label="Optional: Filtering out host DNA" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#downloading-the-masked-human-reference-genome">
    Downloading the masked human reference genome
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#indexing-the-reference-genome-and-read-mapping-with-bbmap">
    Indexing the reference genome and read mapping with BBMap
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="quality-filtering-raw-reads">Quality filtering raw reads<a class="headerlink" href="#quality-filtering-raw-reads" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#visualising-raw-reads">Visualising raw reads with <code>FastQC</code></a></li>
<li><a href="#read-trimming-and-adapter-removal-with-trimmomatic">Read trimming and adapter removal with <code>trimmomatic</code></a></li>
<li><a href="#diagnosing-poor-libraries">Diagnosing poor libraries</a></li>
<li><a href="#understand-common-issues-and-best-practices">Understand common issues and best practices</a></li>
<li><a href="#optional-filtering-out-host-dna"><em>Optional</em>: Filtering out host DNA with <code>BBMap</code></a></li>
</ul>
</div>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../theme_images/qc_image.png"><img alt="image" src="../../theme_images/qc_image.png" width="350"/></a>
</center></p>
<hr/>
<h3 id="visualising-raw-reads">Visualising raw reads<a class="headerlink" href="#visualising-raw-reads" title="Permanent link">¶</a></h3>
<p><code>FastQC</code> is an extremely popular tool for checking your sequencing libraries, as the visual interface makes it easy to identify the following issues:</p>
<ol>
<li>Adapter/barcode sequences</li>
<li>Low quality regions of sequence</li>
<li>Quality drop-off towards the end of read-pair sequence</li>
</ol>
<h4 id="loading-fastqc">Loading <em>FastQC</em><a class="headerlink" href="#loading-fastqc" title="Permanent link">¶</a></h4>
<p>These exercises will take place in the <code>2.fastqc/</code> directory. First, navigate to this directory. Copy the command below into your terminal (logged in to NeSI), replacing <code>&lt;YOUR FOLDER&gt;</code>, and then running the command.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/2.fastqc/
</code></pre></div>
<p>To activate <code>FastQC</code> on NeSI, you need to first load the module using the command</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>module<span class="w"> </span>purge
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>module<span class="w"> </span>load<span class="w"> </span>FastQC/0.11.9
</code></pre></div>
<h4 id="running-fastqc">Running <em>FastQC</em><a class="headerlink" href="#running-fastqc" title="Permanent link">¶</a></h4>
<p>We will run <code>FastQC</code> from the command line as follows:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>fastqc<span class="w"> </span>mock_R1.good.fastq.gz<span class="w"> </span>mock_R2.good.fastq.gz
</code></pre></div>
<h4 id="viewing-the-outputs-from-fastqc">Viewing the outputs from <em>FastQC</em><a class="headerlink" href="#viewing-the-outputs-from-fastqc" title="Permanent link">¶</a></h4>
<p><code>FastQC</code> generates output reports in <code>.html</code> files that can be viewed in a standard web browser. </p>
<p>Fortunately, if you're currently using the terminal within <code>Jupyter hub</code> for today's session, we can open the <code>.html</code> files directly from here:</p>
<ul>
<li>Click on the directory icon in the top left to open the directory navigator pane (if not already open).</li>
<li>The default viewed location will be the overall project that you have logged in to (in this case, the 'Genomics Aotearoa Virtual Lab Training Access (nesi02659)' project). </li>
<li>Click through <code>MGSS_U</code>, into your directory, and then into the <code>2.fastqc/</code> directory. </li>
<li>Double click on the output <code>...fastqc.html</code> files to open them in the a new tab within the <code>Jupyter hub</code>.</li>
</ul>
<p>Examples of the output files are also available for download <a href="../../resources/fastqc_results.zip">here</a>.</p>
<p>Note that <code>FastQC</code> does not load the forward and reverse pairs of a library in the same window, so you need to be mindful of how your samples relate to each other. </p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig1_overview.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig1_overview.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig1_overview.PNG"><img alt="image" src="../../figures/ex3_fig1_overview.PNG"/></a></p>
<p>At a first glance, we can see the follow statistics:</p>
<ol>
<li>The data is stored in Sanger / Illumina 1.9 encoding. <em>This will be important to remember when read trimming.</em></li>
<li>There are 100,000 reads in the file</li>
<li>The maximum sequence length is 251 base pairs. <em>This is good to check, since the data were generated using Illumina
2x250 bp sequencing.</em></li>
</ol>
<p>Have a quick look through the menu sidebar on the left. As you can see, the data has passed most of the basic parameters.</p>
<p><strong>Per-base sequence quality</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig2_quality.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig2_quality.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig2_quality.PNG"><img alt="image" src="../../figures/ex3_fig2_quality.PNG"/></a></p>
<p><strong>Per base sequence content</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig3_content.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig3_content.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig3_content.PNG"><img alt="image" src="../../figures/ex3_fig3_content.PNG"/></a></p>
<p><strong>Adapter Content</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig4_adapter.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig4_adapter.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig4_adapter.PNG"><img alt="image" src="../../figures/ex3_fig4_adapter.PNG"/></a></p>
<p><strong>Per sequence GC content</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig5_gc.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig5_gc.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig5_gc.PNG"><img alt="image" src="../../figures/ex3_fig5_gc.PNG"/></a></p>
<p>The only aspect of the data that <code>FastQC</code> is flagging as potentially problematic is the GC% content of the data set. This observation is expected as we deal with a mixed community and organisms. Therefore, it is unlikely that there will be a perfect normal distribution around an average value. For example, a community comprised of low- and high-GC organisms would manifest a bimodal distribution of peaks which would be a problematic outcome in terms of the expectations of <code>FastQC</code>, but completely consistent with the biology of the system.</p>
<h4 id="fastqc-outputs-for-libraries-with-errors"><em>FastQC</em> outputs for libraries with errors<a class="headerlink" href="#fastqc-outputs-for-libraries-with-errors" title="Permanent link">¶</a></h4>
<p>Lets take a look at a library with significant errors. Process the sequence file <code>mock_R1.adapter_decay.fastq</code> with <code>FastQC</code>.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>fastqc<span class="w"> </span>mock_R1.adapter_decay.fastq.gz
</code></pre></div>
<p>Compare the results with the <code>mock_R1.good.fastq.gz</code> file.</p>
<p>Which of the previous fields we examined are now flagged as problematic? How does this compare with your expectation? Are there any which should be flagged which are not?</p>
<p><strong>Per-base sequence quality</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig6_taildecay.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig6_taildecay.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig6_taildecay.PNG"><img alt="image" src="../../figures/ex3_fig6_taildecay.PNG"/></a></p>
<hr/>
<h3 id="read-trimming-and-adapter-removal-with-trimmomatic"><strong>Read trimming and adapter removal with <em>trimmomatic</em></strong><a class="headerlink" href="#read-trimming-and-adapter-removal-with-trimmomatic" title="Permanent link">¶</a></h3>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../theme_images/cleaning_reads.png"><img alt="image" src="../../theme_images/cleaning_reads.png" width="300"/></a>
</center></p>
<p>There are a multitude of programs which can be used to quality trim sequence data and remove adapter sequence. For this exercise we are going to use <code>trimmomatic</code>, but this should in no way be interpreted as an endorsement of <code>trimmomatic</code> over equivalent tools like <code>BBMap</code>, <code>sickle</code>, <code>cutadapt</code> or any other.</p>
<p>For a first run with <code>trimmomatic</code>, type the following commands into your console:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>module<span class="w"> </span>load<span class="w"> </span>Trimmomatic/0.39-Java-1.8.0_144
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>-threads<span class="w"> </span><span class="m">2</span><span class="w"> </span>-phred33<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="w">               </span>mock_R1.adapter_decay.fastq.gz<span class="w"> </span>mock_R2.adapter_decay.fastq.gz<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="w">               </span>mock_R1.qc.fastq.gz<span class="w"> </span>mock_s1.qc.fastq.gz<span class="w"> </span>mock_R2.qc.fastq.gz<span class="w"> </span>mock_s2.qc.fastq.gz<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="w">               </span>HEADCROP:10<span class="w"> </span>SLIDINGWINDOW:4:30<span class="w"> </span>MINLEN:80
</code></pre></div>
<p>There is a lot going on in this command, so here is a breakdown of the parameters in the command above</p>
<div class="admonition info">
<table>
<thead>
<tr>
<th align="left">Parameter</th>
<th align="center">Type</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">PE</td>
<td align="center"><em>positional</em></td>
<td align="left">Specifies whether we are analysing single- or paired-end reads</td>
</tr>
<tr>
<td align="left">-threads 10</td>
<td align="center"><em>keyword</em></td>
<td align="left">Specifies the number of threads to use when processing</td>
</tr>
<tr>
<td align="left">-phred33</td>
<td align="center"><em>keyword</em></td>
<td align="left">Specifies the fastq encoding used</td>
</tr>
<tr>
<td align="left">mock_R1.adapter_decay.fastq.gz / mock_R2.adapter_decay.fastq.gz</td>
<td align="center"><em>positional</em></td>
<td align="left">The paired forward and reverse reads to trim</td>
</tr>
<tr>
<td align="left">mock_R1.qc.fastq.gz</td>
<td align="center"><em>positional</em></td>
<td align="left">The file to write forward reads which passed quality trimming, if their reverse partner also passed</td>
</tr>
<tr>
<td align="left">mock_s1.qc.fastq.gz</td>
<td align="center"><em>positional</em></td>
<td align="left">The file to write forward reads which passed quality trimming, if their reverse partner failed (orphan reads)</td>
</tr>
<tr>
<td align="left">mock_R2.qc.fastq.gz / mock_s2.qc.fastq.gz</td>
<td align="center"><em>positional</em></td>
<td align="left">The reverse-sequence equivalent of above</td>
</tr>
<tr>
<td align="left">HEADCROP:10</td>
<td align="center"><em>positional</em></td>
<td align="left">Adapter trimming command. Remove the first 10 positions in the sequence</td>
</tr>
<tr>
<td align="left">SLIDINGWINDOW:4:30</td>
<td align="center"><em>positional</em></td>
<td align="left">Quality filtering command. Analyse each sequence in a 4 base pair sliding window and then truncate if the average quality drops below Q30</td>
</tr>
<tr>
<td align="left">MINLEN:80</td>
<td align="center"><em>positional</em></td>
<td align="left">Length filtering command. Discard sequences that are shorter than 80 base pairs after trimming</td>
</tr>
</tbody>
</table>
</div>
<div class="admonition success">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>TrimmomaticPE: Started with arguments:
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a> -threads 10 -phred33 mock_R1.adapter_decay.fastq.gz mock_R2.adapter_decay.fastq.gz mock_R1.qc.fastq.gz mock_s1.qc.fastq.gz mock_R2.qc.fastq.gz mock_s2.qc.fastq.gz HEADCROP:10 SLIDINGWINDOW:4:30 MINLEN:80
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>Input Read Pairs: 100000 Both Surviving: 75684 (75.68%) Forward Only Surviving: 12783 (12.78%) Reverse Only Surviving: 2311 (2.31%) Dropped: 9222 (9.22%)
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>TrimmomaticPE: Completed successfully
</code></pre></div>
</div>
<p>Running the trimmed files back through <code>FastQC</code>, we can see that this signifcantly improves the output.</p>
<p><strong>Quality of reads</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig7_qualtrim.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig7_qualtrim.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig7_qualtrim.PNG"><img alt="image" src="../../figures/ex3_fig7_qualtrim.PNG"/></a></p>
<p><strong>Nucleotide distribution</strong></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig8_headtrim.PNG"><img alt="" src="https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/figures/ex3_fig8_headtrim.PNG"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex3_fig8_headtrim.PNG"><img alt="image" src="../../figures/ex3_fig8_headtrim.PNG"/></a></p>
<hr/>
<h3 id="considerations-when-working-with-trimmomatic">Considerations when working with <em>trimmomatic</em><a class="headerlink" href="#considerations-when-working-with-trimmomatic" title="Permanent link">¶</a></h3>
<p><strong>Order of operations</strong></p>
<p>The basic format for a <code>trimmomatic</code> command is</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>&lt;keyword<span class="w"> </span>flags&gt;<span class="w"> </span>&lt;sequence<span class="w"> </span>input&gt;<span class="w"> </span>&lt;sequence<span class="w"> </span>output&gt;<span class="w"> </span>&lt;trimming<span class="w"> </span>parameters&gt;
</code></pre></div>
<p>The trimming parameters are processed in the order you specify them. This is a deliberate behaviour, but can have some unexpected consequences for new users.</p>
<p>For example, consider these two scenarios:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>&lt;keyword<span class="w"> </span>flags&gt;<span class="w"> </span>&lt;sequence<span class="w"> </span>input&gt;<span class="w"> </span>&lt;sequence<span class="w"> </span>output&gt;<span class="w"> </span>SLIDINGWINDOW:4:30<span class="w"> </span>MINLEN:80
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>&lt;keyword<span class="w"> </span>flags&gt;<span class="w"> </span>&lt;sequence<span class="w"> </span>input&gt;<span class="w"> </span>&lt;sequence<span class="w"> </span>output&gt;<span class="w"> </span>MINLEN:80<span class="w"> </span>SLIDINGWINDOW:4:30
</code></pre></div>
<p>In the first run, we would not expect any sequence shorter than 80 base pairs to exist in the output files. However, we might encounter them in the second command. This is because in the second command we remove sequences shorter than 80 base pairs, <strong><em>then</em></strong> perform quality trimming. If a sequence is trimmed to a length shorter than 80 base pairs <strong><em>after</em></strong> trimming, the <code>MINLEN</code> filtering does not execute a second time. In the first instance, we do not perform trimming <strong><em>before</em></strong> size selection, so any reads that start longer than 80 base pairs, but are trimmed to under 80 base pairs during quality trimming will be caught in the <code>MINLEN</code> run.</p>
<p>A more subtle consequence of this behaviour is the interplay between adapter removal and quality filtering. In the command we ran, we try to identify adapters <strong><em>before</em></strong> quality trimming. Why do you think this is?</p>
<hr/>
<h3 id="optional-working-with-the-illuminaclip-command"><em>Optional:</em> Working with the ILLUMINACLIP command<a class="headerlink" href="#optional-working-with-the-illuminaclip-command" title="Permanent link">¶</a></h3>
<p><code>Trimmomatic</code> also provides the <code>ILLUMINACLIP</code> command, which can be used to pass a FASTA file of adapter and barcode sequences to be found and removed from your data. The format for the <code>ILLUMINACLIP</code> parameter can be quite confusing to work with and so we generally favour a <code>HEADCROP</code> where possible. If you wish to work with the <code>ILLUMINACLIP</code> command, then you can see its use here in the <code>trimmomatic</code> <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf">manual</a>.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>ILLUMINACLIP:iua.fna:1:25:7
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a><span class="w">             </span><span class="p">|</span><span class="w">       </span><span class="p">|</span><span class="w"> </span><span class="p">|</span><span class="w">  </span><span class="p">|</span>
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a><span class="w">             </span><span class="p">|</span><span class="w">       </span><span class="p">|</span><span class="w"> </span><span class="p">|</span><span class="w">  </span><span class="p">|</span>
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a><span class="w">             </span><span class="p">|</span><span class="w">       </span><span class="p">|</span><span class="w"> </span><span class="p">|</span><span class="w">  </span>Simple<span class="w"> </span>clip<span class="w"> </span>threshold
<a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a><span class="w">             </span><span class="p">|</span><span class="w">       </span><span class="p">|</span><span class="w"> </span>Palindrome<span class="w"> </span>clip<span class="w"> </span>threshold
<a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a><span class="w">             </span><span class="p">|</span><span class="w">       </span>Seed<span class="w"> </span>mismatches
<a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a><span class="w">             </span>File<span class="w"> </span>of<span class="w"> </span>expected<span class="w"> </span>sequences
</code></pre></div>
<p>There is always some subjectivity in how sensitive you want your adapter (and barcode) searching to be. If the settings are too strict you might end up discarding real sequence data that only partially overlaps with the Illumina adapters. If your settings are not strict enough then you might leave partial adapters in the sequence. Where possible, we favour the use of simple positional trimming.</p>
<hr/>
<h3 id="diagnosing-poor-libraries">Diagnosing poor libraries<a class="headerlink" href="#diagnosing-poor-libraries" title="Permanent link">¶</a></h3>
<p>Whether a library is 'poor' quality or not can be a bit subjective. These are some aspects of the library that you should be looking for when evaluating <code>FastQC</code>:</p>
<div class="admonition note">
<ol>
<li>Does the sequencing length match what you ordered from the facility?</li>
<li>If the sequences are shorter than expected, is adapter read-through a concern?</li>
<li>What does the sequence quality look like for the whole length of the run? Are there any expected/unexpected regions of quality degradation?</li>
<li>Are adapters and/or barcodes removed? <ul>
<li>Look at the <em>Per base sequence content</em> to diagnose this.</li>
</ul>
</li>
<li>Is there unexpected sequence duplication? <ul>
<li>This can occur when low-input library preparations are used.</li>
</ul>
</li>
<li>Are over-represented <em>k</em>-mers present? <ul>
<li>This can be a sign of adapter and barcode contamination.</li>
</ul>
</li>
</ol>
</div>
<hr/>
<h3 id="understand-common-issues-and-best-practices">Understand common issues and best practices<a class="headerlink" href="#understand-common-issues-and-best-practices" title="Permanent link">¶</a></h3>
<div class="admonition success">
<ol>
<li>Do I need to remove (rare) adapters?</li>
<li>I don’t know if adapters have been removed or not</li>
<li>How do I identify and remove adapter read-through</li>
<li>Identifying incomplete barcode/adapter removal</li>
<li>Over aggressive trimming</li>
<li>GC skew is outside of expected range</li>
</ol>
</div>
<hr/>
<h3 id="optional-filtering-out-host-dna"><em>Optional</em>: Filtering out host DNA<a class="headerlink" href="#optional-filtering-out-host-dna" title="Permanent link">¶</a></h3>
<p>Metagenome data derived from host-associated microbial communities should ideally be filtered to remove any reads originating from host DNA. This may improve the quality and efficiency of downstream data processing (since we will no longer be processing a bunch of data that we are likely not interested in), and is also an important consideration when working with metagenomes that may include data of a sensitive nature (and which may also need to be removed prior to making the data publicly available). This is especially important for any studies involving human subjects or those involving samples derived from Taonga species.</p>
<p>There are several approaches that can be used to achieve this. The general principle is to map your reads to a reference genome (e.g. human genome) and remove those reads that map to the reference from the dataset. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This process may be more complicated if a reference genome for your host taxa is not readily available. In this case an alternative method would need to be employed (for example: predicting taxonomy via <code>Kraken2</code> and then filtering out all reads that map to the phylum or kingdom of your host taxa).</p>
</div>
<p>This exercise provides an example using <code>BBMap</code> to map against a masked human reference genome and retain only those reads that do <em>not</em> map to the reference. Here we are mapping the quality-filtered reads against a pre-prepared human genome that has been processed to mask sections of the genome, including those that:</p>
<ul>
<li>are presumed microbial contaminant in the reference</li>
<li>have high homology to microbial genes/genomes (e.g. ribosomes)</li>
<li>those that are of low complexity </li>
</ul>
<p>This ensures that reads that would normally map to these sections of the human genome are <em>not</em> removed from the dataset (as genuine microbial reads that we wish to retain might also map to these regions), while all reads mapping to the rest of the human genome are removed.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The same process can be used to remove DNA matching other hosts (e.g. mouse), however you would need to search if anyone has prepared (and made available) a masked version of the reference genome, or create a masked version using <code>bbmask</code>. The creator of BBMap has made available masked human, mouse, cat, and dog genomes. More information, including links to these references and instructions on how to generate a masked genome for other taxa, can be found within <a href="http://seqanswers.com/forums/showthread.php?t=42552">this thread</a>.*</p>
</div>
<h4 id="downloading-the-masked-human-reference-genome">Downloading the masked human reference genome<a class="headerlink" href="#downloading-the-masked-human-reference-genome" title="Permanent link">¶</a></h4>
<p>The masked reference genome is available via <a href="https://drive.google.com/file/d/0B3llHR93L14wd0pSSnFULUlhcUk/edit?usp=sharing">Google drive</a> and <a href="https://zenodo.org/record/1208052">Zenodo</a>. For this workshop, we have provided you with the file within the <code>4.evaluation/BBMask_human_reference/</code> directory.</p>
<details class="note">
<summary>Downloading your own copy of the masked genome</summary>
<p>We can use <code>gdown</code> to download this file from Google drive via the command line.</p>
<p>To install <code>gdown</code>, we can use <code>pip</code>. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a><span class="c1"># Install gdown (for downloading from google drive)</span>
<a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a>module<span class="w"> </span>purge
<a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a>module<span class="w"> </span>load<span class="w"> </span>Python/3.10.5-gimkl-2022a
<a href="#__codelineno-9-4" id="__codelineno-9-4" name="__codelineno-9-4"></a>pip<span class="w"> </span>install<span class="w"> </span>--user<span class="w"> </span>gdown
</code></pre></div>
<p>Next, download the reference. It will also be necessary to first add your local <code>bin</code> location to the <code>PATH</code> variable via the <code>export PATH=...</code> command, as this is where <code>gdown</code> is located (modify <code>&lt;your_username&gt;</code> before running the code below).</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a>mkdir<span class="w"> </span>BBMask_human_reference/
<a href="#__codelineno-10-2" id="__codelineno-10-2" name="__codelineno-10-2"></a><span class="nb">cd</span><span class="w"> </span>BBMask_human_reference/
<a href="#__codelineno-10-3" id="__codelineno-10-3" name="__codelineno-10-3"></a>
<a href="#__codelineno-10-4" id="__codelineno-10-4" name="__codelineno-10-4"></a><span class="nb">export</span><span class="w"> </span><span class="nv">PATH</span><span class="o">=</span><span class="s2">"/home/&lt;your_username&gt;/.local/bin:</span><span class="nv">$PATH</span><span class="s2">"</span>
<a href="#__codelineno-10-5" id="__codelineno-10-5" name="__codelineno-10-5"></a>
<a href="#__codelineno-10-6" id="__codelineno-10-6" name="__codelineno-10-6"></a>gdown<span class="w"> </span>https://drive.google.com/uc?id<span class="o">=</span>0B3llHR93L14wd0pSSnFULUlhcUk
</code></pre></div>
</details>
<h4 id="indexing-the-reference-genome-and-read-mapping-with-bbmap">Indexing the reference genome and read mapping with <em>BBMap</em><a class="headerlink" href="#indexing-the-reference-genome-and-read-mapping-with-bbmap" title="Permanent link">¶</a></h4>
<p>We will cover more about read mapping in <a href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day2/ex6_initial_binning/">later exercises</a>. For now, it is important to know that it is first necessary to build an index of the reference using the read mapping tool of choice. Here, we will first build a <code>BBMap</code> index, and then use <code>BBMap</code> to map the quality-filtered reads to that index, ultimately retaining only those reads that do <em>not</em> map to the index.</p>
<p>Build index reference via <code>BBMap</code>. We will do this by submitting the job via slurm. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See <a href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day1/ex3_assembly/">Preparing an assembly job for slurm</a> for more information about how to submit a job via slurm.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Paste or type in the following. Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own directory.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-11-2" id="__codelineno-11-2" name="__codelineno-11-2"></a>
<a href="#__codelineno-11-3" id="__codelineno-11-3" name="__codelineno-11-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-11-4" id="__codelineno-11-4" name="__codelineno-11-4"></a><span class="c1">#SBATCH --job-name      host_filt_bbmap_index</span>
<a href="#__codelineno-11-5" id="__codelineno-11-5" name="__codelineno-11-5"></a><span class="c1">#SBATCH --time          00:20:00</span>
<a href="#__codelineno-11-6" id="__codelineno-11-6" name="__codelineno-11-6"></a><span class="c1">#SBATCH --mem           32GB</span>
<a href="#__codelineno-11-7" id="__codelineno-11-7" name="__codelineno-11-7"></a><span class="c1">#SBATCH --cpus-per-task 12</span>
<a href="#__codelineno-11-8" id="__codelineno-11-8" name="__codelineno-11-8"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a href="#__codelineno-11-9" id="__codelineno-11-9" name="__codelineno-11-9"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a href="#__codelineno-11-10" id="__codelineno-11-10" name="__codelineno-11-10"></a>
<a href="#__codelineno-11-11" id="__codelineno-11-11" name="__codelineno-11-11"></a><span class="c1"># Working directory</span>
<a href="#__codelineno-11-12" id="__codelineno-11-12" name="__codelineno-11-12"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/2.fastqc/BBMask_human_reference/
<a href="#__codelineno-11-13" id="__codelineno-11-13" name="__codelineno-11-13"></a>
<a href="#__codelineno-11-14" id="__codelineno-11-14" name="__codelineno-11-14"></a><span class="c1"># Load BBMap module</span>
<a href="#__codelineno-11-15" id="__codelineno-11-15" name="__codelineno-11-15"></a>module<span class="w"> </span>purge
<a href="#__codelineno-11-16" id="__codelineno-11-16" name="__codelineno-11-16"></a>module<span class="w"> </span>load<span class="w"> </span>BBMap/39.01-GCC-11.3.0
<a href="#__codelineno-11-17" id="__codelineno-11-17" name="__codelineno-11-17"></a>
<a href="#__codelineno-11-18" id="__codelineno-11-18" name="__codelineno-11-18"></a><span class="c1"># Build indexed reference file via BBMap</span>
<a href="#__codelineno-11-19" id="__codelineno-11-19" name="__codelineno-11-19"></a>bbmap.sh<span class="w"> </span><span class="nv">ref</span><span class="o">=</span>hg19_main_mask_ribo_animal_allplant_allfungus.fa.gz
</code></pre></div>
</div>
<p>Finally, map the quality-filtered reads to the reference via <code>BBMap</code>. Here we will submit the job as a slurm array, with one array job per sample. </p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Paste or type in the following. Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own directory.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-12-2" id="__codelineno-12-2" name="__codelineno-12-2"></a>
<a href="#__codelineno-12-3" id="__codelineno-12-3" name="__codelineno-12-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-12-4" id="__codelineno-12-4" name="__codelineno-12-4"></a><span class="c1">#SBATCH --job-name      host_filt_bbmap_map</span>
<a href="#__codelineno-12-5" id="__codelineno-12-5" name="__codelineno-12-5"></a><span class="c1">#SBATCH --time          01:00:00</span>
<a href="#__codelineno-12-6" id="__codelineno-12-6" name="__codelineno-12-6"></a><span class="c1">#SBATCH --mem           27GB</span>
<a href="#__codelineno-12-7" id="__codelineno-12-7" name="__codelineno-12-7"></a><span class="c1">#SBATCH --array         1-4</span>
<a href="#__codelineno-12-8" id="__codelineno-12-8" name="__codelineno-12-8"></a><span class="c1">#SBATCH --cpus-per-task 20</span>
<a href="#__codelineno-12-9" id="__codelineno-12-9" name="__codelineno-12-9"></a><span class="c1">#SBATCH --error         %x_%A_%a.err</span>
<a href="#__codelineno-12-10" id="__codelineno-12-10" name="__codelineno-12-10"></a><span class="c1">#SBATCH --output        %x_%A_%a.out</span>
<a href="#__codelineno-12-11" id="__codelineno-12-11" name="__codelineno-12-11"></a>
<a href="#__codelineno-12-12" id="__codelineno-12-12" name="__codelineno-12-12"></a><span class="c1"># Set up working directories</span>
<a href="#__codelineno-12-13" id="__codelineno-12-13" name="__codelineno-12-13"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/2.fastqc/
<a href="#__codelineno-12-14" id="__codelineno-12-14" name="__codelineno-12-14"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>host_filtered_reads/
<a href="#__codelineno-12-15" id="__codelineno-12-15" name="__codelineno-12-15"></a>
<a href="#__codelineno-12-16" id="__codelineno-12-16" name="__codelineno-12-16"></a><span class="c1"># Load BBMap module</span>
<a href="#__codelineno-12-17" id="__codelineno-12-17" name="__codelineno-12-17"></a>module<span class="w"> </span>purge
<a href="#__codelineno-12-18" id="__codelineno-12-18" name="__codelineno-12-18"></a>module<span class="w"> </span>load<span class="w"> </span>BBMap/39.01-GCC-11.3.0
<a href="#__codelineno-12-19" id="__codelineno-12-19" name="__codelineno-12-19"></a>
<a href="#__codelineno-12-20" id="__codelineno-12-20" name="__codelineno-12-20"></a><span class="c1"># Run bbmap</span>
<a href="#__codelineno-12-21" id="__codelineno-12-21" name="__codelineno-12-21"></a>srun<span class="w"> </span>bbmap.sh<span class="w"> </span>-Xmx27g<span class="w"> </span>-t<span class="o">=</span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-12-22" id="__codelineno-12-22" name="__codelineno-12-22"></a><span class="w">  </span><span class="nv">minid</span><span class="o">=</span><span class="m">0</span>.95<span class="w"> </span><span class="nv">maxindel</span><span class="o">=</span><span class="m">3</span><span class="w"> </span><span class="nv">bwr</span><span class="o">=</span><span class="m">0</span>.16<span class="w"> </span><span class="nv">bw</span><span class="o">=</span><span class="m">12</span><span class="w"> </span>quickmatch<span class="w"> </span>fast<span class="w"> </span><span class="nv">minhits</span><span class="o">=</span><span class="m">2</span><span class="w"> </span><span class="nv">qtrim</span><span class="o">=</span>rl<span class="w"> </span><span class="nv">trimq</span><span class="o">=</span><span class="m">10</span><span class="w"> </span>untrim<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-12-23" id="__codelineno-12-23" name="__codelineno-12-23"></a><span class="w">  </span><span class="nv">in1</span><span class="o">=</span>../3.assembly/sample<span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-12-24" id="__codelineno-12-24" name="__codelineno-12-24"></a><span class="w">  </span><span class="nv">in2</span><span class="o">=</span>../3.assembly/sample<span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-12-25" id="__codelineno-12-25" name="__codelineno-12-25"></a><span class="w">  </span><span class="nv">path</span><span class="o">=</span>BBMask_human_reference/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-12-26" id="__codelineno-12-26" name="__codelineno-12-26"></a><span class="w">  </span><span class="nv">outu1</span><span class="o">=</span>host_filtered_reads/sample<span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>_R1_hostFilt.fastq<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-12-27" id="__codelineno-12-27" name="__codelineno-12-27"></a><span class="w">  </span><span class="nv">outu2</span><span class="o">=</span>host_filtered_reads/sample<span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>_R2_hostFilt.fastq
</code></pre></div>
</div>
<p>Breaking down this command a little:</p>
<div class="admonition quote">
<ul>
<li>We pass the path to the <code>ref</code> directory (the reference we just built) to <code>path=...</code>.</li>
<li>Provide quality-filtered reads as input (i.e. output of the <code>trimmomatic</code> process above). In this case, we will provide the FASTQ files located in <code>../3.assembly/</code> which have been processed via <code>trimmomatic</code> in the same manner as the exercise above. These are four sets of paired reads (representing metagenome data from four "samples") that the remainder of the workshop exercises will be working with.</li>
<li>The flags <code>-Xmx27g</code> and <code>-t=$SLURM_CPUS_PER_TASK</code> set the maximum memory and thread (AKA CPUs) allocations, and must match the <code>--mem</code> and <code>--cpus_per_task</code> allocations in the slurm headers at the top of the script.</li>
<li>The rest of the settings in the <code>BBMap</code> call here are as per the recommendations found within <a href="http://seqanswers.com/forums/showthread.php?t=42552">this thread</a> about processing data to remove host reads.</li>
<li>Finally, the filtered output FASTQ files for downstream use are written to the <code>host_filtered_reads/</code> directory (taken from the outputs <code>outu1=</code> and <code>outu2=</code>, which include only those reads that did not map to the host reference genome).</li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Slurm array jobs automatically create a variable <code>SLURM_ARRAY_TASK_ID</code> for that job, which contains the array task number (i.e. between 1 and 4 in the case above). We use this to run the command on the sample that matches this array task ID. I.e. array job 3 will run the commands on "sample3" (<code>sample${SLURM_ARRAY_TASK_ID}</code> is read in as <code>sample3</code>).</p>
</div>
<p>The filtered reads are now available in <code>host_filtered_reads/</code> for downstream use.</p>
<hr/>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>