
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day3/ex13_gene_annotation_part1/" rel="canonical"/>
<link href="../ex12_gene_prediction/" rel="prev"/>
<link href="../ex14_gene_annotation_part2/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Gene annotation (part 1) - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#gene-annotation-part-1">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Gene annotation (part 1)
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction to shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction to HPC &amp; HPC job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Quality filtering raw reads
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning (continued)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Gene annotation (part 1)
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Gene annotation (part 1)
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#blast-like-gene-annotations-and-domain-annotations">
    BLAST-like gene annotations and domain annotations
  </a>
<nav aria-label="BLAST-like gene annotations and domain annotations" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#blast-like-annotation">
    BLAST-like annotation
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#hmm-profiling-of-domains">
    HMM-profiling of domains
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#annotating-mags-against-the-uniprot-database-with-diamond">
    Annotating MAGs against the UniProt database with diamond
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#annotating-mags-against-the-pfam-database-with-hmmer">
    Annotating MAGs against the Pfam database with hmmer
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#evaluating-the-quality-of-gene-assignment">
    Evaluating the quality of gene assignment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#differences-in-taxonomies">
    Differences in taxonomies
  </a>
<nav aria-label="Differences in taxonomies" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#examples-of-various-genome-level-and-protein-taxonomies">
    Examples of various genome-level and protein taxonomies
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex14_gene_annotation_part2/">
        Gene annotation (part 2) and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation (part 3)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Presentation of data: Per-sample coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#blast-like-gene-annotations-and-domain-annotations">
    BLAST-like gene annotations and domain annotations
  </a>
<nav aria-label="BLAST-like gene annotations and domain annotations" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#blast-like-annotation">
    BLAST-like annotation
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#hmm-profiling-of-domains">
    HMM-profiling of domains
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#annotating-mags-against-the-uniprot-database-with-diamond">
    Annotating MAGs against the UniProt database with diamond
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#annotating-mags-against-the-pfam-database-with-hmmer">
    Annotating MAGs against the Pfam database with hmmer
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#evaluating-the-quality-of-gene-assignment">
    Evaluating the quality of gene assignment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#differences-in-taxonomies">
    Differences in taxonomies
  </a>
<nav aria-label="Differences in taxonomies" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#examples-of-various-genome-level-and-protein-taxonomies">
    Examples of various genome-level and protein taxonomies
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="gene-annotation-part-1">Gene annotation (part 1)<a class="headerlink" href="#gene-annotation-part-1" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#gene-annotation-part-1">Gene annotation (part 1)</a><ul>
<li><a href="#objectives">Objectives</a></li>
<li><a href="#blast-like-gene-annotations-and-domain-annotations"><em>BLAST</em>-like gene annotations and domain annotations</a></li>
<li><a href="#blast-like-annotation">BLAST-like annotation</a></li>
<li><a href="#hmm-profiling-of-domains">HMM-profiling of domains</a></li>
<li><a href="#annotating-mags-against-the-uniprot-database-with-diamond">Annotating MAGs against the <em>UniProt</em> database with <em>diamond</em></a></li>
<li><a href="#annotating-mags-against-the-pfam-database-with-hmmer">Annotating MAGs against the <em>Pfam</em> database with <em>hmmer</em></a></li>
<li><a href="#evaluating-the-quality-of-gene-assignment">Evaluating the quality of gene assignment</a></li>
<li><a href="#differences-in-taxonomies">Differences in taxonomies</a></li>
<li><a href="#examples-of-various-genome-level-and-protein-taxonomies">Examples of various genome-level and protein taxonomies</a></li>
</ul>
</li>
</ul>
</div>
<hr/>
<h3 id="blast-like-gene-annotations-and-domain-annotations"><em>BLAST</em>-like gene annotations and domain annotations<a class="headerlink" href="#blast-like-gene-annotations-and-domain-annotations" title="Permanent link">¶</a></h3>
<p>Broadly speaking, there are two ways we perform gene annotations with protein sequences. Both compare our sequences of interest against a curated set of protein sequences for which function is known, or is strongly suspected. In each case, there are particular strenths to the approach and for particular research questions, one option may be favoured over another.</p>
<h4 id="blast-like-annotation">BLAST-like annotation<a class="headerlink" href="#blast-like-annotation" title="Permanent link">¶</a></h4>
<p>The first of these is the <code>BLAST</code> algorithm for sequence alignment. This approach performs pairwise alignment between the gene of interest (query sequence) and the sequences in the database (target sequence). <code>BLAST</code> searches each potential target sequence for <em>k</em>-mers identified in the query sequence. Where these <em>k</em>-mers are found in targets, the ends are extended out to try to create longer regions of highly similar sequence spans. Across this span, the tool identifies the longest span of characters (nucleotide or amino acid) that match within a scoring framework to return the length of the region (coverage) and the sequence identity over the span (identity).</p>
<p>The original tool for performing this kind of analysis was the <code>BLAST</code> tool. While <code>BLAST</code> and its variants are still excellent tools for performing this kind of sequence annotation, they suffer from a slow runtime speed due to the need to test each query sequence against every target sequence in the database. For this reason, several tools have been published which take the basic approach of <code>BLAST</code>, but augment it with methods to reduce the number of pairwise comparisons needed to identify targets with high sequence similarity to the query. Two popular pieces of software are the tools <code>usearch</code> <a href="http://www.drive5.com/usearch/">here</a> and <code>diamond</code> <a href="https://github.com/bbuchfink/diamond">here</a>.</p>
<h4 id="hmm-profiling-of-domains">HMM-profiling of domains<a class="headerlink" href="#hmm-profiling-of-domains" title="Permanent link">¶</a></h4>
<p>An alternate method for attributing function to query sequences is to consider them as a collection of independently functioning protein folding domains. This is the approach used in the <a href="http://hmmer.org/">HMMer</a> software, and the <em>Pfam</em>, <em>TIGRfam</em>, and <em>PANTHER</em> databases. In these analyses, the database consists not of individual sequences, but of Hidden Markov models built from a collection of proteins that share a common domain. These profiles build out a statistical map of the amino acid transitions (from position to position), variations (differences at a position), and insertions/deletions between positions in the domain across the different observations in the training database and apply these maps to the query data.</p>
<p>These exercises will take place in the <code>10.gene_annotation_and_coverage/</code> folder. </p>
<hr/>
<h3 id="annotating-mags-against-the-uniprot-database-with-diamond">Annotating MAGs against the <em>UniProt</em> database with <em>diamond</em><a class="headerlink" href="#annotating-mags-against-the-uniprot-database-with-diamond" title="Permanent link">¶</a></h3>
<p>For this exercise we are going to use <code>diamond</code> for performing our annotation. We have chosen to use this tool because it is faster than BLAST, and <code>usearch</code> comes with licensing restrictions that make it hard to work with in a shared computing environment like NeSI.</p>
<p>For this exercise we have created a diamond-compatible database from the 2018 release of the UniProt database.</p>
<p>For input files, the <code>predictions/</code> results from the previous gene prediction exercise have been copied over to <code>10.gene_annotation_and_coverage/predictions/</code>.</p>
<p>In general, <code>diamond</code> takes a pair of input files - the protein coding sequences we wish to annotate and the database we will use for this purpose. There are a few parameters that need to be tweaked for obtaining a useful output file, however.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>module<span class="w"> </span>purge
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>module<span class="w"> </span>load<span class="w"> </span>DIAMOND/2.0.15-GCC-11.3.0
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>diamond<span class="w"> </span><span class="nb">help</span>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a><span class="c1"># diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science</span>
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="c1"># Documentation, support and updates available at http://www.diamondsearch.org</span>
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="c1"># Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)</span>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="c1"># Syntax: diamond COMMAND [OPTIONS]</span>
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a>
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a><span class="c1"># Commands:</span>
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a><span class="c1"># ...</span>
<a href="#__codelineno-0-13" id="__codelineno-0-13" name="__codelineno-0-13"></a><span class="c1"># blastp  Align amino acid query sequences against a protein reference database</span>
<a href="#__codelineno-0-14" id="__codelineno-0-14" name="__codelineno-0-14"></a><span class="c1"># ...</span>
<a href="#__codelineno-0-15" id="__codelineno-0-15" name="__codelineno-0-15"></a>
<a href="#__codelineno-0-16" id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="c1"># General options:</span>
<a href="#__codelineno-0-17" id="__codelineno-0-17" name="__codelineno-0-17"></a><span class="c1"># --threads (-p)         number of CPU threads</span>
<a href="#__codelineno-0-18" id="__codelineno-0-18" name="__codelineno-0-18"></a><span class="c1"># --db (-d)              database file</span>
<a href="#__codelineno-0-19" id="__codelineno-0-19" name="__codelineno-0-19"></a><span class="c1"># --out (-o)             output file</span>
<a href="#__codelineno-0-20" id="__codelineno-0-20" name="__codelineno-0-20"></a><span class="c1"># --outfmt (-f)          output format</span>
<a href="#__codelineno-0-21" id="__codelineno-0-21" name="__codelineno-0-21"></a><span class="c1"># ...</span>
</code></pre></div>
<p>There are two output formats we can chose from which are useful for our analysis. We will obtain our output in the BLAST tabular format, which provides the annotation information in a simple-to-parse text file that can be viewed in any text or spreadsheet viewing tool. This will allow us to investigate and evaluate the quality of our annotations. </p>
<p>Awkwardly, <code>diamond</code> does not provide the headers for what the columns in the output table mean. <a href="http://www.metagenomics.wiki/tools/blast/blastn-output-format-6">This table</a> is a handy reference for how to interpret the output.</p>
<p>From here we can view important stastics for each query/target pairing such as the number of identical residues between sequences and the aligned length between query and target.</p>
<p>Before we begin, we need to create an directory for outputs.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>gene_annotations
</code></pre></div>
<p>Now, lets set up a slurm job to annotate each of our MAGs. </p>
<p>Create a new script</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>nano<span class="w"> </span>annotate_uniprot.sl
</code></pre></div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Paste in the script (update <code>&lt;YOUR FOLDER&gt;</code>)</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="c1">#SBATCH --job-name      annotate_uniprot</span>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a><span class="c1">#SBATCH --time          01:00:00</span>
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a><span class="c1">#SBATCH --mem           20GB</span>
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a><span class="c1">#SBATCH --cpus-per-task 20</span>
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a><span class="c1">#SBATCH --array         0-9</span>
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a><span class="c1">#SBATCH --error         %x_%A_%a.err</span>
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a><span class="c1">#SBATCH --output        %x_%A_%a.out</span>
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a><span class="c1"># Load modules</span>
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a>module<span class="w"> </span>purge
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a>module<span class="w"> </span>load<span class="w"> </span>DIAMOND/2.0.15-GCC-11.3.0
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a>
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a><span class="c1"># Working directory</span>
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a>
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a><span class="c1"># Variables</span>
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a><span class="nv">prot_file</span><span class="o">=</span>predictions/bin_<span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>.filtered.genes.no_metadata.faa
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a><span class="nv">out_file</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">prot_file</span><span class="si">}</span><span class="w"> </span>.faa<span class="k">)</span>
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a><span class="nv">db</span><span class="o">=</span>/nesi/nobackup/nesi02659/MGSS_resources_2022/databases/uniprot.20181026.dmnd
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a>
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a><span class="c1"># Run DIAMOND</span>
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>diamond<span class="w"> </span>blastp<span class="w"> </span>--threads<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>--max-target-seqs<span class="w"> </span><span class="m">5</span><span class="w"> </span>--evalue<span class="w"> </span><span class="m">0</span>.001<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-3-26" id="__codelineno-3-26" name="__codelineno-3-26"></a><span class="w">               </span>--db<span class="w"> </span><span class="nv">$db</span><span class="w"> </span>--query<span class="w"> </span><span class="si">${</span><span class="nv">prot_file</span><span class="si">}</span><span class="w"> </span>--outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-3-27" id="__codelineno-3-27" name="__codelineno-3-27"></a><span class="w">               </span>--out<span class="w"> </span>gene_annotations/<span class="si">${</span><span class="nv">out_file</span><span class="si">}</span>.uniprot.txt
</code></pre></div>
</div>
<p>Submit the script</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>sbatch<span class="w"> </span>annotate_uniprot.sl
</code></pre></div>
<hr/>
<h3 id="annotating-mags-against-the-pfam-database-with-hmmer">Annotating MAGs against the <em>Pfam</em> database with <em>hmmer</em><a class="headerlink" href="#annotating-mags-against-the-pfam-database-with-hmmer" title="Permanent link">¶</a></h3>
<p>The standard software for performing HMM-profiling annotation is <a href="http://hmmer.org/">hmmer</a>. Compared to <code>BLAST</code>, <code>FASTA</code>, and other sequence alignment and database search tools based on older scoring methodology, <code>HMMER</code> aims to be significantly more accurate and more able to detect remote homologs because of the strength of its underlying mathematical models. In the past, this strength came at significant computational expense, but in the new <code>HMMER3</code> project, <code>HMMER</code> is now essentially as fast as <code>BLAST</code>. </p>
<p><code>HMMER</code> will search one or more profiles against a sequence database for sequence hommologs, and for making sequence alignments, implementing profile hidden Markov models. In this exercise, we will perform a search using <code>hmmsearch</code>. For each profile in <em>hmmfile</em>, <code>HMMER</code> uses that query profile to search the target database of sequences indicated in <em>seqdb</em>, and output ranked lists of the sequences with the most significant matches to the profile. <code>hmmsearch</code> accepts any <em>fastA</em> file as target database input. It also accepts EMBL/UniProtKB text format, and Genbank format. It will automatically determine what format your file is in so you don’t have to specify it. </p>
<p>As we did with <code>diamond</code>, we will also have to modify some parameters to get the desired ouotput. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>module<span class="w"> </span>load<span class="w"> </span>HMMER/3.3.2-GCC-11.3.0
</code></pre></div>
<details class="abstract">
<summary><code>hmmsearch -h</code></summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="c1"># hmmsearch :: search profile(s) against a sequence database</span>
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a><span class="c1"># HMMER 3.3.2 (Nov 2020); http://hmmer.org/</span>
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a><span class="c1"># Copyright (C) 2020 Howard Hughes Medical Institute.</span>
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="c1"># Freely distributed under the BSD open source license.</span>
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a><span class="c1"># - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a><span class="c1"># Usage: hmmsearch [options] &lt;hmmfile&gt; &lt;seqdb&gt;</span>
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a><span class="c1"># Basic options:</span>
<a href="#__codelineno-6-9" id="__codelineno-6-9" name="__codelineno-6-9"></a><span class="c1">#  -h : show brief help on version and usage</span>
<a href="#__codelineno-6-10" id="__codelineno-6-10" name="__codelineno-6-10"></a>
<a href="#__codelineno-6-11" id="__codelineno-6-11" name="__codelineno-6-11"></a><span class="c1"># Options directing output:</span>
<a href="#__codelineno-6-12" id="__codelineno-6-12" name="__codelineno-6-12"></a><span class="c1"># ...</span>
<a href="#__codelineno-6-13" id="__codelineno-6-13" name="__codelineno-6-13"></a><span class="c1"># --tblout &lt;f&gt;     : save parseable table of per-sequence hits to file &lt;f&gt;</span>
<a href="#__codelineno-6-14" id="__codelineno-6-14" name="__codelineno-6-14"></a><span class="c1"># ....</span>
<a href="#__codelineno-6-15" id="__codelineno-6-15" name="__codelineno-6-15"></a>
<a href="#__codelineno-6-16" id="__codelineno-6-16" name="__codelineno-6-16"></a><span class="c1"># Options controlling reporting thresholds:</span>
<a href="#__codelineno-6-17" id="__codelineno-6-17" name="__codelineno-6-17"></a><span class="c1"># ...</span>
<a href="#__codelineno-6-18" id="__codelineno-6-18" name="__codelineno-6-18"></a><span class="c1"># -E &lt;x&gt;     : report sequences &lt;= this E-value threshold in output  [10.0]  (x&gt;0)</span>
<a href="#__codelineno-6-19" id="__codelineno-6-19" name="__codelineno-6-19"></a><span class="c1"># ...</span>
<a href="#__codelineno-6-20" id="__codelineno-6-20" name="__codelineno-6-20"></a>
<a href="#__codelineno-6-21" id="__codelineno-6-21" name="__codelineno-6-21"></a><span class="c1"># Other expert options:</span>
<a href="#__codelineno-6-22" id="__codelineno-6-22" name="__codelineno-6-22"></a><span class="c1"># ...</span>
<a href="#__codelineno-6-23" id="__codelineno-6-23" name="__codelineno-6-23"></a><span class="c1"># --cpu &lt;n&gt;     : number of parallel CPU workers to use for multithreads</span>
<a href="#__codelineno-6-24" id="__codelineno-6-24" name="__codelineno-6-24"></a><span class="c1"># ...</span>
</code></pre></div>
</details>
<p>We are now going to submit another slurm job to annotate our MAGs using the <a href="https://www.ebi.ac.uk/interpro/download/Pfam/">Pfam database</a>. Pfam used to have a standalone website, but it has recently been integrated into InterPro maintained by the European Bioinformatics Institute (see <a href="https://xfam.wordpress.com/2022/08/04/pfam-website-decommission/">announcement</a>). Matching sequences to a <code>Pfam</code> entry allows us to transfer the functional information from an experimentally characterised sequence to uncharacterised sequences in the same entry. <code>Pfam</code> then provides comprehensive annotation for each entry.</p>
<p>Create a new script</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>nano<span class="w"> </span>annotate_pfam.sl
</code></pre></div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Paste in the script (update <code>&lt;YOUR FOLDER&gt;</code>)</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a>
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a><span class="c1">#SBATCH --job-name      annotate_pfam</span>
<a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a><span class="c1">#SBATCH --time          00:20:00</span>
<a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a><span class="c1">#SBATCH --mem           5GB</span>
<a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a href="#__codelineno-8-8" id="__codelineno-8-8" name="__codelineno-8-8"></a><span class="c1">#SBATCH --array         0-9</span>
<a href="#__codelineno-8-9" id="__codelineno-8-9" name="__codelineno-8-9"></a><span class="c1">#SBATCH --error         %x_%A_%a.err</span>
<a href="#__codelineno-8-10" id="__codelineno-8-10" name="__codelineno-8-10"></a><span class="c1">#SBATCH --output        %x_%A_%a.out</span>
<a href="#__codelineno-8-11" id="__codelineno-8-11" name="__codelineno-8-11"></a>
<a href="#__codelineno-8-12" id="__codelineno-8-12" name="__codelineno-8-12"></a><span class="c1"># Load modules</span>
<a href="#__codelineno-8-13" id="__codelineno-8-13" name="__codelineno-8-13"></a>module<span class="w"> </span>purge
<a href="#__codelineno-8-14" id="__codelineno-8-14" name="__codelineno-8-14"></a>module<span class="w"> </span>load<span class="w"> </span>HMMER/3.3.2-GCC-11.3.0
<a href="#__codelineno-8-15" id="__codelineno-8-15" name="__codelineno-8-15"></a>
<a href="#__codelineno-8-16" id="__codelineno-8-16" name="__codelineno-8-16"></a><span class="c1"># Working directory</span>
<a href="#__codelineno-8-17" id="__codelineno-8-17" name="__codelineno-8-17"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage
<a href="#__codelineno-8-18" id="__codelineno-8-18" name="__codelineno-8-18"></a>
<a href="#__codelineno-8-19" id="__codelineno-8-19" name="__codelineno-8-19"></a><span class="c1"># Variables</span>
<a href="#__codelineno-8-20" id="__codelineno-8-20" name="__codelineno-8-20"></a><span class="nv">prot_file</span><span class="o">=</span>predictions/bin_<span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span>.filtered.genes.no_metadata.faa
<a href="#__codelineno-8-21" id="__codelineno-8-21" name="__codelineno-8-21"></a><span class="nv">out_file</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">prot_file</span><span class="si">}</span><span class="w"> </span>.faa<span class="k">)</span>
<a href="#__codelineno-8-22" id="__codelineno-8-22" name="__codelineno-8-22"></a><span class="nv">db</span><span class="o">=</span>/nesi/nobackup/nesi02659/MGSS_resources_2022/databases/Pfam-A.hmm
<a href="#__codelineno-8-23" id="__codelineno-8-23" name="__codelineno-8-23"></a>
<a href="#__codelineno-8-24" id="__codelineno-8-24" name="__codelineno-8-24"></a><span class="c1"># Run HMMER</span>
<a href="#__codelineno-8-25" id="__codelineno-8-25" name="__codelineno-8-25"></a>hmmsearch<span class="w"> </span>--tblout<span class="w"> </span>gene_annotations/<span class="si">${</span><span class="nv">out_file</span><span class="si">}</span>.pfam.txt<span class="w"> </span>-E<span class="w"> </span><span class="m">0</span>.001<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-8-26" id="__codelineno-8-26" name="__codelineno-8-26"></a><span class="w">          </span>--cpu<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-8-27" id="__codelineno-8-27" name="__codelineno-8-27"></a><span class="w">          </span><span class="si">${</span><span class="nv">db</span><span class="si">}</span><span class="w"> </span><span class="si">${</span><span class="nv">prot_file</span><span class="si">}</span>
</code></pre></div>
</div>
<p>Submit the script</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>sbatch<span class="w"> </span>annotate_pfam.sl
</code></pre></div>
<hr/>
<h3 id="evaluating-the-quality-of-gene-assignment">Evaluating the quality of gene assignment<a class="headerlink" href="#evaluating-the-quality-of-gene-assignment" title="Permanent link">¶</a></h3>
<p>Determining how trustworthy a gene annotation is can be a very tricky process. How similar do protein sequences need to be to perform the same function? The answer is surprisingly low. A bioinformatic analysis performed in 1999 identified that proteins with as little as 20 - 35% sequence identity can still share the same function (<a href="https://doi.org/10.1093/protein/12.2.85">Rost, 1999</a>), but this is not a universal occurrence. When evaluating annotations, consider the following questions:</p>
<div class="admonition quote">
<ol>
<li>What is the amino acid identity along the aligned region?</li>
<li>What is the amino acid <em>similarity</em> between the aligned region?</li>
<li>What is the coverage as a percentage of the query and target genes?</li>
<li>If we infer a phylogeny of this query gene with references from the target family, is a stable tree resolved?</li>
<li>Does the inclusion of this gene function make sense in the context of the organism's taxonomy?</li>
<li>Does the gene sit on a long contig that is core to the MAG, or is it a short contig carrying only a single gene?</li>
<li>If we are uncertain of a particular annotation, does the predicted gene occur in an operon? If so, are the other genes present in the annotation?</li>
</ol>
</div>
<p>We must also remain aware of the potential for incorrectly annotated genes in the annotation database and that proteins can perform multiple functions (and may therefore be attributed multiple, inconsistent annotations). Furthermore, it is also important to consider exactly which part of the target gene the alignment is happening across. There are several catalytic centers of enzymes, such as the Fe-S cofactor, which are shared across many different proteins, and if your annotation is only spanning one of these regions then it may simply be the case that you are identifying a generic electron accepting or donating domain.</p>
<hr/>
<h3 id="differences-in-taxonomies">Differences in taxonomies<a class="headerlink" href="#differences-in-taxonomies" title="Permanent link">¶</a></h3>
<p>Another way to determine if an annotation 'belongs' in the MAG of interest is to consider the predicted taxonomy of the query gene with that of the MAG itself. For example, if you detect a <em>Desulfovibrio</em>-like <em>dsrA</em> seuqence in a bin that has been classified as belonging to the genus <em>Desulfovibrio</em> then it is probably a safe bet that the annotation is correct.</p>
<p>However, when comparing taxonomic assignments, it is important to be aware of the differing taxonomic schemas that are circulating in the microbiological and bioinformatic literature and to know how to reconcile their differences. Similar to how the 16S rRNA gene taxonomies provided by <em>SILVA</em>, <em>Greengenes</em>, and <em>RDP</em> taxonomies all differ in some aspects, there are multiple competing taxonomies in protein databases.</p>
<h4 id="examples-of-various-genome-level-and-protein-taxonomies">Examples of various genome-level and protein taxonomies<a class="headerlink" href="#examples-of-various-genome-level-and-protein-taxonomies" title="Permanent link">¶</a></h4>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI</a></li>
<li><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a></li>
</ul>
<p>This problem exists despite the existance of a formal <a href="https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijsem.0.000778">Code</a> for the naming of bacteria and archaea, because </p>
<div class="admonition quote">
<ol>
<li>There are no rules governing how we define the grouping of these names together, other than for type species</li>
<li>Defunct synonyms and basonyms are not correctly purged from taxonomy lists (this is quite noticable with the NCBI taxonomy)</li>
<li>Valid names cannot be assigned for uncultivate organisms, meaning there are many informal placeholder names in the literature. For example, clades like WPS-2, SAR324, and SAUL are widely cited in the literature despite having no official standing</li>
</ol>
</div>
<p>It is therefore important to periodically sanity check your taxonomic annotations in order to avoid splitting taxa based on spelling differences or the use of historic names that have since been reclassified.</p>
<hr/>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>