
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day2/ex9_refining_bins/" rel="canonical"/>
<link href="../ex8_bin_dereplication/" rel="prev"/>
<link href="../../day3/ex11_coverage_and_taxonomy/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Manual bin refinement - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#manual-bin-refinement">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Manual bin refinement
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day 1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../ex6_initial_binning/">
        Day 2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Day 3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day 4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction I: Shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction II: HPC and job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Filter raw reads by quality
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly I: Assembling contigs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly II: Varying the parameters
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Assembly evaluation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex7_initial_binning/">
        Binning with multiple tools
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Manual bin refinement
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Manual bin refinement
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#preparing-input-files-for-vizbin">
    Preparing input files for VizBin
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#projecting-a-t-sne-and-examining-bin-clusters">
    Projecting a t-SNE and examining bin clusters
  </a>
<nav aria-label="Projecting a t-SNE and examining bin clusters" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#initiate-vizbin-within-the-virtual-desktop-environment">
    Initiate VizBin within the Virtual Desktop environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#load-input-files">
    Load input files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#executing-the-t-sne">
    Executing the t-SNE
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#contigs-coloured-by-bin">
    Contigs coloured by bin
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#picking-refined-bins">
    Picking refined bins
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#1-export-vizbin-clusters">
    1. Export VizBin clusters
  </a>
<nav aria-label="1. Export VizBin clusters" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#highlight-a-cluster-to-zoom-into">
    Highlight a cluster to zoom into
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#select-the-cluster-to-export">
    Select the cluster to export
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#export-the-cluster">
    Export the cluster
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-export-potentially-problematic-contigs">
    2. Export potentially problematic contigs
  </a>
<nav aria-label="2. Export potentially problematic contigs" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#select-problematic-contigs-to-examine">
    Select problematic contigs to examine
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-refining-and-filtering-problematic-contigs-from-bins">
    (Optional) Refining and filtering problematic contigs from bins
  </a>
<nav aria-label="(Optional) Refining and filtering problematic contigs from bins" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#create-a-count-table-of-counts-of-our-problematic-contigs-across-each-bin">
    Create a count table of counts of our problematic contigs across each bin
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#generate-a-list-of-contigs-to-exclude-from-filtering">
    Generate a list of contigs to exclude from filtering
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#create-final-vb_omit_contigs_filteredtxt-list-of-contigs-to-filter-from-bins">
    Create final vb_omit_contigs_filtered.txt list of contigs to filter from bins
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#filter-suspect-contigs-based-on-vizbin-analysis-from-the-bin-data">
    Filter suspect contigs (based on VizBin analysis) from the bin data
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-comparing-pre-and-post-filtered-bins-via-checkm">
    (Optional) Comparing pre- and post-filtered bins via CheckM
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-scripts-for-processing-data-with-esomana">
    (Optional) Scripts for processing data with ESOMana
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11.1_phylogenomics/">
        Phylogenomics
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation I: BLAST-like and HMM
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation II: DRAM and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation III: DRAM distillation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Introduction to data presentation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        CAZy heatmaps
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#preparing-input-files-for-vizbin">
    Preparing input files for VizBin
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#projecting-a-t-sne-and-examining-bin-clusters">
    Projecting a t-SNE and examining bin clusters
  </a>
<nav aria-label="Projecting a t-SNE and examining bin clusters" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#initiate-vizbin-within-the-virtual-desktop-environment">
    Initiate VizBin within the Virtual Desktop environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#load-input-files">
    Load input files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#executing-the-t-sne">
    Executing the t-SNE
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#contigs-coloured-by-bin">
    Contigs coloured by bin
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#picking-refined-bins">
    Picking refined bins
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#1-export-vizbin-clusters">
    1. Export VizBin clusters
  </a>
<nav aria-label="1. Export VizBin clusters" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#highlight-a-cluster-to-zoom-into">
    Highlight a cluster to zoom into
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#select-the-cluster-to-export">
    Select the cluster to export
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#export-the-cluster">
    Export the cluster
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-export-potentially-problematic-contigs">
    2. Export potentially problematic contigs
  </a>
<nav aria-label="2. Export potentially problematic contigs" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#select-problematic-contigs-to-examine">
    Select problematic contigs to examine
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-refining-and-filtering-problematic-contigs-from-bins">
    (Optional) Refining and filtering problematic contigs from bins
  </a>
<nav aria-label="(Optional) Refining and filtering problematic contigs from bins" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#create-a-count-table-of-counts-of-our-problematic-contigs-across-each-bin">
    Create a count table of counts of our problematic contigs across each bin
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#generate-a-list-of-contigs-to-exclude-from-filtering">
    Generate a list of contigs to exclude from filtering
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#create-final-vb_omit_contigs_filteredtxt-list-of-contigs-to-filter-from-bins">
    Create final vb_omit_contigs_filtered.txt list of contigs to filter from bins
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#filter-suspect-contigs-based-on-vizbin-analysis-from-the-bin-data">
    Filter suspect contigs (based on VizBin analysis) from the bin data
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-comparing-pre-and-post-filtered-bins-via-checkm">
    (Optional) Comparing pre- and post-filtered bins via CheckM
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-scripts-for-processing-data-with-esomana">
    (Optional) Scripts for processing data with ESOMana
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="manual-bin-refinement">Manual bin refinement<a class="headerlink" href="#manual-bin-refinement" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#preparing-input-files-for-vizbin">Preparing input files for <code>VizBin</code></a></li>
<li><a href="#projecting-a-t-sne-and-examining-bin-clusters">Projecting a <em>t-SNE</em> using <code>VizBin</code> and examining bin clusters</a></li>
<li><a href="#picking-refined-bins">Picking refined bins</a></li>
<li><a href="#1-export-vizbin-clusters">Export <code>VizBin</code> clusters</a></li>
<li><a href="#2-export-potentially-problematic-contigs">Export potentially problematic contigs</a></li>
<li><a href="#optional-refining-and-filtering-problematic-contigs-from-bins"><em>(Optional)</em> Refining and filtering problematic contigs from bins</a></li>
<li><a href="#optional-comparing-pre--and-post-filtered-bins-via-checkm"><em>(Optional)</em> Comparing pre- and post-filtered bins via <em>CheckM</em></a></li>
<li><a href="#optional-scripts-for-processing-data-with-esomana"><em>(Optional)</em> Scripts for processing data with <code>ESOMana</code></a></li>
</ul>
</div>
<hr/>
<h2 id="preparing-input-files-for-vizbin">Preparing input files for <code>VizBin</code><a class="headerlink" href="#preparing-input-files-for-vizbin" title="Permanent link">¶</a></h2>
<div class="admonition quote">
<p class="admonition-title">What is <code>VizBin</code>?</p>
<p><a href="http://claczny.github.io/VizBin/"><code>VizBin</code></a> is a handy, GUI-based tool for creating ordinations of our binning data using the <a href="https://lvdmaaten.github.io/tsne/">t-Distributed Stochastic Neighbor Embedding (t-SNE)</a> algorithm to project high-dimensional data down into a 2D plot that preserves clustering information. There's a really good video on <a href="https://www.youtube.com/watch?v=NEaUSP4YerM">YouTube</a> that explains how the algorithm works in high-level terms, but for our purposes you can really consider it as a similar approach to a PCA or NMDS.</p>
<p>On its own, <code>VizBin</code> takes a set of contigs and performs the <em>t-SNE</em> projection using compositional data. We can optionally provide it files that annotate contigs as belonging to particular bins and a file that adds coverage data to be considered when clustering. Unfortuantely, at this stage <code>VizBin</code> only allows a single coverage value per contig, which is not ideal. This is because <code>VizBin</code> only uses coverage as a means to modify the visualisation, not the ordination itself. It is possible to create your own <em>t-SNE</em> projection using multiple coverage values, however this is beyond the scope of today's exercise, and here we will be providing <code>VizBin</code> with coverage values for sample1 only. </p>
<p>The only required input file for <code>VizBin</code> is a single <em>.fna</em> file of the concatenated bins. An additional annotation file containing per-contig coverage values and bin IDs can also be provided. Colouring contigs by bin is a really effective way to spot areas that might need refinement.</p>
<p>In the interests of time today, the input files have been generated and are provided in the <code>6.bin_refinement/</code> folder: </p>
<ul>
<li><code>all_bins.fna</code> is a concatenation of the bins of <em>fragmented</em> sub-contigs (fragmented to 20k)</li>
<li><code>all_bins.sample1.vizbin.ann</code> is the annotation file containing per-subcontig coverage, label (bin ID), and length values.</li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Contig fragments as input for <code>VizBin</code></p>
<p>When running <code>VizBin</code>, it is often preferable to split long contigs into smaller pieces in order to increase the density of clustering in the <strong>t-SNE</strong>. The data we are working with today are based on our bins output by <code>DAS_Tool</code> in the last binning exercise, but have been further processed using the <code>cut_up_fasta.py</code> script that comes with the binning tool <code>CONCOCT</code> to cut long contigs into 20k fragments. When reviewing our <code>VizBin</code> plots and outputs, it is important to remember that here we are looking at the <strong>fragmented sub-contigs</strong>, rather than the full complete contigs (the importance of this will be clear when we are reviewing our <code>vb_count_table.txt</code> later in this exercise).</p>
</div>
<p>For future reference, and for working with your own data, a step-by-step process for generating these files from the dereplicated bins generated by <code>DAS_Tool</code> has been provided as an <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">Appendix</a>.</p>
<p>For this section, we will be working within <code>6.bin_refinement/</code>. Let's first have a quick look at the annotation file.</p>
<div class="admonition terminal-2">
<p class="admonition-title">Navigate to working directory</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1"># Navigate to correct directory</span>
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/6.bin_refinement
</code></pre></div>
</div>
<div class="admonition terminal-2">
<p class="admonition-title">Inspect <code>all_bins.sample1.vizbin.ann</code></p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>all_bins.sample1.vizbin.ann
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>coverage,label,length
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>16.5255,bin_0.chopped,20000
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a>17.8777,bin_0.chopped,20000
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>17.6983,bin_0.chopped,20000
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>16.7296,bin_0.chopped,20000
</code></pre></div>
</div>
<p>This file is a comma-delimited table (csv file) that presents the information in the way that <code>VizBin</code> expects it. The order of rows in this file corresponds to the order of contigs in the concatenated FASTA file of our fragmented bins, <code>all_bins.fna</code>.</p>
<p>Create a few variations of the <em>.ann</em> file with various columns removed, in order to examine the different outputs they can generate.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio"><input id="__tabbed_1_2" name="__tabbed_1" type="radio"><div class="tabbed-labels"><label for="__tabbed_1_1">Bin ID only</label><label for="__tabbed_1_2">Bin ID and coverage without length</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>cut<span class="w"> </span>-f2<span class="w"> </span>-d<span class="w"> </span><span class="s1">','</span><span class="w"> </span>all_bins.sample1.vizbin.ann<span class="w"> </span>&gt;<span class="w"> </span>all_bins.sample1.vizbin.bin_only.ann
</code></pre></div>
</div>
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>cut<span class="w"> </span>-f1,2<span class="w"> </span>-d<span class="w"> </span><span class="s1">','</span><span class="w"> </span>all_bins.sample1.vizbin.ann<span class="w"> </span>&gt;<span class="w"> </span>all_bins.sample1.vizbin.no_length.ann
</code></pre></div>
</div>
</div>
</input></input></div>
</div>
<hr/>
<h2 id="projecting-a-t-sne-and-examining-bin-clusters">Projecting a <em>t-SNE</em> and examining bin clusters<a class="headerlink" href="#projecting-a-t-sne-and-examining-bin-clusters" title="Permanent link">¶</a></h2>
<p>We can now use these files in <code>VizBin</code> to curate the contigs in our bins. We will load and view the data in a few different steps.</p>
<p>For this exercise, we will be using the <em>Virtual Desktop</em> on NeSI which allows us to use programmes with graphical user interfaces (GUI) within the NeSI computing environment.</p>
<div class="admonition jupyter">
<p class="admonition-title">Initiate the Virtual Desktop</p>
<ol>
<li>Click on <kbd>File</kbd> on the top left corner of the Jupyter Hub, and then select <kbd>New Launcher</kbd>. You can also click on the small <kbd>+</kbd> on the tabbed bar next to your terminal tab.</li>
<li>Look for the Virtual Desktop icon and then click on centre of the icon.
    <center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/VirtualDesktop.PNG"><img alt="virtual desktop icon" src="../../figures/VirtualDesktop.PNG"/></a>
</center></li>
<li>A new browser tab named 'TurboVNC:...' should appear and the Virtual Desktop should load.</li>
<li>A successful instance of the Virtual Desktop should look like a desktop environment.</li>
</ol>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/VirtualDesktopInitiated.PNG"><img alt="virtual desktop environment" src="../../figures/VirtualDesktopInitiated.PNG" width="600px"/></a>
</center></p>
</div>
<div class="admonition note">
<p class="admonition-title">Running <code>VizBin</code>: local vs remote</p>
<p>Running <code>VizBin</code> remotely (e.g. within NeSI) can be slow with full data sets. Running a GUI (such as a program like <code>VizBin</code>) remotely can also require additional set up on some PCs. For day-to-day work, we recommend installing <code>VizBin</code> on your local machine and downloading the relevant input files (e.g. via <code>scp ...</code>) to run locally.</p>
</div>
<!--

`VizBin` cannot be opened within the NeSI `Jupyter hub`. For today's exercise, open `VizBin` via either:

**A. Install and run `VizBin` locally**

* Download `VizBin-dist.jar` from [here](https://github.com/claczny/VizBin/releases/tag/v1.0.0)
* You may also need to [install Java locally](https://www.java.com/en/download/manual.jsp) (choose the relevant version for your machine).
* Download the required files from NeSI to your machine (via `scp`, or right-click > download, from within the `Jupyter hub`), or download the example files from [this link](../resources/vizbin_files.zip).
* Double-click `VizBin-dist.jar` to open `VizBin`, and then follow the steps below.

**OR**

**B. Log in via a standard terminal.** 

* Log into NeSI via a standard terminal (*not* the terminal within `Jupyter hub`) and run the following command (*NOTE: this may return an error if X11 forwarding is not set up for your machine. In this case, proceed with the first option above*).

<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/6.bin_refinement/
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>java<span class="w"> </span>-jar<span class="w"> </span>../tools/vizbin.jar
</code></pre></div>

If this fails to open on your PC, or if it runs prohibitively slowly, team up with 2-3 others in the workshop to run through this exercise together on one machine.

-->
<h3 id="initiate-vizbin-within-the-virtual-desktop-environment">Initiate <code>VizBin</code> within the Virtual Desktop environment<a class="headerlink" href="#initiate-vizbin-within-the-virtual-desktop-environment" title="Permanent link">¶</a></h3>
<ol>
<li>
<p>In the Virtual Desktop, click on the terminal icon.
    <center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/TerminalEmulator.PNG"><img alt="terminal emulator" src="../../figures/TerminalEmulator.PNG"/></a>
</center></p>
</li>
<li>
<p><code>VizBin</code> is a Java programme, therefore we will need to load the Java module for it to work. In the terminal, type the following to load the Java module</p>
<div class="admonition warning">
<p class="admonition-title">Copy/paste in the Virtual Desktop</p>
<p>You will not be able to copy text from outside the Virtual Desktop and paste into the Virtual Desktop, in which case you will need to manually type these commands.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>module<span class="w"> </span>load<span class="w"> </span>Java/17
</code></pre></div>
</div>
</li>
<li>
<p>In the terminal, navigate to your directory where the Java file resides</p>
<div class="admonition warning">
<p class="admonition-title">Remember to replace <code>YOUR FOLDER</code> with your user name.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/6.bin_refinement/
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a>ls
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>all_bins.fna  all_bins.sample1.vizbin.ann  vizbin_count_table_2022.sh  VizBin-dist.jar  vizbin_example_exports
</code></pre></div>
</div>
</li>
<li>
<p>Type the following into your Virtual Desktop terminal to initiate VizBin.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>java<span class="w"> </span>-jar<span class="w"> </span>VizBin-dist.jar
</code></pre></div>
</div>
</li>
<li>
<p>A successful launch of <code>VizBin</code> will look like the following:
   <center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/SuccessVizBin.PNG"><img alt="vizbin launch" src="../../figures/SuccessVizBin.PNG"/></a>
</center></p>
</li>
</ol>
<h3 id="load-input-files">Load input files<a class="headerlink" href="#load-input-files" title="Permanent link">¶</a></h3>
<p>Once <code>VizBin</code> is open, to get started, click the 'Choose...' button then navigate to the FASTA file <code>all_bins.fna</code>.</p>
<div class="admonition tip">
<p class="admonition-title"><code>VizBin</code> directory</p>
<p>It is likely that when we click on the 'Choose...' button, the directory that is open will be our NeSI home directory. In that case, we can type in the input FASTA file in the 'File Name' bar: <code>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/6.bin_refinement/all_bins.fna</code></p>
</div>
<p><center><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_load_fasta_2022.png"><img alt="image" src="../../figures/ex10_load_fasta_2022.png"/></a></center></p>
<p>Once this is imported, use the 'Show additional options' button to expose the advanced options, and add your <strong>'bin only'</strong> <em>.ann</em> file into the 'Annotation file (optional)' field.</p>
<p><center><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_load_ann_2022.png"><img alt="image" src="../../figures/ex10_load_ann_2022.png"/></a></center></p>
<h3 id="executing-the-t-sne">Executing the <em>t-SNE</em><a class="headerlink" href="#executing-the-t-sne" title="Permanent link">¶</a></h3>
<p>For now leave all other parameters as default. Click the 'Start' button to begin building the ordination. When it completes, you should see an output similar to the following:</p>
<h3 id="contigs-coloured-by-bin">Contigs coloured by bin<a class="headerlink" href="#contigs-coloured-by-bin" title="Permanent link">¶</a></h3>
<p><center><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_bin_only_2022.png"><img alt="image" src="../../figures/ex10_bin_only_2022.png" width="600"/></a></center></p>
<details class="note">
<summary>Additional annotations by length and coverage</summary>
<p>If you input <code>all_bins.sample1.vizbin.ann</code> as your annotation file, you can see that the visualisation takes contig length (represented by point size) and coverage (represented by opacity) into account.</p>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_all_ann_2022.png"><img alt="image" src="../../figures/ex10_all_ann_2022.png"/></a>
</center></p>
</details>
<p>Similar to other projection techniques, we interpret the closeness of points as a proxy for how similar they are, and because of our <em>.ann</em> file we can see which contigs belong to the same bin.</p>
<hr/>
<h2 id="picking-refined-bins">Picking refined bins<a class="headerlink" href="#picking-refined-bins" title="Permanent link">¶</a></h2>
<p>We can use the interactive GUI to pick the boundaries of new bins, or to identify contigs which we do not believe should be retained in the data. Have a play around with the interface, testing out the following commands:</p>
<ol>
<li>Left-click and drag: Highlight an area of the ordination to zoom into</li>
<li>Right-click, 'Zoom Out', 'Both Axes': Rest of the view</li>
<li>Left-click several points: Create a selection of contigs to extract from the data</li>
<li>Right-click, 'Selection', 'Export': Save the selected contigs into a new file</li>
<li>Right-click, 'Selection', 'Clear selection': Clear the current selection</li>
</ol>
<p>How you proceed in this stage is up to you. You can either select bins based on their boundary, and call these the refined bins. Alternatively, you could select outlier contigs and examine these in more detail to determine whether or not they were correctly placed into the bin. Which way you proceed really depends on how well the ordination resolves your bins, and it might be that both approaches are needed.</p>
<p>Today, we will run through an example of selecting potentially problematic (sub)contigs, and then deciding whether or not we want to filter these contigs out of our refined bins. We can use a combination of <code>VizBin</code> and <code>seqmagick</code> to remove contigs from bins where we do not trust the placement of the contig. We are aiming to reduce each bin to a trusted set of contigs.</p>
<h2 id="1-export-vizbin-clusters">1. Export <code>VizBin</code> clusters<a class="headerlink" href="#1-export-vizbin-clusters" title="Permanent link">¶</a></h2>
<p>First, for each <code>VizBin</code> cluster, select the area around the cluster (via multiple left-clicks around the cluster), right-click, 'Selection', 'Export'. Save this output as <code>cluster_1.fna</code>. </p>
<p>Try this for one or two clusters. In practice, we would do this for each <code>VizBin</code> cluster, saving each as a new <code>cluster_n.fna</code> file.</p>
<h3 id="highlight-a-cluster-to-zoom-into">Highlight a cluster to zoom into<a class="headerlink" href="#highlight-a-cluster-to-zoom-into" title="Permanent link">¶</a></h3>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_select_to_zoom_2022.png"><img alt="image" src="../../figures/ex10_select_to_zoom_2022.png" width="600"/></a>
</center></p>
<h3 id="select-the-cluster-to-export">Select the cluster to export<a class="headerlink" href="#select-the-cluster-to-export" title="Permanent link">¶</a></h3>
<p>Left-click several points around the cluster</p>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_select_cluster_2022.png"><img alt="image" src="../../figures/ex10_select_cluster_2022.png" width="600"/></a>
</center></p>
<h3 id="export-the-cluster">Export the cluster<a class="headerlink" href="#export-the-cluster" title="Permanent link">¶</a></h3>
<p>Right-click, 'Selection', 'Export'. Save the output as <code>cluster_1.fna</code>. </p>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_export_2022.png"><img alt="image" src="../../figures/ex10_export_2022.png" width="600"/></a>
</center></p>
<h2 id="2-export-potentially-problematic-contigs">2. Export potentially problematic contigs<a class="headerlink" href="#2-export-potentially-problematic-contigs" title="Permanent link">¶</a></h2>
<h3 id="select-problematic-contigs-to-examine">Select problematic contigs to examine<a class="headerlink" href="#select-problematic-contigs-to-examine" title="Permanent link">¶</a></h3>
<p>Zoom in,  make a selection of potentially problematic contigs, and export as above.</p>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex10_select_outlier_2022.png"><img alt="image" src="../../figures/ex10_select_outlier_2022.png" width="600"/></a>
</center></p>
<p>Try this for one or two problematic contigs (or subsets of contigs). In practice, you could repeat this for all potentially problematic contigs, saving each export as a new <code>contigs_n.fna</code> file.</p>
<div class="admonition abstract">
<p class="admonition-title">Note</p>
<p>For the subsequent step using <code>vizbin_count_table_2022.sh</code>, all exported cluster files must share a common prefix (e.g. <code>cluster...fna</code>), and all files of problematic contigs must also share a common prefix (e.g. <code>contigs...fna</code>).*</p>
</div>
<hr/>
<h2 id="optional-refining-and-filtering-problematic-contigs-from-bins"><em>(Optional)</em> Refining and filtering problematic contigs from bins<a class="headerlink" href="#optional-refining-and-filtering-problematic-contigs-from-bins" title="Permanent link">¶</a></h2>
<h3 id="create-a-count-table-of-counts-of-our-problematic-contigs-across-each-bin">Create a count table of counts of our problematic contigs across each bin<a class="headerlink" href="#create-a-count-table-of-counts-of-our-problematic-contigs-across-each-bin" title="Permanent link">¶</a></h3>
<p>You'll recall that, prior running <code>VizBin</code>, the contigs in our bins were first cut into fragments to improve the density of the clusters in the <em>t-SNE</em> projection. As such, the problematic contigs we have exported from <code>VizBin</code> are <em>sub-contig</em> fragments, rather than full contigs from our bins. It is entirely possible that different fragments of the original contigs have been placed in different clusters during our <code>VizBin</code> analysis - including cases where most sub-contigs have clustered with the bin we expect, and a small number have been identified as "problematic" (i.e. clustered with other bins). Based on the information from these extracted problematic sub-contigs, we now have to carefully consider whether or not we want to remove the <em>full</em> contig from our bin data.</p>
<p>To do this, we will generate a table containing each exported "problematic" sub-contig, and counts of how many of its sister sub-contigs (each of the other sub-contig fragments derived from the same original parent contig) fall into each <code>VizBin</code> cluster.</p>
<p>For this exercise, a folder of the exported files from <code>VizBin</code> for all clusters (<code>cluster_[1-n].fna</code>) and problematic sub-contigs (<code>contigs_[1-n].fna</code>) has been provided at <code>vizbin_example_exports/</code></p>
<p>We will input these files to the shell script <code>vizbin_count_table_2022.sh</code> to generate a count table of the exported subcontigs across each <code>VizBin</code> cluster (<code>vb_count_table.txt</code>), as well as a working list of contigs to potentially remove from our final bin data (<code>vb_omit_contigs_tmp.txt</code>).</p>
<p>For future reference, a copy of this script is available for download <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/tree/master/docs/scripts">here</a>.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a>./vizbin_count_table_2022.sh<span class="w"> </span>-i<span class="w"> </span>vizbin_example_exports/
</code></pre></div>
</div>
<p>The only required input to <code>vizbin_count_table_2022.sh</code> is the path to the cluster and contigs files exported from <code>VizBin</code>. By default, the script looks for the prefix <code>cluster...</code> for the cluster file names, <code>contig...</code> for the files of problematic sub-contigs, and the file extension <code>.fna</code> for each. The arguments <code>-s &lt;contig_file_prefix&gt; -c &lt;cluster_file_prefix&gt; -e &lt;fasta_file_extension&gt;</code> can optionally be provided if your file name formats differ from the default.</p>
<p>View the output count table:</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a>less<span class="w"> </span>vb_count_table.txt
</code></pre></div>
</div>
<p>Example excerpt:</p>
<table>
<thead>
<tr>
<th align="left">Subcontig_ID</th>
<th align="center">Subcontig_vb_cluster</th>
<th align="center">cluster_1_count</th>
<th align="center">cluster_2_count</th>
<th align="center">cluster_3_count</th>
<th align="center">cluster_4_count</th>
<th align="center">cluster_5_count</th>
<th align="center">Total_count</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">&gt;bin_3_NODE_81_length_109410_cov_1.136244.concoct_part_1</td>
<td align="center">cluster_5</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">4</td>
<td align="center">1</td>
<td align="center">5</td>
</tr>
<tr>
<td align="left">&gt;bin_3_NODE_289_length_18049_cov_1.596107.concoct_part_0</td>
<td align="center">cluster_5</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">1</td>
<td align="center">1</td>
</tr>
<tr>
<td align="left">&gt;bin_3_NODE_349_length_12681_cov_1.204936.concoct_part_0</td>
<td align="center">cluster_5</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">1</td>
<td align="center">1</td>
</tr>
</tbody>
</table>
<p>Note that in the case of the first contig from the excerpt above, the 'problematic' contig is only one of 5 sub-contigs, and all other 4 sub-contigs are in the expected cluster. In this case, we likely do <em>not</em> want to remove this contig from the bin.</p>
<h3 id="generate-a-list-of-contigs-to-exclude-from-filtering">Generate a list of contigs to <em>exclude</em> from filtering<a class="headerlink" href="#generate-a-list-of-contigs-to-exclude-from-filtering" title="Permanent link">¶</a></h3>
<p>Create a list of contigs identified from <code>vb_count_table.txt</code> that are <em>not</em> to be filtered out by seqmagick in the next step. For example, those contigs that have sub-contigs split across multiple <code>VizBin</code> clusters, and for which it's reasonable to actually keep the contig (such as when a flagged selected sub-contig exported from vizbin is in one unexpected cluster, but all other sub-contigs from that parent contig are in the expected cluster; in this case, you likely <em>don't</em> want to filter out the parent contig from the data set moving forward). </p>
<p>Below is an example. Simply replace the contig IDs between the quotes for as many lines as necessary for your data. </p>
<div class="admonition note">
<p class="admonition-title">Write/append outputs and extracting contig ID</p>
<ol>
<li>The first line below must always have only one <code>&gt;</code> character, while all subsequent lines must have two (i.e. <code>&gt;&gt;</code>) to append correctly to the list.</li>
<li>We want the original contig ID here, <em>not</em> the sub-contig, so make sure to remove the <code>.concoct_part_n</code> fragment number at the end if there is one.</li>
</ol>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">"bin_3_NODE_81_length_109410_cov_1.136244"</span><span class="w"> </span>&gt;<span class="w"> </span>vb_keep_contigs.txt
</code></pre></div>
</div>
<h3 id="create-final-vb_omit_contigs_filteredtxt-list-of-contigs-to-filter-from-bins">Create final <code>vb_omit_contigs_filtered.txt</code> list of contigs to filter from bins<a class="headerlink" href="#create-final-vb_omit_contigs_filteredtxt-list-of-contigs-to-filter-from-bins" title="Permanent link">¶</a></h3>
<p>Using <code>grep</code>, filter contigs we wish to keep (after assessing <code>vb_count_table.txt</code>) out of the working <code>vb_omit_contigs_tmp.txt</code> list. </p>
<p>This creates <code>vb_omit_contigs_filtered.txt</code>, which we will then pass to <code>seqmagick</code> to filter these contigs out of our actual bin FASTA files.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-13-1" id="__codelineno-13-1" name="__codelineno-13-1"></a>grep<span class="w"> </span>-v<span class="w"> </span>-f<span class="w"> </span>vb_keep_contigs.txt<span class="w"> </span>vb_omit_contigs_tmp.txt<span class="w"> </span>&gt;<span class="w"> </span>vb_omit_contigs_filtered.txt
</code></pre></div>
</div>
<h3 id="filter-suspect-contigs-based-on-vizbin-analysis-from-the-bin-data">Filter suspect contigs (based on <code>VizBin</code> analysis) from the bin data<a class="headerlink" href="#filter-suspect-contigs-based-on-vizbin-analysis-from-the-bin-data" title="Permanent link">¶</a></h3>
<p>Use <code>seqmagick --exclude-from-file ...</code> to filter problematic contigs (those contigs listed in <code>vb_omit_contigs_filtered.txt</code>) out of the initial <em>unchopped</em> bin FASTA files, generating final bins for downstream processing.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-14-1" id="__codelineno-14-1" name="__codelineno-14-1"></a>mkdir<span class="w"> </span>filtered_bins/
<a href="#__codelineno-14-2" id="__codelineno-14-2" name="__codelineno-14-2"></a>
<a href="#__codelineno-14-3" id="__codelineno-14-3" name="__codelineno-14-3"></a><span class="c1"># Load seqmagick</span>
<a href="#__codelineno-14-4" id="__codelineno-14-4" name="__codelineno-14-4"></a>module<span class="w"> </span>purge
<a href="#__codelineno-14-5" id="__codelineno-14-5" name="__codelineno-14-5"></a>module<span class="w"> </span>load<span class="w"> </span>seqmagick/0.8.4-gimkl-2020a-Python-3.8.2
<a href="#__codelineno-14-6" id="__codelineno-14-6" name="__codelineno-14-6"></a>
<a href="#__codelineno-14-7" id="__codelineno-14-7" name="__codelineno-14-7"></a><span class="c1"># filter problematic contigs out of original bin files</span>
<a href="#__codelineno-14-8" id="__codelineno-14-8" name="__codelineno-14-8"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>example_data_unchopped/*.fna<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-14-9" id="__codelineno-14-9" name="__codelineno-14-9"></a><span class="w">    </span><span class="nv">bin_name</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>.fna<span class="k">)</span>
<a href="#__codelineno-14-10" id="__codelineno-14-10" name="__codelineno-14-10"></a><span class="w">    </span>seqmagick<span class="w"> </span>convert<span class="w"> </span>--exclude-from-file<span class="w"> </span>vb_omit_contigs_filtered.txt<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>filtered_bins/<span class="si">${</span><span class="nv">bin_name</span><span class="si">}</span>.filtered.fna
<a href="#__codelineno-14-11" id="__codelineno-14-11" name="__codelineno-14-11"></a><span class="k">done</span>
</code></pre></div>
</div>
<p>Our filtered bins for downstream use are now in <code>filtered_bins/</code></p>
<hr/>
<h2 id="optional-comparing-pre-and-post-filtered-bins-via-checkm"><em>(Optional)</em> Comparing pre- and post-filtered bins via <code>CheckM</code><a class="headerlink" href="#optional-comparing-pre-and-post-filtered-bins-via-checkm" title="Permanent link">¶</a></h2>
<p>The end goal of this process is the generation of a final set of refined bins. Following this, the <code>CheckM</code> procedure should be re-run, this time on the refined <code>filtered_bins/</code>. This provides <code>CheckM</code> metrics for the final actual (filtered) bin set, and also an opportunity to compare between pre- and post-filtering to see if the <code>VizBin</code> bin refinement steps have, for example, improved the degree of contamination in the bins.</p>
<p>For this exercise, a copy of the output from running <code>CheckM</code> on the <code>filtered_bins/</code> is available at <code>6.bin_refinement/filtered_bins_checkm.txt</code>. View the previous <code>CheckM</code> output and the filtered bins output to compare via <code>cat</code>.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-15-1" id="__codelineno-15-1" name="__codelineno-15-1"></a>cat<span class="w"> </span>filtered_bins_checkm.txt<span class="w"> </span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Content of <code>filtered_bins_checkm.txt</code></p>
<table>
<thead>
<tr>
<th align="left">Bin Id</th>
<th align="center">Marker lineage</th>
<th align="center"># genomes</th>
<th align="center"># markers</th>
<th align="center"># marker sets</th>
<th align="center">0</th>
<th align="center">1</th>
<th align="center">2</th>
<th align="center">3</th>
<th align="center">4</th>
<th align="center">5+</th>
<th align="center">Completeness</th>
<th align="center">Contamination</th>
<th align="center">Strain heterogeneity</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">bin_0.filtered</td>
<td align="center">k__Bacteria (UID3060)</td>
<td align="center">138</td>
<td align="center">338</td>
<td align="center">246</td>
<td align="center">1</td>
<td align="center">327</td>
<td align="center">9</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.59</td>
<td align="center">3.79</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_1.filtered</td>
<td align="center">k__Bacteria (UID3060)</td>
<td align="center">138</td>
<td align="center">338</td>
<td align="center">246</td>
<td align="center">1</td>
<td align="center">336</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.59</td>
<td align="center">0.41</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_2.filtered</td>
<td align="center">g__Staphylococcus (UID301)</td>
<td align="center">45</td>
<td align="center">940</td>
<td align="center">178</td>
<td align="center">20</td>
<td align="center">918</td>
<td align="center">2</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">98.32</td>
<td align="center">0.11</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_3.filtered</td>
<td align="center">c__Betaproteobacteria (UID3959)</td>
<td align="center">235</td>
<td align="center">414</td>
<td align="center">211</td>
<td align="center">1</td>
<td align="center">408</td>
<td align="center">5</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.97</td>
<td align="center">0.90</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_4.filtered</td>
<td align="center">c__Deltaproteobacteria (UID3218)</td>
<td align="center">61</td>
<td align="center">284</td>
<td align="center">169</td>
<td align="center">10</td>
<td align="center">274</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">94.08</td>
<td align="center">0.00</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_5.filtered</td>
<td align="center">o__Pseudomonadales (UID4488)</td>
<td align="center">185</td>
<td align="center">813</td>
<td align="center">308</td>
<td align="center">25</td>
<td align="center">787</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">96.87</td>
<td align="center">0.11</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_6.filtered</td>
<td align="center">k__Bacteria (UID2565)</td>
<td align="center">2921</td>
<td align="center">149</td>
<td align="center">91</td>
<td align="center">11</td>
<td align="center">136</td>
<td align="center">2</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">90.66</td>
<td align="center">0.61</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_7.filtered</td>
<td align="center">p__Cyanobacteria (UID2143)</td>
<td align="center">129</td>
<td align="center">471</td>
<td align="center">367</td>
<td align="center">0</td>
<td align="center">470</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">100.00</td>
<td align="center">0.14</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_8.filtered</td>
<td align="center">f__Bradyrhizobiaceae (UID3695)</td>
<td align="center">47</td>
<td align="center">693</td>
<td align="center">296</td>
<td align="center">3</td>
<td align="center">690</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.47</td>
<td align="center">0.00</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">bin_9.filtered</td>
<td align="center">g__Vibrio (UID4878)</td>
<td align="center">67</td>
<td align="center">1130</td>
<td align="center">369</td>
<td align="center">4</td>
<td align="center">1125</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.46</td>
<td align="center">0.03</td>
<td align="center">0.00</td>
</tr>
</tbody>
</table>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-16-1" id="__codelineno-16-1" name="__codelineno-16-1"></a>cat<span class="w"> </span>../5.binning/checkm.txt<span class="w"> </span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Content of <code>checkm.txt</code></p>
<table>
<thead>
<tr>
<th align="left">Bin Id</th>
<th align="center">Marker lineage</th>
<th align="center"># genomes</th>
<th align="center"># markers</th>
<th align="center"># marker sets</th>
<th align="center">0</th>
<th align="center">1</th>
<th align="center">2</th>
<th align="center">3</th>
<th align="center">4</th>
<th align="center">5+</th>
<th align="center">Completeness</th>
<th align="center">Contamination</th>
<th align="center">Strain heterogeneity</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">maxbin.001.fasta</td>
<td align="center">k__Bacteria (UID3060)</td>
<td align="center">138</td>
<td align="center">338</td>
<td align="center">246</td>
<td align="center">1</td>
<td align="center">327</td>
<td align="center">9</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.59</td>
<td align="center">3.79</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">maxbin.002.fasta</td>
<td align="center">k__Bacteria (UID3060)</td>
<td align="center">138</td>
<td align="center">338</td>
<td align="center">246</td>
<td align="center">1</td>
<td align="center">336</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.59</td>
<td align="center">0.41</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.10_sub</td>
<td align="center">g__Staphylococcus (UID301)</td>
<td align="center">45</td>
<td align="center">940</td>
<td align="center">178</td>
<td align="center">20</td>
<td align="center">918</td>
<td align="center">2</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">98.32</td>
<td align="center">0.11</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.11</td>
<td align="center">c__Betaproteobacteria (UID3959)</td>
<td align="center">235</td>
<td align="center">414</td>
<td align="center">211</td>
<td align="center">1</td>
<td align="center">408</td>
<td align="center">5</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.97</td>
<td align="center">0.90</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.12</td>
<td align="center">c__Deltaproteobacteria (UID3218)</td>
<td align="center">61</td>
<td align="center">284</td>
<td align="center">169</td>
<td align="center">10</td>
<td align="center">274</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">94.08</td>
<td align="center">0.00</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.2</td>
<td align="center">o__Pseudomonadales (UID4488)</td>
<td align="center">185</td>
<td align="center">813</td>
<td align="center">308</td>
<td align="center">25</td>
<td align="center">787</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">96.87</td>
<td align="center">0.11</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.3_sub</td>
<td align="center">k__Bacteria (UID2565)</td>
<td align="center">2921</td>
<td align="center">149</td>
<td align="center">91</td>
<td align="center">11</td>
<td align="center">136</td>
<td align="center">2</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">90.66</td>
<td align="center">0.61</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.4</td>
<td align="center">p__Cyanobacteria (UID2143)</td>
<td align="center">129</td>
<td align="center">471</td>
<td align="center">367</td>
<td align="center">0</td>
<td align="center">470</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">100.00</td>
<td align="center">0.14</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.5</td>
<td align="center">f__Bradyrhizobiaceae (UID3695)</td>
<td align="center">47</td>
<td align="center">693</td>
<td align="center">296</td>
<td align="center">3</td>
<td align="center">690</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.47</td>
<td align="center">0.00</td>
<td align="center">0.00</td>
</tr>
<tr>
<td align="left">metabat.7</td>
<td align="center">g__Vibrio (UID4878)</td>
<td align="center">67</td>
<td align="center">1130</td>
<td align="center">369</td>
<td align="center">4</td>
<td align="center">1125</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">99.46</td>
<td align="center">0.03</td>
<td align="center">0.00</td>
</tr>
</tbody>
</table>
</div>
<p>An example of an updated slurm script to run <code>CheckM</code> on the <code>filtered_bins/</code> is as follows:</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-17-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-17-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-17-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-17-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-17-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-17-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-17-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-17-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-17-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-17-10">10</a></span>
<span class="normal"><a href="#__codelineno-17-11">11</a></span>
<span class="normal"><a href="#__codelineno-17-12">12</a></span>
<span class="normal"><a href="#__codelineno-17-13">13</a></span>
<span class="normal"><a href="#__codelineno-17-14">14</a></span>
<span class="normal"><a href="#__codelineno-17-15">15</a></span>
<span class="normal"><a href="#__codelineno-17-16">16</a></span>
<span class="normal"><a href="#__codelineno-17-17">17</a></span>
<span class="normal"><a href="#__codelineno-17-18">18</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-17-2" name="__codelineno-17-2"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-17-3" name="__codelineno-17-3"></a><span class="c1">#SBATCH --job-name      checkm_refined_bins</span>
<a id="__codelineno-17-4" name="__codelineno-17-4"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-17-5" name="__codelineno-17-5"></a><span class="c1">#SBATCH --time          00:20:00</span>
<a id="__codelineno-17-6" name="__codelineno-17-6"></a><span class="c1">#SBATCH --mem           50GB</span>
<a id="__codelineno-17-7" name="__codelineno-17-7"></a><span class="c1">#SBATCH --cpus-per-task 10</span>
<a id="__codelineno-17-8" name="__codelineno-17-8"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-17-9" name="__codelineno-17-9"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-17-10" name="__codelineno-17-10"></a>
<a id="__codelineno-17-11" name="__codelineno-17-11"></a>module<span class="w"> </span>purge
<a id="__codelineno-17-12" name="__codelineno-17-12"></a>module<span class="w"> </span>load<span class="w"> </span>CheckM/1.2.1-gimkl-2022a-Python-3.10.5
<a id="__codelineno-17-13" name="__codelineno-17-13"></a>
<a id="__codelineno-17-14" name="__codelineno-17-14"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/6.bin_refinement/
<a id="__codelineno-17-15" name="__codelineno-17-15"></a>
<a id="__codelineno-17-16" name="__codelineno-17-16"></a>checkm<span class="w"> </span>lineage_wf<span class="w"> </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>--pplacer_threads<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>-x<span class="w"> </span>fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-17" name="__codelineno-17-17"></a><span class="w">                  </span>--tab_table<span class="w"> </span>-f<span class="w"> </span>filtered_bins_checkm.txt<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-17-18" name="__codelineno-17-18"></a><span class="w">                  </span>filtered_bins/<span class="w"> </span>filtered_bins_checkm_out/
</code></pre></div></td></tr></table></div>
</div>
<p>The data you have been working with was created using the <code>cut_up_fasta.py</code> script that comes with the binning tool <code>CONCOCT</code>. It was run to cut contigs into 20k fragments, to better add density to the cluster. If you would like to visualise the data using different contig fragment sizes, you can create these using the following commands (replace <code>YOUR_CONTIG_SIZE</code> with the size of interest, e.g. <code>10000</code>):</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-18-1" id="__codelineno-18-1" name="__codelineno-18-1"></a>module<span class="w"> </span>purge
<a href="#__codelineno-18-2" id="__codelineno-18-2" name="__codelineno-18-2"></a>module<span class="w"> </span>load<span class="w"> </span>CONCOCT/1.0.0-gimkl-2018b-Python-2.7.16
<a href="#__codelineno-18-3" id="__codelineno-18-3" name="__codelineno-18-3"></a>
<a href="#__codelineno-18-4" id="__codelineno-18-4" name="__codelineno-18-4"></a>mkdir<span class="w"> </span>custom_chop/
<a href="#__codelineno-18-5" id="__codelineno-18-5" name="__codelineno-18-5"></a>
<a href="#__codelineno-18-6" id="__codelineno-18-6" name="__codelineno-18-6"></a><span class="c1"># Fragment contigs within each bin, outputting to custom_chop/</span>
<a href="#__codelineno-18-7" id="__codelineno-18-7" name="__codelineno-18-7"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>example_data_unchopped/*<span class="p">;</span>
<a href="#__codelineno-18-8" id="__codelineno-18-8" name="__codelineno-18-8"></a><span class="k">do</span>
<a href="#__codelineno-18-9" id="__codelineno-18-9" name="__codelineno-18-9"></a><span class="w">    </span><span class="nv">bin_name</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>.fna<span class="k">)</span>
<a href="#__codelineno-18-10" id="__codelineno-18-10" name="__codelineno-18-10"></a><span class="w">    </span>cut_up_fasta.py<span class="w"> </span>-c<span class="w"> </span>YOUR_CONTIG_SIZE<span class="w"> </span>-o<span class="w"> </span><span class="m">0</span><span class="w"> </span>--merge_last<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>&gt;<span class="w"> </span>custom_chop/<span class="si">${</span><span class="nv">bin_name</span><span class="si">}</span>.chopped.fna
<a href="#__codelineno-18-11" id="__codelineno-18-11" name="__codelineno-18-11"></a><span class="k">done</span>
<a href="#__codelineno-18-12" id="__codelineno-18-12" name="__codelineno-18-12"></a>
<a href="#__codelineno-18-13" id="__codelineno-18-13" name="__codelineno-18-13"></a><span class="c1"># Concatenate the chopped bins into single .fna</span>
<a href="#__codelineno-18-14" id="__codelineno-18-14" name="__codelineno-18-14"></a>cat<span class="w"> </span>custom_chop/*.fna<span class="w"> </span>&gt;<span class="w"> </span>all_bins_custom_chop.fna
</code></pre></div>
</div>
<p>You can open <code>all_bins_custom_chop.fna</code> in <code>VizBin</code> to view the clustering with this new fragmentation threshold. </p>
<p>If you wish to also provide an annotation file to colour by bin, this can be generated with the following:</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-19-1" id="__codelineno-19-1" name="__codelineno-19-1"></a><span class="c1"># Set up annotation file headers</span>
<a href="#__codelineno-19-2" id="__codelineno-19-2" name="__codelineno-19-2"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">"label"</span><span class="w"> </span>&gt;<span class="w"> </span>custom_chop.vizbin.ann
<a href="#__codelineno-19-3" id="__codelineno-19-3" name="__codelineno-19-3"></a>
<a href="#__codelineno-19-4" id="__codelineno-19-4" name="__codelineno-19-4"></a><span class="c1"># loop through custom_chop .fna files</span>
<a href="#__codelineno-19-5" id="__codelineno-19-5" name="__codelineno-19-5"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>custom_chop/*.fna<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-19-6" id="__codelineno-19-6" name="__codelineno-19-6"></a><span class="w">    </span><span class="c1"># extract bin ID</span>
<a href="#__codelineno-19-7" id="__codelineno-19-7" name="__codelineno-19-7"></a><span class="w">    </span><span class="nv">binID</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>.fna<span class="k">)</span>
<a href="#__codelineno-19-8" id="__codelineno-19-8" name="__codelineno-19-8"></a><span class="w">    </span><span class="c1"># loop through each sequence header in bin_file, adding binID to custom_chop.vizbin.ann for each header present</span>
<a href="#__codelineno-19-9" id="__codelineno-19-9" name="__codelineno-19-9"></a><span class="w">    </span><span class="k">for</span><span class="w"> </span>header<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="sb">`</span>grep<span class="w"> </span><span class="s2">"&gt;"</span><span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="sb">`</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-19-10" id="__codelineno-19-10" name="__codelineno-19-10"></a><span class="w">        </span><span class="c1"># Add binID to vizbin.ann for each header present</span>
<a href="#__codelineno-19-11" id="__codelineno-19-11" name="__codelineno-19-11"></a><span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">"</span><span class="si">${</span><span class="nv">binID</span><span class="si">}</span><span class="s2">"</span><span class="w"> </span>&gt;&gt;<span class="w"> </span>custom_chop.vizbin.ann
<a href="#__codelineno-19-12" id="__codelineno-19-12" name="__codelineno-19-12"></a><span class="w">    </span><span class="k">done</span>
<a href="#__codelineno-19-13" id="__codelineno-19-13" name="__codelineno-19-13"></a><span class="k">done</span>
</code></pre></div>
</div>
<p>If you wish to generate the full annotation file, including coverage and length values, you will need to go through the process outlined in the <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">Appendix for this exercise</a>.  </p>
<hr/>
<h2 id="optional-scripts-for-processing-data-with-esomana"><em>(Optional)</em> Scripts for processing data with <em>ESOMana</em><a class="headerlink" href="#optional-scripts-for-processing-data-with-esomana" title="Permanent link">¶</a></h2>
<p>A suite of tools for creating input files for <code>ESOMana</code> can be found on github <a href="https://github.com/tetramerFreqs/Binning">here</a>.</p>
<p>The tool <code>ESOMana</code> can be downloaded from <a href="http://databionic-esom.sourceforge.net/">SourceForge</a>.</p>
<hr/>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>