
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day4/ex15_gene_annotation_part3/" rel="canonical"/>
<link href="../../day3/ex14_gene_annotation_part2/" rel="prev"/>
<link href="../ex16a_data_presentation_Intro/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Gene annotation (part 3) - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#gene-annotation-part-3">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Gene annotation (part 3)
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="./">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction to shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction to HPC &amp; HPC job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Quality filtering raw reads
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning (continued)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation (part 1)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation (part 2) and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Gene annotation (part 3)
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Gene annotation (part 3)
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#overview-of-drampy-annotate-output">
    Overview of DRAM.py annotate output
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#overview-of-dram-vpy-annotate-output">
    Overview of DRAM-v.py annotate output
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dram-and-dram-v-distillation-of-the-results">
    DRAM and DRAM-v distillation of the results
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#drampy-distill-output-files">
    DRAM.py distill output files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dram-vpy-distill-output-files">
    DRAM-v.py distill output files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#tie-findings-to-your-initial-goal">
    Tie findings to your initial goal
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16b_data_presentation_Coverage/">
        Presentation of data: Per-sample coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16c_OPTIONAL_data_presentation_Ordination/">
        Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#overview-of-drampy-annotate-output">
    Overview of DRAM.py annotate output
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#overview-of-dram-vpy-annotate-output">
    Overview of DRAM-v.py annotate output
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dram-and-dram-v-distillation-of-the-results">
    DRAM and DRAM-v distillation of the results
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#drampy-distill-output-files">
    DRAM.py distill output files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dram-vpy-distill-output-files">
    DRAM-v.py distill output files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#tie-findings-to-your-initial-goal">
    Tie findings to your initial goal
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="gene-annotation-part-3">Gene annotation (part 3)<a class="headerlink" href="#gene-annotation-part-3" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#overview-of-drampy-annotate-and-dramvpy-annotate-output">Overview of <code>DRAM.py annotate</code> and <code>DRAM-v.py annotate</code> output</a></li>
<li><a href="#dram-and-dramv-distillation-of-the-results"><em>DRAM</em> and <em>DRAM-v</em> distillation step and visualization of results</a></li>
<li><a href="#tie-findings-to-your-initial-goal">Tie findings to your initial goal</a></li>
</ul>
</div>
<hr/>
<h3 id="overview-of-drampy-annotate-output">Overview of <em>DRAM.py annotate</em> output<a class="headerlink" href="#overview-of-drampy-annotate-output" title="Permanent link">¶</a></h3>
<p>The submitted jobs from the previous session should now be completed. If we examine the output directory <code>10.gene_annotation_and_coverage/dram_annotations/</code> we will see the following files:</p>
<table>
<thead>
<tr>
<th align="left">File name</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>genes.faa</strong> and <strong>genes.fna</strong></td>
<td align="left">Fasta files with all the genes called by <code>Prodigal</code>, with additional header information gained from the annotation as nucleotide and amino acid records, respectively</td>
</tr>
<tr>
<td align="left"><strong>genes.gff</strong></td>
<td align="left">GFF3 file with the same annotation information as well as gene locations</td>
</tr>
<tr>
<td align="left"><strong>scaffolds.fna</strong></td>
<td align="left">A collection of all scaffolds/contigs given as input to <code>DRAM.py annotate</code> with added bin information</td>
</tr>
<tr>
<td align="left"><strong>annotations.tsv</strong></td>
<td align="left">This file includes all annotation information about every gene from all MAGs</td>
</tr>
<tr>
<td align="left"><strong>trnas.tsv</strong></td>
<td align="left">Summary of the tRNAs found in each MAG</td>
</tr>
<tr>
<td align="left"><strong>rrnas.tsv</strong></td>
<td align="left">Summary of the rRNAs found in each MAG</td>
</tr>
</tbody>
</table>
<p>If we inspect the head of the annotation file we will see the following</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage/
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>dram_annotations/annotations.tsv<span class="w"> </span>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a><span class="c1">#    fasta   scaffold        gene_position   start_position  end_position    strandedness    rank    ko_id   kegg_hit        peptidase_id    peptidase_family        peptidase_hit    peptidase_RBH   peptidase_identity      peptidase_bitScore      peptidase_eVal  pfam_hits       cazy_id cazy_hits       heme_regulatory_motif_count     bin_taxonomy    bin_completeness bin_contamination</span>
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="c1"># bin_0.filtered_bin_0_NODE_11_length_361162_cov_0.994424_1       bin_0.filtered  bin_0_NODE_11_length_361162_cov_0.994424        1       2       277     -1      D               Fumarate hydratase (Fumerase) [PF05681.17]                       0       d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis      99.59   3.79</span>
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="c1"># bin_0.filtered_bin_0_NODE_11_length_361162_cov_0.994424_2       bin_0.filtered  bin_0_NODE_11_length_361162_cov_0.994424        2       474     764     1       E               0d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis     99.59   3.79</span>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a><span class="c1"># bin_0.filtered_bin_0_NODE_11_length_361162_cov_0.994424_3       bin_0.filtered  bin_0_NODE_11_length_361162_cov_0.994424        3       775     1452    1       D               Response regulator receiver domain [PF00072.27]; Transcriptional regulatory protein, C terminal [PF00486.31]                     0       d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis      99.59   3.79</span>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="c1"># bin_0.filtered_bin_0_NODE_11_length_361162_cov_0.994424_4       bin_0.filtered  bin_0_NODE_11_length_361162_cov_0.994424        4       1449    2696    1       D               GHKL domain [PF14501.9]                  0       d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__Arcobacter nitrofigilis      99.59   3.79</span>
</code></pre></div>
</div>
<p>For each gene annotated, <code>DRAM</code> provides a summary rank (from A to E), representing the confidence of the annotation based on reciprocal best hits (RBH). The following figure briefly explains how this summary rank is calculated:</p>
<p><center><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex14_DRAM_annotation_rank.png"><img alt="image" src="../../figures/ex14_DRAM_annotation_rank.png" width="500"/></a></center></p>
<h3 id="overview-of-dram-vpy-annotate-output">Overview of <em>DRAM-v.py annotate</em> output<a class="headerlink" href="#overview-of-dram-vpy-annotate-output" title="Permanent link">¶</a></h3>
<p><em>DRAM-v</em> generates the same output files as <em>DRAM</em>, but this time for the viral contigs. These files can be viewed in the output directory <code>10.gene_annotation_and_coverage/dramv_annotations/</code>. In this case, <code>annotations.tsv</code> also includes some viral-specific columns, including viral gene database matches (<code>vogdb</code>), and categories that are used by <em>DRAM-v.py distill</em> to identify putative auxiliary metabolic genes (AMGs) (<code>virsorter_category</code>, <code>auxiliary_score</code>,  <code>is_transposon</code>, <code>amg_flags</code>)</p>
<hr/>
<h3 id="dram-and-dram-v-distillation-of-the-results"><em>DRAM</em> and <em>DRAM-v</em> distillation of the results<a class="headerlink" href="#dram-and-dram-v-distillation-of-the-results" title="Permanent link">¶</a></h3>
<p>After the annotation is finished, we will summarise and visualise these annotations with the so-called <em>distillation</em> step. We do so by running the following commands directly in the terminal. This will generate the distillate and liquor files for each dataset.</p>
<p>For the viral annotations, we will also include the parameters <code>--remove_transposons</code> ("Do not consider genes on scaffolds with transposons as potential AMGs") and <code>--remove_fs</code> ("Do not consider genes near ends of scaffolds as potential AMGs") to filter out some potential false positives for auxilliary metabolic gene identification.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>module<span class="w"> </span>purge
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>module<span class="w"> </span>load<span class="w"> </span>DRAM/1.3.5-Miniconda3
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage/
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a><span class="c1"># Prokaryote annotations</span>
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a>DRAM.py<span class="w"> </span>distill<span class="w"> </span>-i<span class="w"> </span>dram_annotations/annotations.tsv<span class="w"> </span>-o<span class="w"> </span>dram_distillation<span class="w"> </span>--trna_path<span class="w"> </span>dram_annotations/trnas.tsv<span class="w"> </span>--rrna_path<span class="w"> </span>dram_annotations/rrnas.tsv
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a>
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a><span class="c1"># viral annotations</span>
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a>DRAM-v.py<span class="w"> </span>distill<span class="w"> </span>--remove_transposons<span class="w"> </span>--remove_fs<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-1-11" id="__codelineno-1-11" name="__codelineno-1-11"></a>-i<span class="w"> </span>dramv_annotations/annotations.tsv<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-1-12" id="__codelineno-1-12" name="__codelineno-1-12"></a>-o<span class="w"> </span>dramv_distillation
</code></pre></div>
</div>
<h3 id="drampy-distill-output-files"><em>DRAM.py distill</em> output files<a class="headerlink" href="#drampy-distill-output-files" title="Permanent link">¶</a></h3>
<p>The <em>DRAM</em> distillation step generates the following files that can be found within the <code>dram_distillation</code> directory :</p>
<table>
<thead>
<tr>
<th align="left">File name</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>genome_stats.tsv</strong></td>
<td align="left">Genome quality information required for <a href="https://www.nature.com/articles/nbt.3893">MIMAG</a></td>
</tr>
<tr>
<td align="left"><strong>metabolism_summary.xlsx</strong></td>
<td align="left">Summarised metabolism table containing number of genes with specific metabolic function identifiers</td>
</tr>
<tr>
<td align="left"><strong>product.html</strong></td>
<td align="left">HTML file displaying a heatmap summarising pathway coverage, electron transport chain component completion, and presence/absence of specific functions</td>
</tr>
<tr>
<td align="left"><strong>product.tsv</strong></td>
<td align="left">Data table visualised in product.html</td>
</tr>
</tbody>
</table>
<p>First, let's have a look at the <code>genome_stats.tsv</code> file to check the assembly quality of our bins by double-clicking the file within the <code>Jupyter</code> environment, viewing from the terminal via <code>less</code> or <code>cat</code>, or downloading the files from <a href="../../resources/dram_distillation.zip">here</a> and opening locally (e.g. via excel).</p>
<details class="magnifying-glass">
<summary>Content of <code>genome_stats.tsv</code></summary>
<table>
<thead>
<tr>
<th align="left">genome</th>
<th align="left">number of scaffolds</th>
<th align="left">taxonomy</th>
<th align="left">completeness score</th>
<th align="left">contamination score</th>
<th align="left">5S rRNA</th>
<th align="left">16S rRNA</th>
<th align="left">23S rRNA</th>
<th align="left">tRNA count</th>
<th align="left">assembly quality</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">bin_0</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Cyanobacteria;c__Cyanobacteriia;o__Synechococcales;f__Cyanobiaceae;g__Prochlorococcus_C;s__</td>
<td align="left">100</td>
<td align="left">0.14</td>
<td align="left">-</td>
<td align="left">2 present</td>
<td align="left">2 present</td>
<td align="left">47</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_1</td>
<td align="left">4</td>
<td align="left">d__Bacteria;p__Firmicutes;c__Bacilli;o__Staphylococcales;f__Staphylococcaceae;g__Staphylococcus;s__</td>
<td align="left">99.51</td>
<td align="left">0.08</td>
<td align="left">6 present</td>
<td align="left">5 present</td>
<td align="left">5 present</td>
<td align="left">60</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_2</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Pseudomonadales;f__Pseudomonadaceae;g__Pseudomonas;s__</td>
<td align="left">96.45</td>
<td align="left">0.11</td>
<td align="left">3 present</td>
<td align="left">-</td>
<td align="left">-</td>
<td align="left">43</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_3</td>
<td align="left">3</td>
<td align="left">d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Vibrionaceae;g__Vibrio;s__</td>
<td align="left">99.73</td>
<td align="left">0.03</td>
<td align="left">9 present</td>
<td align="left">8 present</td>
<td align="left">8 present</td>
<td align="left">98</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_4</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Nitrosomonadaceae;g__Nitrosomonas;s__</td>
<td align="left">99.97</td>
<td align="left">0.74</td>
<td align="left">bin_4, (74043, 74150)</td>
<td align="left">bin_4, (69143, 70676)</td>
<td align="left">bin_4, (71085, 73967)</td>
<td align="left">42</td>
<td align="left">high</td>
</tr>
<tr>
<td align="left">bin_5</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhizobiales;f__Xanthobacteraceae;g__Nitrobacter;s__</td>
<td align="left">99.8</td>
<td align="left">0</td>
<td align="left">bin_5, (643507, 643615)</td>
<td align="left">bin_5, (638306, 639791)</td>
<td align="left">bin_5, (640621, 643431)</td>
<td align="left">49</td>
<td align="left">high</td>
</tr>
<tr>
<td align="left">bin_6</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Nautiliales;f__Nautiliaceae;g__;s__</td>
<td align="left">99.59</td>
<td align="left">0.41</td>
<td align="left">4 present</td>
<td align="left">4 present</td>
<td align="left">4 present</td>
<td align="left">49</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_7</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Campylobacterota;c__Campylobacteria;o__Campylobacterales;f__Arcobacteraceae;g__Arcobacter;s__</td>
<td align="left">99.59</td>
<td align="left">2.98</td>
<td align="left">4 present</td>
<td align="left">4 present</td>
<td align="left">4 present</td>
<td align="left">54</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_8</td>
<td align="left">19</td>
<td align="left">d__Bacteria;p__Desulfobacterota_A;c__Desulfovibrionia;o__Desulfovibrionales;f__Desulfovibrionaceae;g__Desulfovibrio;s__</td>
<td align="left">99.41</td>
<td align="left">0</td>
<td align="left">2 present</td>
<td align="left">bin_8, (3744, 5289)</td>
<td align="left">bin_8, (379, 3300)</td>
<td align="left">57</td>
<td align="left">med</td>
</tr>
<tr>
<td align="left">bin_9</td>
<td align="left">1</td>
<td align="left">d__Bacteria;p__Planctomycetota;c__Brocadiae;o__Brocadiales;f__Brocadiaceae;g__;s__</td>
<td align="left">97.8</td>
<td align="left">1.65</td>
<td align="left">bin_9, (1066028, 1066130)</td>
<td align="left">bin_9, (1069811, 1071397)</td>
<td align="left">bin_9, (1066309, 1069302)</td>
<td align="left">46</td>
<td align="left">high</td>
</tr>
</tbody>
</table>
</details>
<p>To finish, we visualize the <em>Product</em>, an .HTML file produced in the distillation step, by double-clicking on it in our <em>Jupyter</em> lab notebook or downloading from <a href="../../resources/dram_distillation.zip">here</a>. The <em>Product</em> has three primary parts:</p>
<ol>
<li><strong>Modules.</strong> Central metabolism pathways coverage. Completion of pathways is based on the structure of KEGG modules, with the pathway coverage calculated as the percent of steps with at least one gene present.</li>
</ol>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex15_fig1_DRAM_product_met_2022.png"><img alt="image" src="../../figures/ex15_fig1_DRAM_product_met_2022.png"/></a>
</center></p>
<ol>
<li><strong>ETC Complexes.</strong> Electron Transport Chain component completion</li>
</ol>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex15_fig2_DRAM_product_ETC_2022.png"><img alt="image" src="../../figures/ex15_fig2_DRAM_product_ETC_2022.png"/></a>
</center></p>
<ol>
<li>Presence of specific functions, including CAZy, Nitrogen metabolism, Sulfur metabolism and Photosynthesis. Note that the taxonomic classification of each of the bins is also shown in the first figure</li>
</ol>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/ex15_fig3_DRAM_product_2022.png"><img alt="image" src="../../figures/ex15_fig3_DRAM_product_2022.png"/></a>
</center></p>
<h3 id="dram-vpy-distill-output-files"><em>DRAM-v.py distill</em> output files<a class="headerlink" href="#dram-vpy-distill-output-files" title="Permanent link">¶</a></h3>
<p>The <em>DRAM-v</em> distillation step for the viral contigs generates the following files that can be found within the <code>dramv_distillation/</code> directory :</p>
<table>
<thead>
<tr>
<th align="left">File name</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>vMAG_stats.tsv</strong></td>
<td align="left">"Genome" (in this case viral contigs of varying completeness) information including: total gene counts, viral vs host gene counts, and counts of genes related to viral replication, structure, and those with presumed viral or host benefits</td>
</tr>
<tr>
<td align="left"><strong>amg_summary.tsv</strong></td>
<td align="left">Genes identified as putative auxiliary metabolic genes (AMGs) and various columns for metabolic characterisation of each gene</td>
</tr>
<tr>
<td align="left"><strong>product.html</strong></td>
<td align="left">HTML file displaying a heatmap summarising AMG counts and presence/absence for different broad metabolic categories for each viral contig</td>
</tr>
</tbody>
</table>
<p>When viewing these files, see if you can find the following information:</p>
<div class="admonition quote">
<ul>
<li>What are some annotations of interest within the output annotations file? </li>
<li><em>NOTE: the *VirSorter2</em> annotations file includes multiple columns for both <strong>prokaryote</strong> and <strong>viral</strong> protein predictions. Be careful as to which column you are looking at (as well as its associated confidence score) when assessing viral annotations vs. AMGs*.</li>
<li>Among these annotations, how many were flagged as AMGs by <em>DRAM-v</em>?</li>
<li>What broad metabolic categories did the AMGs fall into? </li>
<li>Discussion point: How might we investigate whether identified putative AMGs are actually <em>within</em> the viral genomes, rather than residual contaminating host genomic sequence attached to the end of integrated prophage (but incompletely trimmed off in the excision process)?</li>
</ul>
</div>
<hr/>
<h3 id="tie-findings-to-your-initial-goal">Tie findings to your initial goal<a class="headerlink" href="#tie-findings-to-your-initial-goal" title="Permanent link">¶</a></h3>
<p>It is now time to explore the genomes and try to address your original goal!</p>
<p>You were tasked with identifying one of the following.</p>
<div class="admonition abstract">
<ol>
<li>Denitrification (Nitrate or nitrite to nitrogen)</li>
<li>Ammonia oxidation (Ammonia to nitrite or nitrate)</li>
<li>Anammox (Ammonia and nitrite to nitrogen)</li>
<li>Sulfur oxidation (SOX pathway, thiosulfate to sulfate)</li>
<li>Sulfur reduction (DSR pathway, sulfate to sulfide)</li>
<li>Photosynthetic carbon fixation</li>
<li>Non-photosynthetic carbon fixation (Reverse TCA or Wood-Ljundahl)</li>
<li>Non-polar flagella expression due to a chromosomal deletion</li>
<li>Plasmid-encoded antibiotic resistance</li>
<li>Aerobic (versus anaerobic) metabolism</li>
</ol>
</div>
<p>Depending on what you are looking for, you will either be trying to find gene(s) of relevance to a particular functional pathway, or the omission of genes that might be critical in function. In either case, make sure to use the taxonomy of each MAG to determine whether it is likely to be a worthwhile candidate for exploration, as some of these traits are quite restricted in terms of which organisms carry them.</p>
<p>To conduct this exersise, you should use the information generated with <code>DRAM</code> as well as the annotation files we created yesterday and that are available in the directory <code>10.gene_annotation_and_coverage/gene_annotations</code>. </p>
<p>Please note that we have also provided further annotation files within the directory <code>10.gene_annotation_and_coverage/example_annotation_tables</code> that contain information obtained after annotating the MAGs against additional databases (UniProt, UniRef100, KEGG, PFAM and TIGRfam). These example files can also be downloaded from <a href="../../resources/example_annotation_tables.zip">here</a>. </p>
<hr/>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>