
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day2/ex10.1_viruses/">
      
      
        <link rel="prev" href="../ex9_refining_bins/">
      
      
        <link rel="next" href="../../day3/ex11_coverage_and_taxonomy/">
      
      
      <link rel="icon" href="../../theme_images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.34">
    
    
      
        <title>Identifying viral contigs in metagenomic data - Metagenomics Summer School</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.35f28582.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#identifying-viral-contigs-in-metagenomic-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Metagenomics Summer School" class="md-header__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics Summer School
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Identifying viral contigs in metagenomic data
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day1/ex1_bash_and_scheduler/" class="md-tabs__link">
          
  
  Day 1

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../ex6_initial_binning/" class="md-tabs__link">
          
  
  Day 2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-tabs__link">
          
  
  Day 3

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../day4/ex15_gene_annotation_part3/" class="md-tabs__link">
          
  
  Day 4

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-tabs__link">
          
  
  Resources

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../supplementary/supplementary_1/" class="md-tabs__link">
          
  
  Supplementary

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Metagenomics Summer School" class="md-nav__button md-logo" aria-label="Metagenomics Summer School" data-md-component="logo">
      
  <img src="../../theme_images/nesi_ga.png" alt="logo">

    </a>
    Metagenomics Summer School
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 1
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex1_bash_and_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction I: Shell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_1_intro_to_scheduler/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction II: HPC and job scheduler
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex2_quality_filtering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filter raw reads by quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex3_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly I: Assembling contigs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex4_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly II: Varying the parameters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/ex5_evaluating_assemblies/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assembly evaluation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Day 2
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex6_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to binning
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex7_initial_binning/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Binning with multiple tools
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex8_bin_dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Bin dereplication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ex9_refining_bins/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Manual bin refinement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Identifying viral contigs in metagenomic data
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Identifying viral contigs in metagenomic data
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#identifying-viral-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      Identifying viral contigs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#identifying-viral-contigs-using-virsorter2" class="md-nav__link">
    <span class="md-ellipsis">
      Identifying viral contigs using VirSorter2
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#checking-quality-and-estimate-completeness-of-the-viral-contigs-via-checkv" class="md-nav__link">
    <span class="md-ellipsis">
      Checking quality and estimate completeness of the viral contigs via CheckV
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Checking quality and estimate completeness of the viral contigs via CheckV">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-virsorter2-and-checkv" class="md-nav__link">
    <span class="md-ellipsis">
      Run VirSorter2 and CheckV
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 3
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11_coverage_and_taxonomy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Assigning taxonomy to refined prokaryotic bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex11.1_phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex10.2_viruses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Virus taxonomy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex12_gene_prediction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene prediction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex13_gene_annotation_part1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation I: BLAST-like and HMM
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/ex14_gene_annotation_part2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation II: DRAM and coverage calculation
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Day 4
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex15_gene_annotation_part3/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene annotation III: DRAM distillation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16a_data_presentation_Intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction to data presentation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16b_data_presentation_Coverage/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Coverage heatmaps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Ordinations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16d_data_presentation_KEGG_pathways/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    KEGG pathway maps
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16e_data_presentation_Gene_synteny/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene synteny
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    CAZy heatmaps
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Resources
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/1_APPENDIX_ex8_Dereplication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex8): Dereplicating data from multiple assemblies
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex9): Generating input files for VizBin from DAS_Tool curated bins
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    APPENDIX (ex11) : Viral taxonomy prediction via vContact2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/7_command_line_shortcuts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Useful command line shortcuts
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Supplementary
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Supplementary
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_1/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI Setup
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../supplementary/supplementary_2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NeSI File system, Working directory and Symlinks
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#identifying-viral-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      Identifying viral contigs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#identifying-viral-contigs-using-virsorter2" class="md-nav__link">
    <span class="md-ellipsis">
      Identifying viral contigs using VirSorter2
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#checking-quality-and-estimate-completeness-of-the-viral-contigs-via-checkv" class="md-nav__link">
    <span class="md-ellipsis">
      Checking quality and estimate completeness of the viral contigs via CheckV
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Checking quality and estimate completeness of the viral contigs via CheckV">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-virsorter2-and-checkv" class="md-nav__link">
    <span class="md-ellipsis">
      Run VirSorter2 and CheckV
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="identifying-viral-contigs-in-metagenomic-data">Identifying viral contigs in metagenomic data<a class="headerlink" href="#identifying-viral-contigs-in-metagenomic-data" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#identifying-viral-contigs">Identifying viral contigs</a></li>
<li><a href="#identifying-viral-contigs-using-virsorter2">Identifying viral contigs using <code>VirSorter2</code></a></li>
</ul>
</div>
<h2 id="identifying-viral-contigs">Identifying viral contigs<a class="headerlink" href="#identifying-viral-contigs" title="Permanent link">&para;</a></h2>
<p>Viral metagenomics is a rapidly progressing field, and new software are constantly being developed and released each year that aim to better identify and characterise viral genomic sequences from assembled metagenomic sequence reads. </p>
<p>Currently, the most commonly used methods are <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00990-y"><code>VirSorter2</code></a>, <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00867-0"><code>VIBRANT</code></a>, and <a href="https://link.springer.com/epdf/10.1186/s40168-017-0283-5?author_access_token=YQgkTWibFIFPtRICkTjZF2_BpE1tBhCbnbw3BuzI2RMCpVMGldKV8DA9scozc7Z-db3ufPFz9-pswHsYVHyEsCrziBuECllLPOgZ6ANHsMeKF5KejrdDKdeASyDkxB5wfFDq523QSd01cnqxCLqCiQ%3D%3D"><code>VirFinder</code></a> (or the machine learning implementation of this, <a href="https://github.com/jessieren/DeepVirFinder"><code>DeepVirFinder</code></a>). A number of recent studies use one of these tools or a combination of several at once.</p>
<div class="admonition info">
<div class="tabbed-set tabbed-alternate" data-tabs="1:3"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1"><code>VirSorter2</code></label><label for="__tabbed_1_2"><code>VIBRANT</code></label><label for="__tabbed_1_3"><code>DeepVirFinder</code></label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>Uses a predicted protein homology reference database-based approach, together with searching for a number of pre-defined metrics based on known viral genomic features. <code>VirSorter2</code> includes dsDNAphage, ssDNA, and RNA viruses, and the viral groups Nucleocytoviricota and lavidaviridae.* </p>
<p><strong>More info</strong> </p>
<ul>
<li><a href="https://github.com/jiarong/VirSorter2">VirSorter2 GitHub</a></li>
<li><a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00990-y">Guo <em>et al.</em> (2021) VirSorter2: a multi-classifier, expert-guided approach to detect diverse DNA and RNA viruses</a></li>
</ul>
</div>
<div class="tabbed-block">
<p>Uses a machine learning approach based on protein similarity (non-reference-based similarity searches with multiple HMM sets), and is in principle applicable to bacterial and archaeal DNA and RNA viruses, integrated proviruses (which are excised from contigs by <code>VIBRANT</code>), and eukaryotic viruses. </p>
<p><strong>More info</strong> </p>
<ul>
<li><a href="https://github.com/AnantharamanLab/VIBRANT">VIBRANT GitHub</a></li>
<li><a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00867-0">Kieft, Zhou, and Anantharaman (2020) VIBRANT: automated recovery, annotation and curation of microbial viruses, and evaluation of viral community function from genomic sequences</a></li>
</ul>
</div>
<div class="tabbed-block">
<p>Uses a machine learning based approach based on <em>k</em>-mer frequencies. Having developed a database of the differences in <em>k</em>-mer frequencies between prokaryote and viral genomes, <em>VirFinder</em> examines assembled contigs and identifies whether their <em>k</em>-mer frequencies are comparable to known viruses in the database, using this to predict viral genomic sequence. This method has some limitations based on the viruses that were included when building the database (bacterial DNA viruses, but very few archaeal viruses, and, at least in some versions of the software, no eukaryotic viruses). However, tools are also provided to build your own database should you wish to develop an expanded one. Due to its distinctive <em>k</em>-mer frequency-based approach, <em>VirFinder</em> may also have the capability of identifying some novel viruses overlooked by tools such as <em>VIBRANT</em> or <em>VirSorter</em>.</p>
<p><strong>More info</strong> </p>
<ul>
<li><a href="https://github.com/jessieren/DeepVirFinder">DeepVirFinder GitHub</a></li>
</ul>
</div>
</div>
</div>
</div>
<hr />
<h2 id="identifying-viral-contigs-using-virsorter2">Identifying viral contigs using <code>VirSorter2</code><a class="headerlink" href="#identifying-viral-contigs-using-virsorter2" title="Permanent link">&para;</a></h2>
<p>For this exercise, we will use <code>VirSorter2</code> to identify viral contigs from our assembled contigs. We can also use <code>VirSorter2</code> to prepare files for later use with the gene annotation tool <code>DRAM-v</code>, which we'll run later in the day.</p>
<h2 id="checking-quality-and-estimate-completeness-of-the-viral-contigs-via-checkv">Checking quality and estimate completeness of the viral contigs via <code>CheckV</code><a class="headerlink" href="#checking-quality-and-estimate-completeness-of-the-viral-contigs-via-checkv" title="Permanent link">&para;</a></h2>
<p><a href="https://www.biorxiv.org/content/10.1101/2020.05.06.081778v1.abstract"><code>CheckV</code></a> was developed as an analogue to <code>CheckM</code>. <code>CheckV</code> first performs a 'contaminating sequence' trim, removing any retained (prokaryote) host sequence on the end of contigs with integrated prophage, and then assesses the quality and completeness of the assembled viral contigs. The quality of the contigs are also categoriesed based on the recently developed <a href="https://www.nature.com/articles/nbt.4306">Minimum Information about an Unclutivated Virus Genome</a> (MIUViG) standards for reporting sequences of unclutivated virus geneomes (such as those recovered from metagenomic sequencing data). The MIUViG were developed as an extension of the <a href="https://www.nature.com/articles/nbt.1823">Minimum Information about any (x) Sequence</a> (<a href="https://gensc.org/mixs/">MIxS</a>) standards, which include, among others, standards for Metagenome-Assembled Genomes (MIMAG).</p>
<h3 id="run-virsorter2-and-checkv">Run <code>VirSorter2</code> and <code>CheckV</code><a class="headerlink" href="#run-virsorter2-and-checkv" title="Permanent link">&para;</a></h3>
<p>These exercises will take place in the <code>7.viruses/</code> folder.</p>
<div class="admonition terminal-2">
<p class="admonition-title">Navigate to working directory</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/7.viruses/
</code></pre></div>
</div>
<p>For <code>VirSorter2</code>, we will input the assembled contigs from the <code>SPAdes</code> assembly we performed earlier. These assembly files have been copied to <code>7.viruses/spades_assembly/</code> for this exercise.</p>
<p>We will then run <code>CheckV</code> in the same script, providing the FASTA file of viral contigs output by <code>VirSorter2</code> as input (<code>final-viral-combined.fa</code>).</p>
<div class="admonition terminal-2">
<p class="admonition-title">Create a script named <code>VirSorter2_and_checkv.sl</code></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>nano<span class="w"> </span>VirSorter2_and_checkv.sl
</code></pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Remember to update <code>&lt;YOUR FOLDER&gt;</code> to your own folder.</p>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-2-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-2-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-2-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-2-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-2-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-2-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-2-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-2-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-2-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-2-10">10</a></span>
<span class="normal"><a href="#__codelineno-2-11">11</a></span>
<span class="normal"><a href="#__codelineno-2-12">12</a></span>
<span class="normal"><a href="#__codelineno-2-13">13</a></span>
<span class="normal"><a href="#__codelineno-2-14">14</a></span>
<span class="normal"><a href="#__codelineno-2-15">15</a></span>
<span class="normal"><a href="#__codelineno-2-16">16</a></span>
<span class="normal"><a href="#__codelineno-2-17">17</a></span>
<span class="normal"><a href="#__codelineno-2-18">18</a></span>
<span class="normal"><a href="#__codelineno-2-19">19</a></span>
<span class="normal"><a href="#__codelineno-2-20">20</a></span>
<span class="normal"><a href="#__codelineno-2-21">21</a></span>
<span class="normal"><a href="#__codelineno-2-22">22</a></span>
<span class="normal"><a href="#__codelineno-2-23">23</a></span>
<span class="normal"><a href="#__codelineno-2-24">24</a></span>
<span class="normal"><a href="#__codelineno-2-25">25</a></span>
<span class="normal"><a href="#__codelineno-2-26">26</a></span>
<span class="normal"><a href="#__codelineno-2-27">27</a></span>
<span class="normal"><a href="#__codelineno-2-28">28</a></span>
<span class="normal"><a href="#__codelineno-2-29">29</a></span>
<span class="normal"><a href="#__codelineno-2-30">30</a></span>
<span class="normal"><a href="#__codelineno-2-31">31</a></span>
<span class="normal"><a href="#__codelineno-2-32">32</a></span>
<span class="normal"><a href="#__codelineno-2-33">33</a></span>
<span class="normal"><a href="#__codelineno-2-34">34</a></span>
<span class="normal"><a href="#__codelineno-2-35">35</a></span>
<span class="normal"><a href="#__codelineno-2-36">36</a></span>
<span class="normal"><a href="#__codelineno-2-37">37</a></span>
<span class="normal"><a href="#__codelineno-2-38">38</a></span>
<span class="normal"><a href="#__codelineno-2-39">39</a></span>
<span class="normal"><a href="#__codelineno-2-40">40</a></span>
<span class="normal"><a href="#__codelineno-2-41">41</a></span>
<span class="normal"><a href="#__codelineno-2-42">42</a></span>
<span class="normal"><a href="#__codelineno-2-43">43</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-2-2" name="__codelineno-2-2"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-2-3" name="__codelineno-2-3"></a><span class="c1">#SBATCH --job-name      VirSorter2_and_checkv</span>
<a id="__codelineno-2-4" name="__codelineno-2-4"></a><span class="c1">#SBATCH --partition     milan</span>
<a id="__codelineno-2-5" name="__codelineno-2-5"></a><span class="c1">#SBATCH --time          05:00:00</span>
<a id="__codelineno-2-6" name="__codelineno-2-6"></a><span class="c1">#SBATCH --mem           2GB</span>
<a id="__codelineno-2-7" name="__codelineno-2-7"></a><span class="c1">#SBATCH --cpus-per-task 28</span>
<a id="__codelineno-2-8" name="__codelineno-2-8"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-2-9" name="__codelineno-2-9"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-2-10" name="__codelineno-2-10"></a>
<a id="__codelineno-2-11" name="__codelineno-2-11"></a><span class="c1"># Working directory</span>
<a id="__codelineno-2-12" name="__codelineno-2-12"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/7.viruses/
<a id="__codelineno-2-13" name="__codelineno-2-13"></a>
<a id="__codelineno-2-14" name="__codelineno-2-14"></a><span class="c1"># VirSorter2</span>
<a id="__codelineno-2-15" name="__codelineno-2-15"></a><span class="c1">## Load modules</span>
<a id="__codelineno-2-16" name="__codelineno-2-16"></a>module<span class="w"> </span>purge
<a id="__codelineno-2-17" name="__codelineno-2-17"></a>module<span class="w"> </span>unload<span class="w"> </span>XALT
<a id="__codelineno-2-18" name="__codelineno-2-18"></a>module<span class="w"> </span>load<span class="w"> </span>VirSorter/2.2.3-gimkl-2020a-Python-3.8.2
<a id="__codelineno-2-19" name="__codelineno-2-19"></a>
<a id="__codelineno-2-20" name="__codelineno-2-20"></a><span class="c1">## Output directory</span>
<a id="__codelineno-2-21" name="__codelineno-2-21"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>VirSorter2
<a id="__codelineno-2-22" name="__codelineno-2-22"></a>
<a id="__codelineno-2-23" name="__codelineno-2-23"></a><span class="c1">## Run VirSorter2</span>
<a id="__codelineno-2-24" name="__codelineno-2-24"></a>virsorter<span class="w"> </span>run<span class="w"> </span>-j<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-25" name="__codelineno-2-25"></a><span class="w">  </span>-i<span class="w"> </span>spades_assembly/spades_assembly.m1000.fna<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-26" name="__codelineno-2-26"></a><span class="w">  </span>-d<span class="w"> </span>/nesi/project/nesi02659/MGSS_2024/resources/databases/virsorter2_20210909<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-27" name="__codelineno-2-27"></a><span class="w">  </span>--min-score<span class="w"> </span><span class="m">0</span>.7<span class="w"> </span>--include-groups<span class="w"> </span>dsDNAphage,NCLDV,RNA,ssDNA,lavidaviridae<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-28" name="__codelineno-2-28"></a><span class="w">  </span>--prep-for-dramv<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-29" name="__codelineno-2-29"></a><span class="w">  </span>-l<span class="w"> </span>mgss<span class="w"> </span>-w<span class="w"> </span>VirSorter2<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-30" name="__codelineno-2-30"></a><span class="w">  </span>--tmpdir<span class="w"> </span><span class="si">${</span><span class="nv">SLURM_JOB_ID</span><span class="si">}</span>.tmp<span class="w"> </span>--rm-tmpdir<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-31" name="__codelineno-2-31"></a><span class="w">  </span>all<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-32" name="__codelineno-2-32"></a><span class="w">  </span>--config<span class="w"> </span><span class="nv">LOCAL_SCRATCH</span><span class="o">=</span><span class="si">${</span><span class="nv">TMPDIR</span><span class="k">:-</span><span class="p">/tmp</span><span class="si">}</span>
<a id="__codelineno-2-33" name="__codelineno-2-33"></a>
<a id="__codelineno-2-34" name="__codelineno-2-34"></a><span class="c1"># CheckV</span>
<a id="__codelineno-2-35" name="__codelineno-2-35"></a><span class="c1">## Load module</span>
<a id="__codelineno-2-36" name="__codelineno-2-36"></a>module<span class="w"> </span>purge
<a id="__codelineno-2-37" name="__codelineno-2-37"></a>module<span class="w"> </span>load<span class="w"> </span>CheckV/1.0.1-gimkl-2022a-Python-3.10.5
<a id="__codelineno-2-38" name="__codelineno-2-38"></a>
<a id="__codelineno-2-39" name="__codelineno-2-39"></a><span class="c1">## Output directory</span>
<a id="__codelineno-2-40" name="__codelineno-2-40"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>checkv_out
<a id="__codelineno-2-41" name="__codelineno-2-41"></a>
<a id="__codelineno-2-42" name="__codelineno-2-42"></a><span class="c1">## Run CheckV</span>
<a id="__codelineno-2-43" name="__codelineno-2-43"></a>checkv<span class="w"> </span>end_to_end<span class="w"> </span>VirSorter2/mgss-final-viral-combined.fa<span class="w"> </span>checkv_out/<span class="w"> </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span>
</code></pre></div></td></tr></table></div>
</div>
<div class="admonition warning">
<p class="admonition-title"><code>module unload XALT</code></p>
<p>For the <code>VirSorter/2.2.3-gimkl-2020a-Python-3.8.2</code> NeSI module to work properly, <strong>we must also include <code>module unload XALT</code></strong> in the script above.</p>
</div>
<div class="admonition tip">
<p class="admonition-title"><code>VirSorter2</code> parameters</p>
<p>The key parameters you may want to consider altering for your own work are <code>--min-score</code> and <code>--include-groups</code>. For today's excersice we will include all available groups (<code>--include-groups dsDNAphage,NCLDV,RNA,ssDNA,lavidaviridae</code>), and will set the min-score to 0.7. You can experiment with this value for your own data (see the <a href="https://github.com/jiarong/VirSorter2">Virsorter2 github page</a> for more information).</p>
</div>
<div class="admonition terminal-2">
<p class="admonition-title">Submit the script as a slurm job</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>sbatch<span class="w"> </span>VirSorter2_and_checkv.sl
</code></pre></div>
</div>
<div class="admonition note">
<p class="admonition-title"><code>VirSorter2</code> for your own work</p>
<p>The required databases for <code>VirSorter2</code> are not loaded with the NeSI module. For your own work, you will need to first download these databases and provide the path to the <code>-d</code> flag. For today's workshop this is already set up.</p>
</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.07f07601.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.56dfad97.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>