
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day1/ex3_assembly/" rel="canonical"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.1, mkdocs-material-8.3.9" name="generator"/>
<title>Assembly - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.1d29e8d0.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.cbb835fc.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gdesc-inner { font-size: 0.75rem; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="" data-md-color-primary="indigo" data-md-color-scheme="" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#assembly">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Assembly
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-tabs__inner md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day1" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../ex1_bash_and_scheduler/">
        Introduction to shell &amp; HPC Job Scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex2_quality_filtering/">
        Quality filtering raw reads
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" data-md-toggle="toc" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Assembly
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Assembly
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#the-standard-input-files-for-spades-and-idba-ud">
    The standard input files for SPAdes and IDBA-UD
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#basic-assembly-parameters">
    Basic assembly parameters
  </a>
<nav aria-label="Basic assembly parameters" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#setting-the-k-mer-size">
    Setting the k-mer size
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#specifying-the-number-of-threads">
    Specifying the number of threads
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#setting-a-memory-limit">
    Setting a memory limit
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#preparing-an-assembly-job-for-slurm">
    Preparing an assembly job for slurm
  </a>
<nav aria-label="Preparing an assembly job for slurm" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#submitting-a-spades-job-to-nesi-using-slurm">
    Submitting a SPAdes job to NeSI using slurm
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#monitoring-job-progress">
    Monitoring job progress
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#submitting-an-idba-ud-job-to-nesi-using-slurm">
    Submitting an IDBA-UD job to NeSI using slurm
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day2" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_4" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day3" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Per-sample coverage and assigning taxonomy
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation (part 1)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation (part 2)
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Day4" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation (part 3)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Presentation of data: Per-sample coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        (*Optional*) Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: *Optional exercise*: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_6" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Resources" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8):Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): How to generate the blast files provided in [ex15_gene_synteny](../day4/ex16e_data_presentation_Gene_synteny.md)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/6_APPENDIX_ex15_gene_synteny_grab_GOI/">
        APPENDIX (ex15): Grab Gene of Interest (generate the *cys.txt* files provided in [ex15_gene_synteny](https://github.com/GenomicsAotearoa/metagenomics_summer_school/blob/master/materials/day4/ex16e_data_presentation_Gene_synteny.md))
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#the-standard-input-files-for-spades-and-idba-ud">
    The standard input files for SPAdes and IDBA-UD
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#basic-assembly-parameters">
    Basic assembly parameters
  </a>
<nav aria-label="Basic assembly parameters" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#setting-the-k-mer-size">
    Setting the k-mer size
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#specifying-the-number-of-threads">
    Specifying the number of threads
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#setting-a-memory-limit">
    Setting a memory limit
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#preparing-an-assembly-job-for-slurm">
    Preparing an assembly job for slurm
  </a>
<nav aria-label="Preparing an assembly job for slurm" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#submitting-a-spades-job-to-nesi-using-slurm">
    Submitting a SPAdes job to NeSI using slurm
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#monitoring-job-progress">
    Monitoring job progress
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#submitting-an-idba-ud-job-to-nesi-using-slurm">
    Submitting an IDBA-UD job to NeSI using slurm
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school/edit/master/docs/day1/ex3_assembly.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg>
</a>
<h1 id="assembly">Assembly<a class="headerlink" href="#assembly" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#the-standard-input-files-for-spades-and-idba-ud">Become familiar with the standard input files for <code>SPAdes</code> and <code>IDBA-UD</code></a></li>
<li><a href="#basic-assembly-parameters">Understand the basic parameters that should be modified when using these assemblers</a></li>
<li><a href="#preparing-an-assembly-job-for-slurm">Prepare an assembly job to run under slurm</a></li>
</ul>
</div>
<p>All work for this exercise will occur in the <code>3.assembly/</code> directory.</p>
<hr/>
<h3 id="the-standard-input-files-for-spades-and-idba-ud">The standard input files for <em>SPAdes</em> and <em>IDBA-UD</em><a class="headerlink" href="#the-standard-input-files-for-spades-and-idba-ud" title="Permanent link">¶</a></h3>
<p>Although they both make use of the same types of data, both <code>SPAdes</code> and <code>IDBA-UD</code> have their own preferences for how sequence data is provided to them. To begin, we will look at the types of data accepted by <code>SPAdes</code>:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>module purge
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>module load SPAdes/3.15.4-gimkl-2022a-Python-3.10.5
</code></pre></div>
<details class="note">
<summary><code>spades -h</code></summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>spades.py -h
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="c1"># ...</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="c1">#Input data:</span>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="c1">#-1      &lt;filename&gt;      file with forward paired-end reads</span>
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a><span class="c1">#-2      &lt;filename&gt;      file with reverse paired-end reads</span>
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a><span class="c1">#-s      &lt;filename&gt;      file with unpaired reads</span>
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a><span class="c1">#--mp&lt;#&gt;-1       &lt;filename&gt;      file with forward reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,..,9)</span>
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a><span class="c1">#--mp&lt;#&gt;-2       &lt;filename&gt;      file with reverse reads for mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,..,9)</span>
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a><span class="c1">#--hqmp&lt;#&gt;-1     &lt;filename&gt;      file with forward reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,..,9)</span>
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a><span class="c1">#--hqmp&lt;#&gt;-2     &lt;filename&gt;      file with reverse reads for high-quality mate-pair library number &lt;#&gt; (&lt;#&gt; = 1,2,..,9)</span>
<a href="#__codelineno-1-11" id="__codelineno-1-11" name="__codelineno-1-11"></a><span class="c1">#--nxmate&lt;#&gt;-1   &lt;filename&gt;      file with forward reads for Lucigen NxMate library number &lt;#&gt; (&lt;#&gt; = 1,2,..,9)</span>
<a href="#__codelineno-1-12" id="__codelineno-1-12" name="__codelineno-1-12"></a><span class="c1">#--nxmate&lt;#&gt;-2   &lt;filename&gt;      file with reverse reads for Lucigen NxMate library number &lt;#&gt; (&lt;#&gt; = 1,2,..,9)</span>
<a href="#__codelineno-1-13" id="__codelineno-1-13" name="__codelineno-1-13"></a><span class="c1">#--sanger        &lt;filename&gt;      file with Sanger reads</span>
<a href="#__codelineno-1-14" id="__codelineno-1-14" name="__codelineno-1-14"></a><span class="c1">#--pacbio        &lt;filename&gt;      file with PacBio reads</span>
<a href="#__codelineno-1-15" id="__codelineno-1-15" name="__codelineno-1-15"></a><span class="c1">#--nanopore      &lt;filename&gt;      file with Nanopore reads</span>
<a href="#__codelineno-1-16" id="__codelineno-1-16" name="__codelineno-1-16"></a><span class="c1">#--tslr  &lt;filename&gt;      file with TSLR-contigs</span>
<a href="#__codelineno-1-17" id="__codelineno-1-17" name="__codelineno-1-17"></a><span class="c1">#--trusted-contigs       &lt;filename&gt;      file with trusted contigs</span>
<a href="#__codelineno-1-18" id="__codelineno-1-18" name="__codelineno-1-18"></a><span class="c1">#--untrusted-contigs     &lt;filename&gt;      file with untrusted contigs</span>
</code></pre></div>
</details>
<p>At a glance, you could provide any of the following data types to <code>SPAdes</code> and have it perform an assembly:</p>
<ol>
<li>Illumina paired-end sequencing data, either as standard library or Mate Pairs</li>
<li>Sanger sequences</li>
<li>PacBio reads</li>
<li>Oxford Nanopore reads</li>
<li>Pre-assembled scaffolds for guiding the assembly</li>
</ol>
<p>Awkwardly, while <code>SPAdes</code> accepts multiple input libraries (i.e. samples) in a single assembly, this behaviour does not work with the <code>-meta</code> flag enabled, which is needed in our example to activate metagenome assembly mode. We therefore need to concatenate our four individual samples together ready for sequencing.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>cat sample1_R1.fastq.gz sample2_R1.fastq.gz sample3_R1.fastq.gz sample4_R1.fastq.gz &gt; for_spades_R1.fq.gz
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>cat sample1_R2.fastq.gz sample2_R2.fastq.gz sample3_R2.fastq.gz sample4_R2.fastq.gz &gt; for_spades_R2.fq.gz
</code></pre></div>
<p>Note that these <em>fastQ</em> files are compressed, yet we can concatenate them together with the <code>cat</code> command regardless. This is a nice feature of <em>.gz</em> files that is handy to remember.</p>
<p>By contrast, what does <code>IDBA-UD</code> accept?</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>module purge
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>module load IDBA-UD/1.1.3-GCC-11.3.0
</code></pre></div>
<details class="abstract">
<summary><code>idba_ud --help</code></summary>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>  -o, --out arg <span class="o">(=</span>out<span class="o">)</span>                   output directory
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>  -r, --read arg                         fasta <span class="nb">read</span> file <span class="o">(</span>&lt;<span class="o">=</span><span class="m">128</span><span class="o">)</span>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>      --read_level_2 arg                 paired-end reads fasta <span class="k">for</span> second level scaffolds
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a>      --read_level_3 arg                 paired-end reads fasta <span class="k">for</span> third level scaffolds
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a>      --read_level_4 arg                 paired-end reads fasta <span class="k">for</span> fourth level scaffolds
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a>      --read_level_5 arg                 paired-end reads fasta <span class="k">for</span> fifth level scaffolds
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a>  -l, --long_read arg                    fasta long <span class="nb">read</span> file <span class="o">(</span>&gt;128<span class="o">)</span>
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a>      --mink arg <span class="o">(=</span><span class="m">20</span><span class="o">)</span>                   minimum k value <span class="o">(</span>&lt;<span class="o">=</span><span class="m">124</span><span class="o">)</span>
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a>      --maxk arg <span class="o">(=</span><span class="m">100</span><span class="o">)</span>                  maximum k value <span class="o">(</span>&lt;<span class="o">=</span><span class="m">124</span><span class="o">)</span>
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a>      --step arg <span class="o">(=</span><span class="m">20</span><span class="o">)</span>                   increment of k-mer of each iteration
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a>      --inner_mink arg <span class="o">(=</span><span class="m">10</span><span class="o">)</span>             inner minimum k value
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a>      --inner_step arg <span class="o">(=</span><span class="m">5</span><span class="o">)</span>              inner increment of k-mer
<a href="#__codelineno-4-13" id="__codelineno-4-13" name="__codelineno-4-13"></a>      --prefix arg <span class="o">(=</span><span class="m">3</span><span class="o">)</span>                  prefix length used to build sub k-mer table
<a href="#__codelineno-4-14" id="__codelineno-4-14" name="__codelineno-4-14"></a>      --min_count arg <span class="o">(=</span><span class="m">2</span><span class="o">)</span>               minimum multiplicity <span class="k">for</span> filtering k-mer when building the graph
<a href="#__codelineno-4-15" id="__codelineno-4-15" name="__codelineno-4-15"></a>      --min_support arg <span class="o">(=</span><span class="m">1</span><span class="o">)</span>             minimum supoort <span class="k">in</span> each iteration
<a href="#__codelineno-4-16" id="__codelineno-4-16" name="__codelineno-4-16"></a>      --num_threads arg <span class="o">(=</span><span class="m">0</span><span class="o">)</span>             number of threads
<a href="#__codelineno-4-17" id="__codelineno-4-17" name="__codelineno-4-17"></a>      --seed_kmer arg <span class="o">(=</span><span class="m">30</span><span class="o">)</span>              seed kmer size <span class="k">for</span> alignment
<a href="#__codelineno-4-18" id="__codelineno-4-18" name="__codelineno-4-18"></a>      --min_contig arg <span class="o">(=</span><span class="m">200</span><span class="o">)</span>            minimum size of contig
<a href="#__codelineno-4-19" id="__codelineno-4-19" name="__codelineno-4-19"></a>      --similar arg <span class="o">(=</span><span class="m">0</span>.95<span class="o">)</span>              similarity <span class="k">for</span> alignment
<a href="#__codelineno-4-20" id="__codelineno-4-20" name="__codelineno-4-20"></a>      --max_mismatch arg <span class="o">(=</span><span class="m">3</span><span class="o">)</span>            max mismatch of error correction
<a href="#__codelineno-4-21" id="__codelineno-4-21" name="__codelineno-4-21"></a>      --min_pairs arg <span class="o">(=</span><span class="m">3</span><span class="o">)</span>               minimum number of pairs
<a href="#__codelineno-4-22" id="__codelineno-4-22" name="__codelineno-4-22"></a>      --no_bubble                        <span class="k">do</span> not merge bubble
<a href="#__codelineno-4-23" id="__codelineno-4-23" name="__codelineno-4-23"></a>      --no_local                         <span class="k">do</span> not use <span class="nb">local</span> assembly
<a href="#__codelineno-4-24" id="__codelineno-4-24" name="__codelineno-4-24"></a>      --no_coverage                      <span class="k">do</span> not iterate on coverage
<a href="#__codelineno-4-25" id="__codelineno-4-25" name="__codelineno-4-25"></a>      --no_correct                       <span class="k">do</span> not <span class="k">do</span> correction
<a href="#__codelineno-4-26" id="__codelineno-4-26" name="__codelineno-4-26"></a>      --pre_correction                   perform pre-correction before assembly
</code></pre></div>
</details>
<p>'Short' or 'long' reads, and only a single file for each. This means that if we want to assemble our community data using <code>IDBA-UD</code> we will need to pool the paired-end data into a single, interleaved <em>fastA</em> file. Interleaved means that instead of having a pair of files that contain the separate forward and reverse sequences, the read pairs are in a single file in alternating order. For example</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="c1"># Paired-end file, forward</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>&gt;read1_1
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>...
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>&gt;read2_1
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>...
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a>
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a><span class="c1"># Paired-end file, forward</span>
<a href="#__codelineno-5-8" id="__codelineno-5-8" name="__codelineno-5-8"></a>&gt;read1_2
<a href="#__codelineno-5-9" id="__codelineno-5-9" name="__codelineno-5-9"></a>...
<a href="#__codelineno-5-10" id="__codelineno-5-10" name="__codelineno-5-10"></a>&gt;read2_2
<a href="#__codelineno-5-11" id="__codelineno-5-11" name="__codelineno-5-11"></a>...
<a href="#__codelineno-5-12" id="__codelineno-5-12" name="__codelineno-5-12"></a>
<a href="#__codelineno-5-13" id="__codelineno-5-13" name="__codelineno-5-13"></a><span class="c1"># Interleaved file</span>
<a href="#__codelineno-5-14" id="__codelineno-5-14" name="__codelineno-5-14"></a>&gt;read1_1
<a href="#__codelineno-5-15" id="__codelineno-5-15" name="__codelineno-5-15"></a>...
<a href="#__codelineno-5-16" id="__codelineno-5-16" name="__codelineno-5-16"></a>&gt;read1_2
<a href="#__codelineno-5-17" id="__codelineno-5-17" name="__codelineno-5-17"></a>...
<a href="#__codelineno-5-18" id="__codelineno-5-18" name="__codelineno-5-18"></a>&gt;read2_1
<a href="#__codelineno-5-19" id="__codelineno-5-19" name="__codelineno-5-19"></a>...
<a href="#__codelineno-5-20" id="__codelineno-5-20" name="__codelineno-5-20"></a>&gt;read2_2
<a href="#__codelineno-5-21" id="__codelineno-5-21" name="__codelineno-5-21"></a>...
</code></pre></div>
<p>Fortunately, the <code>IDBA</code> set of tools comes with some helper scripts to achieve just this. Unfortunately we cannot apply this shuffling operation to compressed data, so we must decompress the data first.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>module load pigz/2.7
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a>
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a><span class="k">for</span> i <span class="k">in</span> sample1 sample2 sample3 sample4<span class="p">;</span>
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="k">do</span>
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a>  pigz --keep --decompress <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R1.fastq.gz <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R2.fastq.gz
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a>  fq2fa --merge <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R1.fastq <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_R2.fastq <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.fna
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a><span class="k">done</span>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a>
<a href="#__codelineno-6-9" id="__codelineno-6-9" name="__codelineno-6-9"></a>cat sample1.fna sample2.fna sample3.fna sample4.fna &gt; for_idba.fna
</code></pre></div>
<hr/>
<h3 id="basic-assembly-parameters">Basic assembly parameters<a class="headerlink" href="#basic-assembly-parameters" title="Permanent link">¶</a></h3>
<p>For any assembler, there are a <strong><em>lot</em></strong> of parameters that can be fine-tuned depending on your data. As no two data sets are the same, it is almost impossible to predict which parameter combinations will yield the best outcome for your dataset. That said, an assembly can be quite a resource-intensive process and it is generally not practical to test every permutation of parameter values with your data. In genomics, the saying goes that the best assembly is the one that answers your question. As long as the data you are receiving is meaningful to the hypothesis you are seeking to address, then your assembly is as good as it needs to be.</p>
<p>Generally speaking, assemblers are developed in a way where they run with default parameters that have been empirically demonstrated to produce the best outcome <strong><em>on average</em></strong> across multiple data sets. For most purposes, there is not a lot of need to change these, but some parameters that we would always want to look at include:</p>
<div class="admonition quote">
<ol>
<li><em>k</em>-mer sizes to be assembled over, and step size if using a range</li>
<li>Number of threads to use during assembly</li>
<li>Memory limit to prevent the assembler from using up all available RAM and forcing the computer to use its <a href="https://web.mit.edu/rhel-doc/5/RHEL-5-manual/Deployment_Guide-en-US/ch-swapspace.html">swap space</a></li>
</ol>
</div>
<h4 id="setting-the-k-mer-size">Setting the <em>k</em>-mer size<a class="headerlink" href="#setting-the-k-mer-size" title="Permanent link">¶</a></h4>
<p>Depending on which assembler you are using, the commands for choosing the <em>k</em>-mer sizes for the assembly vary slightly, but they are recognisable between programs. In <code>SPAdes</code>, you can set the <em>k</em>-mer size using either</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>spades.py -k <span class="m">21</span>,33,55,77,99,121 ...
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a>spades.py -k auto ...
</code></pre></div>
<p>The first command lets us specify the <em>k</em>-mers ourselves, or we are letting <code>SPAdes</code> automatically pick the most appropriate size. For <code>IDBA-UD</code>, we can select the <em>k</em>-mer size using</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>idba_ud --mink <span class="m">21</span> --maxk <span class="m">121</span> --step <span class="m">22</span>
</code></pre></div>
<p>Unlike <code>SPAdes</code>, we do not have fine-scale control over the <em>k</em>-mer sizes used in the assembly. We instead provide <code>IDBA-UD</code> with the first and last <em>k</em>-mer size to use, then specify the increment to use between these. In either case, it is important that we are always assembling using a <em>k</em>-mer of uneven (odd) length in order to avoid the creation of palindromic <em>k</em>-mers.</p>
<h4 id="specifying-the-number-of-threads">Specifying the number of threads<a class="headerlink" href="#specifying-the-number-of-threads" title="Permanent link">¶</a></h4>
<p>This is simple in either assembler:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>spades.py -t <span class="m">20</span> ...
<a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a>
<a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a>idba_ud --num_threads <span class="m">20</span> ...
</code></pre></div>
<p>The only thing to keep in mind is that these tools have different default behaviour. If no thread count is specified by the user, <code>SPAdes</code> will assemble with 16 threads. <code>IDBA-UD</code> will use all available threads, which can be problematic if you are using a shared compute environment that does not use a resource management system like slurm.</p>
<h4 id="setting-a-memory-limit">Setting a memory limit<a class="headerlink" href="#setting-a-memory-limit" title="Permanent link">¶</a></h4>
<p>By far, the worst feature of <code>SPAdes</code> is the high memory requirement for performing an assembly. In the absence of monitoring, <code>SPAdes</code> will request more and more memory as it proceeds. If this requires more memory than is available on your computer, your system will start to store memory to disk space. This is an extremely slow operation and can render your computer effectively unusable. In managed environments such as NeSI a memory limit is imposed upon all running jobs, but if you are not using such a system you are advised to set a memory limit when executing <code>SPAdes</code>:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a>spades.py -m 400GB ...
</code></pre></div>
<p>No such parameter exists in <code>IDBA-UD</code>, but it requires far less RAM than <code>SPAdes</code>, so you are less likely to need it.</p>
<hr/>
<h3 id="preparing-an-assembly-job-for-slurm">Preparing an assembly job for slurm<a class="headerlink" href="#preparing-an-assembly-job-for-slurm" title="Permanent link">¶</a></h3>
<p>NeSI does not allow users to execute large jobs interactively on the terminal. Instead, the node that we have logged in to (<em>lander02</em>) has only a small fraction of the computing resources that NeSI houses. The <em>lander</em> node is used to write small command scripts, which are then deployed to the large compute nodes by a system called <strong>slurm</strong>. The ins and outs of working in slurm are well beyond the scope of this workshop (and may not be relevant if your institution uses a different resource allocation system). In this workshop, we will therefore only be showing you how to write minimal slurm scripts sufficient to achieve our goals. By the end of the workshop, you should have built up a small collection of slurm scripts for performing the necessary stages of our workflow and with experience you will be able to modify these to suit your own needs.</p>
<h4 id="submitting-a-spades-job-to-nesi-using-slurm">Submitting a <em>SPAdes</em> job to NeSI using slurm<a class="headerlink" href="#submitting-a-spades-job-to-nesi-using-slurm" title="Permanent link">¶</a></h4>
<p>...
To begin, we need to open a text file using the <code>nano</code> text editor. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a><span class="nb">cd</span> <span class="m">3</span>.assembly/
<a href="#__codelineno-11-2" id="__codelineno-11-2" name="__codelineno-11-2"></a>
<a href="#__codelineno-11-3" id="__codelineno-11-3" name="__codelineno-11-3"></a>nano assembly_spades.sl
</code></pre></div>
<p>Into this file, either write or copy/paste the following commands:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-12-2" id="__codelineno-12-2" name="__codelineno-12-2"></a>
<a href="#__codelineno-12-3" id="__codelineno-12-3" name="__codelineno-12-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-12-4" id="__codelineno-12-4" name="__codelineno-12-4"></a><span class="c1">#SBATCH --job-name      spades_assembly</span>
<a href="#__codelineno-12-5" id="__codelineno-12-5" name="__codelineno-12-5"></a><span class="c1">#SBATCH --res           SummerSchool</span>
<a href="#__codelineno-12-6" id="__codelineno-12-6" name="__codelineno-12-6"></a><span class="c1">#SBATCH --time          00:50:00</span>
<a href="#__codelineno-12-7" id="__codelineno-12-7" name="__codelineno-12-7"></a><span class="c1">#SBATCH --mem           10GB</span>
<a href="#__codelineno-12-8" id="__codelineno-12-8" name="__codelineno-12-8"></a><span class="c1">#SBATCH --cpus-per-task 12</span>
<a href="#__codelineno-12-9" id="__codelineno-12-9" name="__codelineno-12-9"></a><span class="c1">#SBATCH --error         spades_assembly.err</span>
<a href="#__codelineno-12-10" id="__codelineno-12-10" name="__codelineno-12-10"></a><span class="c1">#SBATCH --output        spades_assembly.out</span>
<a href="#__codelineno-12-11" id="__codelineno-12-11" name="__codelineno-12-11"></a>
<a href="#__codelineno-12-12" id="__codelineno-12-12" name="__codelineno-12-12"></a>module purge
<a href="#__codelineno-12-13" id="__codelineno-12-13" name="__codelineno-12-13"></a>module load SPAdes/3.15.4-gimkl-2022a-Python-3.10.5
<a href="#__codelineno-12-14" id="__codelineno-12-14" name="__codelineno-12-14"></a>
<a href="#__codelineno-12-15" id="__codelineno-12-15" name="__codelineno-12-15"></a>spades.py --meta -k <span class="m">33</span>,55,77,99,121 -t <span class="m">12</span> -1 for_spades_R1.fq.gz -2 for_spades_R2.fq.gz -o spades_assembly/
</code></pre></div>
<p>To save your file, use <code>Ctrl + O</code> to save the file, then <code>Ctrl + X</code> to exit <code>nano</code>. Going through those lines one by one;</p>
<details class="abstract">
<summary>Review above submission script :Slurm parameters and their functions</summary>
<table>
<thead>
<tr>
<th align="left">Slurm parameter</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>#!/bin/bash -e</strong></td>
<td align="left">Header for the file, letting NeSI know how to interpret the following commands. The <code>-e</code> flag means that the slurm run will halt at the first failed command (rather than pushing through and trying to execute subsequent ones)</td>
</tr>
<tr>
<td align="left"><strong>--account</strong></td>
<td align="left">The name of the project account to run the run under. You are provided with this when you create a project on NeSI</td>
</tr>
<tr>
<td align="left"><strong>--job-name</strong></td>
<td align="left">The name of the job, to display when using the <code>squeue</code> command</td>
</tr>
<tr>
<td align="left"><strong>--res</strong></td>
<td align="left">This is a parameter that we need due to a system reservation for this workshop. For your own work, ignore this flag</td>
</tr>
<tr>
<td align="left"><strong>--time</strong></td>
<td align="left">Maximum run time for the job before it is killed</td>
</tr>
<tr>
<td align="left"><strong>--mem</strong></td>
<td align="left">Amount of server memory to allocate to the job. If this is exceeded, the job will be terminated</td>
</tr>
<tr>
<td align="left"><strong>--cpus-per-task</strong></td>
<td align="left">number of processing cores to assign to the job. This should match with the number used by your assembler</td>
</tr>
<tr>
<td align="left"><strong>--error</strong></td>
<td align="left">File to log the <a href="https://en.wikipedia.org/wiki/Standard_streams">standard error</a> stream of the program.<br/>This is typically used to prove error reports, or to just inform the user of job progress</td>
</tr>
<tr>
<td align="left"><strong>--output</strong></td>
<td align="left">File to log the <a href="https://en.wikipedia.org/wiki/Standard_streams">standard output</a> stream of the program.<br/>This is typically used to inform the user of job progress and supply messages</td>
</tr>
</tbody>
</table>
</details>
<p>The <code>module load</code> command needs to be invoked within your slurm script. It is also a good idea to explicitly set the path to your files within the job so that</p>
<ol>
<li>There is no chance of having the job fail immediately because it cannot find the relevant files</li>
<li>When looking back through your slurm logs, you know where the data is meant to be</li>
</ol>
<div class="admonition abstract">
<p class="admonition-title">When executing the <code>SPAdes</code> command, there are a few parameters to note here:</p>
<table>
<thead>
<tr>
<th align="left">Parameter</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>--meta</strong></td>
<td align="left">Activate metagenome assembly mode. Default is to assemble your metagenome using single genome assembly assumptions</td>
</tr>
<tr>
<td align="left"><strong>-k</strong></td>
<td align="left"><em>k</em>-mer sizes for assembly. These choices will provide the output we will use in the <strong>Binning</strong> session, but feel free to experiment with these to see if you can improve the assembly</td>
</tr>
<tr>
<td align="left"><strong>-1</strong></td>
<td align="left">Forward reads, matched to their reverse partners</td>
</tr>
<tr>
<td align="left"><strong>-2</strong></td>
<td align="left">Reverse reads, matched to their forward partners</td>
</tr>
<tr>
<td align="left"><strong>-o</strong></td>
<td align="left">Output directory for all files</td>
</tr>
</tbody>
</table>
</div>
<p>Note that we also prefix the command (<code>spades.py</code>) with the <code>srun</code> command. This is a command specific to slurm and allows NeSI to track the resource usage of the <code>SPAdes</code> job.</p>
<p>We don't explicitly set memory or thread counts for this job, simply for the sake of keeping the command uncluttered. The default memory limit of <code>SPAdes</code> (250 GB) is much higher than the 10 GB we have allowed our job here. If the memory cap was violated then both slurm and <code>SPAdes</code> will terminate the assembly. We have also left the number of threads at the default value of 16, which matches the number specified in the slurm header.</p>
<p>It is a good idea to match your number of threads request in the slurm script with what you intend to use with <code>SPAdes</code> because your project usage is calculated based off what you request in your slurm scripts rather than what you actually use. Requesting many unused threads simply drives your project down the priority queue. By contrast, requesting fewer threads than you attempt to use in the program (i.e. request 10 in slurm, set thread count to 30 in <code>SPAdes</code>) will result in reduced performance, as your <code>SPAdes</code> job will divide up jobs as though it has 30 threads, but only 10 will be provided. This is discussed <a href="https://www.codeguru.com/cpp/sample_chapter/article.php/c13533/Why-Too-Many-Threads-Hurts-Performance-and-What-to-do-About-It.htm">in this blog post</a>.</p>
<p>Once you are happy with your slurm script, execute the job by navigating to the location of your script and entering the command</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-13-1" id="__codelineno-13-1" name="__codelineno-13-1"></a>sbatch assembly_spades.sl
</code></pre></div>
<p>You will receive a message telling you the job identifier for your assembly. Record this number, as we will use it in the next exercise.</p>
<h4 id="monitoring-job-progress">Monitoring job progress<a class="headerlink" href="#monitoring-job-progress" title="Permanent link">¶</a></h4>
<p>You can view the status of your current jobs using the command</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-14-1" id="__codelineno-14-1" name="__codelineno-14-1"></a>squeue --me
<a href="#__codelineno-14-2" id="__codelineno-14-2" name="__codelineno-14-2"></a>
<a href="#__codelineno-14-3" id="__codelineno-14-3" name="__codelineno-14-3"></a>  JOBID     USER ACCOUNT            NAME  ST REASON    START_TIME                TIME TIME_LEFT NODES CPUS
<a href="#__codelineno-14-4" id="__codelineno-14-4" name="__codelineno-14-4"></a><span class="m">8744675</span>  dwai012 ga02676 spades_assembly  PD Priority  <span class="m">2019</span>-11-29T16:37:43       <span class="m">0</span>:00 <span class="m">00</span>:30:00      <span class="m">1</span>   <span class="m">16</span>
</code></pre></div>
<p>We can see here that the job has not yet begun, as NeSI is waiting for resources to come available. At the stage the <code>START_TIME</code> is an estimation of when the resources are expected to become available. When they do, the output will change to</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-15-1" id="__codelineno-15-1" name="__codelineno-15-1"></a>  JOBID     USER ACCOUNT            NAME  ST REASON    START_TIME                TIME TIME_LEFT NODES CPUS
<a href="#__codelineno-15-2" id="__codelineno-15-2" name="__codelineno-15-2"></a><span class="m">8744675</span>  dwai012 ga02676 spades_assembly   R None      <span class="m">2019</span>-11-29T16:40:00       <span class="m">1</span>:20 <span class="m">00</span>:28:40      <span class="m">1</span>   <span class="m">16</span>
</code></pre></div>
<p>Which allows us to track how far into our run we are, and see the remaining time for the job. The <code>START_TIME</code> column now reports the time the job actually began.</p>
<h4 id="submitting-an-idba-ud-job-to-nesi-using-slurm">Submitting an <em>IDBA-UD</em> job to NeSI using slurm<a class="headerlink" href="#submitting-an-idba-ud-job-to-nesi-using-slurm" title="Permanent link">¶</a></h4>
<p>To run an equivalent assembly with <code>IDBA-UD</code>, create a new slurm script as follows</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-16-1" id="__codelineno-16-1" name="__codelineno-16-1"></a>nano idbaud_assembly.sl
</code></pre></div>
<p>Paste or type in the following:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-17-1" id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="ch">#!/bin/bash -e</span>
<a href="#__codelineno-17-2" id="__codelineno-17-2" name="__codelineno-17-2"></a>
<a href="#__codelineno-17-3" id="__codelineno-17-3" name="__codelineno-17-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a href="#__codelineno-17-4" id="__codelineno-17-4" name="__codelineno-17-4"></a><span class="c1">#SBATCH --job-name      idbaud_assembly</span>
<a href="#__codelineno-17-5" id="__codelineno-17-5" name="__codelineno-17-5"></a><span class="c1">#SBATCH --res           SummerSchool</span>
<a href="#__codelineno-17-6" id="__codelineno-17-6" name="__codelineno-17-6"></a><span class="c1">#SBATCH --time          00:35:00</span>
<a href="#__codelineno-17-7" id="__codelineno-17-7" name="__codelineno-17-7"></a><span class="c1">#SBATCH --mem           4GB</span>
<a href="#__codelineno-17-8" id="__codelineno-17-8" name="__codelineno-17-8"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-17-9" id="__codelineno-17-9" name="__codelineno-17-9"></a><span class="c1">#SBATCH --error         idbaud_assembly.err</span>
<a href="#__codelineno-17-10" id="__codelineno-17-10" name="__codelineno-17-10"></a><span class="c1">#SBATCH --output        idbaud_assembly.out</span>
<a href="#__codelineno-17-11" id="__codelineno-17-11" name="__codelineno-17-11"></a>
<a href="#__codelineno-17-12" id="__codelineno-17-12" name="__codelineno-17-12"></a>module purge
<a href="#__codelineno-17-13" id="__codelineno-17-13" name="__codelineno-17-13"></a>module load IDBA-UD/1.1.3-GCC-11.3.0
<a href="#__codelineno-17-14" id="__codelineno-17-14" name="__codelineno-17-14"></a>
<a href="#__codelineno-17-15" id="__codelineno-17-15" name="__codelineno-17-15"></a>idba_ud --num_threads <span class="m">8</span> --mink <span class="m">33</span> --maxk <span class="m">99</span> --step <span class="m">22</span> -r for_idba.fna -o idbaud_assembly/
</code></pre></div>
<p>And submit the script as a slurm job:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-18-1" id="__codelineno-18-1" name="__codelineno-18-1"></a>sbatch idbaud_assembly.sl
</code></pre></div>
<p>Remember to record your job identification number.</p>
<hr/>
</article>
</div>
</div>
<a class="md-top md-icon" data-md-component="top" hidden="" href="#">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </a>
</main>
<footer class="md-footer">
<nav aria-label="Footer" class="md-footer__inner md-grid">
<a aria-label="Previous: Quality filtering raw reads" class="md-footer__link md-footer__link--prev" href="../ex2_quality_filtering/" rel="prev">
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</div>
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Previous
              </span>
              Quality filtering raw reads
            </div>
</div>
</a>
<a aria-label="Next: Assembly (part 2)" class="md-footer__link md-footer__link--next" href="../ex4_assembly/" rel="next">
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Next
              </span>
              Assembly (part 2)
            </div>
</div>
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"></path></svg>
</div>
</a>
</nav>
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.6c7ad80a.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom"});})</script></body>
</html>