
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day4/ex16b_data_presentation_Coverage/" rel="canonical"/>
<link href="../ex16a_data_presentation_Intro/" rel="prev"/>
<link href="../ex16c_OPTIONAL_data_presentation_Ordination/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Presentation of data: Per-sample coverage heatmaps - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#presentation-of-data-per-sample-coverage-heatmaps">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Presentation of data: Per-sample coverage heatmaps
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex10_viruses/">
        Day3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../ex15_gene_annotation_part3/">
        Day4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction to shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction to HPC &amp; HPC job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Quality filtering raw reads
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly (part 2)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Evaluating the assemblies
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning (continued)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manually refining bins
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation (part 1)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation (part 2) and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../ex15_gene_annotation_part3/">
        Gene annotation (part 3)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16a_data_presentation_Intro/">
        Presentation of data: Introduction
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Presentation of data: Per-sample coverage heatmaps
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Presentation of data: Per-sample coverage heatmaps
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#build-a-heatmap-of-average-coverage-per-sample-using-r">
    Build a heatmap of average coverage per sample using R
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#part-1-building-a-heatmap-of-mag-coverage-per-sample">
    Part 1 - Building a heatmap of MAG coverage per sample
  </a>
<nav aria-label="Part 1 - Building a heatmap of MAG coverage per sample" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#11-prepare-environment">
    1.1 Prepare environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#12-wrangle-data">
    1.2 Wrangle data
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#13-order-the-heatmap">
    1.3 Order the heatmap
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#14-set-colour-palettes">
    1.4 Set colour palettes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#15-build-heatmap">
    1.5 Build heatmap
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#part-2-building-a-heatmap-of-viral-contigs-per-sample">
    Part 2 - Building a heatmap of viral contigs per sample
  </a>
<nav aria-label="Part 2 - Building a heatmap of viral contigs per sample" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#21-prepare-environment">
    2.1 Prepare environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#22-wrangle-data">
    2.2 Wrangle data
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#23-order-the-heatmap">
    2.3 Order the heatmap
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#24-set-colour-palettes">
    2.4 Set colour palettes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#25-build-heatmap">
    2.5 Build heatmap
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16c_OPTIONAL_data_presentation_Ordination/">
        Presentation of data: Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16d_data_presentation_KEGG_pathways/">
        Presentation of data: KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16e_data_presentation_Gene_synteny/">
        Presentation of data: Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        Presentation of data: CAZy annotations heatmap
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#build-a-heatmap-of-average-coverage-per-sample-using-r">
    Build a heatmap of average coverage per sample using R
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#part-1-building-a-heatmap-of-mag-coverage-per-sample">
    Part 1 - Building a heatmap of MAG coverage per sample
  </a>
<nav aria-label="Part 1 - Building a heatmap of MAG coverage per sample" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#11-prepare-environment">
    1.1 Prepare environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#12-wrangle-data">
    1.2 Wrangle data
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#13-order-the-heatmap">
    1.3 Order the heatmap
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#14-set-colour-palettes">
    1.4 Set colour palettes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#15-build-heatmap">
    1.5 Build heatmap
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#part-2-building-a-heatmap-of-viral-contigs-per-sample">
    Part 2 - Building a heatmap of viral contigs per sample
  </a>
<nav aria-label="Part 2 - Building a heatmap of viral contigs per sample" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#21-prepare-environment">
    2.1 Prepare environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#22-wrangle-data">
    2.2 Wrangle data
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#23-order-the-heatmap">
    2.3 Order the heatmap
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#24-set-colour-palettes">
    2.4 Set colour palettes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#25-build-heatmap">
    2.5 Build heatmap
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="presentation-of-data-per-sample-coverage-heatmaps">Presentation of data: Per-sample coverage heatmaps<a class="headerlink" href="#presentation-of-data-per-sample-coverage-heatmaps" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#part-1---building-a-heatmap-of-mag-coverage-per-sample">Building a heatmap of MAG coverage per sample</a></li>
<li><a href="#part-2---building-a-heatmap-of-viral-contigs-per-sample">Building a heatmap of viral contigs per sample</a></li>
</ul>
</div>
<hr/>
<h3 id="build-a-heatmap-of-average-coverage-per-sample-using-r">Build a heatmap of average coverage per sample using <em>R</em><a class="headerlink" href="#build-a-heatmap-of-average-coverage-per-sample-using-r" title="Permanent link">¶</a></h3>
<div class="admonition quote">
<p>One of the first questions we often ask when studying the ecology of a system is: What are the pattens of abundance and distribution of taxa across the different samples? In the previous <a href="../../day3/ex14_gene_annotation_part2/">coverage calculation</a> exercises we generated per-sample coverage tables by mapping the quality-filtered unassembled reads back to the refined bins and the viral contigs to then generate coverage profiles for each. </p>
<p>As a reminder:</p>
<blockquote>
<p>Genomes in higher abundance in a sample will contribute more genomic sequence to the metagenome, and so the average depth of sequencing coverage for each of the different genomes provides a proxy for abundance in each sample.</p>
</blockquote>
<p>A simple way to present this information is via a heatmap. In this exercise we will build a clustered heatmap of these coverage profiles in <code>R</code>. Since we also have tables of taxonomy assignments (via <code>gtdb-tk</code> for MAGs) and/or predictions (via <code>vContact2</code> for viral contigs), we will also use these to add taxonomy information to the plot.</p>
<p>The coverage and taxonomy tables generated in earlier exercises have been copied to <code>11.data_presentation/coverage/</code> for use in these exercises.</p>
<p>In addition to this, a simple mapping file has also been created (<code>11.data_presentation/coverage/mapping_file.txt</code>). This is a tab-delimited file listing each sample ID in the first column, and the sample "Group" in the second column (<em>Group_A</em>, <em>Group_B</em>, and <em>Group_C</em>). This grouping might represent, for example, three different sampling sites that we want to compare between. If you had other data (such as environmental measures, community diversity measures, etc.) that you wish to incorporate in other downstream analyses (such an fitting environmental variables to an ordination, or correlation analyses) you could also add new columns to this file and load them at the same time.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As discussed in the <a href="../../day3/ex11_coverage_and_taxonomy/">coverage and taxonomy exercises</a>, it is usually necessary to normalise coverage values across samples based on equal sequencing depth. This isn't necessary with the mock metagenome data we're working with, but if you include this step in your own work you would read the <strong>normalised</strong> coverage tables into the steps outlined below.*</p>
</div>
<hr/>
<h3 id="part-1-building-a-heatmap-of-mag-coverage-per-sample">Part 1 - Building a heatmap of MAG coverage per sample<a class="headerlink" href="#part-1-building-a-heatmap-of-mag-coverage-per-sample" title="Permanent link">¶</a></h3>
<p>To get started, if you're not already, log back in to NeSI's <a href="https://jupyter.nesi.org.nz/hub/login">Jupyter hub</a> and open a <code>Notebook</code> running the <code>R 4.0.1</code> module as the kernel (or, outside the context of this workshop, open <code>RStudio</code> with the required packages installed (see the <a href="../ex16a_data_presentation_Intro/">data presentation intro</a> docs for more information)).</p>
<h4 id="11-prepare-environment">1.1 Prepare environment<a class="headerlink" href="#11-prepare-environment" title="Permanent link">¶</a></h4>
<p>First, set the working directory and load the required libraries.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1"># Set working directory ----</span>
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nf">setwd</span><span class="p">(</span><span class="s">'/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/11.data_presentation/coverage'</span><span class="p">)</span>
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="c1"># Load libraries ----</span>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a><span class="c1"># Tidyverse packages </span>
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a><span class="nf">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a><span class="nf">library</span><span class="p">(</span><span class="n">tibble</span><span class="p">)</span>
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a>
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a><span class="c1"># Visualisation package</span>
<a href="#__codelineno-0-13" id="__codelineno-0-13" name="__codelineno-0-13"></a><span class="nf">library</span><span class="p">(</span><span class="n">gplots</span><span class="p">)</span>
<a href="#__codelineno-0-14" id="__codelineno-0-14" name="__codelineno-0-14"></a>
<a href="#__codelineno-0-15" id="__codelineno-0-15" name="__codelineno-0-15"></a><span class="c1"># Ecological analyses</span>
<a href="#__codelineno-0-16" id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="nf">library</span><span class="p">(</span><span class="n">vegan</span><span class="p">)</span>
</code></pre></div>
</div>
<p><em>NOTE: after copying this code into the empty script file in <code>Rstudio</code>, remember that, to run the code, press <code>&lt;shift&gt; + &lt;enter&gt;</code> with the code block selected.</em></p>
<p>Import all relevant data as follows:</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="c1"># Read files ----</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="n">contig_cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">"bins_cov_table.txt"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Bin contig coverage table</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="n">gtdbtk_out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">"gtdbtk.bac120.summary.tsv"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Prokaryotic taxonomy from GTDB-Tk</span>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="n">virus_cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">"viruses_cov_table.txt"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Viral contig coverage table</span>
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a><span class="n">checkv_summary</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">"checkv_quality_summary.tsv"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Viral contig quality from CheckV</span>
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a><span class="n">vContact2_out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">"tax_predict_table.txt"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Viral taxonomy from vContact2</span>
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a><span class="n">metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">"mapping_file.txt"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Metadata/mapping file of environmental parameters</span>
</code></pre></div>
</div>
<h4 id="12-wrangle-data">1.2 Wrangle data<a class="headerlink" href="#12-wrangle-data" title="Permanent link">¶</a></h4>
<p>After importing the data tables, we need to subset the tables to only relevant columns. As noted during the <a href="../../day3/ex14_gene_annotation_part2/">coverage</a> exercises, it is important to remember that we currently have a table of coverage values for all <em>contigs</em> contained within each MAG. Since we're aiming to present coverage for each <em>MAG</em>, we need to reduce these contig coverages into a single mean coverage value per MAG per sample.</p>
<p>In the following code, we first <code>select()</code> columns of interest (i.e. the contig ID and sample coverages). We then remove the <code>.bam</code> suffix using the <code>rename_with()</code> function. Given that we require mean coverages per MAG, we create a column of MAG/bin IDs (via <code>mutate</code>) by extracting the relevant text from the contig ID using <code>str_replace()</code>. Finally, we group the data by the bin ID (via <code>group_by()</code>) then use a combination of <code>summarise()</code> and <code>across()</code> to obtain a mean of coverage values per bin per sample. Here, <code>summarise()</code> calculates the <code>mean</code> based on grouping variables set by <code>group_by()</code>, and this is done <code>across()</code> columns that have the column header/name which <code>contains("sample")</code>.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a><span class="c1">## 1. Obtain average coverage per MAG per sample ----</span>
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a><span class="n">MAG_cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">contig_cov</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a><span class="w">  </span><span class="c1"># Pick relevant columns</span>
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">contigName</span><span class="p">,</span><span class="w"> </span><span class="nf">ends_with</span><span class="p">(</span><span class="s">".bam"</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a><span class="w">  </span><span class="c1"># Remove ".bam" from sample columns</span>
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a><span class="w">  </span><span class="nf">rename_with</span><span class="p">(</span>
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a><span class="w">    </span><span class="n">.fn</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">sample_name</span><span class="p">)</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">sample_name</span><span class="p">,</span><span class="w"> </span><span class="s">".bam"</span><span class="p">),</span>
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a><span class="w">    </span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ends_with</span><span class="p">(</span><span class="s">".bam"</span><span class="p">)</span>
<a href="#__codelineno-2-9" id="__codelineno-2-9" name="__codelineno-2-9"></a><span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-2-10" id="__codelineno-2-10" name="__codelineno-2-10"></a><span class="w">  </span><span class="c1"># Add bin ID to data</span>
<a href="#__codelineno-2-11" id="__codelineno-2-11" name="__codelineno-2-11"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-2-12" id="__codelineno-2-12" name="__codelineno-2-12"></a><span class="w">    </span><span class="n">binID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_replace</span><span class="p">(</span><span class="n">contigName</span><span class="p">,</span><span class="w"> </span><span class="s">"(.*)_NODE_.*"</span><span class="p">,</span><span class="w"> </span><span class="s">"\\1"</span><span class="p">)</span>
<a href="#__codelineno-2-13" id="__codelineno-2-13" name="__codelineno-2-13"></a><span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-2-14" id="__codelineno-2-14" name="__codelineno-2-14"></a><span class="w">  </span><span class="c1"># Designate "binID" as a grouping factor</span>
<a href="#__codelineno-2-15" id="__codelineno-2-15" name="__codelineno-2-15"></a><span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">binID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-2-16" id="__codelineno-2-16" name="__codelineno-2-16"></a><span class="w">  </span><span class="c1"># Calculate average coverage per sample per bin</span>
<a href="#__codelineno-2-17" id="__codelineno-2-17" name="__codelineno-2-17"></a><span class="w">  </span><span class="nf">summarise</span><span class="p">(</span>
<a href="#__codelineno-2-18" id="__codelineno-2-18" name="__codelineno-2-18"></a><span class="w">    </span><span class="nf">across</span><span class="p">(</span>
<a href="#__codelineno-2-19" id="__codelineno-2-19" name="__codelineno-2-19"></a><span class="w">      </span><span class="nf">contains</span><span class="p">(</span><span class="s">"sample"</span><span class="p">),</span><span class="w"> </span><span class="n">mean</span>
<a href="#__codelineno-2-20" id="__codelineno-2-20" name="__codelineno-2-20"></a><span class="w">    </span><span class="p">)</span>
<a href="#__codelineno-2-21" id="__codelineno-2-21" name="__codelineno-2-21"></a><span class="w">  </span><span class="p">)</span>
</code></pre></div>
</div>
<p>Next, we would also like to append the lowest identified taxonomy to each MAG. But before that, we will need to tidy up the taxonomy table to make it more readable.</p>
<p>In the following code, we first <code>select()</code> the <code>user_genome</code> (which is equivalent to <code>binID</code> above) and the relevant taxonomic <code>classification</code> columns. We then <code>separate()</code> the taxonomic <code>classification</code> column from a long semicolon-separated (<code>;</code>) string to 7 columns describing the MAG's <code>domain</code>, <code>phylum</code>, <code>class</code>, <code>order</code>, <code>family</code>, <code>genus</code>, and <code>species</code>. Finally, we polish the taxonomy annotations. We remove the hierarchical prefixes by retaining everything after the double underscores (<code>__</code>) via <code>str_replace</code> across all columns except <code>user_genome</code> (via inverse selection).</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="c1">## 2. Clean up taxonomy table ----</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a><span class="n">bin_taxonomy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gtdbtk_out</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a><span class="w">  </span><span class="c1"># Select relevant columns</span>
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">user_genome</span><span class="p">,</span><span class="w"> </span><span class="n">classification</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a><span class="w">  </span><span class="c1"># Split taxonomic classifications into individual columns</span>
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a><span class="w">  </span><span class="nf">separate</span><span class="p">(</span>
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a><span class="w">    </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">classification</span><span class="p">,</span>
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a><span class="w">    </span><span class="n">into</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">"domain"</span><span class="p">,</span><span class="w"> </span><span class="s">"phylum"</span><span class="p">,</span><span class="w"> </span><span class="s">"class"</span><span class="p">,</span><span class="w"> </span><span class="s">"order"</span><span class="p">,</span><span class="w"> </span><span class="s">"family"</span><span class="p">,</span><span class="w"> </span><span class="s">"genus"</span><span class="p">,</span><span class="w"> </span><span class="s">"species"</span><span class="p">),</span>
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a><span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">';'</span>
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a><span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a><span class="w">    </span><span class="c1"># Remove ".filtered" from the bin ID</span>
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a><span class="w">    </span><span class="n">user_genome</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">user_genome</span><span class="p">,</span><span class="w"> </span><span class="s">".filtered"</span><span class="p">),</span>
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a><span class="w">    </span><span class="c1"># Remove "d__" from all taxonomy columns</span>
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a><span class="w">    </span><span class="nf">across</span><span class="p">(</span>
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a><span class="w">      </span><span class="c1"># Negative selection: change all columns except "user_genome"</span>
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a><span class="w">      </span><span class="o">-</span><span class="n">user_genome</span><span class="p">,</span>
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a><span class="w">      </span><span class="n">.fns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">taxa</span><span class="p">)</span><span class="w"> </span><span class="nf">str_replace</span><span class="p">(</span><span class="n">taxa</span><span class="p">,</span><span class="w"> </span><span class="s">"[a-z]__(.*)"</span><span class="p">,</span><span class="w"> </span><span class="s">"\\1"</span><span class="p">)</span>
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a><span class="w">    </span><span class="p">)</span>
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a><span class="w">  </span><span class="p">)</span>
</code></pre></div>
</div>
<p>After obtaining a tidy taxonomy table, we append the species annotations to the bin IDs as follows:</p>
<ol>
<li><code>left_join()</code> a subset of the columns (<code>user_genome</code>, <code>species</code>) from <code>bin_taxonomy</code> generated above based on bin IDs. Bin IDs is <code>binID</code> in <code>MAG_cov</code> and in <code>user_genome</code> in <code>bin_taxonomy</code>, so we must specifically tell <code>left_join()</code> which columns to join <code>by</code>.</li>
<li>We <code>unite()</code> the columns <code>binID</code> and <code>species</code> as 1 column and place an underscore <code>_</code> betwee the strings.</li>
<li>We then convert the <code>binID</code> column into <code>rownames</code> for the <code>data.frame</code> in preparation for the other steps.</li>
</ol>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="c1">## 3. Add species to bin ID ----</span>
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a><span class="n">MAG_cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">MAG_cov</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a><span class="w">  </span><span class="nf">left_join</span><span class="p">(</span>
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="n">bin_taxonomy</span><span class="p">,</span><span class="w"> </span><span class="n">user_genome</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">),</span>
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="w">    </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">"binID"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"user_genome"</span><span class="p">)</span>
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="w">  </span><span class="nf">unite</span><span class="p">(</span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"binID"</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">binID</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"_"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="w">  </span><span class="nf">column_to_rownames</span><span class="p">(</span><span class="s">"binID"</span><span class="p">)</span>
</code></pre></div>
</div>
<details class="note">
<summary>Too much data?</summary>
<p>In your own work, if you have too many MAGs, it can be difficult to present them concisely in a heatmap. Thus, you may want to only retain MAGs that fulfill some criteria. </p>
<p>For example, you may only be interested in the top 10 MAGs based on coverage:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="c1"># Create a vector of top 10 MAGs</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a><span class="n">top10</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">MAG_cov</span><span class="p">),</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">]</span>
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="c1"># Retain only top 10 MAGs</span>
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a><span class="n">MAG_cov_top10</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">MAG_cov</span><span class="p">[</span><span class="nf">rownames</span><span class="p">(</span><span class="n">MAG_cov</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">top10</span><span class="p">),</span><span class="w"> </span><span class="p">]</span>
</code></pre></div>
<p>Perhaps you want to filter MAGs based on their prevalence across your sampling regime? Assuming you would like to retain MAGs that are present in at least 50% of your samples:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="c1"># Create a presence/absence matrix</span>
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a><span class="n">MAG_prevalence</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">MAG_cov</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a>
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="c1"># Create a logical vector that fulfill criteria</span>
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a><span class="n">top_prev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="n">MAG_prevalence</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="p">(</span><span class="m">0.5</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">MAG_prevalence</span><span class="p">))</span>
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a>
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a><span class="c1"># Retain only MAGs in coverage table that fulfill criteria</span>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a><span class="n">MAG_cov_top_prev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">MAG_cov</span><span class="p">[</span><span class="n">top_prev</span><span class="p">,</span><span class="w"> </span><span class="p">]</span>
</code></pre></div>
</details>
<p>Very often, coverage data generated using metagenomic methods can be sparse and/or have values that differ by large orders of magnitude. This can hamper effective visualisation of the data. To remedy this, we can perform numeric transformations that can enhance visualisation of relative differences between values. Here, we will use a logarithm (base 2) transform with 1 added as a pseudocount (because <span class="arithmatex">\(\log 0\)</span> is undefined).</p>
<div class="admonition note">
<p class="admonition-title">On data transformation</p>
<p>Transformations applied to enhance data visualisation are not necessarily suitable for downstream statistical analyses. Depending on the attributes of your data (shape, sparseness, potential biases, etc.) and choice of analyses, it is recommended that you become familiar with field and tool-specifc assumptions, norms and requirements for data transformation.</p>
</div>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="c1">## 4. Transform coverage data ----</span>
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a><span class="n">MAG_cov_log2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">log2</span><span class="p">(</span><span class="n">MAG_cov</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</code></pre></div>
<h4 id="13-order-the-heatmap">1.3 Order the heatmap<a class="headerlink" href="#13-order-the-heatmap" title="Permanent link">¶</a></h4>
<p>We then need to prepare some form of ordering for our heatmap. This is usually presented in the form of a dendrogram based on results from hierarchical clustering. To do this, we first need to generate two dissimilarity/distance matrices based on transformed coverage values:</p>
<ul>
<li>Sample-wise dissimilarities</li>
<li>MAG-wise dissimilarities</li>
</ul>
<p>Here, we calculate <a href="https://en.wikipedia.org/wiki/Bray%E2%80%93Curtis_dissimilarity">Bray-Curtis dissimilarity</a> using <code>vegdist()</code> to generate dissimilarities between samples and MAGs. Then, the matrices are used as input to perform hierarchical clustering and plotted.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a><span class="c1"># Perform hierarchical clustering ----</span>
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a><span class="c1">## Dissimilarities between samples</span>
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a><span class="n">sample_dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vegdist</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">MAG_cov_log2</span><span class="p">),</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"bray"</span><span class="p">)</span>
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a><span class="n">sample_hclust</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">hclust</span><span class="p">(</span><span class="n">sample_dist</span><span class="p">,</span><span class="w"> </span><span class="s">"average"</span><span class="p">)</span>
<a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a>
<a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a><span class="nf">plot</span><span class="p">(</span>
<a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a><span class="w">  </span><span class="n">sample_hclust</span><span class="p">,</span>
<a href="#__codelineno-8-8" id="__codelineno-8-8" name="__codelineno-8-8"></a><span class="w">  </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Bray-Curtis dissimilarities between samples\n(MAG)"</span><span class="p">,</span>
<a href="#__codelineno-8-9" id="__codelineno-8-9" name="__codelineno-8-9"></a><span class="w">  </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Sample"</span><span class="p">,</span>
<a href="#__codelineno-8-10" id="__codelineno-8-10" name="__codelineno-8-10"></a><span class="w">  </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Height"</span><span class="p">,</span>
<a href="#__codelineno-8-11" id="__codelineno-8-11" name="__codelineno-8-11"></a><span class="w">  </span><span class="n">sub</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Method: average linkage"</span><span class="p">,</span>
<a href="#__codelineno-8-12" id="__codelineno-8-12" name="__codelineno-8-12"></a><span class="w">  </span><span class="n">hang</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span>
<a href="#__codelineno-8-13" id="__codelineno-8-13" name="__codelineno-8-13"></a><span class="p">)</span>
<a href="#__codelineno-8-14" id="__codelineno-8-14" name="__codelineno-8-14"></a>
<a href="#__codelineno-8-15" id="__codelineno-8-15" name="__codelineno-8-15"></a><span class="c1">## Dissimilarities between MAGs</span>
<a href="#__codelineno-8-16" id="__codelineno-8-16" name="__codelineno-8-16"></a><span class="n">MAG_dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vegdist</span><span class="p">(</span><span class="n">MAG_cov_log2</span><span class="p">)</span>
<a href="#__codelineno-8-17" id="__codelineno-8-17" name="__codelineno-8-17"></a><span class="n">MAG_hclust</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">hclust</span><span class="p">(</span><span class="n">MAG_dist</span><span class="p">,</span><span class="w"> </span><span class="s">"average"</span><span class="p">)</span>
<a href="#__codelineno-8-18" id="__codelineno-8-18" name="__codelineno-8-18"></a>
<a href="#__codelineno-8-19" id="__codelineno-8-19" name="__codelineno-8-19"></a><span class="nf">plot</span><span class="p">(</span>
<a href="#__codelineno-8-20" id="__codelineno-8-20" name="__codelineno-8-20"></a><span class="w">  </span><span class="n">MAG_hclust</span><span class="p">,</span>
<a href="#__codelineno-8-21" id="__codelineno-8-21" name="__codelineno-8-21"></a><span class="w">  </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Bray-Curtis dissimilarities between MAGs"</span><span class="p">,</span>
<a href="#__codelineno-8-22" id="__codelineno-8-22" name="__codelineno-8-22"></a><span class="w">  </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"MAGs"</span><span class="p">,</span>
<a href="#__codelineno-8-23" id="__codelineno-8-23" name="__codelineno-8-23"></a><span class="w">  </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Height"</span><span class="p">,</span>
<a href="#__codelineno-8-24" id="__codelineno-8-24" name="__codelineno-8-24"></a><span class="w">  </span><span class="n">sub</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Method: average linkage"</span><span class="p">,</span>
<a href="#__codelineno-8-25" id="__codelineno-8-25" name="__codelineno-8-25"></a><span class="w">  </span><span class="n">hang</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span><span class="p">,</span>
<a href="#__codelineno-8-26" id="__codelineno-8-26" name="__codelineno-8-26"></a><span class="w">  </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.75</span>
<a href="#__codelineno-8-27" id="__codelineno-8-27" name="__codelineno-8-27"></a><span class="p">)</span>
</code></pre></div>
</div>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.01.cov_hmp.hclust.MAG_sample.png"><img alt="image" src="../../figures/day4_coverage.01.cov_hmp.hclust.MAG_sample.png"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.02.cov_hmp.hclust.MAG.png"><img alt="image" src="../../figures/day4_coverage.02.cov_hmp.hclust.MAG.png"/></a></p>
<div class="admonition note">
<p class="admonition-title">Exporting figures</p>
<p>To export figures you have made, navigate your mouse pointer to the bottom right pane, then click on 'Export' on the second toolbar of the pane. You can export your figure as an image (e.g. TIFF, JPEG, PNG, BMP), a vector image (i.e. PDF, SVG) or PostScript (EPS). </p>
<p>If you would like to do this via code, you can wrap the plot code in between functions for graphic devices (i.e. <code>png()</code>, <code>jpeg()</code>, <code>tiff()</code>, <code>bmp()</code>, <code>pdf()</code>, and <code>postscript()</code>) and <code>dev.off()</code>. The following is an example:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a><span class="nf">png</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"file.png"</span><span class="p">,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span>
<a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a><span class="nf">plot</span><span class="p">(</span><span class="kc">...</span><span class="p">)</span>
<a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a><span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
</div>
<h4 id="14-set-colour-palettes">1.4 Set colour palettes<a class="headerlink" href="#14-set-colour-palettes" title="Permanent link">¶</a></h4>
<p>The penultimate step before building our heatmap is to set the colours that will be used to represent annotations and the cell/tile values. In this case, annotations are sample groups as designated in the metadata (a.k.a. mapping) file and MAG taxonomy. In the code below, we will set 3 palettes:</p>
<ul>
<li>Sample groups</li>
<li>MAG phylum</li>
<li>Cell/tile values</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Pre-installed colour palettes</p>
<p>Base <code>R</code> has many colour palettes that come pre-installed. Older versions of R had colour palettes (also known as palette <code>R3</code>) that were not ideal for people with colour vision deficiencies. However, newer versions (4.0+) now carry better palettes that are more colour-blind friendly (see <a href="https://developer.r-project.org/Blog/public/2019/11/21/a-new-palette-for-r/">here</a>).</p>
<p>You can also quickly check what these colour palettes are:</p>
<p><div class="highlight"><pre><span></span><code><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a><span class="c1"># What colour palettes come pre-installed?</span>
<a href="#__codelineno-10-2" id="__codelineno-10-2" name="__codelineno-10-2"></a><span class="nf">palette.pals</span><span class="p">()</span>
<a href="#__codelineno-10-3" id="__codelineno-10-3" name="__codelineno-10-3"></a>
<a href="#__codelineno-10-4" id="__codelineno-10-4" name="__codelineno-10-4"></a><span class="c1"># Plotting the "Okabe-Ito" palette</span>
<a href="#__codelineno-10-5" id="__codelineno-10-5" name="__codelineno-10-5"></a><span class="c1"># (Requires the "scales" package)</span>
<a href="#__codelineno-10-6" id="__codelineno-10-6" name="__codelineno-10-6"></a><span class="n">scales</span><span class="o">::</span><span class="nf">show_col</span><span class="p">(</span>
<a href="#__codelineno-10-7" id="__codelineno-10-7" name="__codelineno-10-7"></a><span class="w">  </span><span class="nf">palette.colors</span><span class="p">(</span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Okabe-Ito"</span><span class="p">)</span>
<a href="#__codelineno-10-8" id="__codelineno-10-8" name="__codelineno-10-8"></a><span class="p">)</span>
</code></pre></div>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.03.show_col.Okabe-Ito.png"><img alt="image" src="../../figures/day4_coverage.03.show_col.Okabe-Ito.png"/></a></p>
</div>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a><span class="c1"># Set colour palette ----</span>
<a href="#__codelineno-11-2" id="__codelineno-11-2" name="__codelineno-11-2"></a><span class="c1">## Prepare sample groups and colour</span>
<a href="#__codelineno-11-3" id="__codelineno-11-3" name="__codelineno-11-3"></a><span class="n">metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">metadata</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-11-4" id="__codelineno-11-4" name="__codelineno-11-4"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-11-5" id="__codelineno-11-5" name="__codelineno-11-5"></a><span class="w">    </span><span class="n">Group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.factor</span><span class="p">(</span><span class="n">Group</span><span class="p">)</span>
<a href="#__codelineno-11-6" id="__codelineno-11-6" name="__codelineno-11-6"></a><span class="w">  </span><span class="p">)</span>
<a href="#__codelineno-11-7" id="__codelineno-11-7" name="__codelineno-11-7"></a>
<a href="#__codelineno-11-8" id="__codelineno-11-8" name="__codelineno-11-8"></a><span class="n">group_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<a href="#__codelineno-11-9" id="__codelineno-11-9" name="__codelineno-11-9"></a><span class="w">  </span><span class="n">Group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">levels</span><span class="p">(</span><span class="n">metadata</span><span class="o">$</span><span class="n">Group</span><span class="p">),</span>
<a href="#__codelineno-11-10" id="__codelineno-11-10" name="__codelineno-11-10"></a><span class="w">  </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">palette.colors</span><span class="p">(</span>
<a href="#__codelineno-11-11" id="__codelineno-11-11" name="__codelineno-11-11"></a><span class="w">    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">levels</span><span class="p">(</span><span class="n">metadata</span><span class="o">$</span><span class="n">Group</span><span class="p">)),</span>
<a href="#__codelineno-11-12" id="__codelineno-11-12" name="__codelineno-11-12"></a><span class="w">    </span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Okabe-Ito"</span><span class="p">)</span>
<a href="#__codelineno-11-13" id="__codelineno-11-13" name="__codelineno-11-13"></a><span class="p">)</span>
<a href="#__codelineno-11-14" id="__codelineno-11-14" name="__codelineno-11-14"></a>
<a href="#__codelineno-11-15" id="__codelineno-11-15" name="__codelineno-11-15"></a><span class="n">sample_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="n">metadata</span><span class="p">,</span><span class="w"> </span><span class="n">group_colour</span><span class="p">)</span>
<a href="#__codelineno-11-16" id="__codelineno-11-16" name="__codelineno-11-16"></a>
<a href="#__codelineno-11-17" id="__codelineno-11-17" name="__codelineno-11-17"></a><span class="c1">## Prepare MAG colours based on taxonomic class</span>
<a href="#__codelineno-11-18" id="__codelineno-11-18" name="__codelineno-11-18"></a><span class="c1">### Remember that the new labels (bin ID) are 'binID_species'</span>
<a href="#__codelineno-11-19" id="__codelineno-11-19" name="__codelineno-11-19"></a><span class="n">bin_phylum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">bin_taxonomy</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-11-20" id="__codelineno-11-20" name="__codelineno-11-20"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">user_genome</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">phylum</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-11-21" id="__codelineno-11-21" name="__codelineno-11-21"></a><span class="w">  </span><span class="nf">unite</span><span class="p">(</span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"binID"</span><span class="p">,</span><span class="w"> </span><span class="n">user_genome</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"_"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-11-22" id="__codelineno-11-22" name="__codelineno-11-22"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-11-23" id="__codelineno-11-23" name="__codelineno-11-23"></a><span class="w">    </span><span class="n">phylum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.factor</span><span class="p">(</span><span class="n">phylum</span><span class="p">)</span>
<a href="#__codelineno-11-24" id="__codelineno-11-24" name="__codelineno-11-24"></a><span class="w">  </span><span class="p">)</span>
<a href="#__codelineno-11-25" id="__codelineno-11-25" name="__codelineno-11-25"></a>
<a href="#__codelineno-11-26" id="__codelineno-11-26" name="__codelineno-11-26"></a><span class="n">phylum_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<a href="#__codelineno-11-27" id="__codelineno-11-27" name="__codelineno-11-27"></a><span class="w">  </span><span class="n">phylum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">levels</span><span class="p">(</span><span class="n">bin_phylum</span><span class="o">$</span><span class="n">phylum</span><span class="p">),</span>
<a href="#__codelineno-11-28" id="__codelineno-11-28" name="__codelineno-11-28"></a><span class="w">  </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">palette.colors</span><span class="p">(</span>
<a href="#__codelineno-11-29" id="__codelineno-11-29" name="__codelineno-11-29"></a><span class="w">    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">levels</span><span class="p">(</span><span class="n">bin_phylum</span><span class="o">$</span><span class="n">phylum</span><span class="p">)),</span>
<a href="#__codelineno-11-30" id="__codelineno-11-30" name="__codelineno-11-30"></a><span class="w">    </span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"R4"</span><span class="p">)</span>
<a href="#__codelineno-11-31" id="__codelineno-11-31" name="__codelineno-11-31"></a><span class="p">)</span>
<a href="#__codelineno-11-32" id="__codelineno-11-32" name="__codelineno-11-32"></a>
<a href="#__codelineno-11-33" id="__codelineno-11-33" name="__codelineno-11-33"></a><span class="n">MAG_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="n">bin_phylum</span><span class="p">,</span><span class="w"> </span><span class="n">phylum_colour</span><span class="p">)</span>
<a href="#__codelineno-11-34" id="__codelineno-11-34" name="__codelineno-11-34"></a>
<a href="#__codelineno-11-35" id="__codelineno-11-35" name="__codelineno-11-35"></a><span class="c1">## Set a grayscale for cell colours</span>
<a href="#__codelineno-11-36" id="__codelineno-11-36" name="__codelineno-11-36"></a><span class="n">cell_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colorRampPalette</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">"white"</span><span class="p">,</span><span class="w"> </span><span class="s">"black"</span><span class="p">),</span><span class="w"> </span><span class="n">space</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Lab"</span><span class="p">)(</span><span class="m">2</span><span class="o">^</span><span class="m">8</span><span class="p">)</span>
</code></pre></div>
</div>
<h4 id="15-build-heatmap">1.5 Build heatmap<a class="headerlink" href="#15-build-heatmap" title="Permanent link">¶</a></h4>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a><span class="nf">heatmap.2</span><span class="p">(</span>
<a href="#__codelineno-12-2" id="__codelineno-12-2" name="__codelineno-12-2"></a><span class="w">    </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">MAG_cov_log2</span><span class="p">),</span><span class="w"> </span><span class="c1"># Log2 transformed coverage table</span>
<a href="#__codelineno-12-3" id="__codelineno-12-3" name="__codelineno-12-3"></a><span class="w">    </span><span class="n">Colv</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.dendrogram</span><span class="p">(</span><span class="n">sample_hclust</span><span class="p">),</span><span class="w"> </span><span class="c1"># Arrange columns based on sample-wise dissimilarities</span>
<a href="#__codelineno-12-4" id="__codelineno-12-4" name="__codelineno-12-4"></a><span class="w">    </span><span class="n">Rowv</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.dendrogram</span><span class="p">(</span><span class="n">MAG_hclust</span><span class="p">),</span><span class="w"> </span><span class="c1"># Arrange rows based on MAG-wise dissimilarities</span>
<a href="#__codelineno-12-5" id="__codelineno-12-5" name="__codelineno-12-5"></a><span class="w">    </span><span class="n">margins</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="m">20</span><span class="p">),</span><span class="w"> </span><span class="c1"># Numbers represent margins for bottom &amp; right side</span>
<a href="#__codelineno-12-6" id="__codelineno-12-6" name="__codelineno-12-6"></a><span class="w">    </span><span class="n">RowSideColors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">MAG_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span><span class="w"> </span><span class="c1"># Colours on the row represent MAG phylum</span>
<a href="#__codelineno-12-7" id="__codelineno-12-7" name="__codelineno-12-7"></a><span class="w">    </span><span class="n">ColSideColors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sample_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span><span class="w"> </span><span class="c1"># Colours on the column represent sample groups</span>
<a href="#__codelineno-12-8" id="__codelineno-12-8" name="__codelineno-12-8"></a><span class="w">    </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"none"</span><span class="p">,</span><span class="w"> </span><span class="c1"># Do not scale values provided</span>
<a href="#__codelineno-12-9" id="__codelineno-12-9" name="__codelineno-12-9"></a><span class="w">    </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cell_colour</span><span class="p">,</span><span class="w"> </span><span class="c1"># Colour shading based on values</span>
<a href="#__codelineno-12-10" id="__codelineno-12-10" name="__codelineno-12-10"></a><span class="w">    </span><span class="n">trace</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"none"</span><span class="p">,</span><span class="w"> </span>
<a href="#__codelineno-12-11" id="__codelineno-12-11" name="__codelineno-12-11"></a><span class="w">    </span><span class="n">density.info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"none"</span>
<a href="#__codelineno-12-12" id="__codelineno-12-12" name="__codelineno-12-12"></a><span class="p">)</span>
<a href="#__codelineno-12-13" id="__codelineno-12-13" name="__codelineno-12-13"></a><span class="c1">## Legend for MAG phyla</span>
<a href="#__codelineno-12-14" id="__codelineno-12-14" name="__codelineno-12-14"></a><span class="nf">legend</span><span class="p">(</span>
<a href="#__codelineno-12-15" id="__codelineno-12-15" name="__codelineno-12-15"></a><span class="w">  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"bottomleft"</span><span class="p">,</span><span class="w"> </span><span class="c1"># Legend position</span>
<a href="#__codelineno-12-16" id="__codelineno-12-16" name="__codelineno-12-16"></a><span class="w">  </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">phylum_colour</span><span class="o">$</span><span class="n">phylum</span><span class="p">,</span><span class="w"> </span><span class="c1"># Legend text</span>
<a href="#__codelineno-12-17" id="__codelineno-12-17" name="__codelineno-12-17"></a><span class="w">  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">phylum_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span><span class="w"> </span><span class="c1"># Legend colour</span>
<a href="#__codelineno-12-18" id="__codelineno-12-18" name="__codelineno-12-18"></a><span class="w">  </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="c1"># Line type</span>
<a href="#__codelineno-12-19" id="__codelineno-12-19" name="__codelineno-12-19"></a><span class="w">  </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="c1"># Line width</span>
<a href="#__codelineno-12-20" id="__codelineno-12-20" name="__codelineno-12-20"></a><span class="w">  </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="c1"># Number of columns for legend,</span>
<a href="#__codelineno-12-21" id="__codelineno-12-21" name="__codelineno-12-21"></a><span class="w">  </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"n"</span><span class="p">,</span><span class="w"> </span><span class="c1"># Border type (set to no border)</span>
<a href="#__codelineno-12-22" id="__codelineno-12-22" name="__codelineno-12-22"></a><span class="w">  </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Y-axis key: MAG taxonomy"</span>
<a href="#__codelineno-12-23" id="__codelineno-12-23" name="__codelineno-12-23"></a><span class="p">)</span>
<a href="#__codelineno-12-24" id="__codelineno-12-24" name="__codelineno-12-24"></a><span class="c1">## Legend for sample grouping</span>
<a href="#__codelineno-12-25" id="__codelineno-12-25" name="__codelineno-12-25"></a><span class="nf">legend</span><span class="p">(</span>
<a href="#__codelineno-12-26" id="__codelineno-12-26" name="__codelineno-12-26"></a><span class="w">  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"bottomright"</span><span class="p">,</span>
<a href="#__codelineno-12-27" id="__codelineno-12-27" name="__codelineno-12-27"></a><span class="w">  </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">group_colour</span><span class="o">$</span><span class="n">Group</span><span class="p">,</span>
<a href="#__codelineno-12-28" id="__codelineno-12-28" name="__codelineno-12-28"></a><span class="w">  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">group_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span>
<a href="#__codelineno-12-29" id="__codelineno-12-29" name="__codelineno-12-29"></a><span class="w">  </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-12-30" id="__codelineno-12-30" name="__codelineno-12-30"></a><span class="w">  </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span>
<a href="#__codelineno-12-31" id="__codelineno-12-31" name="__codelineno-12-31"></a><span class="w">  </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-12-32" id="__codelineno-12-32" name="__codelineno-12-32"></a><span class="w">  </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"n"</span><span class="p">,</span>
<a href="#__codelineno-12-33" id="__codelineno-12-33" name="__codelineno-12-33"></a><span class="w">  </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"X-axis key: Sample group"</span>
<a href="#__codelineno-12-34" id="__codelineno-12-34" name="__codelineno-12-34"></a><span class="p">)</span>
</code></pre></div>
</div>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.04.heatmap.MAG.png"><img alt="image" src="../../figures/day4_coverage.04.heatmap.MAG.png"/></a>
</center></p>
<div class="admonition vial-virus">
<p class="admonition-title">Results at a glance</p>
<p>The heatmap shows us that:</p>
<ul>
<li>4 MAGs belong to the phylum Proteobacteria</li>
<li>Sample 4 belonging to Group C is somewhat of an outlier, with only 2 MAGs recovered from this sample</li>
<li><code>bin_8</code>, a Nitrobacter, is only found in sample 4</li>
<li>Presumably, sample 4 has a high abundance of nitrogen cycling MAGs</li>
<li>Samples from Groups A and B recovered the same bins</li>
</ul>
</div>
<p>You can and should experiment with different paramters and arguments within the <code>heatmap.2</code> function. A good starting point is <code>cex = ...</code> which controls "character expansion" (i.e. text size). Here, we use the <code>...SideColors</code> argument to annotate MAG phylum and sample grouping. We also use <code>legend</code> to provide additional information on what those colours represent. Read the help page for <code>heatmap.2()</code> by typing <code>?heatmap.2</code> into the console. The help manual will appear on the bottom right pane.</p>
<p>Here, we arrange our columns based Bray-Curtis dissimilarities between samples. However, you may want to enforce or preserve some form of sample order (samples from the same transect, chronological/topological order, etc.). You can do this by setting <code>Colv = FALSE</code>.</p>
<div class="admonition note">
<p class="admonition-title">Arranging figure elements</p>
<p>As observed in the above plot, the positioning and arrangement of the colour key and legends relative to the main heatmap may not be ideal. Often times, it is desirable to manually arrange or resize elements of a figure (legends, keys, text). This is usually done by exporting the figure in a format capable of preseving vectors (e.g. SVG or PDF) and then post-processing them in vector graphics software such as Adobe Illustrator or Inkscape.</p>
</div>
<hr/>
<h3 id="part-2-building-a-heatmap-of-viral-contigs-per-sample">Part 2 - Building a heatmap of viral contigs per sample<a class="headerlink" href="#part-2-building-a-heatmap-of-viral-contigs-per-sample" title="Permanent link">¶</a></h3>
<p>We can run through the same process for the viral contigs. Many of the steps are as outlined above, so we will work through these a bit quicker and with less commentary along the way. However, we will highlight a handful of differences compared to the commands for the MAGs above, for example:  steps for selecting and/or formatting the taxonomy; importing the quality output from <code>CheckV</code>; and the (optional) addition of filtering out low quality contigs.</p>
<h4 id="21-prepare-environment">2.1 Prepare environment<a class="headerlink" href="#21-prepare-environment" title="Permanent link">¶</a></h4>
<p>This has already been done above. We will immediately begin wrangling the data.</p>
<h4 id="22-wrangle-data">2.2 Wrangle data<a class="headerlink" href="#22-wrangle-data" title="Permanent link">¶</a></h4>
<p>To start, we need to identify viral contigs of at least medium quality. THis will be used as a filtering vector.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-13-1" id="__codelineno-13-1" name="__codelineno-13-1"></a><span class="c1">## 1. Clean up viral taxonomy annotation quality ----</span>
<a href="#__codelineno-13-2" id="__codelineno-13-2" name="__codelineno-13-2"></a><span class="n">virus_quality</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">checkv_summary</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-13-3" id="__codelineno-13-3" name="__codelineno-13-3"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-13-4" id="__codelineno-13-4" name="__codelineno-13-4"></a><span class="w">    </span><span class="n">checkv_quality</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span>
<a href="#__codelineno-13-5" id="__codelineno-13-5" name="__codelineno-13-5"></a><span class="w">      </span><span class="n">checkv_quality</span><span class="p">,</span>
<a href="#__codelineno-13-6" id="__codelineno-13-6" name="__codelineno-13-6"></a><span class="w">      </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<a href="#__codelineno-13-7" id="__codelineno-13-7" name="__codelineno-13-7"></a><span class="w">        </span><span class="s">"Not-determined"</span><span class="p">,</span>
<a href="#__codelineno-13-8" id="__codelineno-13-8" name="__codelineno-13-8"></a><span class="w">        </span><span class="s">"Low-quality"</span><span class="p">,</span>
<a href="#__codelineno-13-9" id="__codelineno-13-9" name="__codelineno-13-9"></a><span class="w">        </span><span class="s">"Medium-quality"</span><span class="p">,</span>
<a href="#__codelineno-13-10" id="__codelineno-13-10" name="__codelineno-13-10"></a><span class="w">        </span><span class="s">"High-quality"</span><span class="p">,</span>
<a href="#__codelineno-13-11" id="__codelineno-13-11" name="__codelineno-13-11"></a><span class="w">        </span><span class="s">"Complete"</span>
<a href="#__codelineno-13-12" id="__codelineno-13-12" name="__codelineno-13-12"></a><span class="w">      </span><span class="p">))</span>
<a href="#__codelineno-13-13" id="__codelineno-13-13" name="__codelineno-13-13"></a><span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-13-14" id="__codelineno-13-14" name="__codelineno-13-14"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">contig_id</span><span class="p">,</span><span class="w"> </span><span class="n">checkv_quality</span><span class="p">)</span>
<a href="#__codelineno-13-15" id="__codelineno-13-15" name="__codelineno-13-15"></a>
<a href="#__codelineno-13-16" id="__codelineno-13-16" name="__codelineno-13-16"></a><span class="c1">## Subset good quality predictions</span>
<a href="#__codelineno-13-17" id="__codelineno-13-17" name="__codelineno-13-17"></a><span class="n">virus_hq</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">virus_quality</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-13-18" id="__codelineno-13-18" name="__codelineno-13-18"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">checkv_quality</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">"Not-determined"</span><span class="p">,</span><span class="w"> </span><span class="s">"Low-quality"</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-13-19" id="__codelineno-13-19" name="__codelineno-13-19"></a><span class="w">  </span><span class="nf">pull</span><span class="p">(</span><span class="n">contig_id</span><span class="p">)</span>
<a href="#__codelineno-13-20" id="__codelineno-13-20" name="__codelineno-13-20"></a>
<a href="#__codelineno-13-21" id="__codelineno-13-21" name="__codelineno-13-21"></a><span class="n">virus_taxonomy_hq</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">vContact2_out</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-13-22" id="__codelineno-13-22" name="__codelineno-13-22"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">Genome</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">virus_hq</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-13-23" id="__codelineno-13-23" name="__codelineno-13-23"></a><span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-13-24" id="__codelineno-13-24" name="__codelineno-13-24"></a><span class="w">    </span><span class="n">Order_VC_predicted</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.factor</span><span class="p">(</span><span class="n">Order_VC_predicted</span><span class="p">)</span>
<a href="#__codelineno-13-25" id="__codelineno-13-25" name="__codelineno-13-25"></a><span class="w">  </span><span class="p">)</span>
</code></pre></div>
</div>
<p>We then obtain the coverage matrix and transform the values to enhance visualisation.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-14-1" id="__codelineno-14-1" name="__codelineno-14-1"></a><span class="c1">## 2. Obtain viral contig coverage matrix ----</span>
<a href="#__codelineno-14-2" id="__codelineno-14-2" name="__codelineno-14-2"></a><span class="c1">## Unlike MAGs, we do not need to average coverage values.</span>
<a href="#__codelineno-14-3" id="__codelineno-14-3" name="__codelineno-14-3"></a><span class="c1">## However, we do need to filter out annotations of lesser quality.</span>
<a href="#__codelineno-14-4" id="__codelineno-14-4" name="__codelineno-14-4"></a><span class="c1">## We find 15 contigs that needs to be kept</span>
<a href="#__codelineno-14-5" id="__codelineno-14-5" name="__codelineno-14-5"></a><span class="n">virus_cov_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">virus_cov</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-14-6" id="__codelineno-14-6" name="__codelineno-14-6"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">contigName</span><span class="p">,</span><span class="w"> </span><span class="nf">ends_with</span><span class="p">(</span><span class="s">".bam"</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-14-7" id="__codelineno-14-7" name="__codelineno-14-7"></a><span class="w">  </span><span class="nf">rename_with</span><span class="p">(</span>
<a href="#__codelineno-14-8" id="__codelineno-14-8" name="__codelineno-14-8"></a><span class="w">    </span><span class="n">.fn</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">name</span><span class="p">)</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">name</span><span class="p">,</span><span class="w"> </span><span class="s">".bam"</span><span class="p">),</span>
<a href="#__codelineno-14-9" id="__codelineno-14-9" name="__codelineno-14-9"></a><span class="w">    </span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">everything</span><span class="p">()</span>
<a href="#__codelineno-14-10" id="__codelineno-14-10" name="__codelineno-14-10"></a><span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-14-11" id="__codelineno-14-11" name="__codelineno-14-11"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">contigName</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">virus_hq</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-14-12" id="__codelineno-14-12" name="__codelineno-14-12"></a><span class="w">  </span><span class="nf">column_to_rownames</span><span class="p">(</span><span class="s">"contigName"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-14-13" id="__codelineno-14-13" name="__codelineno-14-13"></a><span class="w">  </span><span class="nf">as.matrix</span><span class="p">()</span>
<a href="#__codelineno-14-14" id="__codelineno-14-14" name="__codelineno-14-14"></a>
<a href="#__codelineno-14-15" id="__codelineno-14-15" name="__codelineno-14-15"></a><span class="c1">## 3. Log2 transform coverage matrix ----</span>
<a href="#__codelineno-14-16" id="__codelineno-14-16" name="__codelineno-14-16"></a><span class="n">virus_cov_matrix_log2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">log2</span><span class="p">(</span><span class="n">virus_cov_matrix</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</code></pre></div>
</div>
<h4 id="23-order-the-heatmap">2.3 Order the heatmap<a class="headerlink" href="#23-order-the-heatmap" title="Permanent link">¶</a></h4>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-15-1" id="__codelineno-15-1" name="__codelineno-15-1"></a><span class="c1"># Perform hierarchical clustering (viral contigs) ----</span>
<a href="#__codelineno-15-2" id="__codelineno-15-2" name="__codelineno-15-2"></a><span class="c1">## Dissimilarities between samples</span>
<a href="#__codelineno-15-3" id="__codelineno-15-3" name="__codelineno-15-3"></a><span class="n">virus_sample_dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vegdist</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">virus_cov_matrix_log2</span><span class="p">))</span>
<a href="#__codelineno-15-4" id="__codelineno-15-4" name="__codelineno-15-4"></a><span class="n">virus_sample_hclust</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">hclust</span><span class="p">(</span><span class="n">virus_sample_dist</span><span class="p">,</span><span class="w"> </span><span class="s">"average"</span><span class="p">)</span>
<a href="#__codelineno-15-5" id="__codelineno-15-5" name="__codelineno-15-5"></a>
<a href="#__codelineno-15-6" id="__codelineno-15-6" name="__codelineno-15-6"></a><span class="nf">plot</span><span class="p">(</span>
<a href="#__codelineno-15-7" id="__codelineno-15-7" name="__codelineno-15-7"></a><span class="w">  </span><span class="n">sample_hclust</span><span class="p">,</span>
<a href="#__codelineno-15-8" id="__codelineno-15-8" name="__codelineno-15-8"></a><span class="w">  </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Bray-Curtis dissimilarities between samples\n(virus)"</span><span class="p">,</span>
<a href="#__codelineno-15-9" id="__codelineno-15-9" name="__codelineno-15-9"></a><span class="w">  </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Sample"</span><span class="p">,</span>
<a href="#__codelineno-15-10" id="__codelineno-15-10" name="__codelineno-15-10"></a><span class="w">  </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Height"</span><span class="p">,</span>
<a href="#__codelineno-15-11" id="__codelineno-15-11" name="__codelineno-15-11"></a><span class="w">  </span><span class="n">sub</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Method: average linkage"</span><span class="p">,</span>
<a href="#__codelineno-15-12" id="__codelineno-15-12" name="__codelineno-15-12"></a><span class="w">  </span><span class="n">hang</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span>
<a href="#__codelineno-15-13" id="__codelineno-15-13" name="__codelineno-15-13"></a><span class="p">)</span>
<a href="#__codelineno-15-14" id="__codelineno-15-14" name="__codelineno-15-14"></a>
<a href="#__codelineno-15-15" id="__codelineno-15-15" name="__codelineno-15-15"></a><span class="c1">## Dissimilarities between viruses</span>
<a href="#__codelineno-15-16" id="__codelineno-15-16" name="__codelineno-15-16"></a><span class="n">virus_dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vegdist</span><span class="p">(</span><span class="n">virus_cov_matrix_log2</span><span class="p">)</span>
<a href="#__codelineno-15-17" id="__codelineno-15-17" name="__codelineno-15-17"></a><span class="n">virus_hclust</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">hclust</span><span class="p">(</span><span class="n">virus_dist</span><span class="p">,</span><span class="w"> </span><span class="s">"average"</span><span class="p">)</span>
<a href="#__codelineno-15-18" id="__codelineno-15-18" name="__codelineno-15-18"></a>
<a href="#__codelineno-15-19" id="__codelineno-15-19" name="__codelineno-15-19"></a><span class="nf">plot</span><span class="p">(</span>
<a href="#__codelineno-15-20" id="__codelineno-15-20" name="__codelineno-15-20"></a><span class="w">  </span><span class="n">virus_hclust</span><span class="p">,</span>
<a href="#__codelineno-15-21" id="__codelineno-15-21" name="__codelineno-15-21"></a><span class="w">  </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Bray-Curtis dissimilarities between viruses"</span><span class="p">,</span>
<a href="#__codelineno-15-22" id="__codelineno-15-22" name="__codelineno-15-22"></a><span class="w">  </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"MAGs"</span><span class="p">,</span>
<a href="#__codelineno-15-23" id="__codelineno-15-23" name="__codelineno-15-23"></a><span class="w">  </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Height"</span><span class="p">,</span>
<a href="#__codelineno-15-24" id="__codelineno-15-24" name="__codelineno-15-24"></a><span class="w">  </span><span class="n">sub</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Method: average linkage"</span><span class="p">,</span>
<a href="#__codelineno-15-25" id="__codelineno-15-25" name="__codelineno-15-25"></a><span class="w">  </span><span class="n">hang</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span><span class="p">,</span>
<a href="#__codelineno-15-26" id="__codelineno-15-26" name="__codelineno-15-26"></a><span class="w">  </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.75</span>
<a href="#__codelineno-15-27" id="__codelineno-15-27" name="__codelineno-15-27"></a><span class="p">)</span>
</code></pre></div>
</div>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.05.cov_hmp.hclust.vir_sample.png"><img alt="image" src="../../figures/day4_coverage.05.cov_hmp.hclust.vir_sample.png"/></a>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.06.cov_hmp.hclust.vir.png"><img alt="image" src="../../figures/day4_coverage.06.cov_hmp.hclust.vir.png"/></a></p>
<h4 id="24-set-colour-palettes">2.4 Set colour palettes<a class="headerlink" href="#24-set-colour-palettes" title="Permanent link">¶</a></h4>
<p>For colours based on sample group, we will retain the colours set as above. Here, we will only set a new palette (Tableau 10) for viruses based on the predicted taxonomy.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-16-1" id="__codelineno-16-1" name="__codelineno-16-1"></a><span class="c1"># Prepare grouping variables and colour palettes ----</span>
<a href="#__codelineno-16-2" id="__codelineno-16-2" name="__codelineno-16-2"></a><span class="c1"># Check palette colours</span>
<a href="#__codelineno-16-3" id="__codelineno-16-3" name="__codelineno-16-3"></a><span class="n">scales</span><span class="o">::</span><span class="nf">show_col</span><span class="p">(</span>
<a href="#__codelineno-16-4" id="__codelineno-16-4" name="__codelineno-16-4"></a><span class="w">  </span><span class="nf">palette.colors</span><span class="p">(</span>
<a href="#__codelineno-16-5" id="__codelineno-16-5" name="__codelineno-16-5"></a><span class="w">    </span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Tableau 10"</span>
<a href="#__codelineno-16-6" id="__codelineno-16-6" name="__codelineno-16-6"></a><span class="w">  </span><span class="p">)</span>
<a href="#__codelineno-16-7" id="__codelineno-16-7" name="__codelineno-16-7"></a><span class="p">)</span>
</code></pre></div>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.07.show_col.Tableau10.png"><img alt="image" src="../../figures/day4_coverage.07.show_col.Tableau10.png"/></a></p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-17-1" id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="c1">## Set colours for viral orders</span>
<a href="#__codelineno-17-2" id="__codelineno-17-2" name="__codelineno-17-2"></a><span class="n">virus_order_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<a href="#__codelineno-17-3" id="__codelineno-17-3" name="__codelineno-17-3"></a><span class="w">  </span><span class="s">"virus_order"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">levels</span><span class="p">(</span><span class="n">virus_taxonomy_hq</span><span class="o">$</span><span class="n">Order_VC_predicted</span><span class="p">),</span>
<a href="#__codelineno-17-4" id="__codelineno-17-4" name="__codelineno-17-4"></a><span class="w">  </span><span class="s">"colour"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">palette.colors</span><span class="p">(</span>
<a href="#__codelineno-17-5" id="__codelineno-17-5" name="__codelineno-17-5"></a><span class="w">    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">levels</span><span class="p">(</span><span class="n">virus_taxonomy_hq</span><span class="o">$</span><span class="n">Order_VC_predicted</span><span class="p">)),</span>
<a href="#__codelineno-17-6" id="__codelineno-17-6" name="__codelineno-17-6"></a><span class="w">    </span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Tableau 10"</span>
<a href="#__codelineno-17-7" id="__codelineno-17-7" name="__codelineno-17-7"></a><span class="w">  </span><span class="p">)</span>
<a href="#__codelineno-17-8" id="__codelineno-17-8" name="__codelineno-17-8"></a><span class="p">)</span>
<a href="#__codelineno-17-9" id="__codelineno-17-9" name="__codelineno-17-9"></a>
<a href="#__codelineno-17-10" id="__codelineno-17-10" name="__codelineno-17-10"></a><span class="c1">## Set colours for viruses</span>
<a href="#__codelineno-17-11" id="__codelineno-17-11" name="__codelineno-17-11"></a><span class="n">virus_colour</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">virus_taxonomy_hq</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-17-12" id="__codelineno-17-12" name="__codelineno-17-12"></a><span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">Genome</span><span class="p">,</span><span class="w"> </span><span class="n">Order_VC_predicted</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-17-13" id="__codelineno-17-13" name="__codelineno-17-13"></a><span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">virus_order_colour</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">"Order_VC_predicted"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"virus_order"</span><span class="p">))</span>
</code></pre></div>
</div>
<h4 id="25-build-heatmap">2.5 Build heatmap<a class="headerlink" href="#25-build-heatmap" title="Permanent link">¶</a></h4>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-18-1" id="__codelineno-18-1" name="__codelineno-18-1"></a><span class="nf">heatmap.2</span><span class="p">(</span>
<a href="#__codelineno-18-2" id="__codelineno-18-2" name="__codelineno-18-2"></a><span class="w">  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">virus_cov_matrix_log2</span><span class="p">,</span>
<a href="#__codelineno-18-3" id="__codelineno-18-3" name="__codelineno-18-3"></a><span class="w">  </span><span class="n">Colv</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.dendrogram</span><span class="p">(</span><span class="n">virus_sample_hclust</span><span class="p">),</span>
<a href="#__codelineno-18-4" id="__codelineno-18-4" name="__codelineno-18-4"></a><span class="w">  </span><span class="n">Rowv</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.dendrogram</span><span class="p">(</span><span class="n">virus_hclust</span><span class="p">),</span>
<a href="#__codelineno-18-5" id="__codelineno-18-5" name="__codelineno-18-5"></a><span class="w">  </span><span class="n">margins</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="m">20</span><span class="p">),</span>
<a href="#__codelineno-18-6" id="__codelineno-18-6" name="__codelineno-18-6"></a><span class="w">  </span><span class="n">RowSideColors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">virus_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span>
<a href="#__codelineno-18-7" id="__codelineno-18-7" name="__codelineno-18-7"></a><span class="w">  </span><span class="n">ColSideColors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sample_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span>
<a href="#__codelineno-18-8" id="__codelineno-18-8" name="__codelineno-18-8"></a><span class="w">  </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"none"</span><span class="p">,</span>
<a href="#__codelineno-18-9" id="__codelineno-18-9" name="__codelineno-18-9"></a><span class="w">  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cell_colour</span><span class="p">,</span>
<a href="#__codelineno-18-10" id="__codelineno-18-10" name="__codelineno-18-10"></a><span class="w">  </span><span class="n">trace</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"none"</span><span class="p">,</span>
<a href="#__codelineno-18-11" id="__codelineno-18-11" name="__codelineno-18-11"></a><span class="w">  </span><span class="n">keysize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-18-12" id="__codelineno-18-12" name="__codelineno-18-12"></a><span class="w">  </span><span class="n">density.info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"none"</span>
<a href="#__codelineno-18-13" id="__codelineno-18-13" name="__codelineno-18-13"></a><span class="p">)</span>
<a href="#__codelineno-18-14" id="__codelineno-18-14" name="__codelineno-18-14"></a><span class="c1">## Legends</span>
<a href="#__codelineno-18-15" id="__codelineno-18-15" name="__codelineno-18-15"></a><span class="nf">legend</span><span class="p">(</span>
<a href="#__codelineno-18-16" id="__codelineno-18-16" name="__codelineno-18-16"></a><span class="w">  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"bottomleft"</span><span class="p">,</span>
<a href="#__codelineno-18-17" id="__codelineno-18-17" name="__codelineno-18-17"></a><span class="w">  </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">virus_order_colour</span><span class="o">$</span><span class="n">virus_order</span><span class="p">,</span>
<a href="#__codelineno-18-18" id="__codelineno-18-18" name="__codelineno-18-18"></a><span class="w">  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">virus_order_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span>
<a href="#__codelineno-18-19" id="__codelineno-18-19" name="__codelineno-18-19"></a><span class="w">  </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-18-20" id="__codelineno-18-20" name="__codelineno-18-20"></a><span class="w">  </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span>
<a href="#__codelineno-18-21" id="__codelineno-18-21" name="__codelineno-18-21"></a><span class="w">  </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-18-22" id="__codelineno-18-22" name="__codelineno-18-22"></a><span class="w">  </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"n"</span><span class="p">,</span>
<a href="#__codelineno-18-23" id="__codelineno-18-23" name="__codelineno-18-23"></a><span class="w">  </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Y-axis key: Virus taxonomy"</span>
<a href="#__codelineno-18-24" id="__codelineno-18-24" name="__codelineno-18-24"></a><span class="p">)</span>
<a href="#__codelineno-18-25" id="__codelineno-18-25" name="__codelineno-18-25"></a><span class="nf">legend</span><span class="p">(</span>
<a href="#__codelineno-18-26" id="__codelineno-18-26" name="__codelineno-18-26"></a><span class="w">  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"bottomright"</span><span class="p">,</span>
<a href="#__codelineno-18-27" id="__codelineno-18-27" name="__codelineno-18-27"></a><span class="w">  </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">group_colour</span><span class="o">$</span><span class="n">Group</span><span class="p">,</span>
<a href="#__codelineno-18-28" id="__codelineno-18-28" name="__codelineno-18-28"></a><span class="w">  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">group_colour</span><span class="o">$</span><span class="n">colour</span><span class="p">,</span>
<a href="#__codelineno-18-29" id="__codelineno-18-29" name="__codelineno-18-29"></a><span class="w">  </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-18-30" id="__codelineno-18-30" name="__codelineno-18-30"></a><span class="w">  </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span>
<a href="#__codelineno-18-31" id="__codelineno-18-31" name="__codelineno-18-31"></a><span class="w">  </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<a href="#__codelineno-18-32" id="__codelineno-18-32" name="__codelineno-18-32"></a><span class="w">  </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"n"</span><span class="p">,</span>
<a href="#__codelineno-18-33" id="__codelineno-18-33" name="__codelineno-18-33"></a><span class="w">  </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"X-axis key: Sample group"</span>
<a href="#__codelineno-18-34" id="__codelineno-18-34" name="__codelineno-18-34"></a><span class="p">)</span>
</code></pre></div>
</div>
<p><center>
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_coverage.08.heatmap.vir.png"><img alt="image" src="../../figures/day4_coverage.08.heatmap.vir.png"/></a>
</center></p>
<hr/>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>