
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/" rel="canonical"/>
<link href="../4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/" rel="prev"/>
<link href="../5_APPENDIX_ex15_gene_synteny_Generate_blast_files/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>APPENDIX (ex15): Prepare input for gene synteny visualisation - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#appendix-ex15-prepare-input-for-gene-synteny-visualisation">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              APPENDIX (ex15): Prepare input for gene synteny visualisation
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day 1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day 2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Day 3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day4/ex15_gene_annotation_part3/">
        Day 4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction I: Shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction II: HPC and job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Filter raw reads by quality
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly I: Assembling contigs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly II: Varying the parameters
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Assembly evaluation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning with multiple tools
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manual bin refinement
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11.1_phylogenomics/">
        Phylogenomics
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation I: BLAST-like and HMM
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation II: DRAM and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex15_gene_annotation_part3/">
        Gene annotation III: DRAM distillation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16a_data_presentation_Intro/">
        Introduction to data presentation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16b_data_presentation_Coverage/">
        Coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16c_OPTIONAL_data_presentation_Ordination/">
        Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16d_data_presentation_KEGG_pathways/">
        KEGG pathway maps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16e_data_presentation_Gene_synteny/">
        Gene synteny
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day4/ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        CAZy heatmaps
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          APPENDIX (ex15): Prepare input for gene synteny visualisation
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#1-navigating-the-working-directory">
    1. Navigating the working directory
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-obtain-annotations-for-genes-of-interest">
    2. Obtain annotations for genes of interest
  </a>
<nav aria-label="2. Obtain annotations for genes of interest" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#21-subset-annotations">
    2.1 Subset annotations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#22-check-for-contiguous-and-sequential-outputs">
    2.2 Check for contiguous and sequential outputs
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#23-create-subsets-of-annotation-files-with-correct-genes">
    2.3 Create subsets of annotation files with correct genes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#24-clean-up-annotation-tables">
    2.4 Clean up annotation tables
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#3-create-comparison-tables">
    3. Create comparison tables
  </a>
<nav aria-label="3. Create comparison tables" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#31-subset-contigs-per-bin">
    3.1 Subset contigs per bin
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#33-run-tblastx">
    3.3 Run tBLASTx
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#1-navigating-the-working-directory">
    1. Navigating the working directory
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-obtain-annotations-for-genes-of-interest">
    2. Obtain annotations for genes of interest
  </a>
<nav aria-label="2. Obtain annotations for genes of interest" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#21-subset-annotations">
    2.1 Subset annotations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#22-check-for-contiguous-and-sequential-outputs">
    2.2 Check for contiguous and sequential outputs
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#23-create-subsets-of-annotation-files-with-correct-genes">
    2.3 Create subsets of annotation files with correct genes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#24-clean-up-annotation-tables">
    2.4 Clean up annotation tables
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#3-create-comparison-tables">
    3. Create comparison tables
  </a>
<nav aria-label="3. Create comparison tables" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#31-subset-contigs-per-bin">
    3.1 Subset contigs per bin
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#33-run-tblastx">
    3.3 Run tBLASTx
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="appendix-ex15-prepare-input-for-gene-synteny-visualisation">APPENDIX (ex15): Prepare input for gene synteny visualisation<a class="headerlink" href="#appendix-ex15-prepare-input-for-gene-synteny-visualisation" title="Permanent link">¶</a></h1>
<p>In order to produce gene synteny plots using <code>genoPlotR</code> as outlined in <a href="../../day4/ex16e_data_presentation_Gene_synteny/">Gene synteny</a>, we need to know the annotations and relative nucleotide positions of our genes of interest. We can use our annotations file generated via <a href="../../day3/ex13_gene_annotation_part1/">homology and domain searches</a> (here, we have parsed and aggregated these annotations using an <a href="https://github.com/GenomicsAotearoa/environmental_metagenomics/tree/master/analysis_tools/annotationaggregator_v0.1">in-house custom script</a>) or <a href="../../day3/ex14_gene_annotation_part2/">DRAM</a> in order to obtain relevant genes and their labels. Gene nucleotide positions relative to assembled contigs/scaffolds can be obtained from <code>prodigal</code> outputs. </p>
<p>Additionally, we need BLAST outputs for comparing between genes along their contigs. For this, we rely on outputs from pairwise <code>tBLASTx</code> (translates a nucleotide database then searches it using a translated nucleotide query) to perform sequential comparisons across different bins.</p>
<p>For this example, we use our aggregated annotations (provided in <code>10.gene_annotation_and_coverage/example_annotation_tables</code>) and prodigal outputs to generate input files for visualising the synteny of genes that encode the <a href="https://metacyc.org/META/NEW-IMAGE?type=ENZYME&amp;object=ABC-70-CPLX">sulfate/thiosulfate ABC transporter</a> involved in <a href="https://www.kegg.jp/module/M00616">assimilating extracellular sulfate</a>. We will begin by subsetting our genes of interest based on KEGG orthology numbers and relevant annotation labels, followed by a pairwise <code>tBLASTx</code> of our contigs.</p>
<h2 id="1-navigating-the-working-directory">1. Navigating the working directory<a class="headerlink" href="#1-navigating-the-working-directory" title="Permanent link">¶</a></h2>
<p>We will be working in <code>10.gene_annotation_and_coverage/</code> where you will find the <code>example_annotation_tables/</code>, <code>filtered_bins/</code> and <code>predictions/</code> sub-directories already prepared for you. </p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1"># Navigate to working directory</span>
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/10.gene_annotation_and_coverage
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="c1"># View relevant sub-directories</span>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>ls<span class="w"> </span>example_annotation_tables/
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>bin_0.annotation.aa  bin_3.annotation.aa  bin_6.annotation.aa  bin_9.annotation.aa
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>bin_1.annotation.aa  bin_4.annotation.aa  bin_7.annotation.aa
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>bin_2.annotation.aa  bin_5.annotation.aa  bin_8.annotation.aa
</code></pre></div>
</div>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>ls predictions/
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>bin_0.filtered.genes.faa              bin_5.filtered.genes.faa
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>bin_0.filtered.genes.fna              bin_5.filtered.genes.fna
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>bin_0.filtered.genes.gbk              bin_5.filtered.genes.gbk
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>bin_0.filtered.genes.no_metadata.faa  bin_5.filtered.genes.no_metadata.faa
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>bin_0.filtered.genes.no_metadata.fna  bin_5.filtered.genes.no_metadata.fna
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a>bin_1.filtered.genes.faa              bin_6.filtered.genes.faa
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a>bin_1.filtered.genes.fna              bin_6.filtered.genes.fna
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a>bin_1.filtered.genes.gbk              bin_6.filtered.genes.gbk
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a>bin_1.filtered.genes.no_metadata.faa  bin_6.filtered.genes.no_metadata.faa
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a>bin_1.filtered.genes.no_metadata.fna  bin_6.filtered.genes.no_metadata.fna
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>bin_2.filtered.genes.faa              bin_7.filtered.genes.faa
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a>bin_2.filtered.genes.fna              bin_7.filtered.genes.fna
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a>bin_2.filtered.genes.gbk              bin_7.filtered.genes.gbk
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a>bin_2.filtered.genes.no_metadata.faa  bin_7.filtered.genes.no_metadata.faa
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a>bin_2.filtered.genes.no_metadata.fna  bin_7.filtered.genes.no_metadata.fna
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a>bin_3.filtered.genes.faa              bin_8.filtered.genes.faa
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a>bin_3.filtered.genes.fna              bin_8.filtered.genes.fna
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a>bin_3.filtered.genes.gbk              bin_8.filtered.genes.gbk
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a>bin_3.filtered.genes.no_metadata.faa  bin_8.filtered.genes.no_metadata.faa
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a>bin_3.filtered.genes.no_metadata.fna  bin_8.filtered.genes.no_metadata.fna
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a>bin_4.filtered.genes.faa              bin_9.filtered.genes.faa
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a>bin_4.filtered.genes.fna              bin_9.filtered.genes.fna
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a>bin_4.filtered.genes.gbk              bin_9.filtered.genes.gbk
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a>bin_4.filtered.genes.no_metadata.faa  bin_9.filtered.genes.no_metadata.faa
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>bin_4.filtered.genes.no_metadata.fna  bin_9.filtered.genes.no_metadata.fna
</code></pre></div>
</div>
<h2 id="2-obtain-annotations-for-genes-of-interest">2. Obtain annotations for genes of interest<a class="headerlink" href="#2-obtain-annotations-for-genes-of-interest" title="Permanent link">¶</a></h2>
<h3 id="21-subset-annotations">2.1 Subset annotations<a class="headerlink" href="#21-subset-annotations" title="Permanent link">¶</a></h3>
<p>Based on the KEGG module for <a href="https://www.kegg.jp/module/M00616">sulfate-sulfur assimilation</a>, we need annotations that match <code>K02048</code>, <code>K23163</code>, <code>K02046</code>, <code>K02047</code>, and <code>K02045</code>. Out of all the columns, we are only interested in the gene ID and the KEGG annotations. We know that column 1 is the gene ID, so lets find the column index that contains the KEGG annotations.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>grep<span class="w"> </span>-P<span class="w"> </span><span class="s2">"K0204[5-8]|K23163"</span><span class="w"> </span>example_annotation_tables/*.aa<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a><span class="w">  </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s1">'\t'</span><span class="w"> </span><span class="s1">'</span>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a><span class="s1">      {</span>
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="s1">        for (i = 1; i &lt;= NF; i++) {</span>
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="s1">          if (match($i, "K[0-9]{5}"))</span>
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="s1">            printf("Column %d contains KO of interest.\n", i)</span>
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="s1">        }</span>
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="s1">      }</span>
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a><span class="s1">    '</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a><span class="w">  </span><span class="p">|</span><span class="w"> </span>uniq
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a>
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a><span class="c1"># Column 26 contains KO of interest.</span>
</code></pre></div>
</div>
<p>Deconstruct the code block above:</p>
<table>
<thead>
<tr>
<th align="left">Code</th>
<th align="left">Action</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><code>grep -P "K0204[5-8]\|K23163" *.aa</code></td>
<td align="left">Finds the rows that contain the pattern for our KO of interest in all annotation files within this directory.</td>
</tr>
<tr>
<td align="left"><code>awk -F '\t' '</code></td>
<td align="left">Starts the <code>awk</code> programme to parse the output of <code>grep</code> and tells it that the outputs have tab-delimited columns.</td>
</tr>
<tr>
<td align="left"><code>for (i = 1; i &lt;= NF; i++) {</code></td>
<td align="left">Starts the C-style <code>for</code> loop within <code>awk</code> by assigning it an initial value of 1 (<code>i = 1</code>), continue the loop as long as the value of <code>i</code> is lower than the total number of rows (<code>i &lt;= NF</code>), and increment by 1 as the loop progresses (<code>i++</code>).</td>
</tr>
<tr>
<td align="left"><code>if (match($i, "K[0-9]{5}"))</code></td>
<td align="left">Starts an <code>if</code> loop and evaluates if a column contains string that matches the pattern <code>K[0-9]{5}</code> (which is the regex for a KO number). This is evaluated per row.</td>
</tr>
<tr>
<td align="left"><code>printf("Column %d contains KO of interest.\n", i)</code></td>
<td align="left">If there is a match in the row, print out the resulting column index.</td>
</tr>
<tr>
<td align="left"><code>uniq</code></td>
<td align="left">Returns only unique results.</td>
</tr>
</tbody>
</table>
<p>The code above says that all KO annotations are in column 26. This is true for all annotation files. Lets move on to subset to the KO numbers we want and cut out the columns we need. We will do this per bin.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="k">for</span><span class="w"> </span>bin_number<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="o">{</span><span class="m">0</span>..9<span class="o">}</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a><span class="w">  </span>grep<span class="w"> </span>-P<span class="w"> </span><span class="s2">"K0204[5-8]|K23163"</span><span class="w"> </span>example_annotation_tables/bin_<span class="si">${</span><span class="nv">bin_number</span><span class="si">}</span>.annotation.aa<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>,26<span class="w"> </span>&gt;<span class="w"> </span>bin_<span class="si">${</span><span class="nv">bin_number</span><span class="si">}</span>.goi.aa
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a><span class="w">  </span><span class="k">if</span><span class="w"> </span><span class="o">[</span><span class="w"> </span>!<span class="w"> </span>-s<span class="w"> </span>bin_<span class="si">${</span><span class="nv">bin_number</span><span class="si">}</span>.goi.aa<span class="w"> </span><span class="o">]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a><span class="w">    </span>rm<span class="w"> </span>bin_<span class="si">${</span><span class="nv">bin_number</span><span class="si">}</span>.goi.aa
<a href="#__codelineno-5-8" id="__codelineno-5-8" name="__codelineno-5-8"></a><span class="w">  </span><span class="k">fi</span>
<a href="#__codelineno-5-9" id="__codelineno-5-9" name="__codelineno-5-9"></a>
<a href="#__codelineno-5-10" id="__codelineno-5-10" name="__codelineno-5-10"></a><span class="k">done</span>
</code></pre></div>
</div>
<details class="info">
<summary>Deconstructing the code block</summary>
<p>The above code block loops through each bin annotation file to subset relevant rows from the main annotation files (<code>grep -P "K0204[5-8]|K23163" example_annotation_tables/bin_${bin_number}.annotation.aa</code>) then selects the first and 26<sup>th</sup> columns (<code>cut -f 1,26</code>). </p>
<p>It also evaluates if the output is an empty file (in the case where the KO of interest is not found in the bin annotation) (<code>if [ ! -s bin_${bin_number}.goi.aa ]</code>) and removes it if it is. </p>
</details>
<p>We have 4 files after running the above code.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>ls<span class="w"> </span>*.aa
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>bin_3.goi.aa  bin_5.goi.aa  bin_8.goi.aa  bin_9.goi.aa
</code></pre></div>
</div>
<h3 id="22-check-for-contiguous-and-sequential-outputs">2.2 Check for contiguous and sequential outputs<a class="headerlink" href="#22-check-for-contiguous-and-sequential-outputs" title="Permanent link">¶</a></h3>
<p>We need to check that the genes are contiguous (all on the same contig) and sequential (identify gaps between genes and fill them). Remember that prodigal headers (for which were propagated through our annotation workflows) always have <code>contigID_geneID</code> where <code>geneID</code> is always relative to the order it was found on the contig's nucleotide sequence. With that in mind, lets check for gene contiguity first.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a><span class="k">for</span><span class="w"> </span>aa_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>*goi.aa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">aa_file</span><span class="si">}</span><span class="w"> </span>.goi.aa<span class="k">)</span>
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a><span class="w">  </span><span class="nv">contig</span><span class="o">=</span><span class="k">$(</span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="si">${</span><span class="nv">aa_file</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-E<span class="w"> </span><span class="s1">'s/_[0-9]+$//g'</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sort<span class="w"> </span>-u<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l<span class="k">)</span>
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a>
<a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a><span class="w">  </span><span class="k">if</span><span class="w"> </span><span class="o">[</span><span class="w"> </span><span class="nv">$contig</span><span class="w"> </span>-gt<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a><span class="w">    </span><span class="nb">printf</span><span class="w"> </span><span class="s2">"%s has %d contig(s).\n"</span><span class="w"> </span><span class="nv">$bin</span><span class="w"> </span><span class="nv">$contig</span>
<a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a><span class="w">  </span><span class="k">fi</span>
<a href="#__codelineno-8-8" id="__codelineno-8-8" name="__codelineno-8-8"></a>
<a href="#__codelineno-8-9" id="__codelineno-8-9" name="__codelineno-8-9"></a><span class="k">done</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>bin_5 has 2 contig(s).
</code></pre></div>
</div>
<details class="note">
<summary>Deconstructing the code block</summary>
<p>The code above loops through all annotation subsets and finds the number of contigs in each file. It does this by using <code>cut -f 1</code> which selects the first columns, then <code>sed -E 's/_[0-9]+$//g'</code> removes the trailing gene numbers of the gene ID, followed by a dereplication of contigs via <code>sort -u</code> and finally counts the number of lines of the results using <code>wc -l</code>.</p>
<p>if the bin annotation subset has more than 1 contig (<code>if [ $contig -gt 1 ]; then</code>), then it reports/prints the bin and number of contigs found (<code>printf "%s has %d contig(s).\n" $bin $contig</code>). </p>
</details>
<p>The code block above shows that we have non-contiguous genes in bin 5. We will need to remove the extra gene(s) originating from another contig. Lets check which contig we need to remove.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a>cat<span class="w"> </span>bin_5.goi.aa
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a>bin_5_NODE_52_length_158668_cov_0.373555_136    K02048: cysP; sulfate ABC transporter substrate-binding protein
<a href="#__codelineno-11-2" id="__codelineno-11-2" name="__codelineno-11-2"></a>bin_5_NODE_95_length_91726_cov_0.379302_67      K02048: sbp; sulfate-binding protein
<a href="#__codelineno-11-3" id="__codelineno-11-3" name="__codelineno-11-3"></a>bin_5_NODE_95_length_91726_cov_0.379302_68      K02046: cysT; sulfate transporter CysT
<a href="#__codelineno-11-4" id="__codelineno-11-4" name="__codelineno-11-4"></a>bin_5_NODE_95_length_91726_cov_0.379302_69      K02047: cysW; sulfate transporter CysW
<a href="#__codelineno-11-5" id="__codelineno-11-5" name="__codelineno-11-5"></a>bin_5_NODE_95_length_91726_cov_0.379302_70      K02045: cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA
</code></pre></div>
</div>
<p>We find that contig <code>bin_5_NODE_52_length_158668_cov_0.373555</code> is not contiguous with the set of genes in contig <code>bin_5_NODE_95_length_91726_cov_0.379302</code>. We will make a note of that. Now lets check for continuity of genes.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a><span class="k">for</span><span class="w"> </span>aa_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>*goi.aa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-12-2" id="__codelineno-12-2" name="__codelineno-12-2"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">aa_file</span><span class="si">}</span><span class="w"> </span>.goi.aa<span class="k">)</span>
<a href="#__codelineno-12-3" id="__codelineno-12-3" name="__codelineno-12-3"></a><span class="w">  </span><span class="nv">gene_number</span><span class="o">=</span><span class="k">$(</span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="si">${</span><span class="nv">aa_file</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-E<span class="w"> </span><span class="s1">'s/.*_([0-9]+$)/\1/g'</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>uniq<span class="k">)</span>
<a href="#__codelineno-12-4" id="__codelineno-12-4" name="__codelineno-12-4"></a><span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">"</span><span class="nv">$bin</span><span class="s2">: </span><span class="k">$(</span><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">gene_number</span><span class="p">[@]</span><span class="si">}</span><span class="k">)</span><span class="s2">"</span>
<a href="#__codelineno-12-5" id="__codelineno-12-5" name="__codelineno-12-5"></a><span class="k">done</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-13-1" id="__codelineno-13-1" name="__codelineno-13-1"></a>bin_3: 128 132 133 134
<a href="#__codelineno-13-2" id="__codelineno-13-2" name="__codelineno-13-2"></a>bin_5: 136 67 68 69 70
<a href="#__codelineno-13-3" id="__codelineno-13-3" name="__codelineno-13-3"></a>bin_8: 55 56 57 58
<a href="#__codelineno-13-4" id="__codelineno-13-4" name="__codelineno-13-4"></a>bin_9: 147 148 149 150
</code></pre></div>
</div>
<details class="info">
<summary>Deconstructing the code block</summary>
<p>We loop through each of the annotation subset to select the first column (<code>cut -f 1</code>), extract the trailing gene order/number (<code>sed -E 's/.*_([0-9]+$)/\1/g'</code>) and returns unique entries (<code>uniq</code>). The result is stored as an array named <code>gene_number</code>.</p>
<p>It also reports the results of each bin and the array of gene numbers.</p>
</details>
<p>Results from the above code block shows that bin 3 has a gap between genes. We need to make sure we get the genes in between gene number 128 and 132.</p>
<h3 id="23-create-subsets-of-annotation-files-with-correct-genes">2.3 Create subsets of annotation files with correct genes<a class="headerlink" href="#23-create-subsets-of-annotation-files-with-correct-genes" title="Permanent link">¶</a></h3>
<p>Based on previous checks, we find that we need to:</p>
<ul>
<li>remove contig <code>bin_5_NODE_52_length_158668_cov_0.373555</code> from bin 5 when we create the final annotation file.</li>
<li>fill in gaps in genes for bin 3.</li>
</ul>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-14-1" id="__codelineno-14-1" name="__codelineno-14-1"></a><span class="c1"># Remove non-contiguous gene from bin_5.goi.aa</span>
<a href="#__codelineno-14-2" id="__codelineno-14-2" name="__codelineno-14-2"></a>sed<span class="w"> </span>-i<span class="w"> </span><span class="s1">'/bin_5_NODE_52_length_158668_cov_0.373555/d'</span><span class="w"> </span>bin_5.goi.aa
<a href="#__codelineno-14-3" id="__codelineno-14-3" name="__codelineno-14-3"></a>
<a href="#__codelineno-14-4" id="__codelineno-14-4" name="__codelineno-14-4"></a><span class="c1"># Obtain genes between 129 and 131 in contig bin_3_NODE_53_length_158395_cov_1.135272 and add to annotations</span>
<a href="#__codelineno-14-5" id="__codelineno-14-5" name="__codelineno-14-5"></a>cat<span class="w"> </span>example_annotation_tables/bin_3.annotation.aa<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-14-6" id="__codelineno-14-6" name="__codelineno-14-6"></a><span class="w">  </span><span class="p">|</span><span class="w"> </span>grep<span class="w"> </span>-E<span class="w"> </span><span class="s2">"bin_3_NODE_53_length_158395_cov_1.135272_"</span><span class="w"> </span><span class="se">\ </span><span class="c1"># Search for required contig</span>
<a href="#__codelineno-14-7" id="__codelineno-14-7" name="__codelineno-14-7"></a><span class="w">  </span><span class="p">|</span><span class="w"> </span>grep<span class="w"> </span>-E<span class="w"> </span><span class="s2">"_129|_130|_131"</span><span class="w"> </span><span class="se">\ </span><span class="w">                           </span><span class="c1"># Search for required gene numbers</span>
<a href="#__codelineno-14-8" id="__codelineno-14-8" name="__codelineno-14-8"></a><span class="w">  </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>,26<span class="w"> </span>&gt;&gt;<span class="w"> </span>bin_3.goi.aa<span class="w">                           </span><span class="c1"># Select columns 1 and 26</span>
</code></pre></div>
</div>
<h3 id="24-clean-up-annotation-tables">2.4 Clean up annotation tables<a class="headerlink" href="#24-clean-up-annotation-tables" title="Permanent link">¶</a></h3>
<p>Now we have gene annotations that are contiguous and continuous. Lets clean up the tables by:</p>
<ol>
<li>Sorting entries by gene order</li>
<li>Separating KO from other annotation information</li>
<li>Adding headers</li>
<li>Replace anything that doesn't have KO numbers with an annotation so we do not get empty fields</li>
</ol>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-15-1" id="__codelineno-15-1" name="__codelineno-15-1"></a><span class="k">for</span><span class="w"> </span>aa_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>*.aa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-15-2" id="__codelineno-15-2" name="__codelineno-15-2"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">aa_file</span><span class="si">}</span><span class="w"> </span>.goi.aa<span class="k">)</span>
<a href="#__codelineno-15-3" id="__codelineno-15-3" name="__codelineno-15-3"></a><span class="w">  </span>sort<span class="w"> </span>-u<span class="w"> </span>-k<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="si">${</span><span class="nv">aa_file</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-15-4" id="__codelineno-15-4" name="__codelineno-15-4"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">'s/: /\t/g'</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-15-5" id="__codelineno-15-5" name="__codelineno-15-5"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span><span class="s1">'1i\Query gene\tKO\tAnnotation'</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-15-6" id="__codelineno-15-6" name="__codelineno-15-6"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s1">'\t'</span><span class="w"> </span>-v<span class="w"> </span><span class="nv">OFS</span><span class="o">=</span><span class="s1">'\t'</span><span class="w"> </span><span class="s1">'</span>
<a href="#__codelineno-15-7" id="__codelineno-15-7" name="__codelineno-15-7"></a><span class="s1">        {</span>
<a href="#__codelineno-15-8" id="__codelineno-15-8" name="__codelineno-15-8"></a><span class="s1">          if (!match($2, "K")) {</span>
<a href="#__codelineno-15-9" id="__codelineno-15-9" name="__codelineno-15-9"></a><span class="s1">            $3=$2</span>
<a href="#__codelineno-15-10" id="__codelineno-15-10" name="__codelineno-15-10"></a><span class="s1">          }</span>
<a href="#__codelineno-15-11" id="__codelineno-15-11" name="__codelineno-15-11"></a>
<a href="#__codelineno-15-12" id="__codelineno-15-12" name="__codelineno-15-12"></a><span class="s1">          print</span>
<a href="#__codelineno-15-13" id="__codelineno-15-13" name="__codelineno-15-13"></a><span class="s1">        }</span>
<a href="#__codelineno-15-14" id="__codelineno-15-14" name="__codelineno-15-14"></a><span class="s1">      '</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.txt
<a href="#__codelineno-15-15" id="__codelineno-15-15" name="__codelineno-15-15"></a><span class="k">done</span>
</code></pre></div>
</div>
<details class="info">
<summary>Deconstructing the code block</summary>
<p>We loop through each annotation subset, then:</p>
<table>
<thead>
<tr>
<th align="left">Code</th>
<th align="left">Action</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><code>sort -u -k 1</code></td>
<td align="left">Dereplicate and sort entries based on the first column</td>
</tr>
<tr>
<td align="left"><code>sed '1i\Query gene\tKO\tAnnotation'</code></td>
<td align="left">Add header</td>
</tr>
<tr>
<td align="left"><code>awk -F '\t' -v OFS='\t'</code></td>
<td align="left">Initiate the <code>awk</code> programme by setting the input and output field separator as tabs</td>
</tr>
<tr>
<td align="left"><code>if (!match($2, "K")) {$3=$2}</code></td>
<td align="left">If entries in the second column that <strong>do not</strong> have a "K", use value in the second column as values for the third column</td>
</tr>
<tr>
<td align="left"><code>print</code></td>
<td align="left">Return/print out all lines</td>
</tr>
</tbody>
</table>
</details>
<h2 id="3-create-comparison-tables">3. Create comparison tables<a class="headerlink" href="#3-create-comparison-tables" title="Permanent link">¶</a></h2>
<p>After obtaining relevant annotations for our genes of interest, we need to align the sequences against each other in order to identify sequence homology and directionality across our genes of interest. This can be achieved via pairwise sequence alignment using BLAST (or BLAST-like algorithms). Here, we will use <code>tBLASTx</code> to obtain requisite comparison tables. Here, we will do this using the command line. However, you can also use online BLAST as <a href="../5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">outlined here</a>. This is a 2 step process:</p>
<ol>
<li>Obtain nucleotide sequences for relevant contigs in each bin</li>
<li>Run (sequentially) a pairwise <code>tBLASTx</code> between each bin</li>
</ol>
<details class="note">
<summary>Which BLAST algorithms to use?</summary>
<p>For visualising gene synteny using <code>genoPlotR</code>, one can use outputs of <code>BLASTn</code> or <code>tBLASTx</code>. The determination of which algorithm to use depends on the phylogenetic relationship between MAGs. For closely related genomes, you can use <code>BLASTn</code> (or its variants) given that the genes will likely be conserved at the nucleotide sequence level. However, if your genomes are phylogenetically distant, you will need to use <code>tBLASTx</code>. This allows us to compare gene homology at the amino acid sequence level and retain nucleotide position information.</p>
</details>
<h3 id="31-subset-contigs-per-bin">3.1 Subset contigs per bin<a class="headerlink" href="#31-subset-contigs-per-bin" title="Permanent link">¶</a></h3>
<p>We begin by getting contig headers from our annotation files then using that to search and subset binned contigs. To subset sequences in FASTA format, we will be using <code>seqtk</code></p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-16-1" id="__codelineno-16-1" name="__codelineno-16-1"></a>module<span class="w"> </span>purge
<a href="#__codelineno-16-2" id="__codelineno-16-2" name="__codelineno-16-2"></a>module<span class="w"> </span>load<span class="w"> </span>seqtk/1.3-gimkl-2018b
<a href="#__codelineno-16-3" id="__codelineno-16-3" name="__codelineno-16-3"></a>
<a href="#__codelineno-16-4" id="__codelineno-16-4" name="__codelineno-16-4"></a><span class="k">for</span><span class="w"> </span>gene_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>*_cys.txt<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-16-5" id="__codelineno-16-5" name="__codelineno-16-5"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">gene_file</span><span class="si">}</span><span class="w"> </span>_cys.txt<span class="k">)</span>
<a href="#__codelineno-16-6" id="__codelineno-16-6" name="__codelineno-16-6"></a><span class="w">  </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="si">${</span><span class="nv">gene_file</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tail<span class="w"> </span>-n+2<span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">'s/_[0-9]*$//g'</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>uniq<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.contigID
<a href="#__codelineno-16-7" id="__codelineno-16-7" name="__codelineno-16-7"></a><span class="w">  </span>seqtk<span class="w"> </span>subseq<span class="w"> </span>filtered_bins/<span class="si">${</span><span class="nv">bin</span><span class="si">}</span>.filtered.fna<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.contigID<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys_contig.fna
<a href="#__codelineno-16-8" id="__codelineno-16-8" name="__codelineno-16-8"></a><span class="k">done</span>
</code></pre></div>
</div>
<h3 id="33-run-tblastx">3.3 Run <code>tBLASTx</code><a class="headerlink" href="#33-run-tblastx" title="Permanent link">¶</a></h3>
<p>We do not need to perform pairwise comparisons for all bin combinations (you can, if you want to). We will run tBLASTx in the order which we want to observe the genes in: <code>bin_3</code>, <code>bin_5</code>, <code>bin_8</code>, then <code>bin_9</code>. To do this, we set an array for bin IDs then run them through sequentially.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-17-1" id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="c1"># Set array</span>
<a href="#__codelineno-17-2" id="__codelineno-17-2" name="__codelineno-17-2"></a><span class="nv">bin_array</span><span class="o">=(</span>bin_3<span class="w"> </span>bin_5<span class="w"> </span>bin_8<span class="w"> </span>bin_9<span class="o">)</span>
<a href="#__codelineno-17-3" id="__codelineno-17-3" name="__codelineno-17-3"></a>
<a href="#__codelineno-17-4" id="__codelineno-17-4" name="__codelineno-17-4"></a><span class="c1"># Initiate index</span>
<a href="#__codelineno-17-5" id="__codelineno-17-5" name="__codelineno-17-5"></a><span class="nv">i</span><span class="o">=</span><span class="m">0</span>
<a href="#__codelineno-17-6" id="__codelineno-17-6" name="__codelineno-17-6"></a>
<a href="#__codelineno-17-7" id="__codelineno-17-7" name="__codelineno-17-7"></a><span class="c1"># Set maximum array index</span>
<a href="#__codelineno-17-8" id="__codelineno-17-8" name="__codelineno-17-8"></a><span class="nv">max_i</span><span class="o">=</span><span class="k">$((</span><span class="w"> </span><span class="si">${#</span><span class="nv">bin_array</span><span class="p">[@]</span><span class="si">}</span><span class="o">-</span><span class="m">1</span><span class="w"> </span><span class="k">))</span>
<a href="#__codelineno-17-9" id="__codelineno-17-9" name="__codelineno-17-9"></a>
<a href="#__codelineno-17-10" id="__codelineno-17-10" name="__codelineno-17-10"></a><span class="c1"># Run tBLASTx, comparing bins sequentially in the order of the array</span>
<a href="#__codelineno-17-11" id="__codelineno-17-11" name="__codelineno-17-11"></a><span class="k">while</span><span class="w"> </span><span class="o">[</span><span class="w"> </span><span class="nv">$i</span><span class="w"> </span>-lt<span class="w"> </span><span class="nv">$max_i</span><span class="w"> </span><span class="o">]</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-17-12" id="__codelineno-17-12" name="__codelineno-17-12"></a><span class="w">  </span><span class="nv">qbin</span><span class="o">=</span><span class="si">${</span><span class="nv">bin_array</span><span class="p">[</span><span class="nv">$i</span><span class="p">]</span><span class="si">}</span>
<a href="#__codelineno-17-13" id="__codelineno-17-13" name="__codelineno-17-13"></a><span class="w">  </span><span class="nv">sbin</span><span class="o">=</span><span class="si">${</span><span class="nv">bin_array</span><span class="p">[</span><span class="k">$((</span><span class="nv">$i</span><span class="o">+</span><span class="m">1</span><span class="k">))</span><span class="p">]</span><span class="si">}</span>
<a href="#__codelineno-17-14" id="__codelineno-17-14" name="__codelineno-17-14"></a><span class="w">  </span><span class="nb">printf</span><span class="w"> </span><span class="s2">"Comparing %s against %s\n"</span><span class="w"> </span><span class="nv">$qbin</span><span class="w"> </span><span class="nv">$sbin</span>
<a href="#__codelineno-17-15" id="__codelineno-17-15" name="__codelineno-17-15"></a><span class="w">  </span>tblastx<span class="w"> </span>-query<span class="w"> </span><span class="si">${</span><span class="nv">qbin</span><span class="si">}</span>_cys_contig.fna<span class="w"> </span>-subject<span class="w"> </span><span class="si">${</span><span class="nv">sbin</span><span class="si">}</span>_cys_contig.fna<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-17-16" id="__codelineno-17-16" name="__codelineno-17-16"></a><span class="w">          </span>-out<span class="w"> </span>blast_<span class="si">${</span><span class="nv">qbin</span><span class="si">}</span>_<span class="si">${</span><span class="nv">sbin</span><span class="si">}</span>.txt<span class="w"> </span>-outfmt<span class="w"> </span><span class="s2">"7 std ppos frames"</span>
<a href="#__codelineno-17-17" id="__codelineno-17-17" name="__codelineno-17-17"></a><span class="w">  </span><span class="o">((</span>i++<span class="o">))</span><span class="w"> </span><span class="c1"># Increment i after each comparison</span>
<a href="#__codelineno-17-18" id="__codelineno-17-18" name="__codelineno-17-18"></a><span class="k">done</span>
</code></pre></div>
</div>
<hr/>
<p>The above workflow should generate the required files for the annotation subset (<code>&lt;binID&gt;_cys.txt</code>) and pairwise <code>BLAST</code> comparisons (<code>blast_&lt;query binID&gt;_&lt;subject binID&gt;.txt</code>). Along with these outputs, copy the <code>prodigal</code> predictions (either <code>*.faa</code> or <code>*.fna</code>; do not use <code>no_metadata</code> files) from relevant bins into your working directory and follow the steps outlined in <a href="../../day4/ex16e_data_presentation_Gene_synteny/">Presentation of data: gene synteny</a> to generate synteny plots.</p>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>