
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa &amp; NeSI" name="author"/>
<link href="https://genomicsaotearoa.github.io/metagenomics_summer_school/day4/ex16e_data_presentation_Gene_synteny/" rel="canonical"/>
<link href="../ex16d_data_presentation_KEGG_pathways/" rel="prev"/>
<link href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/" rel="next"/>
<link href="../../theme_images/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.15" name="generator"/>
<title>Gene synteny - Metagenomics Summer School</title>
<link href="../../assets/stylesheets/main.26e3688c.min.css" rel="stylesheet"/>
<link href="../../assets/stylesheets/palette.ecc896b0.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#gene-synteny">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Metagenomics Summer School" class="md-header__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Metagenomics Summer School
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Gene synteny
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
</nav>
<nav aria-label="Tabs" class="md-tabs" data-md-component="tabs">
<div class="md-grid">
<ul class="md-tabs__list">
<li class="md-tabs__item">
<a class="md-tabs__link" href="../..">
      Home
    </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day1/ex1_bash_and_scheduler/">
        Day 1
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day2/ex6_initial_binning/">
        Day 2
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Day 3
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link md-tabs__link--active" href="../ex15_gene_annotation_part3/">
        Day 4
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        Resources
      </a>
</li>
<li class="md-tabs__item">
<a class="md-tabs__link" href="../../supplementary/supplementary_1/">
        Supplementary
      </a>
</li>
</ul>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Metagenomics Summer School" class="md-nav__button md-logo" data-md-component="logo" href="../.." title="Metagenomics Summer School">
<img alt="logo" src="../../theme_images/nesi_ga.png"/>
</a>
    Metagenomics Summer School
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/metagenomics_summer_school" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/metagenomics_summer_school
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../..">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Day 1
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
          Day 1
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex1_bash_and_scheduler/">
        Introduction I: Shell
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_1_intro_to_scheduler/">
        Introduction II: HPC and job scheduler
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex2_quality_filtering/">
        Filter raw reads by quality
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex3_assembly/">
        Assembly I: Assembling contigs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex4_assembly/">
        Assembly II: Varying the parameters
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day1/ex5_evaluating_assemblies/">
        Assembly evaluation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          Day 2
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Day 2
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex6_initial_binning/">
        Introduction to binning
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex7_initial_binning/">
        Binning with multiple tools
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex8_bin_dereplication/">
        Bin dereplication
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day2/ex9_refining_bins/">
        Manual bin refinement
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox"/>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          Day 3
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
          Day 3
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11_coverage_and_taxonomy/">
        Assigning taxonomy to refined prokaryotic bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex11.1_phylogenomics/">
        Phylogenomics
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex10_viruses/">
        Identifying viral contigs in metagenomic data
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex12_gene_prediction/">
        Gene prediction
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex13_gene_annotation_part1/">
        Gene annotation I: BLAST-like and HMM
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../day3/ex14_gene_annotation_part2/">
        Gene annotation II: DRAM and coverage calculation
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5" type="checkbox"/>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          Day 4
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
          Day 4
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../ex15_gene_annotation_part3/">
        Gene annotation III: DRAM distillation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16a_data_presentation_Intro/">
        Introduction to data presentation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16b_data_presentation_Coverage/">
        Coverage heatmaps
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16c_OPTIONAL_data_presentation_Ordination/">
        Ordinations
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16d_data_presentation_KEGG_pathways/">
        KEGG pathway maps
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Gene synteny
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Gene synteny
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r">
    Build a sulfur assimilation gene alignment figure to investigate gene synteny using R
  </a>
<nav aria-label="Build a sulfur assimilation gene alignment figure to investigate gene synteny using R" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#part-1-parsing-files-in-bash">
    Part 1 - Parsing files in bash
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#part-2-producing-the-figure-in-r">
    Part 2 - Producing the figure in R
  </a>
<nav aria-label="Part 2 - Producing the figure in R" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#21-prepare-environment">
    2.1 Prepare environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#22-import-files">
    2.2 Import files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#23-wrangle-data">
    2.3 Wrangle data
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#24-plot-data">
    2.4 Plot data
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ex16f_OPTIONAL_data_presentation_CAZy_annotations/">
        CAZy heatmaps
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox"/>
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
          Resources
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
          Resources
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/1_APPENDIX_ex8_Dereplication/">
        APPENDIX (ex8): Dereplicating data from multiple assemblies
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/2_APPENDIX_ex9_Generating_input_files_for_VizBin/">
        APPENDIX (ex9): Generating input files for *VizBin* from *DAS_Tool* curated bins
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/3_APPENDIX_ex11_Normalise_coverage_example/">
        APPENDIX (ex11): Normalise per-sample coverage values by average library size (example)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/4_APPENDIX_ex11_viral_taxonomy_prediction_via_vContact2/">
        APPENDIX (ex11) : Viral taxonomy prediction via *vContact2*
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">
        APPENDIX (ex15): Prepare input for gene synteny visualisation
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/5_APPENDIX_ex15_gene_synteny_Generate_blast_files/">
        APPENDIX (ex15): Generating pairwise contig comparisons using online BLAST
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../resources/7_command_line_shortcuts/">
        Useful command line shortcuts
      </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox"/>
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_1/">
        NeSI Setup
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../supplementary/supplementary_2/">
        NeSI File system, Working directory and Symlinks
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r">
    Build a sulfur assimilation gene alignment figure to investigate gene synteny using R
  </a>
<nav aria-label="Build a sulfur assimilation gene alignment figure to investigate gene synteny using R" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#part-1-parsing-files-in-bash">
    Part 1 - Parsing files in bash
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#part-2-producing-the-figure-in-r">
    Part 2 - Producing the figure in R
  </a>
<nav aria-label="Part 2 - Producing the figure in R" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#21-prepare-environment">
    2.1 Prepare environment
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#22-import-files">
    2.2 Import files
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#23-wrangle-data">
    2.3 Wrangle data
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#24-plot-data">
    2.4 Plot data
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="gene-synteny">Gene synteny<a class="headerlink" href="#gene-synteny" title="Permanent link">¶</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
<ul>
<li><a href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r">Build a sulfur assimilation gene alignment figure to investigate gene synteny using <code>R</code></a></li>
<li><a href="#part-1---parsing-files-in-bash">Parsing files in <code>bash</code></a></li>
<li><a href="#part-2---producing-the-figure-in-r">Import and wrangle data in <code>R</code></a></li>
</ul>
</div>
<hr/>
<h2 id="build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r">Build a sulfur assimilation gene alignment figure to investigate gene synteny using <code>R</code><a class="headerlink" href="#build-a-sulfur-assimilation-gene-alignment-figure-to-investigate-gene-synteny-using-r" title="Permanent link">¶</a></h2>
<p>When investigating the evolution of genomes, we sometimes want to consider not only the presence/absence of genes in a genome, but also how they are arranged in an operon. For this exercise, we are going to visualise several sulfur assimilation genes from our MAGs and compare their arrangements.</p>
<p>This exercise involves the use of commands in <code>bash</code> and <code>R</code>. In the first part, we will need to find and parse gene coordinates in <code>bash</code>. This is followed by data wrangling and plot generation in <code>R</code>.</p>
<p>For this exercise, navigate to the directory <code>11.data_presentation/gene_synteny/</code>. You have been provided with a copy of the <code>prodigal</code> gene predictions for each of the bins (<code>.faa</code> files), an annotation output table using multiple databases (<code>.aa</code> files), a small table of the annotation of some key genes of interest (<code>cys.txt</code> files), and tBLASTx output (<code>blast*.txt</code>) comparing the genes of interest from these organisms. The annotation files were created by manually searching the annotations obtained in the previous exercises.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Refer to <a href="../../resources/5_APPENDIX_ex15_Prepare_gene_synteny_inputs/">this appendix</a> for detailed information on how to generate input data.</p>
</div>
<h3 id="part-1-parsing-files-in-bash">Part 1 - Parsing files in <em>bash</em><a class="headerlink" href="#part-1-parsing-files-in-bash" title="Permanent link">¶</a></h3>
<p>We will be performing this exercise in two stages. Firstly, in <code>bash</code>, we will use <code>cut</code> and <code>tail</code> to pull out the genes of interest listed in the <code>*cys.txt</code> files from the <code>prodigal</code> files. The gene names will then be used to create a table of gene coordinates from the <code>prodigal</code> output using <code>grep</code>, <code>cut</code>, and <code>sed</code>.</p>
<p>For these <code>bash</code> steps, we will need to return to our logged in NeSI terminal. Switch over to a NeSI <code>Jupyter hub</code> terminal or log in to a fresh session in a new terminal. </p>
<p>Navigate to the <code>11.data_presentation/gene_synteny/</code> folder, and then perform the <code>cut</code> and <code>tail</code> steps outlined above.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="nb">cd</span><span class="w"> </span>/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR<span class="w"> </span>FOLDER&gt;/11.data_presentation/gene_synteny/
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a><span class="k">for</span><span class="w"> </span>bin_file<span class="w"> </span><span class="k">in</span><span class="w"> </span>*_cys.txt<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span>_cys.txt<span class="k">)</span>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a><span class="w">  </span>cut<span class="w"> </span>-f1<span class="w"> </span><span class="si">${</span><span class="nv">bin_file</span><span class="si">}</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tail<span class="w"> </span>-n+2<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.genes
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="k">done</span>
</code></pre></div>
</div>
<p>We now have three new files, ending with the <code>.genes</code> suffix which are simply a list of the genes that we wish to extract from the <code>prodigal</code> files. We can then use each of these files as input for <code>grep</code> to pull out the FASTA entries that correspond to these genes.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="k">for</span><span class="w"> </span>gene_list<span class="w"> </span><span class="k">in</span><span class="w"> </span>*.genes<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="w">  </span><span class="nv">bin</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">gene_list</span><span class="si">}</span><span class="w"> </span>_cys.genes<span class="k">)</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="w">  </span>grep<span class="w"> </span>-f<span class="w"> </span><span class="si">${</span><span class="nv">gene_list</span><span class="si">}</span><span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>.filtered.genes.faa<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">'s/ # /\t/g'</span><span class="w"> </span>-e<span class="w"> </span><span class="s1">'s/&gt;//g'</span><span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>-4<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bin</span><span class="si">}</span>_cys.coords
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a><span class="k">done</span>
</code></pre></div>
</div>
<p>As previously, we will quickly go through the steps of this command:</p>
<ul>
<li><code>grep -f ${gene_list} ${bin}.filtered.genes.faa</code> searches the <code>prodigal</code> FASTA amino acid sequence output for headers that match those in our <code>.genes</code> file</li>
<li><code>sed -e 's/ # /\t/g' -e 's/&gt;//g'</code> replaces the default <code>prodigal</code> header line delimiter " # " with a tab, then removes <code>&gt;</code> from the header.</li>
<li><code>cut -f 1-4</code> selects the first 4 columns delimited by a tab (default for <code>cut</code>).</li>
</ul>
<p>Check the content of the <code>.coords</code> files now. You should see something like the following:</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>cat<span class="w"> </span>bin_3_cys.coords
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Terminal output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>bin_3_NODE_53_length_158395_cov_1.135272_128    135928  136935  1
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>bin_3_NODE_53_length_158395_cov_1.135272_129    136994  137299  1
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>bin_3_NODE_53_length_158395_cov_1.135272_130    137411  138322  1
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>bin_3_NODE_53_length_158395_cov_1.135272_131    138413  139201  1
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>bin_3_NODE_53_length_158395_cov_1.135272_132    139267  140100  1
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a>bin_3_NODE_53_length_158395_cov_1.135272_133    140110  140988  1
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a>bin_3_NODE_53_length_158395_cov_1.135272_134    140985  142073  1
</code></pre></div>
</div>
<p>If you recall from the previous exercise on gene prediction, we have taken the first four entries from each line of the <code>prodigal</code> output, which consists of:</p>
<div class="admonition quote">
<ol>
<li>Gene name, written as [CONTIG]_[GENE]</li>
<li>Start position of the gene</li>
<li>Stop position of the gene</li>
<li>Orientation of the gene</li>
</ol>
</div>
<p>We will now use these tables, together with the annotation tables to create the gene synteny view in <code>R</code>.</p>
<h3 id="part-2-producing-the-figure-in-r">Part 2 - Producing the figure in <code>R</code><a class="headerlink" href="#part-2-producing-the-figure-in-r" title="Permanent link">¶</a></h3>
<p>First, move back to the <a href="https://jupyter.nesi.org.nz/hub/login">Jupyter hub</a> pane and start an <code>RStudio</code> session</p>
<h4 id="21-prepare-environment">2.1 Prepare environment<a class="headerlink" href="#21-prepare-environment" title="Permanent link">¶</a></h4>
<p>Similar to previous sessions, we will begin by preparing the environment by setting our working directory and loading required libraries.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="c1"># Set working directory</span>
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a><span class="nf">setwd</span><span class="p">(</span><span class="s">'/nesi/nobackup/nesi02659/MGSS_U/&lt;YOUR FOLDER&gt;/11.data_presentation/gene_synteny/'</span><span class="p">)</span>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="c1"># Load libraries</span>
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="c1"># Tidyverse packages</span>
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="nf">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="nf">library</span><span class="p">(</span><span class="n">purrr</span><span class="p">)</span>
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a><span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a><span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a>
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a><span class="c1"># Gene synteny</span>
<a href="#__codelineno-4-13" id="__codelineno-4-13" name="__codelineno-4-13"></a><span class="nf">library</span><span class="p">(</span><span class="n">genoPlotR</span><span class="p">)</span>
</code></pre></div>
</div>
<h4 id="22-import-files">2.2 Import files<a class="headerlink" href="#22-import-files" title="Permanent link">¶</a></h4>
<p>Using genoPlotR requires files with 3 different types of information:</p>
<ul>
<li>Gene coordinates (from <code>prodigal</code>)</li>
<li>Gene annotations (from our annotation workflows)</li>
<li>Contig comparisons (from tBLASTx output)</li>
</ul>
<p>First, we will import the <code>.coords</code> files, and set column names to the files.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="c1"># Gene coordinates</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a><span class="n">coords_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">".coords"</span><span class="p">)</span>
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a><span class="n">coords_header</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">"name"</span><span class="p">,</span><span class="w"> </span><span class="s">"start"</span><span class="p">,</span><span class="w"> </span><span class="s">"end"</span><span class="p">,</span><span class="w"> </span><span class="s">"strand"</span><span class="p">)</span>
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="n">coords_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">coords_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a><span class="w">  </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">coords_header</span><span class="p">)</span>
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">coords_files</span><span class="p">,</span><span class="w"> </span><span class="s">"cys."</span><span class="p">))</span>
</code></pre></div>
</div>
<p>Take a look at the content of each of these data.frames, by entering their names into the terminal. You should notice that the coordinates occur at quite different positions between the genomes. If we were looking at complete genomes, this would indicate their position relative to the <em>origin of replication</em>, but as these are unclosed genomes obtained from MAGs, they reflect the coordinates upon their particular <em>contig</em>.</p>
<p>Lets continue to import the other files</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="c1"># Gene annotations</span>
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a><span class="n">annot_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"^bin_.*.txt"</span><span class="p">)</span>
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a><span class="n">annot_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">annot_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="w">  </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">str_replace</span><span class="p">(</span><span class="n">annot_files</span><span class="p">,</span><span class="w"> </span><span class="s">"cys.txt"</span><span class="p">,</span><span class="w"> </span><span class="s">"annot"</span><span class="p">))</span>
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a><span class="c1"># BLAST outputs</span>
<a href="#__codelineno-6-9" id="__codelineno-6-9" name="__codelineno-6-9"></a><span class="n">blast_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"^blast_.*.txt"</span><span class="p">)</span>
<a href="#__codelineno-6-10" id="__codelineno-6-10" name="__codelineno-6-10"></a><span class="n">blast_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">blast_files</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-6-11" id="__codelineno-6-11" name="__codelineno-6-11"></a><span class="w">  </span><span class="nf">read_comparison_from_blast</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<a href="#__codelineno-6-12" id="__codelineno-6-12" name="__codelineno-6-12"></a><span class="p">})</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-6-13" id="__codelineno-6-13" name="__codelineno-6-13"></a><span class="w">  </span><span class="nf">setNames</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="n">blast_files</span><span class="p">,</span><span class="w"> </span><span class="s">".txt"</span><span class="p">))</span>
</code></pre></div>
</div>
<p>Notice that the file reading function for BLAST files are different (<code>read_comparison_from_blast()</code> versus <code>read_tsv()</code>). This is a function specific to <code>genoPlotR</code> for parsing BLAST outputs. If you have some custom comparison pipeline, <code>genoPlotR</code> can also read tab-delimited files via <code>read_comparison_from_tab()</code></p>
<h4 id="23-wrangle-data">2.3 Wrangle data<a class="headerlink" href="#23-wrangle-data" title="Permanent link">¶</a></h4>
<p>We now need to do the following:</p>
<ul>
<li>Parse coordinate data into a format genoPlotR can use</li>
<li>Tidy up annotation tables so they will show the correct labels</li>
</ul>
<p>We start by converting our coordinate data into a <code>dna_seg</code> (DNA segment) data class, then check what is in one of the objects in the list.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="c1"># Transform coordinate tables into "dna_seg" objects</span>
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a><span class="n">ds_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">coords_list</span><span class="p">,</span><span class="w"> </span><span class="n">dna_seg</span><span class="p">)</span>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a>
<a href="#__codelineno-7-4" id="__codelineno-7-4" name="__codelineno-7-4"></a><span class="n">ds_list</span><span class="o">$</span><span class="n">bin_5_coords</span>
</code></pre></div>
</div>
<table>
<thead>
<tr>
<th align="right">name</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">strand</th>
<th align="right">col</th>
<th align="right">fill</th>
<th align="right">lty</th>
<th align="right">lwd</th>
<th align="right">pch</th>
<th align="right">cex</th>
<th align="right">gene_type</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">bin_5_NODE_95_length_91726_cov_0.379302_67</td>
<td align="right">71608</td>
<td align="right">72606</td>
<td align="right">1</td>
<td align="right">blue</td>
<td align="right">blue</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">1</td>
<td align="right">arrows</td>
</tr>
<tr>
<td align="right">bin_5_NODE_95_length_91726_cov_0.379302_68</td>
<td align="right">72768</td>
<td align="right">73586</td>
<td align="right">1</td>
<td align="right">blue</td>
<td align="right">blue</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">1</td>
<td align="right">arrows</td>
</tr>
<tr>
<td align="right">bin_5_NODE_95_length_91726_cov_0.379302_69</td>
<td align="right">73597</td>
<td align="right">74466</td>
<td align="right">1</td>
<td align="right">blue</td>
<td align="right">blue</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">1</td>
<td align="right">arrows</td>
</tr>
<tr>
<td align="right">bin_5_NODE_95_length_91726_cov_0.379302_70</td>
<td align="right">74470</td>
<td align="right">75459</td>
<td align="right">1</td>
<td align="right">blue</td>
<td align="right">blue</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">1</td>
<td align="right">arrows</td>
</tr>
</tbody>
</table>
<p>Now we tidy our annotations table and create a joint coordinate-annotation table. But first, lets take a look at our annotation tables to get a feel for what needs to change.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a><span class="n">annot_list</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>$bin_3_annot
<a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a># A tibble: 7 × 3
<a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a>  `Query gene`                                 KO                                      Annotation                                 
<a href="#__codelineno-9-4" id="__codelineno-9-4" name="__codelineno-9-4"></a>  &lt;chr&gt;                                        &lt;chr&gt;                                   &lt;chr&gt;                                      
<a href="#__codelineno-9-5" id="__codelineno-9-5" name="__codelineno-9-5"></a>1 bin_3_NODE_53_length_158395_cov_1.135272_128 K02048                                  sbp1; Prokaryotic sulfate-/thiosulfate-bin…
<a href="#__codelineno-9-6" id="__codelineno-9-6" name="__codelineno-9-6"></a>2 bin_3_NODE_53_length_158395_cov_1.135272_129 possible predicted diverged CheY-domain possible predicted diverged CheY-domain    
<a href="#__codelineno-9-7" id="__codelineno-9-7" name="__codelineno-9-7"></a>3 bin_3_NODE_53_length_158395_cov_1.135272_130 Domain of unknown function 2            Domain of unknown function 2               
<a href="#__codelineno-9-8" id="__codelineno-9-8" name="__codelineno-9-8"></a>4 bin_3_NODE_53_length_158395_cov_1.135272_131 Domain of unknown function 2            Domain of unknown function 2               
<a href="#__codelineno-9-9" id="__codelineno-9-9" name="__codelineno-9-9"></a>5 bin_3_NODE_53_length_158395_cov_1.135272_132 K02046                                  cysU; cysU; sulfate transport ABC transpor…
<a href="#__codelineno-9-10" id="__codelineno-9-10" name="__codelineno-9-10"></a>6 bin_3_NODE_53_length_158395_cov_1.135272_133 K02047                                  cysW; cysW; sulfate transport ABC transpor…
<a href="#__codelineno-9-11" id="__codelineno-9-11" name="__codelineno-9-11"></a>7 bin_3_NODE_53_length_158395_cov_1.135272_134 K02045                                  cysA; cysA; sulfate transport ATP-binding …
<a href="#__codelineno-9-12" id="__codelineno-9-12" name="__codelineno-9-12"></a>
<a href="#__codelineno-9-13" id="__codelineno-9-13" name="__codelineno-9-13"></a>$bin_5_annot
<a href="#__codelineno-9-14" id="__codelineno-9-14" name="__codelineno-9-14"></a># A tibble: 4 × 3
<a href="#__codelineno-9-15" id="__codelineno-9-15" name="__codelineno-9-15"></a>  `Query gene`                               KO     Annotation                                                        
<a href="#__codelineno-9-16" id="__codelineno-9-16" name="__codelineno-9-16"></a>  &lt;chr&gt;                                      &lt;chr&gt;  &lt;chr&gt;                                                             
<a href="#__codelineno-9-17" id="__codelineno-9-17" name="__codelineno-9-17"></a>1 bin_5_NODE_95_length_91726_cov_0.379302_67 K02048 sbp; sulfate-binding protein                                      
<a href="#__codelineno-9-18" id="__codelineno-9-18" name="__codelineno-9-18"></a>2 bin_5_NODE_95_length_91726_cov_0.379302_68 K02046 cysT; sulfate transporter CysT                                    
<a href="#__codelineno-9-19" id="__codelineno-9-19" name="__codelineno-9-19"></a>3 bin_5_NODE_95_length_91726_cov_0.379302_69 K02047 cysW; sulfate transporter CysW                                    
<a href="#__codelineno-9-20" id="__codelineno-9-20" name="__codelineno-9-20"></a>4 bin_5_NODE_95_length_91726_cov_0.379302_70 K02045 cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA
<a href="#__codelineno-9-21" id="__codelineno-9-21" name="__codelineno-9-21"></a>
<a href="#__codelineno-9-22" id="__codelineno-9-22" name="__codelineno-9-22"></a>$bin_8_annot
<a href="#__codelineno-9-23" id="__codelineno-9-23" name="__codelineno-9-23"></a># A tibble: 4 × 3
<a href="#__codelineno-9-24" id="__codelineno-9-24" name="__codelineno-9-24"></a>  `Query gene`                                KO     Annotation                                    
<a href="#__codelineno-9-25" id="__codelineno-9-25" name="__codelineno-9-25"></a>  &lt;chr&gt;                                       &lt;chr&gt;  &lt;chr&gt;                                         
<a href="#__codelineno-9-26" id="__codelineno-9-26" name="__codelineno-9-26"></a>1 bin_8_NODE_60_length_149231_cov_0.651774_55 K02048 Thiosulphate-binding protein                  
<a href="#__codelineno-9-27" id="__codelineno-9-27" name="__codelineno-9-27"></a>2 bin_8_NODE_60_length_149231_cov_0.651774_56 K02046 sulfate ABC transporter permease subunit CysT 
<a href="#__codelineno-9-28" id="__codelineno-9-28" name="__codelineno-9-28"></a>3 bin_8_NODE_60_length_149231_cov_0.651774_57 K02047 Sulfate ABC transporter, permease protein CysW
<a href="#__codelineno-9-29" id="__codelineno-9-29" name="__codelineno-9-29"></a>4 bin_8_NODE_60_length_149231_cov_0.651774_58 K02045 sulphate transport system permease protein 1  
<a href="#__codelineno-9-30" id="__codelineno-9-30" name="__codelineno-9-30"></a>
<a href="#__codelineno-9-31" id="__codelineno-9-31" name="__codelineno-9-31"></a>$bin_9_annot
<a href="#__codelineno-9-32" id="__codelineno-9-32" name="__codelineno-9-32"></a># A tibble: 4 × 3
<a href="#__codelineno-9-33" id="__codelineno-9-33" name="__codelineno-9-33"></a>  `Query gene`                                 KO     Annotation                                           
<a href="#__codelineno-9-34" id="__codelineno-9-34" name="__codelineno-9-34"></a>  &lt;chr&gt;                                        &lt;chr&gt;  &lt;chr&gt;                                                
<a href="#__codelineno-9-35" id="__codelineno-9-35" name="__codelineno-9-35"></a>1 bin_9_NODE_12_length_355673_cov_0.516990_147 K02048 thiosulfate ABC transporter substrate-binding protein
<a href="#__codelineno-9-36" id="__codelineno-9-36" name="__codelineno-9-36"></a>2 bin_9_NODE_12_length_355673_cov_0.516990_148 K02046 sulfate ABC transporter permease                     
<a href="#__codelineno-9-37" id="__codelineno-9-37" name="__codelineno-9-37"></a>3 bin_9_NODE_12_length_355673_cov_0.516990_149 K02047 sulfate ABC transporter permease                     
<a href="#__codelineno-9-38" id="__codelineno-9-38" name="__codelineno-9-38"></a>4 bin_9_NODE_12_length_355673_cov_0.516990_150 K02045 sulfate ABC transporter ATP-binding protein   
</code></pre></div>
</div>
<p>Immediately, we can see that there are some inconsistencies in the annotations. </p>
<ul>
<li>In the annotations for bin 3, there are repeat gene names in the same annotation line.</li>
<li>Annotations for the same KO number is inconsistent across tables. For instance, K02046 is cysU in bin 3 and cysT in bins 5 and 8.</li>
<li>There is a complete lack of gene names in bin 9 annotations. </li>
</ul>
<p>We need to remedy this by having a consistent label system. First, create a data frame of labels you want to show in your synteny plot.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a><span class="n">KO_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<a href="#__codelineno-10-2" id="__codelineno-10-2" name="__codelineno-10-2"></a><span class="w">  </span><span class="s">"KO"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">"K0204"</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="o">:</span><span class="m">8</span><span class="p">),</span>
<a href="#__codelineno-10-3" id="__codelineno-10-3" name="__codelineno-10-3"></a><span class="w">  </span><span class="s">"gene_name"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">"cysA"</span><span class="p">,</span><span class="w"> </span><span class="s">"cysU"</span><span class="p">,</span><span class="w"> </span><span class="s">"cysW"</span><span class="p">,</span><span class="w"> </span><span class="s">"sbp1"</span><span class="p">)</span>
<a href="#__codelineno-10-4" id="__codelineno-10-4" name="__codelineno-10-4"></a><span class="p">)</span>
<a href="#__codelineno-10-5" id="__codelineno-10-5" name="__codelineno-10-5"></a>
<a href="#__codelineno-10-6" id="__codelineno-10-6" name="__codelineno-10-6"></a><span class="c1"># Check what we have created</span>
<a href="#__codelineno-10-7" id="__codelineno-10-7" name="__codelineno-10-7"></a><span class="n">KO_genes</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a>      KO gene_name
<a href="#__codelineno-11-2" id="__codelineno-11-2" name="__codelineno-11-2"></a>1 K02045      cysA
<a href="#__codelineno-11-3" id="__codelineno-11-3" name="__codelineno-11-3"></a>2 K02046      cysU
<a href="#__codelineno-11-4" id="__codelineno-11-4" name="__codelineno-11-4"></a>3 K02047      cysW
<a href="#__codelineno-11-5" id="__codelineno-11-5" name="__codelineno-11-5"></a>4 K02048      sbp1
</code></pre></div>
</div>
<p>Then, we join the KO_genes, annotation, and coordinate tables. We also replace rows that contain <code>NA</code> with <code>"Unknown domain"</code>.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a><span class="n">ann_coords_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map2</span><span class="p">(</span><span class="n">annot_list</span><span class="p">,</span><span class="w"> </span><span class="n">coords_list</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">ann</span><span class="p">,</span><span class="w"> </span><span class="n">coord</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-12-2" id="__codelineno-12-2" name="__codelineno-12-2"></a><span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">ann</span><span class="p">,</span><span class="w"> </span><span class="n">coord</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">'Query gene'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">'name'</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-12-3" id="__codelineno-12-3" name="__codelineno-12-3"></a><span class="w">    </span><span class="nf">left_join</span><span class="p">(</span><span class="n">KO_genes</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-12-4" id="__codelineno-12-4" name="__codelineno-12-4"></a><span class="w">    </span><span class="nf">mutate</span><span class="p">(</span>
<a href="#__codelineno-12-5" id="__codelineno-12-5" name="__codelineno-12-5"></a><span class="w">      </span><span class="n">gene_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">replace_na</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">"Unknown domain"</span><span class="p">)</span>
<a href="#__codelineno-12-6" id="__codelineno-12-6" name="__codelineno-12-6"></a><span class="w">    </span><span class="p">)</span>
<a href="#__codelineno-12-7" id="__codelineno-12-7" name="__codelineno-12-7"></a><span class="p">})</span>
<a href="#__codelineno-12-8" id="__codelineno-12-8" name="__codelineno-12-8"></a>
<a href="#__codelineno-12-9" id="__codelineno-12-9" name="__codelineno-12-9"></a><span class="c1"># Check one of the tables</span>
<a href="#__codelineno-12-10" id="__codelineno-12-10" name="__codelineno-12-10"></a><span class="nf">View</span><span class="p">(</span><span class="n">ann_coords_list</span><span class="o">$</span><span class="n">bin_3_annot</span><span class="p">)</span><span class="w"> </span><span class="c1"># In RStudio, View() will open a data viewing tab on the top left pane!</span>
</code></pre></div>
</div>
<p>Much better! We can use the <code>gene_name</code> column as our annotation on our gene synteny plot.</p>
<p>We also need to convert the tidy annotation table into something that <code>genoPlotR</code> can use (i.e. an <code>annotations</code> class object).</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-13-1" id="__codelineno-13-1" name="__codelineno-13-1"></a><span class="n">annotations_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">ann_coords_list</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-13-2" id="__codelineno-13-2" name="__codelineno-13-2"></a><span class="w">  </span><span class="nf">annotation</span><span class="p">(</span>
<a href="#__codelineno-13-3" id="__codelineno-13-3" name="__codelineno-13-3"></a><span class="w">    </span><span class="n">x1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="c1"># Where each gene starts</span>
<a href="#__codelineno-13-4" id="__codelineno-13-4" name="__codelineno-13-4"></a><span class="w">    </span><span class="n">x2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">end</span><span class="p">,</span><span class="w"> </span><span class="c1"># Gene end coordinate</span>
<a href="#__codelineno-13-5" id="__codelineno-13-5" name="__codelineno-13-5"></a><span class="w">    </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">gene_name</span><span class="w"> </span><span class="c1"># Labels to show</span>
<a href="#__codelineno-13-6" id="__codelineno-13-6" name="__codelineno-13-6"></a><span class="w">  </span><span class="p">)</span>
<a href="#__codelineno-13-7" id="__codelineno-13-7" name="__codelineno-13-7"></a><span class="p">})</span>
<a href="#__codelineno-13-8" id="__codelineno-13-8" name="__codelineno-13-8"></a>
<a href="#__codelineno-13-9" id="__codelineno-13-9" name="__codelineno-13-9"></a><span class="c1"># Check one of the annotations_list objects</span>
<a href="#__codelineno-13-10" id="__codelineno-13-10" name="__codelineno-13-10"></a><span class="n">annotations_list</span><span class="o">$</span><span class="n">bin_5_annot</span>
</code></pre></div>
</div>
<div class="admonition circle-check">
<p class="admonition-title">Console output</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-14-1" id="__codelineno-14-1" name="__codelineno-14-1"></a>     x1    x2 text color rot
<a href="#__codelineno-14-2" id="__codelineno-14-2" name="__codelineno-14-2"></a>1 71608 72606 sbp1 black   0
<a href="#__codelineno-14-3" id="__codelineno-14-3" name="__codelineno-14-3"></a>2 72768 73586 cysU black   0
<a href="#__codelineno-14-4" id="__codelineno-14-4" name="__codelineno-14-4"></a>3 73597 74466 cysW black   0
<a href="#__codelineno-14-5" id="__codelineno-14-5" name="__codelineno-14-5"></a>4 74470 75459 cysA black   0
</code></pre></div>
</div>
<h4 id="24-plot-data">2.4 Plot data<a class="headerlink" href="#24-plot-data" title="Permanent link">¶</a></h4>
<p>Lets plot our data to see what it looks like and if it need tweaking.</p>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-15-1" id="__codelineno-15-1" name="__codelineno-15-1"></a><span class="nf">plot_gene_map</span><span class="p">(</span>
<a href="#__codelineno-15-2" id="__codelineno-15-2" name="__codelineno-15-2"></a><span class="w">  </span><span class="n">dna_segs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ds_list</span><span class="p">,</span>
<a href="#__codelineno-15-3" id="__codelineno-15-3" name="__codelineno-15-3"></a><span class="w">  </span><span class="n">comparisons</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast_list</span><span class="p">,</span>
<a href="#__codelineno-15-4" id="__codelineno-15-4" name="__codelineno-15-4"></a><span class="w">  </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotations_list</span><span class="p">,</span>
<a href="#__codelineno-15-5" id="__codelineno-15-5" name="__codelineno-15-5"></a><span class="w">  </span><span class="n">dna_seg_labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">ds_list</span><span class="p">),</span><span class="w"> </span><span class="s">"_coords"</span><span class="p">),</span>
<a href="#__codelineno-15-6" id="__codelineno-15-6" name="__codelineno-15-6"></a><span class="w">  </span><span class="n">dna_seg_scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<a href="#__codelineno-15-7" id="__codelineno-15-7" name="__codelineno-15-7"></a><span class="w">  </span><span class="n">gene_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"arrows"</span>
<a href="#__codelineno-15-8" id="__codelineno-15-8" name="__codelineno-15-8"></a><span class="p">)</span>
</code></pre></div>
</div>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_synteny.01.synteny_all.png"><img alt="image" src="../../figures/day4_synteny.01.synteny_all.png"/></a></p>
<p>We have a plot! However, it is rather cluttered and could be further improved:</p>
<ul>
<li>Remove gene names in the middle tracks</li>
<li>Concatenate the "Unknown domain" portion into a single annotation</li>
<li>Showing only the best tBLASTx hits, reducing short alignments</li>
</ul>
<div class="admonition r-project">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-16-1" id="__codelineno-16-1" name="__codelineno-16-1"></a><span class="c1"># Filter BLAST results</span>
<a href="#__codelineno-16-2" id="__codelineno-16-2" name="__codelineno-16-2"></a><span class="n">filt_blast_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">blast_list</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-16-3" id="__codelineno-16-3" name="__codelineno-16-3"></a><span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-16-4" id="__codelineno-16-4" name="__codelineno-16-4"></a><span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">name1</span><span class="p">,</span><span class="w"> </span><span class="n">name2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-16-5" id="__codelineno-16-5" name="__codelineno-16-5"></a><span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="n">e_value</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">e_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<a href="#__codelineno-16-6" id="__codelineno-16-6" name="__codelineno-16-6"></a><span class="w">    </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-16-7" id="__codelineno-16-7" name="__codelineno-16-7"></a><span class="w">    </span><span class="nf">as.comparison</span><span class="p">()</span>
<a href="#__codelineno-16-8" id="__codelineno-16-8" name="__codelineno-16-8"></a><span class="p">})</span>
<a href="#__codelineno-16-9" id="__codelineno-16-9" name="__codelineno-16-9"></a>
<a href="#__codelineno-16-10" id="__codelineno-16-10" name="__codelineno-16-10"></a><span class="c1"># Concatenate "Unknown domain" in bin 3</span>
<a href="#__codelineno-16-11" id="__codelineno-16-11" name="__codelineno-16-11"></a><span class="n">annotations_bin_3_mod</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotations_list</span><span class="o">$</span><span class="n">bin_3_annot</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-16-12" id="__codelineno-16-12" name="__codelineno-16-12"></a><span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">"Unknown domain"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<a href="#__codelineno-16-13" id="__codelineno-16-13" name="__codelineno-16-13"></a><span class="w">  </span><span class="nf">add_row</span><span class="p">(</span>
<a href="#__codelineno-16-14" id="__codelineno-16-14" name="__codelineno-16-14"></a><span class="w">    </span><span class="n">x1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">136994</span><span class="p">,</span><span class="w"> </span><span class="n">x2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">139201</span><span class="p">,</span><span class="w"> </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"Unknown domain"</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">rot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span>
<a href="#__codelineno-16-15" id="__codelineno-16-15" name="__codelineno-16-15"></a><span class="w">  </span><span class="p">)</span>
<a href="#__codelineno-16-16" id="__codelineno-16-16" name="__codelineno-16-16"></a>
<a href="#__codelineno-16-17" id="__codelineno-16-17" name="__codelineno-16-17"></a><span class="c1"># Remove gene names in middle tracks</span>
<a href="#__codelineno-16-18" id="__codelineno-16-18" name="__codelineno-16-18"></a><span class="nf">plot_gene_map</span><span class="p">(</span>
<a href="#__codelineno-16-19" id="__codelineno-16-19" name="__codelineno-16-19"></a><span class="w">  </span><span class="n">dna_segs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ds_list</span><span class="p">,</span>
<a href="#__codelineno-16-20" id="__codelineno-16-20" name="__codelineno-16-20"></a><span class="w">  </span><span class="n">comparisons</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">filt_blast_list</span><span class="p">,</span>
<a href="#__codelineno-16-21" id="__codelineno-16-21" name="__codelineno-16-21"></a><span class="w">  </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">annotations_bin_3_mod</span><span class="p">,</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">annotations_list</span><span class="p">[[</span><span class="m">4</span><span class="p">]]),</span>
<a href="#__codelineno-16-22" id="__codelineno-16-22" name="__codelineno-16-22"></a><span class="w">  </span><span class="n">dna_seg_labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_remove</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">ds_list</span><span class="p">),</span><span class="w"> </span><span class="s">"_coords"</span><span class="p">),</span>
<a href="#__codelineno-16-23" id="__codelineno-16-23" name="__codelineno-16-23"></a><span class="w">  </span><span class="n">dna_seg_scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<a href="#__codelineno-16-24" id="__codelineno-16-24" name="__codelineno-16-24"></a><span class="w">  </span><span class="n">gene_type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">"arrows"</span>
<a href="#__codelineno-16-25" id="__codelineno-16-25" name="__codelineno-16-25"></a><span class="p">)</span>
</code></pre></div>
</div>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-width="100%" href="../../figures/day4_synteny.03.synteny_final.png"><img alt="image" src="../../figures/day4_synteny.03.synteny_final.png"/></a></p>
<p>This looks much cleaner!</p>
<details class="success">
<summary>Results at a glance</summary>
<p>From the last plot, we see that:</p>
<ul>
<li>The arrangement sulfur assimilation genes are well conserved</li>
<li>Gene sequences of <em>sbp1</em>, <em>cysW</em>, and <em>cysA</em> are very conserved between bins 3, 5, and 8</li>
<li>Bin 3 has 3 unknown domains between its sulfur binding protein and the other sulfur assimilation genes</li>
<li><em>cysW</em> in bin 9 shows some sequence homology with <em>cysU</em> from bin 8</li>
</ul>
</details>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.tabs", "navigation.tabs.sticky", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../../assets/javascripts/bundle.b4d07000.min.js"></script>
<script src="../../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>